// keynote-archives/lib/utils/reader.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Uint8ArrayReader = void 0;
class Uint8ArrayReader {
    constructor(buf, start = 0, end = buf.length) {
        this.buf = buf;
        this.start = start;
        this.end = end;
        this.pos = start;
    }
    eof() {
        return this.pos >= this.buf.length;
    }
    readBytes(length) {
        this.assertBounds(length);
        const result = new Uint8ArrayReader(this.buf, this.pos, this.pos + length - 1);
        this.pos += length;
        return result;
    }
    assertBounds(length) {
        if (this.pos < this.start || this.pos + length > this.buf.length) {
            throw new Error('Reading operation will lead to out of bounds.');
        }
    }
    readVarint32() {
        let result = 0;
        let index = 0;
        for (; index < 4; index++) {
            this.assertBounds(index + 1);
            const byte = this.buf[this.pos + index];
            result |= (byte & 0x7f) << (7 * index);
            if ((byte & 0x80) === 0) {
                break;
            }
        }
        this.pos += index + 1;
        return result;
    }
    readUint24() {
        this.assertBounds(4);
        const result = this.buf[this.pos] | (this.buf[this.pos + 1] << 8) | (this.buf[this.pos + 2] << 16);
        this.pos += 3;
        return result;
    }
    readUint8() {
        this.assertBounds(1);
        const result = this.buf[this.pos];
        this.pos += 1;
        return result;
    }
    slice() {
        return this.buf.slice(this.start, this.end + 1);
    }
}
exports.Uint8ArrayReader = Uint8ArrayReader;
