// keynote-archives/lib/utils/split.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.splitObjectsAs = void 0;
const TSPArchiveMessages_1 = require("../generated/TSPArchiveMessages");
const reader_1 = require("./reader");
async function* splitObjectsAs(chunk, registry) {
    const reader = new reader_1.Uint8ArrayReader(chunk);
    while (!reader.eof()) {
        const offset = reader.pos;
        let archiveInfo;
        try {
            const length = reader.readVarint32();
            const buffer = reader.readBytes(length).slice();
            archiveInfo = TSPArchiveMessages_1.ArchiveInfo.fromBinary(buffer);
        }
        catch (e) {
            console.error('Error while parsing archive info! ', e);
            return;
        }
        const messages = [];
        for (const [index, messageInfo] of Object.entries(archiveInfo.messageInfos)) {
            try {
                const offset = reader.pos;
                const messageType = registry[messageInfo.type];
                const messagePayload = reader.readBytes(messageInfo.length).slice();
                const message = messageType.fromBinary(messagePayload);
                messages.push({
                    info: messageInfo,
                    offset,
                    length: reader.pos - offset,
                    data: message
                });
            }
            catch (e) {
                console.error('Error while parsing message! ', index, e);
            }
        }
        const length = reader.pos - offset;
        const object = {
            identifier: archiveInfo.identifier,
            shouldMerge: archiveInfo.shouldMerge,
            offset,
            length,
            messages
        };
        yield object;
    }
}
exports.splitObjectsAs = splitObjectsAs;
