// keynote-archives/lib/generated/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TSDCommandArchives$Archives = exports.TSDArchives$Archives = exports.TSCHPreUFFArchives$Archives = exports.TSCHArchives_sos$Archives = exports.TSCHCommandArchives$Archives = exports.TSCHArchives_GEN$Archives = exports.TSCHArchives_Common$Archives = exports.TSCHArchives$Archives = exports.TSCH3DArchives$Archives = exports.TSCEArchives$Archives = exports.TSACommandArchives_sos$Archives = exports.TSAArchives_sos$Archives = exports.TSAArchives$Archives = exports.KNCommandArchives_sos$Archives = exports.KNArchives_sos$Archives = exports.KNCommandArchives$Archives = exports.KNArchives$Archives = exports.KeynoteArchives = exports.TSWPArchives_sos = exports.TSWPCommandArchives = exports.TSWPArchives = exports.TSTArchives_sos = exports.TSTStylePropertyArchiving = exports.TSTCommandArchives = exports.TSTArchives = exports.TSSArchives_sos = exports.TSSArchives = exports.TSPMessages = exports.TSPDatabaseMessages = exports.TSPArchiveMessages = exports.TSKArchives_sos = exports.TSKArchives = exports.TSDArchives_sos = exports.TSDCommandArchives = exports.TSDArchives = exports.TSCHPreUFFArchives = exports.TSCHArchives_sos = exports.TSCHCommandArchives = exports.TSCHArchives_GEN = exports.TSCHArchives_Common = exports.TSCHArchives = exports.TSCH3DArchives = exports.TSCEArchives = exports.TSACommandArchives_sos = exports.TSAArchives_sos = exports.TSAArchives = exports.KNCommandArchives_sos = exports.KNArchives_sos = exports.KNCommandArchives = exports.KNArchives = void 0;
exports.AllArchives = exports.TSWPArchives_sos$Archives = exports.TSWPCommandArchives$Archives = exports.TSWPArchives$Archives = exports.TSTArchives_sos$Archives = exports.TSTStylePropertyArchiving$Archives = exports.TSTCommandArchives$Archives = exports.TSTArchives$Archives = exports.TSSArchives_sos$Archives = exports.TSSArchives$Archives = exports.TSPMessages$Archives = exports.TSPDatabaseMessages$Archives = exports.TSPArchiveMessages$Archives = exports.TSKArchives_sos$Archives = exports.TSKArchives$Archives = exports.TSDArchives_sos$Archives = void 0;
const tslib_1 = require("tslib");
tslib_1.__exportStar(require("../utils"), exports);
exports.KNArchives = require("./KNArchives");
exports.KNCommandArchives = require("./KNCommandArchives");
exports.KNArchives_sos = require("./KNArchives_sos");
exports.KNCommandArchives_sos = require("./KNCommandArchives_sos");
exports.TSAArchives = require("./TSAArchives");
exports.TSAArchives_sos = require("./TSAArchives_sos");
exports.TSACommandArchives_sos = require("./TSACommandArchives_sos");
exports.TSCEArchives = require("./TSCEArchives");
exports.TSCH3DArchives = require("./TSCH3DArchives");
exports.TSCHArchives = require("./TSCHArchives");
exports.TSCHArchives_Common = require("./TSCHArchives_Common");
exports.TSCHArchives_GEN = require("./TSCHArchives_GEN");
exports.TSCHCommandArchives = require("./TSCHCommandArchives");
exports.TSCHArchives_sos = require("./TSCHArchives_sos");
exports.TSCHPreUFFArchives = require("./TSCHPreUFFArchives");
exports.TSDArchives = require("./TSDArchives");
exports.TSDCommandArchives = require("./TSDCommandArchives");
exports.TSDArchives_sos = require("./TSDArchives_sos");
exports.TSKArchives = require("./TSKArchives");
exports.TSKArchives_sos = require("./TSKArchives_sos");
exports.TSPArchiveMessages = require("./TSPArchiveMessages");
exports.TSPDatabaseMessages = require("./TSPDatabaseMessages");
exports.TSPMessages = require("./TSPMessages");
exports.TSSArchives = require("./TSSArchives");
exports.TSSArchives_sos = require("./TSSArchives_sos");
exports.TSTArchives = require("./TSTArchives");
exports.TSTCommandArchives = require("./TSTCommandArchives");
exports.TSTStylePropertyArchiving = require("./TSTStylePropertyArchiving");
exports.TSTArchives_sos = require("./TSTArchives_sos");
exports.TSWPArchives = require("./TSWPArchives");
exports.TSWPCommandArchives = require("./TSWPCommandArchives");
exports.TSWPArchives_sos = require("./TSWPArchives_sos");
const KNArchives = require("./KNArchives");
const KNCommandArchives = require("./KNCommandArchives");
const KNArchives_sos = require("./KNArchives_sos");
const KNCommandArchives_sos = require("./KNCommandArchives_sos");
const TSAArchives = require("./TSAArchives");
const TSAArchives_sos = require("./TSAArchives_sos");
const TSACommandArchives_sos = require("./TSACommandArchives_sos");
const TSCEArchives = require("./TSCEArchives");
const TSCH3DArchives = require("./TSCH3DArchives");
const TSCHArchives = require("./TSCHArchives");
const TSCHArchives_Common = require("./TSCHArchives_Common");
const TSCHArchives_GEN = require("./TSCHArchives_GEN");
const TSCHCommandArchives = require("./TSCHCommandArchives");
const TSCHArchives_sos = require("./TSCHArchives_sos");
const TSCHPreUFFArchives = require("./TSCHPreUFFArchives");
const TSDArchives = require("./TSDArchives");
const TSDCommandArchives = require("./TSDCommandArchives");
const TSDArchives_sos = require("./TSDArchives_sos");
const TSKArchives = require("./TSKArchives");
const TSKArchives_sos = require("./TSKArchives_sos");
const TSPArchiveMessages = require("./TSPArchiveMessages");
const TSPDatabaseMessages = require("./TSPDatabaseMessages");
const TSPMessages = require("./TSPMessages");
const TSSArchives = require("./TSSArchives");
const TSSArchives_sos = require("./TSSArchives_sos");
const TSTArchives = require("./TSTArchives");
const TSTCommandArchives = require("./TSTCommandArchives");
const TSTStylePropertyArchiving = require("./TSTStylePropertyArchiving");
const TSTArchives_sos = require("./TSTArchives_sos");
const TSWPArchives = require("./TSWPArchives");
const TSWPCommandArchives = require("./TSWPCommandArchives");
const TSWPArchives_sos = require("./TSWPArchives_sos");
exports.KeynoteArchives = {
    1: KNArchives.DocumentArchive,
    2: KNArchives.ShowArchive,
    3: KNArchives.UIStateArchive,
    4: KNArchives.SlideNodeArchive,
    5: KNArchives.SlideArchive,
    6: KNArchives.SlideArchive,
    7: KNArchives.PlaceholderArchive,
    8: KNArchives.BuildArchive,
    9: KNArchives.SlideStyleArchive,
    10: KNArchives.ThemeArchive,
    11: KNArchives.PasteboardNativeStorageArchive,
    12: KNArchives.PlaceholderArchive,
    14: TSWPArchives.TextualAttachmentArchive,
    15: KNArchives.NoteArchive,
    16: KNArchives.RecordingArchive,
    17: KNArchives.RecordingEventTrackArchive,
    18: KNArchives.RecordingMovieTrackArchive,
    19: KNArchives.ClassicStylesheetRecordArchive,
    20: KNArchives.ClassicThemeRecordArchive,
    21: KNArchives.Soundtrack,
    22: KNArchives.SlideNumberAttachmentArchive,
    23: KNArchives.DesktopUILayoutArchive,
    24: KNArchives.CanvasSelectionArchive,
    25: KNArchives.SlideCollectionSelectionArchive,
    26: KNArchives.MotionBackgroundStyleArchive,
    100: KNCommandArchives.CommandBuildSetValueArchive,
    101: KNCommandArchives.CommandShowInsertSlideArchive,
    102: KNCommandArchives.CommandShowMoveSlideArchive,
    103: KNCommandArchives.CommandShowRemoveSlideArchive,
    104: KNCommandArchives.CommandSlideInsertDrawablesArchive,
    105: KNCommandArchives.CommandSlideRemoveDrawableArchive,
    106: KNCommandArchives.CommandSlideNodeSetPropertyArchive,
    107: KNCommandArchives.CommandSlideInsertBuildArchive,
    109: KNCommandArchives.CommandSlideRemoveBuildArchive,
    110: KNCommandArchives.CommandSlideInsertBuildChunkArchive,
    111: KNCommandArchives.CommandSlideMoveBuildChunksArchive,
    112: KNCommandArchives.CommandSlideRemoveBuildChunkArchive,
    114: KNCommandArchives.CommandTransitionSetValueArchive,
    118: KNCommandArchives.CommandSlideMoveDrawableZOrderArchive,
    119: KNCommandArchives.CommandChangeTemplateSlideArchive,
    123: KNCommandArchives.CommandShowSetSlideNumberVisibilityArchive,
    124: KNCommandArchives.CommandShowSetValueArchive,
    128: KNCommandArchives.CommandShowMarkOutOfSyncRecordingArchive,
    129: KNCommandArchives.CommandShowRemoveRecordingArchive,
    130: KNCommandArchives.CommandShowReplaceRecordingArchive,
    131: KNCommandArchives.CommandShowSetSoundtrack,
    132: KNCommandArchives.CommandSoundtrackSetValue,
    134: KNCommandArchives.CommandMoveTemplatesArchive,
    135: KNCommandArchives.CommandInsertTemplateArchive,
    136: KNCommandArchives.CommandSlideSetStyleArchive,
    137: KNCommandArchives.CommandSlideSetPlaceholdersForTagsArchive,
    138: KNCommandArchives.CommandBuildChunkSetValueArchive,
    140: KNCommandArchives.CommandRemoveTemplateArchive,
    142: KNCommandArchives.CommandTemplateSetThumbnailTextArchive,
    143: KNCommandArchives.CommandShowChangeThemeArchive,
    144: KNCommandArchives.CommandSlidePrimitiveSetTemplateArchive,
    145: KNCommandArchives.CommandTemplateSetBodyStylesArchive,
    146: KNCommandArchives_sos.CommandSlideReapplyTemplateSlideArchive,
    148: KNCommandArchives.ChartInfoGeometryCommandArchive,
    150: KNCommandArchives.CommandSlideUpdateTemplateDrawables,
    152: KNCommandArchives.CommandSlideSetBackgroundFillArchive,
    153: KNArchives.BuildChunkArchive,
    156: KNCommandArchives.CommandSlideNodeSetViewStatePropertyArchive,
    157: KNCommandArchives.CommandBuildUpdateChunkCountArchive,
    158: KNCommandArchives.CommandBuildUpdateChunkReferentsArchive,
    159: KNArchives.BuildAttributeTupleArchive,
    160: KNCommandArchives.CommandSetThemeCustomEffectTimingCurveArchive,
    161: KNCommandArchives.CommandShowChangeSlideSizeArchive,
    162: KNCommandArchives.InsertBuildDescriptionArchive,
    163: KNCommandArchives.RemoveBuildDescriptionArchive,
    164: KNCommandArchives.DocumentSelectionTransformerArchive,
    165: KNCommandArchives.SlideCollectionSelectionTransformerArchive,
    166: KNCommandArchives.OutlineCanvasSelectionTransformerArchive,
    167: KNCommandArchives.NoteCanvasSelectionTransformerArchive,
    168: KNCommandArchives.CanvasSelectionTransformerArchive,
    169: KNCommandArchives.OutlineSelectionTransformerArchive,
    170: KNCommandArchives.UndoObjectArchive,
    172: KNArchives.PrototypeForUndoTemplateChangeArchive,
    173: KNCommandArchives.CommandShowMarkOutOfSyncRecordingIfNeededArchive,
    174: KNCommandArchives_sos.InducedVerifyDocumentWithServerCommandArchive,
    175: KNCommandArchives_sos.InducedVerifyDrawableZOrdersWithServerCommandArchive,
    176: KNCommandArchives.CommandPrimitiveInsertTemplateArchive,
    177: KNCommandArchives.CommandPrimitiveRemoveTemplateArchive,
    178: KNCommandArchives.CommandTemplateSlideSetPlaceholderForTagArchive,
    179: KNCommandArchives.CommandSlidePropagateSetPlaceholderForTagArchive,
    180: KNArchives.ActionGhostSelectionArchive,
    181: KNCommandArchives.ActionGhostSelectionTransformerArchive,
    182: KNCommandArchives.CommandSlideResetTemplateBackgroundObjectsArchive,
    184: KNArchives.LiveVideoSource,
    185: KNArchives.LiveVideoSourceCollection,
    186: KNCommandArchives.CommandLiveVideoInfoApplyPreset,
    187: KNCommandArchives.CommandLiveVideoInfoSetSource,
    188: KNCommandArchives.CommandLiveVideoInfoSetValue,
    189: KNCommandArchives.CommandLiveVideoSourceSetValue,
    190: KNCommandArchives.CommandLiveVideoStyleSetValue,
    191: KNCommandArchives.CommandThemeAddLiveVideoSource,
    192: KNCommandArchives.CommandThemeRemoveLiveVideoSource,
    194: KNCommandArchives.CommandMotionBackgroundStyleSetValueArchive,
    195: KNCommandArchives.CommandMotionBackgroundStyleUpdatePosterFrameDataArchive,
    200: TSKArchives.DocumentArchive,
    201: TSKArchives.LocalCommandHistory,
    202: TSKArchives.CommandGroupArchive,
    203: TSKArchives.CommandContainerArchive,
    205: TSKArchives.TreeNode,
    210: TSKArchives.ViewStateArchive,
    211: TSKArchives.DocumentSupportArchive,
    212: TSKArchives.AnnotationAuthorArchive,
    213: TSKArchives.AnnotationAuthorStorageArchive,
    215: TSKArchives.SetAnnotationAuthorColorCommandArchive,
    218: TSKArchives.CollaborationCommandHistory,
    219: TSKArchives.DocumentSelectionArchive,
    220: TSKArchives.CommandSelectionBehaviorArchive,
    221: TSKArchives.NullCommandArchive,
    222: TSKArchives.CustomFormatListArchive,
    223: TSKArchives.GroupCommitCommandArchive,
    224: TSKArchives.InducedCommandCollectionArchive,
    225: TSKArchives.InducedCommandCollectionCommitCommandArchive,
    226: TSKArchives.CollaborationDocumentSessionState,
    227: TSKArchives.CollaborationCommandHistoryCoalescingGroup,
    228: TSKArchives.CollaborationCommandHistoryCoalescingGroupNode,
    229: TSKArchives.CollaborationCommandHistoryOriginatingCommandAcknowledgementObserver,
    230: TSKArchives.DocumentSupportCollaborationState,
    231: TSKArchives.ChangeDocumentPackageTypeCommandArchive,
    232: TSKArchives.UpgradeDocPostProcessingCommandArchive,
    233: TSKArchives.FinalCommandPairArchive,
    234: TSKArchives.OutgoingCommandQueueItem,
    235: TSKArchives.TransformerEntry,
    238: TSKArchives.CreateLocalStorageSnapshotCommandArchive,
    240: TSKArchives.SelectionPathTransformerArchive,
    241: TSKArchives.NativeContentDescription,
    242: TSDArchives.PencilAnnotationStorageArchive,
    245: TSKArchives.OperationStorage,
    246: TSKArchives.OperationStorageEntryArray,
    247: TSKArchives.OperationStorageEntryArraySegment,
    248: TSKArchives.BlockDiffsAtCurrentRevisionCommand,
    249: TSKArchives.OutgoingCommandQueue,
    250: TSKArchives.OutgoingCommandQueueSegment,
    251: TSKArchives.PropagatedCommandCollectionArchive,
    252: TSKArchives.LocalCommandHistoryItem,
    253: TSKArchives.LocalCommandHistoryArray,
    254: TSKArchives.LocalCommandHistoryArraySegment,
    255: TSKArchives.CollaborationCommandHistoryItem,
    256: TSKArchives.CollaborationCommandHistoryArray,
    257: TSKArchives.CollaborationCommandHistoryArraySegment,
    258: TSKArchives.PencilAnnotationUIState,
    259: TSKArchives_sos.FixCorruptedDataCommandArchive,
    260: TSKArchives.CommandAssetChunkArchive,
    261: TSKArchives.AssetUploadStatusCommandArchive,
    262: TSKArchives.AssetUnmaterializedOnServerCommandArchive,
    263: TSKArchives.CommandBehaviorArchive,
    264: TSKArchives.CommandBehaviorSelectionPathStorageArchive,
    265: TSKArchives.CommandActivityBehaviorArchive,
    273: TSKArchives.ActivityOnlyCommandArchive,
    275: TSKArchives.SetActivityAuthorShareParticipantIDCommandArchive,
    279: TSKArchives.ActivityAuthorCacheArchive,
    280: TSKArchives.ActivityStreamArchive,
    281: TSKArchives.ActivityArchive,
    282: TSKArchives.ActivityCommitCommandArchive,
    283: TSKArchives.ActivityStreamActivityArray,
    284: TSKArchives.ActivityStreamActivityArraySegment,
    285: TSKArchives.ActivityStreamRemovedAuthorAuditorPendingStateArchive,
    286: TSKArchives.ActivityAuthorArchive,
    287: TSKArchives_sos.ResetActivityStreamCommandArchive,
    288: TSKArchives_sos.RemoveAuthorIdentifiersCommandArchive,
    289: TSKArchives.ActivityCursorCollectionPersistenceWrapperArchive,
    400: TSSArchives.StyleArchive,
    401: TSSArchives.StylesheetArchive,
    402: TSSArchives.ThemeArchive,
    412: TSSArchives.StyleUpdatePropertyMapCommandArchive,
    413: TSSArchives.ThemeReplacePresetCommandArchive,
    414: TSSArchives.ThemeAddStylePresetCommandArchive,
    415: TSSArchives.ThemeRemoveStylePresetCommandArchive,
    416: TSSArchives.ThemeReplaceColorPresetCommandArchive,
    417: TSSArchives.ThemeMovePresetCommandArchive,
    419: TSSArchives.ThemeReplaceStylePresetAndDisconnectStylesCommandArchive,
    600: TSAArchives.DocumentArchive,
    601: TSAArchives.FunctionBrowserStateArchive,
    602: TSAArchives.PropagatePresetCommandArchive,
    603: TSAArchives.ShortcutControllerArchive,
    604: TSAArchives.ShortcutCommandArchive,
    605: TSAArchives.AddCustomFormatCommandArchive,
    606: TSAArchives.UpdateCustomFormatCommandArchive,
    607: TSAArchives.ReplaceCustomFormatCommandArchive,
    611: TSACommandArchives_sos.VerifyObjectsWithServerCommandArchive,
    612: TSAArchives.InducedVerifyObjectsWithServerCommandArchive,
    613: TSACommandArchives_sos.VerifyDocumentWithServerCommandArchive,
    614: TSACommandArchives_sos.VerifyDrawableZOrdersWithServerCommandArchive,
    615: TSACommandArchives_sos.InducedVerifyDrawableZOrdersWithServerCommandArchive,
    616: TSAArchives.NeedsMediaCompatibilityUpgradeCommandArchive,
    617: TSAArchives.ChangeDocumentLocaleCommandArchive,
    618: TSAArchives.StyleUpdatePropertyMapCommandArchive,
    619: TSAArchives.RemoteDataChangeCommandArchive,
    623: TSAArchives.GalleryItem,
    624: TSAArchives.GallerySelectionTransformer,
    625: TSAArchives.GalleryItemSelection,
    626: TSAArchives.GalleryItemSelectionTransformer,
    627: TSAArchives.GalleryInfoSetValueCommandArchive,
    628: TSAArchives.GalleryItemSetGeometryCommand,
    629: TSAArchives.GalleryItemSetValueCommand,
    630: TSAArchives.InducedVerifyTransformHistoryWithServerCommandArchive,
    631: TSACommandArchives_sos.CommandReapplyMasterArchive,
    632: TSACommandArchives_sos.PropagateMasterChangeCommandArchive,
    633: TSAArchives.CaptionInfoArchive,
    634: TSAArchives.CaptionPlacementArchive,
    635: TSAArchives.TitlePlacementCommandArchive,
    636: TSAArchives.GalleryInfoInsertItemsCommandArchive,
    637: TSAArchives.GalleryInfoRemoveItemsCommandArchive,
    638: TSACommandArchives_sos.VerifyActivityStreamWithServerCommandArchive,
    639: TSAArchives_sos.InducedVerifyActivityStreamWithServerCommandArchive,
    2001: TSWPArchives.StorageArchive,
    2002: TSWPArchives.SelectionArchive,
    2003: TSWPArchives.DrawableAttachmentArchive,
    2004: TSWPArchives.TextualAttachmentArchive,
    2005: TSWPArchives.StorageArchive,
    2006: TSWPArchives.UIGraphicalAttachment,
    2007: TSWPArchives.TextualAttachmentArchive,
    2008: TSWPArchives.FootnoteReferenceAttachmentArchive,
    2009: TSWPArchives.TextualAttachmentArchive,
    2010: TSWPArchives.TSWPTOCPageNumberAttachmentArchive,
    2011: TSWPArchives.ShapeInfoArchive,
    2013: TSWPArchives.HighlightArchive,
    2014: TSWPArchives.CommentInfoArchive,
    2015: TSWPArchives.EquationInfoArchive,
    2016: TSWPArchives.PencilAnnotationArchive,
    2021: TSWPArchives.CharacterStyleArchive,
    2022: TSWPArchives.ParagraphStyleArchive,
    2023: TSWPArchives.ListStyleArchive,
    2024: TSWPArchives.ColumnStyleArchive,
    2025: TSWPArchives.ShapeStyleArchive,
    2026: TSWPArchives.TOCEntryStyleArchive,
    2031: TSWPArchives.PlaceholderSmartFieldArchive,
    2032: TSWPArchives.HyperlinkFieldArchive,
    2033: TSWPArchives.FilenameSmartFieldArchive,
    2034: TSWPArchives.DateTimeSmartFieldArchive,
    2035: TSWPArchives.BookmarkFieldArchive,
    2036: TSWPArchives.MergeSmartFieldArchive,
    2037: TSWPArchives.CitationRecordArchive,
    2038: TSWPArchives.CitationSmartFieldArchive,
    2039: TSWPArchives.UnsupportedHyperlinkFieldArchive,
    2040: TSWPArchives.BibliographySmartFieldArchive,
    2041: TSWPArchives.TOCSmartFieldArchive,
    2042: TSWPArchives.RubyFieldArchive,
    2043: TSWPArchives.NumberAttachmentArchive,
    2050: TSWPArchives.TextStylePresetArchive,
    2051: TSWPArchives.TOCSettingsArchive,
    2052: TSWPArchives.TOCEntryInstanceArchive,
    2053: TSWPArchives_sos.StyleDiffArchive,
    2060: TSWPArchives.ChangeArchive,
    2061: TSKArchives.DeprecatedChangeAuthorArchive,
    2062: TSWPArchives.ChangeSessionArchive,
    2101: TSWPCommandArchives.TextCommandArchive,
    2107: TSWPCommandArchives.ApplyPlaceholderTextCommandArchive,
    2116: TSWPCommandArchives.ApplyRubyTextCommandArchive,
    2118: TSWPCommandArchives.ModifyRubyTextCommandArchive,
    2119: TSWPCommandArchives.UpdateDateTimeFieldCommandArchive,
    2120: TSWPCommandArchives.ModifyTOCSettingsBaseCommandArchive,
    2121: TSWPCommandArchives.ModifyTOCSettingsForTOCInfoCommandArchive,
    2123: TSWPCommandArchives.SetObjectPropertiesCommandArchive,
    2124: TSWPCommandArchives.UpdateFlowInfoCommandArchive,
    2125: TSWPCommandArchives.AddFlowInfoCommandArchive,
    2126: TSWPCommandArchives.RemoveFlowInfoCommandArchive,
    2127: TSWPCommandArchives.ContainedObjectsCommandArchive,
    2128: TSWPCommandArchives.EquationInfoGeometryCommandArchive,
    2206: TSWPCommandArchives.AnchorAttachmentCommandArchive,
    2217: TSWPCommandArchives.TextCommentReplyCommandArchive,
    2231: TSWPCommandArchives.ShapeApplyPresetCommandArchive,
    2240: TSWPArchives.TOCInfoArchive,
    2241: TSWPArchives.TOCAttachmentArchive,
    2242: TSWPArchives.TOCLayoutHintArchive,
    2400: TSWPCommandArchives.StyleBaseCommandArchive,
    2401: TSWPCommandArchives.StyleCreateCommandArchive,
    2402: TSWPCommandArchives.StyleRenameCommandArchive,
    2404: TSWPCommandArchives.StyleDeleteCommandArchive,
    2405: TSWPCommandArchives.StyleReorderCommandArchive,
    2406: TSWPCommandArchives.StyleUpdatePropertyMapCommandArchive,
    2407: TSWPCommandArchives.StorageActionCommandArchive,
    2408: TSWPCommandArchives.ShapeStyleSetValueCommandArchive,
    2409: TSWPArchives.HyperlinkSelectionArchive,
    2410: TSWPArchives.FlowInfoArchive,
    2411: TSWPArchives.FlowInfoContainerArchive,
    2412: TSWPCommandArchives.PencilAnnotationSelectionTransformerArchive,
    3002: TSDArchives.DrawableArchive,
    3003: TSDArchives.ContainerArchive,
    3004: TSDArchives.ShapeArchive,
    3005: TSDArchives.ImageArchive,
    3006: TSDArchives.MaskArchive,
    3007: TSDArchives.MovieArchive,
    3008: TSDArchives.GroupArchive,
    3009: TSDArchives.ConnectionLineArchive,
    3015: TSDArchives.ShapeStyleArchive,
    3016: TSDArchives.MediaStyleArchive,
    3021: TSDCommandArchives.InfoGeometryCommandArchive,
    3022: TSDCommandArchives.DrawablePathSourceCommandArchive,
    3024: TSDCommandArchives.ImageMaskCommandArchive,
    3025: TSDCommandArchives.ImageMediaCommandArchive,
    3026: TSDCommandArchives.ImageReplaceCommandArchive,
    3027: TSDCommandArchives.MediaOriginalSizeCommandArchive,
    3028: TSDCommandArchives.ShapeStyleSetValueCommandArchive,
    3030: TSDCommandArchives.MediaStyleSetValueCommandArchive,
    3031: TSDCommandArchives.ShapeApplyPresetCommandArchive,
    3032: TSDCommandArchives.MediaApplyPresetCommandArchive,
    3034: TSDCommandArchives.MovieSetValueCommandArchive,
    3036: TSDCommandArchives.ExteriorTextWrapCommandArchive,
    3037: TSDCommandArchives.MediaFlagsCommandArchive,
    3040: TSDCommandArchives.DrawableHyperlinkCommandArchive,
    3041: TSDCommandArchives.ConnectionLineConnectCommandArchive,
    3042: TSDCommandArchives.InstantAlphaCommandArchive,
    3043: TSDCommandArchives.DrawableLockCommandArchive,
    3044: TSDCommandArchives.ImageNaturalSizeCommandArchive,
    3045: TSDArchives.CanvasSelectionArchive,
    3047: TSDArchives.GuideStorageArchive,
    3048: TSDCommandArchives.StyledInfoSetStyleCommandArchive,
    3049: TSDCommandArchives.DrawableInfoCommentCommandArchive,
    3050: TSDCommandArchives.GuideCommandArchive,
    3051: TSDCommandArchives.DrawableAspectRatioLockedCommandArchive,
    3052: TSDCommandArchives.ContainerRemoveChildrenCommandArchive,
    3053: TSDCommandArchives.ContainerInsertChildrenCommandArchive,
    3054: TSDCommandArchives.ContainerReorderChildrenCommandArchive,
    3055: TSDCommandArchives.ImageAdjustmentsCommandArchive,
    3056: TSDArchives.CommentStorageArchive,
    3057: TSDArchives.ThemeReplaceFillPresetCommandArchive,
    3058: TSDCommandArchives.DrawableAccessibilityDescriptionCommandArchive,
    3059: TSDCommandArchives.PasteStyleCommandArchive,
    3061: TSDArchives.DrawableSelectionArchive,
    3062: TSDArchives.GroupSelectionArchive,
    3063: TSDArchives.PathSelectionArchive,
    3064: TSDCommandArchives.CommentInvalidatingCommandSelectionBehaviorArchive,
    3065: TSDCommandArchives.ImageInfoAbstractGeometryCommandArchive,
    3066: TSDCommandArchives.ImageInfoGeometryCommandArchive,
    3067: TSDCommandArchives.ImageInfoMaskGeometryCommandArchive,
    3068: TSDCommandArchives.UndoObjectArchive,
    3070: TSDArchives.ReplaceAnnotationAuthorCommandArchive,
    3071: TSDCommandArchives.DrawableSelectionTransformerArchive,
    3072: TSDCommandArchives.GroupSelectionTransformerArchive,
    3073: TSDCommandArchives.ShapeSelectionTransformerArchive,
    3074: TSDCommandArchives.PathSelectionTransformerArchive,
    3080: TSDCommandArchives.MediaInfoGeometryCommandArchive,
    3082: TSDCommandArchives.GroupUngroupInformativeCommandArchive,
    3083: TSDArchives.DrawableContentDescription,
    3084: TSDCommandArchives.ContainerRemoveDrawablesCommandArchive,
    3085: TSDCommandArchives.ContainerInsertDrawablesCommandArchive,
    3086: TSDArchives.PencilAnnotationArchive,
    3087: TSDCommandArchives.FreehandDrawingOpacityCommandArchive,
    3088: TSDCommandArchives.DrawablePencilAnnotationCommandArchive,
    3089: TSDArchives.PencilAnnotationSelectionArchive,
    3090: TSDArchives.FreehandDrawingContentDescription,
    3091: TSDArchives.FreehandDrawingToolkitUIState,
    3092: TSDCommandArchives.PencilAnnotationSelectionTransformerArchive,
    3094: TSDCommandArchives.FreehandDrawingAnimationCommandArchive,
    3095: TSDCommandArchives.InsertCaptionOrTitleCommandArchive,
    3096: TSDCommandArchives.RemoveCaptionOrTitleCommandArchive,
    3097: TSDArchives.StandinCaptionArchive,
    3098: TSDCommandArchives.SetCaptionOrTitleVisibilityCommandArchive,
    4000: TSCEArchives.CalculationEngineArchive,
    4001: TSCEArchives.FormulaRewriteCommandArchive,
    4003: TSCEArchives.NamedReferenceManagerArchive,
    4004: TSCEArchives.TrackedReferenceStoreArchive,
    4005: TSCEArchives.TrackedReferenceArchive,
    4007: TSCEArchives.RemoteDataStoreArchive,
    4008: TSCEArchives.FormulaOwnerDependenciesArchive,
    4009: TSCEArchives.CellRecordTileArchive,
    4010: TSCEArchives.RangePrecedentsTileArchive,
    4011: TSCEArchives.ReferencesToDirtyArchive,
    5000: TSCHPreUFFArchives.ChartInfoArchive,
    5002: TSCHPreUFFArchives.ChartGridArchive,
    5004: TSCHArchives.ChartMediatorArchive,
    5010: TSCHPreUFFArchives.ChartStyleArchive,
    5011: TSCHPreUFFArchives.ChartSeriesStyleArchive,
    5012: TSCHPreUFFArchives.ChartAxisStyleArchive,
    5013: TSCHPreUFFArchives.LegendStyleArchive,
    5014: TSCHPreUFFArchives.ChartNonStyleArchive,
    5015: TSCHPreUFFArchives.ChartSeriesNonStyleArchive,
    5016: TSCHPreUFFArchives.ChartAxisNonStyleArchive,
    5017: TSCHPreUFFArchives.LegendNonStyleArchive,
    5020: TSCHArchives.ChartStylePreset,
    5021: TSCHArchives.ChartDrawableArchive,
    5022: TSCHArchives_Common.ChartStyleArchive,
    5023: TSCHArchives_Common.ChartNonStyleArchive,
    5024: TSCHArchives_Common.LegendStyleArchive,
    5025: TSCHArchives_Common.LegendNonStyleArchive,
    5026: TSCHArchives_Common.ChartAxisStyleArchive,
    5027: TSCHArchives_Common.ChartAxisNonStyleArchive,
    5028: TSCHArchives_Common.ChartSeriesStyleArchive,
    5029: TSCHArchives_Common.ChartSeriesNonStyleArchive,
    5030: TSCHArchives_Common.ReferenceLineStyleArchive,
    5031: TSCHArchives_Common.ReferenceLineNonStyleArchive,
    5103: TSCHCommandArchives.CommandSetChartTypeArchive,
    5104: TSCHCommandArchives.CommandSetSeriesNameArchive,
    5105: TSCHCommandArchives.CommandSetCategoryNameArchive,
    5107: TSCHCommandArchives.CommandSetScatterFormatArchive,
    5108: TSCHCommandArchives.CommandSetLegendFrameArchive,
    5109: TSCHCommandArchives.CommandSetGridValueArchive,
    5110: TSCHCommandArchives.CommandSetGridDirectionArchive,
    5115: TSCHCommandArchives.CommandAddGridRowsArchive,
    5116: TSCHCommandArchives.CommandAddGridColumnsArchive,
    5118: TSCHCommandArchives.CommandMoveGridRowsArchive,
    5119: TSCHCommandArchives.CommandMoveGridColumnsArchive,
    5122: TSCHCommandArchives.CommandSetPieWedgeExplosion,
    5123: TSCHCommandArchives.CommandStyleSwapArchive,
    5125: TSCHCommandArchives.CommandChartApplyPreset,
    5126: TSCHCommandArchives.ChartCommandArchive,
    5127: TSCHCommandArchives.CommandReplaceGridValuesArchive,
    5129: TSCHArchives.StylePasteboardDataArchive,
    5130: TSCHCommandArchives.CommandSetMultiDataSetIndexArchive,
    5131: TSCHCommandArchives.CommandReplaceThemePresetArchive,
    5132: TSCHCommandArchives.CommandInvalidateWPCaches,
    5135: TSCHCommandArchives.CommandMutatePropertiesArchive,
    5136: TSCHCommandArchives.CommandScaleAllTextArchive,
    5137: TSCHCommandArchives.CommandSetFontFamilyArchive,
    5138: TSCHCommandArchives.CommandApplyFillSetArchive,
    5139: TSCHCommandArchives.CommandReplaceCustomFormatArchive,
    5140: TSCHCommandArchives.CommandAddReferenceLineArchive,
    5141: TSCHCommandArchives.CommandDeleteReferenceLineArchive,
    5142: TSCHCommandArchives.CommandDeleteGridColumnsArchive,
    5143: TSCHCommandArchives.CommandDeleteGridRowsArchive,
    5145: TSCHArchives.ChartSelectionArchive,
    5146: TSCHCommandArchives.ChartTextSelectionTransformerArchive,
    5147: TSCHCommandArchives.ChartSubselectionTransformerArchive,
    5148: TSCHCommandArchives.ChartDrawableSelectionTransformerArchive,
    5149: TSCHCommandArchives.ChartSubselectionTransformerHelperArchive,
    5150: TSCHCommandArchives.ChartRefLineSubselectionTransformerHelperArchive,
    5151: TSCHCommandArchives.CDESelectionTransformerArchive,
    5152: TSCHCommandArchives.ChartSubselectionIdentityTransformerHelperArchive,
    5154: TSCHCommandArchives.CommandPasteStyleArchive,
    5155: TSCHCommandArchives.CommandInducedReplaceChartGrid,
    5156: TSCHCommandArchives.CommandReplaceImageDataArchive,
    5157: TSCHCommandArchives.CommandInduced3DChartGeometry,
    6000: TSTArchives.TableInfoArchive,
    6001: TSTArchives.TableModelArchive,
    6002: TSTArchives.Tile,
    6003: TSTArchives.TableStyleArchive,
    6004: TSTArchives.CellStyleArchive,
    6005: TSTArchives.TableDataList,
    6006: TSTArchives.HeaderStorageBucket,
    6007: TSTArchives.WPTableInfoArchive,
    6008: TSTStylePropertyArchiving.TableStylePresetArchive,
    6009: TSTStylePropertyArchiving.TableStrokePresetArchive,
    6010: TSTArchives.ConditionalStyleSetArchive,
    6011: TSTArchives.TableDataListSegment,
    6030: TSTArchives.SelectionArchive,
    6031: TSTArchives.CellMapArchive,
    6032: TSTArchives.DeathhawkRdar39989167CellSelectionArchive,
    6033: TSTArchives.ConcurrentCellMapArchive,
    6034: TSTArchives.ConcurrentCellListArchive,
    6100: TSTCommandArchives.TableCommandArchive,
    6101: TSTCommandArchives.CommandDeleteCellsArchive,
    6102: TSTCommandArchives.CommandInsertColumnsOrRowsArchive,
    6103: TSTCommandArchives.CommandRemoveColumnsOrRowsArchive,
    6104: TSTCommandArchives.CommandResizeColumnOrRowArchive,
    6107: TSTCommandArchives.CommandSetTableNameArchive,
    6111: TSTCommandArchives.CommandChangeFreezeHeaderStateArchive,
    6114: TSTCommandArchives.CommandSetTableNameEnabledArchive,
    6117: TSTCommandArchives.CommandApplyTableStylePresetArchive,
    6120: TSTCommandArchives.CommandSetRepeatingHeaderEnabledArchive,
    6123: TSTCommandArchives.CommandSortArchive,
    6125: TSTCommandArchives.CommandStyleTableArchive,
    6126: TSTCommandArchives.CommandSetNumberOfDecimalPlacesArchive,
    6127: TSTCommandArchives.CommandSetShowThousandsSeparatorArchive,
    6128: TSTCommandArchives.CommandSetNegativeNumberStyleArchive,
    6129: TSTCommandArchives.CommandSetFractionAccuracyArchive,
    6131: TSTCommandArchives.CommandSetCurrencyCodeArchive,
    6132: TSTCommandArchives.CommandSetUseAccountingStyleArchive,
    6136: TSTCommandArchives.CommandSetTableFontNameArchive,
    6137: TSTCommandArchives.CommandSetTableFontSizeArchive,
    6142: TSTCommandArchives.CommandSetTableNameHeightArchive,
    6144: TSTArchives.MergeRegionMapArchive,
    6145: TSTCommandArchives.CommandHideShowArchive,
    6146: TSTCommandArchives.CommandSetBaseArchive,
    6147: TSTCommandArchives.CommandSetBasePlacesArchive,
    6148: TSTCommandArchives.CommandSetBaseUseMinusSignArchive,
    6149: TSTCommandArchives.CommandSetTextStylePropertiesArchive,
    6150: TSTCommandArchives.CommandCategoryChangeSummaryAggregateType,
    6152: TSTCommandArchives.CommandCategoryResizeColumnOrRowArchive,
    6153: TSTCommandArchives.CommandCategoryMoveRowsArchive,
    6156: TSTCommandArchives.CommandSetPencilAnnotationsArchive,
    6157: TSTCommandArchives.CommandCategoryWillChangeGroupValue,
    6158: TSTCommandArchives.CommandApplyConcurrentCellMapArchive,
    6159: TSTCommandArchives.CommandSetGroupSortOrderArchive,
    6179: TSTArchives.FormulaEqualsTokenAttachmentArchive,
    6181: TSTArchives.TokenAttachmentArchive,
    6182: TSTArchives.ExpressionNodeArchive,
    6183: TSTArchives.BooleanNodeArchive,
    6184: TSTArchives.NumberNodeArchive,
    6185: TSTArchives.StringNodeArchive,
    6186: TSTArchives.ArrayNodeArchive,
    6187: TSTArchives.ListNodeArchive,
    6188: TSTArchives.OperatorNodeArchive,
    6189: TSTArchives.FunctionNodeArchive,
    6190: TSTArchives.DateNodeArchive,
    6191: TSTArchives.ReferenceNodeArchive,
    6192: TSTArchives.DurationNodeArchive,
    6193: TSTArchives.ArgumentPlaceholderNodeArchive,
    6194: TSTArchives.PostfixOperatorNodeArchive,
    6195: TSTArchives.PrefixOperatorNodeArchive,
    6196: TSTArchives.FunctionEndNodeArchive,
    6197: TSTArchives.EmptyExpressionNodeArchive,
    6198: TSTArchives.LayoutHintArchive,
    6199: TSTArchives.CompletionTokenAttachmentArchive,
    6201: TSTArchives.TableDataList,
    6204: TSTArchives.HiddenStateFormulaOwnerArchive,
    6205: TSTCommandArchives.CommandSetAutomaticDurationUnitsArchive,
    6206: TSTArchives.PopUpMenuModel,
    6218: TSTArchives.RichTextPayloadArchive,
    6220: TSTArchives.FilterSetArchive,
    6221: TSTCommandArchives.CommandSetFiltersEnabledArchive,
    6224: TSTCommandArchives.CommandRewriteFilterFormulasForTableResizeArchive,
    6226: TSTCommandArchives.CommandTextPreflightInsertCellArchive,
    6228: TSTCommandArchives.CommandDeleteCellContentsArchive,
    6229: TSTCommandArchives.CommandPostflightSetCellArchive,
    6235: TSTArchives.IdentifierNodeArchive,
    6238: TSTCommandArchives.CommandSetDateTimeFormatArchive,
    6239: TSTCommandArchives.TableCommandSelectionBehaviorArchive,
    6244: TSTCommandArchives.CommandApplyCellCommentArchive,
    6246: TSTCommandArchives.CommandSetFormulaTokenizationArchive,
    6247: TSTArchives.TableStyleNetworkArchive,
    6250: TSTCommandArchives.CommandSetFilterSetTypeArchive,
    6255: TSTCommandArchives.CommandSetTextStyleArchive,
    6256: TSTCommandArchives.CommandJustForNotifyingArchive,
    6258: TSTCommandArchives.CommandSetSortOrderArchive,
    6262: TSTCommandArchives.CommandAddTableStylePresetArchive,
    6264: TSTArchives.CellDiffMapArchive,
    6265: TSTCommandArchives.CommandApplyCellContentsArchive,
    6266: TSTCommandArchives.CommandRemoveTableStylePresetArchive,
    6267: TSTArchives.ColumnRowUIDMapArchive,
    6268: TSTCommandArchives.CommandMoveColumnsOrRowsArchive,
    6269: TSTCommandArchives.CommandReplaceCustomFormatArchive,
    6270: TSTCommandArchives.CommandReplaceTableStylePresetArchive,
    6271: TSTArchives.FormulaSelectionArchive,
    6273: TSTArchives.CellListArchive,
    6275: TSTCommandArchives.CommandApplyCellDiffMapArchive,
    6276: TSTCommandArchives.CommandSetFilterSetArchive,
    6277: TSTCommandArchives.CommandMutateCellFormatArchive,
    6278: TSTCommandArchives.CommandSetStorageLanguageArchive,
    6280: TSTCommandArchives.CommandMergeArchive,
    6281: TSTCommandArchives.CommandUnmergeArchive,
    6282: TSTCommandArchives.CommandApplyCellMapArchive,
    6283: TSTArchives.ControlCellSelectionArchive,
    6284: TSTArchives.TableNameSelectionArchive,
    6285: TSTCommandArchives.CommandRewriteFormulasForTransposeArchive,
    6287: TSTCommandArchives.CommandTransposeTableArchive,
    6289: TSTCommandArchives.CommandSetDurationStyleArchive,
    6290: TSTCommandArchives.CommandSetDurationUnitSmallestLargestArchive,
    6291: TSTCommandArchives.CommandRewriteTableFormulasForRewriteSpecArchive,
    6292: TSTCommandArchives.CommandRewriteConditionalStylesForRewriteSpecArchive,
    6293: TSTCommandArchives.CommandRewriteFilterFormulasForRewriteSpecArchive,
    6294: TSTCommandArchives.CommandRewriteSortOrderForRewriteSpecArchive,
    6295: TSTArchives.StrokeSelectionArchive,
    6297: TSTArchives.LetNodeArchive,
    6298: TSTArchives.VariableNodeArchive,
    6299: TSTArchives.InNodeArchive,
    6300: TSTCommandArchives.CommandInverseMergeArchive,
    6301: TSTCommandArchives.CommandMoveCellsArchive,
    6302: TSTArchives.DefaultCellStylesContainerArchive,
    6303: TSTCommandArchives.CommandRewriteMergeFormulasArchive,
    6304: TSTCommandArchives.CommandChangeTableAreaForColumnOrRowArchive,
    6305: TSTArchives.StrokeSidecarArchive,
    6306: TSTArchives.StrokeLayerArchive,
    6307: TSTCommandArchives.CommandChooseTableIdRemapperArchive,
    6310: TSTCommandArchives.CommandSetWasCutArchive,
    6311: TSTArchives.AutofillSelectionArchive,
    6312: TSTArchives.StockCellSelectionArchive,
    6313: TSTCommandArchives.CommandSetNowArchive,
    6314: TSTCommandArchives.CommandSetStructuredTextImportRecordArchive,
    6315: TSTCommandArchives.CommandRewriteCategoryFormulasArchive,
    6316: TSTArchives.SummaryModelArchive,
    6317: TSTArchives.SummaryCellVendorArchive,
    6318: TSTArchives.CategoryOrderArchive,
    6320: TSTCommandArchives.CommandCategoryCollapseExpandGroupArchive,
    6321: TSTCommandArchives.CommandCategorySetGroupingColumnsArchive,
    6323: TSTCommandArchives.CommandRewriteHiddenStatesForGroupByChangeArchive,
    6350: TSTCommandArchives.IdempotentSelectionTransformerArchive,
    6351: TSTCommandArchives.TableSubSelectionTransformerBaseArchive,
    6352: TSTCommandArchives.TableNameSelectionTransformerArchive,
    6353: TSTCommandArchives.RegionSelectionTransformerArchive,
    6354: TSTCommandArchives.RowColumnSelectionTransformerArchive,
    6355: TSTCommandArchives.ControlCellSelectionTransformerArchive,
    6357: TSTArchives.ChangePropagationMapWrapper,
    6358: TSTCommandArchives.WPSelectionTransformerArchive,
    6359: TSTCommandArchives.StockCellSelectionTransformerArchive,
    6360: TSTCommandArchives.CommandSetRangeControlMinMaxIncArchive,
    6361: TSTCommandArchives.CommandCategorySetLabelRowVisibility,
    6362: TSTCommandArchives.CommandRewritePencilAnnotationFormulasArchive,
    6363: TSTArchives.PencilAnnotationArchive,
    6364: TSTCommandArchives.StrokeSelectionTransformerArchive,
    6365: TSTArchives.HeaderNameMgrTileArchive,
    6366: TSTArchives.HeaderNameMgrArchive,
    6367: TSTArchives.CellDiffArray,
    6368: TSTArchives.CellDiffArraySegment,
    6369: TSTArchives.PivotOrderArchive,
    6370: TSTArchives.PivotOwnerArchive,
    6371: TSTCommandArchives.CommandPivotSetPivotRulesArchive,
    6372: TSTArchives.CategoryOwnerRefArchive,
    6373: TSTArchives.GroupByArchive,
    6374: TSTArchives.PivotGroupingColumnOptionsMapArchive,
    6375: TSTCommandArchives.CommandPivotSetGroupingColumnOptionsArchive,
    6376: TSTCommandArchives.CommandPivotHideShowGrandTotalsArchive,
    6377: TSTCommandArchives.CommandPivotSortArchive,
    6379: TSTCommandArchives.CommandRewritePivotOwnerFormulasArchive,
    6380: TSTCommandArchives.CommandRewriteTrackedReferencesArchive,
    6381: TSTCommandArchives.CommandExtendTableIDHistoryArchive,
    6382: TSTArchives.GroupByArchive_AggregatorArchive,
    6383: TSTArchives.GroupByArchive_GroupNodeArchive,
    10011: TSWPArchives.SectionPlaceholderArchive,
    10020: TSWPCommandArchives.ShapeSelectionTransformerArchive,
    10021: TSWPCommandArchives.SelectionTransformerArchive,
    10022: TSWPCommandArchives.ShapeContentDescription,
    10023: TSWPArchives.TateChuYokoFieldArchive,
    10024: TSWPArchives.DropCapStyleArchive,
    11000: TSPMessages.PasteboardObject,
    11006: TSPArchiveMessages.PackageMetadata,
    11007: TSPArchiveMessages.PasteboardMetadata,
    11008: TSPMessages.ObjectContainer,
    11009: TSPArchiveMessages.ViewStateMetadata,
    11010: TSPMessages.ObjectCollection,
    11011: TSPArchiveMessages.DocumentMetadata,
    11012: TSPArchiveMessages.SupportMetadata,
    11013: TSPArchiveMessages.ObjectSerializationMetadata,
    11014: TSPArchiveMessages.DataMetadata,
    11015: TSPArchiveMessages.DataMetadataMap,
    11016: TSPMessages.LargeNumberArraySegment,
    11017: TSPMessages.LargeStringArraySegment,
    11018: TSPMessages.LargeLazyObjectArraySegment,
    11019: TSPMessages.LargeNumberArray,
    11020: TSPMessages.LargeStringArray,
    11021: TSPMessages.LargeLazyObjectArray,
    11024: TSPMessages.LargeUUIDArraySegment,
    11025: TSPMessages.LargeUUIDArray,
    11026: TSPMessages.LargeObjectArraySegment,
    11027: TSPMessages.LargeObjectArray,
};
exports.KNArchives$Archives = [
    KNArchives.AnimationAttributesArchive,
    KNArchives.TransitionAttributesArchive,
    KNArchives.TransitionArchive,
    KNArchives.BuildChunkArchive,
    KNArchives.BuildChunkIdentifierArchive,
    KNArchives.BuildAttributeValueArchive,
    KNArchives.BuildAttributeTupleArchive,
    KNArchives.BuildAttributesArchive,
    KNArchives.BuildArchive,
    KNArchives.PlaceholderArchive,
    KNArchives.NoteArchive,
    KNArchives.ClassicStylesheetRecordArchive,
    KNArchives.ClassicThemeRecordArchive,
    KNArchives.SlideArchive,
    KNArchives.SlideArchive_SageTagMapEntry,
    KNArchives.SlideArchive_InstructionalTextMap,
    KNArchives.SlideArchive_InstructionalTextMap_InstructionalTextMapEntry,
    KNArchives.SlideNodeArchive,
    KNArchives.SlideNodeArchive_SlideSpecificHyperlinkMapEntry,
    KNArchives.DesktopUILayoutArchive,
    KNArchives.UIStateArchive,
    KNArchives.IOSRestorableViewStateRootArchive,
    KNArchives.IOSSavedPlaybackStateArchive,
    KNArchives.CanvasSelectionArchive,
    KNArchives.ActionGhostSelectionArchive,
    KNArchives.ThemeCustomTimingCurveArchive,
    KNArchives.ThemeArchive,
    KNArchives.SlideTreeArchive,
    KNArchives.ShowArchive,
    KNArchives.DocumentArchive,
    KNArchives.SlideStylePropertiesArchive,
    KNArchives.SlideStyleArchive,
    KNArchives.PasteboardNativeStorageArchive,
    KNArchives.PasteboardNativeStorageArchive_LiveVideoSourcePair,
    KNArchives.PrototypeForUndoTemplateChangeArchive,
    KNArchives.RecordingArchive,
    KNArchives.RecordingSyncState,
    KNArchives.RecordingCorrectionHistory,
    KNArchives.RecordingEventTrackArchive,
    KNArchives.RecordingEventArchive,
    KNArchives.RecordingNavigationEventArchive,
    KNArchives.RecordingLaserEventArchive,
    KNArchives.RecordingPauseEventArchive,
    KNArchives.RecordingMovieEventArchive,
    KNArchives.RecordingMovieTrackArchive,
    KNArchives.MovieSegmentArchive,
    KNArchives.Soundtrack,
    KNArchives.SlideNumberAttachmentArchive,
    KNArchives.SlideCollectionSelectionArchive,
    KNArchives.SlideCollectionSelectionArchive_OutlineSelection,
    KNArchives.PresenterNotesSelectionArchive,
    KNArchives.MixedIdOperationArgs,
    KNArchives.LiveVideoInfo,
    KNArchives.LiveVideoSource,
    KNArchives.LiveVideoSourceCollaborationCommandUsageState,
    KNArchives.LiveVideoCaptureDeviceDescription,
    KNArchives.LiveVideoSourceCollection,
    KNArchives.LiveVideoSourceUsageEntry,
    KNArchives.MotionBackgroundStylePropertiesArchive,
    KNArchives.MotionBackgroundStyleArchive,
    KNArchives.MotionBackgroundFillArchive,
];
exports.KNCommandArchives$Archives = [
    KNCommandArchives.UIStateCommandGroupArchive,
    KNCommandArchives.CommandSlideInsertDrawablesArchive,
    KNCommandArchives.CommandSlideSetPlaceholdersForTagsArchive,
    KNCommandArchives.CommandTemplateSlideSetPlaceholderForTagArchive,
    KNCommandArchives.CommandSlidePropagateSetPlaceholderForTagArchive,
    KNCommandArchives.CommandSlideRemoveDrawableArchive,
    KNCommandArchives.CommandSlideMoveDrawableZOrderArchive,
    KNCommandArchives.CommandSlideNodeSetPropertyArchive,
    KNCommandArchives.CommandSlideNodeSetViewStatePropertyArchive,
    KNCommandArchives.CommandSlideSetBackgroundFillArchive,
    KNCommandArchives.CommandSlideSetStyleArchive,
    KNCommandArchives.CommandShowMarkOutOfSyncRecordingArchive,
    KNCommandArchives.CommandShowMarkOutOfSyncRecordingIfNeededArchive,
    KNCommandArchives.CommandShowRemoveRecordingArchive,
    KNCommandArchives.CommandShowReplaceRecordingArchive,
    KNCommandArchives.CommandShowSetValueArchive,
    KNCommandArchives.CommandShowSetValueArchive_PropertyValue,
    KNCommandArchives.CommandShowInsertSlideArchive,
    KNCommandArchives.CommandShowMoveSlideArchive,
    KNCommandArchives.CommandShowRemoveSlideArchive,
    KNCommandArchives.CommandShowChangeThemeArchive,
    KNCommandArchives.CommandShowChangeThemeArchive_IdentifierToStyleMap,
    KNCommandArchives.CommandShowChangeSlideSizeArchive,
    KNCommandArchives.CommandSetThemeCustomEffectTimingCurveArchive,
    KNCommandArchives.CommandMoveTemplatesArchive,
    KNCommandArchives.CommandInsertTemplateArchive,
    KNCommandArchives.CommandPrimitiveInsertTemplateArchive,
    KNCommandArchives.CommandRemoveTemplateArchive,
    KNCommandArchives.CommandPrimitiveRemoveTemplateArchive,
    KNCommandArchives.CommandSlideInsertBuildArchive,
    KNCommandArchives.InsertBuildDescriptionArchive,
    KNCommandArchives.UndoObjectArchive,
    KNCommandArchives.CommandSlideRemoveBuildArchive,
    KNCommandArchives.RemoveBuildDescriptionArchive,
    KNCommandArchives.CommandSlideInsertBuildChunkArchive,
    KNCommandArchives.CommandSlideRemoveBuildChunkArchive,
    KNCommandArchives.CommandSlideMoveBuildChunksArchive,
    KNCommandArchives.CommandBuildSetValueArchive,
    KNCommandArchives.CommandBuildUpdateChunkReferentsArchive,
    KNCommandArchives.CommandBuildUpdateChunkCountArchive,
    KNCommandArchives.CommandBuildChunkSetValueArchive,
    KNCommandArchives.CommandTransitionSetValueArchive,
    KNCommandArchives.CommandShowSetSlideNumberVisibilityArchive,
    KNCommandArchives.CommandTemplateSetThumbnailTextArchive,
    KNCommandArchives.CommandTemplateSetBodyStylesArchive,
    KNCommandArchives.CommandChangeTemplateSlideArchive,
    KNCommandArchives.CommandSlidePrimitiveSetTemplateArchive,
    KNCommandArchives.CommandSlideResetTemplateBackgroundObjectsArchive,
    KNCommandArchives.CommandShowSetSoundtrack,
    KNCommandArchives.CommandSoundtrackSetValue,
    KNCommandArchives.CommandSoundtrackSetValue_PropertyValue,
    KNCommandArchives.CommandSlideUpdateTemplateDrawables,
    KNCommandArchives.ChartInfoGeometryCommandArchive,
    KNCommandArchives.CanvasSelectionTransformerArchive,
    KNCommandArchives.ActionGhostSelectionTransformerArchive,
    KNCommandArchives.DocumentSelectionTransformerArchive,
    KNCommandArchives.NoteCanvasSelectionTransformerArchive,
    KNCommandArchives.OutlineCanvasSelectionTransformerArchive,
    KNCommandArchives.SlideCollectionSelectionTransformerHelperArchive,
    KNCommandArchives.SlideCollectionSelectionTransformerArchive,
    KNCommandArchives.OutlineSelectionTransformerArchive,
    KNCommandArchives.CommandLiveVideoInfoApplyPreset,
    KNCommandArchives.CommandLiveVideoInfoSetSource,
    KNCommandArchives.CommandLiveVideoInfoSetValue,
    KNCommandArchives.CommandLiveVideoInfoSetValue_PropertyValue,
    KNCommandArchives.CommandLiveVideoSourceSetValue,
    KNCommandArchives.CommandLiveVideoSourceSetValue_PropertyValue,
    KNCommandArchives.CommandLiveVideoStyleSetValue,
    KNCommandArchives.CommandThemeAddLiveVideoSource,
    KNCommandArchives.CommandThemeRemoveLiveVideoSource,
    KNCommandArchives.CommandMotionBackgroundStyleSetValueArchive,
    KNCommandArchives.CommandMotionBackgroundStyleUpdatePosterFrameDataArchive,
];
exports.KNArchives_sos$Archives = [
    KNArchives_sos.SpecSetTransitionAttributesArchive,
    KNArchives_sos.SpecTransitionAttributesArchive,
    KNArchives_sos.MotionBackgroundStylePropertyChangeSetArchive,
    KNArchives_sos.SlideStylePropertyChangeSetArchive,
];
exports.KNCommandArchives_sos$Archives = [
    KNCommandArchives_sos.InducedVerifyDocumentWithServerCommandArchive,
    KNCommandArchives_sos.InducedVerifyDrawableZOrdersWithServerCommandArchive,
    KNCommandArchives_sos.CommandSlideReapplyTemplateSlideArchive,
];
exports.TSAArchives$Archives = [
    TSAArchives.DocumentArchive,
    TSAArchives.FunctionBrowserStateArchive,
    TSAArchives.ThemePresetsArchive,
    TSAArchives.ShortcutControllerArchive,
    TSAArchives.ShortcutControllerArchive_ShortcutMapEntry,
    TSAArchives.PropagatePresetCommandArchive,
    TSAArchives.ShortcutCommandArchive,
    TSAArchives.AddCustomFormatCommandArchive,
    TSAArchives.UpdateCustomFormatCommandArchive,
    TSAArchives.ReplaceCustomFormatCommandArchive,
    TSAArchives.NeedsMediaCompatibilityUpgradeCommandArchive,
    TSAArchives.ChangeDocumentLocaleCommandArchive,
    TSAArchives.InducedVerifyObjectsWithServerCommandArchive,
    TSAArchives.InducedVerifyTransformHistoryWithServerCommandArchive,
    TSAArchives.StyleUpdatePropertyMapCommandArchive,
    TSAArchives.RemoteDataChangeCommandArchive,
    TSAArchives.GalleryInfoSetValueCommandArchive,
    TSAArchives.GalleryInfoSetValueCommandArchive_PropertyValue,
    TSAArchives.GalleryInfoInsertItemsCommandArchive,
    TSAArchives.GalleryInfoRemoveItemsCommandArchive,
    TSAArchives.GalleryItemSetGeometryCommand,
    TSAArchives.GalleryItem,
    TSAArchives.GalleryInfo,
    TSAArchives.GallerySelectionTransformer,
    TSAArchives.GalleryItemSelection,
    TSAArchives.GalleryItemSelectionTransformer,
    TSAArchives.GalleryItemSelectionTransformerHelper,
    TSAArchives.GalleryItemSetValueCommand,
    TSAArchives.GalleryItemSetValueCommand_PropertyValue,
    TSAArchives.CollaboratorGalleryItemCursor,
    TSAArchives.WebVideoInfo,
    TSAArchives.CaptionPlacementArchive,
    TSAArchives.CaptionInfoArchive,
    TSAArchives.TitlePlacementCommandArchive,
];
exports.TSAArchives_sos$Archives = [
    TSAArchives_sos.InducedVerifyActivityStreamWithServerCommandArchive,
    TSAArchives_sos.DrawableZOrderListArchive,
];
exports.TSACommandArchives_sos$Archives = [
    TSACommandArchives_sos.InducedVerifyDrawableZOrdersWithServerCommandArchive,
    TSACommandArchives_sos.PropagateMasterChangeCommandArchive,
    TSACommandArchives_sos.CommandReapplyMasterArchive,
    TSACommandArchives_sos.VerifyActivityStreamWithServerCommandArchive,
    TSACommandArchives_sos.VerifyDocumentWithServerCommandArchive,
    TSACommandArchives_sos.VerifyDrawableZOrdersWithServerCommandArchive,
    TSACommandArchives_sos.VerifyObjectsWithServerCommandArchive,
    TSACommandArchives_sos.VerifyTransformHistoryWithServerCommandArchive,
];
exports.TSCEArchives$Archives = [
    TSCEArchives.CellReferenceArchive,
    TSCEArchives.InternalCellReferenceArchive,
    TSCEArchives.RelativeCellRefArchive,
    TSCEArchives.RangeReferenceArchive,
    TSCEArchives.InternalRangeReferenceArchive,
    TSCEArchives.IndexSetArchive,
    TSCEArchives.IndexSetArchive_IndexSetEntry,
    TSCEArchives.CellCoordSetArchive,
    TSCEArchives.CellCoordSetArchive_ColumnEntry,
    TSCEArchives.InternalCellRefSetArchive,
    TSCEArchives.InternalCellRefSetArchive_OwnerEntry,
    TSCEArchives.ColumnRowSize,
    TSCEArchives.CellRectArchive,
    TSCEArchives.CellRefSetArchive,
    TSCEArchives.CellRefSetArchive_OwnerEntry,
    TSCEArchives.UidCoordSetArchive,
    TSCEArchives.UidCoordSetArchive_ColumnEntry,
    TSCEArchives.UidCellRefSetArchive,
    TSCEArchives.UidCellRefSetArchive_OwnerEntry,
    TSCEArchives.ReferencesToDirtyArchive,
    TSCEArchives.CoordMapperArchive,
    TSCEArchives.CoordMapperArchive_BaseToViewEntry,
    TSCEArchives.CoordMapperArchive_SummaryToViewEntry,
    TSCEArchives.ReferenceArchive,
    TSCEArchives.EdgeArchive,
    TSCEArchives.EdgesArchive,
    TSCEArchives.ExpandedEdgesArchive,
    TSCEArchives.CellRecordArchive,
    TSCEArchives.CellDependenciesArchive,
    TSCEArchives.CellRecordExpandedArchive,
    TSCEArchives.CellDependenciesExpandedArchive,
    TSCEArchives.CellRecordTileArchive,
    TSCEArchives.CellDependenciesTiledArchive,
    TSCEArchives.VolatileDependenciesArchive,
    TSCEArchives.VolatileDependenciesExpandedArchive,
    TSCEArchives.RTreeInternalNodeContentsArchive,
    TSCEArchives.RTreeLeafNodeContentsArchive,
    TSCEArchives.RTreeNodeArchive,
    TSCEArchives.RTreeArchive,
    TSCEArchives.RangeBackDependencyArchive,
    TSCEArchives.RangeDependenciesArchive,
    TSCEArchives.RangePrecedentsTileArchive,
    TSCEArchives.RangePrecedentsTileArchive_FromToRangeArchive,
    TSCEArchives.RangeDependenciesTiledArchive,
    TSCEArchives.SpanningDependenciesArchive,
    TSCEArchives.SpanningDependenciesArchive_ReferringColumnToLocalCells,
    TSCEArchives.SpanningDependenciesArchive_ReferringColumnToRemoteCells,
    TSCEArchives.SpanningDependenciesExpandedArchive,
    TSCEArchives.SpanningDependenciesExpandedArchive_ExtentRange,
    TSCEArchives.SpanningDependenciesExpandedArchive_ExtentRangeWithTableWithContext,
    TSCEArchives.SpanningDependenciesExpandedArchive_CellCoordRefersToExtents,
    TSCEArchives.WholeOwnerDependenciesArchive,
    TSCEArchives.WholeOwnerDependenciesExpandedArchive,
    TSCEArchives.CellErrorsArchive,
    TSCEArchives.CellErrorsArchive_ErrorForCell,
    TSCEArchives.UuidReferencesArchive,
    TSCEArchives.UuidReferencesArchive_TableRef,
    TSCEArchives.UuidReferencesArchive_UuidRef,
    TSCEArchives.UuidReferencesArchive_TableWithUuidRef,
    TSCEArchives.FormulaOwnerDependenciesArchive,
    TSCEArchives.FormulaOwnerInfoArchive,
    TSCEArchives.SubFormulaOwnerIDArchive,
    TSCEArchives.OwnerIDMapArchive,
    TSCEArchives.OwnerIDMapArchive_OwnerIDMapArchiveEntry,
    TSCEArchives.UidLookupListArchive,
    TSCEArchives.DependencyTrackerArchive,
    TSCEArchives.RemoteDataSpecifierArchive,
    TSCEArchives.RemoteDataValueMapArchive,
    TSCEArchives.RemoteDataValueMapArchive_RemoteDataMapEntry,
    TSCEArchives.StockArchive,
    TSCEArchives.StockArchive_AttributeEntry,
    TSCEArchives.RemoteDataStoreArchive,
    TSCEArchives.NameTrackedReferencePair,
    TSCEArchives.NamesByTrackedReferenceArchive,
    TSCEArchives.NamedReferenceManagerArchive,
    TSCEArchives.UuidSetStoreArchive,
    TSCEArchives.UuidSetStoreArchive_UuidSet,
    TSCEArchives.UuidReferenceMapArchive,
    TSCEArchives.UuidReferenceMapArchive_CellRefsForUuid,
    TSCEArchives.GroupByNodeMapArchive,
    TSCEArchives.GroupByNodeMapArchive_CellRefsForGroupNode,
    TSCEArchives.GroupByNodeMapArchive_GroupNodesForGroupBy,
    TSCEArchives.CalculationEngineArchive,
    TSCEArchives.PreserveColumnRowFlagsArchive,
    TSCEArchives.CategoryReferenceArchive,
    TSCEArchives.CategoryReferenceArchive_CatRefUidList,
    TSCEArchives.ASTNodeArrayArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTLocalCellReferenceNodeArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTCrossTableCellReferenceNodeArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTColumnCoordinateArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTRowCoordinateArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTUidList,
    TSCEArchives.ASTNodeArrayArchive_ASTUidTract,
    TSCEArchives.ASTNodeArrayArchive_ASTStickyBits,
    TSCEArchives.ASTNodeArrayArchive_ASTUidTractList,
    TSCEArchives.ASTNodeArrayArchive_ASTUidCoordinateArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTCategoryLevels,
    TSCEArchives.ASTNodeArrayArchive_ASTCategoryReferenceArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTColonTractArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTColonTractArchive_ASTColonTractRelativeRangeArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTColonTractArchive_ASTColonTractAbsoluteRangeArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTCrossTableReferenceExtraInfoArchive,
    TSCEArchives.ASTNodeArrayArchive_ASTLetNodeWhitespace,
    TSCEArchives.ASTNodeArrayArchive_ASTNodeArchive,
    TSCEArchives.FormulaTranslationFlagsArchive,
    TSCEArchives.FormulaArchive,
    TSCEArchives.FunctorArchive,
    TSCEArchives.FormatStructArchive,
    TSCEArchives.RangeCoordinateArchive,
    TSCEArchives.TableTransposeInfoArchive,
    TSCEArchives.MergeOriginMovedArchive,
    TSCEArchives.OwnerUIDMapperArchive,
    TSCEArchives.RewriteTableUIDInfoArchive,
    TSCEArchives.RewriteTableUIDInfoArchive_TableUIDMapEntryArchive,
    TSCEArchives.GroupByChangeArchive,
    TSCEArchives.GroupByChangeArchive_GroupingColumnChangeArchive,
    TSCEArchives.RewriteGroupNodeUIDInfoArchive,
    TSCEArchives.IndexedUidsArchive,
    TSCEArchives.IndexedUidsArchive_IndexedUid,
    TSCEArchives.RewriteRangeEntryArchive,
    TSCEArchives.ColumnOrRowUuidsInfoArchive,
    TSCEArchives.RegionInfoArchive,
    TSCEArchives.RegionMovedInfoArchive,
    TSCEArchives.MergeSourceArchive,
    TSCEArchives.HauntedOwnerArchive,
    TSCEArchives.CellCoordinateArchive,
    TSCEArchives.CellCoordinateVectorArchive,
    TSCEArchives.ExpandedCellRefObjectPairArchive,
    TSCEArchives.ExpandedCellRefObjectMapArchive,
    TSCEArchives.FormulaAtCoordArchive,
    TSCEArchives.FormulaCoordPairsByOwnerArchive,
    TSCEArchives.FormulasForUndoArchive,
    TSCEArchives.FormulaRewriteCommandArchive,
    TSCEArchives.TrackedReferenceArchive,
    TSCEArchives.ExpandedTrackedReferenceArchive,
    TSCEArchives.TrackedReferenceStoreArchive,
    TSCEArchives.ViewTractRefArchive,
    TSCEArchives.BooleanCellValueArchive,
    TSCEArchives.DateCellValueArchive,
    TSCEArchives.NumberCellValueArchive,
    TSCEArchives.StringCellValueArchive,
    TSCEArchives.ErrorCellValueArchive,
    TSCEArchives.CellValueArchive,
];
exports.TSCH3DArchives$Archives = [
    TSCH3DArchives.Chart3DEnvironmentPackageArchive,
    TSCH3DArchives.Chart3DFillArchive,
    TSCH3DArchives.Chart3DPointLightArchive,
    TSCH3DArchives.Chart3DDirectionalLightArchive,
    TSCH3DArchives.Chart3DSpotLightArchive,
    TSCH3DArchives.Chart3DLightArchive,
    TSCH3DArchives.Chart3DLightingModelArchive,
    TSCH3DArchives.Chart3DLightingPackageArchive,
    TSCH3DArchives.Chart3DTexturesMaterialArchive,
    TSCH3DArchives.Chart3DEmissiveMaterialArchive,
    TSCH3DArchives.Chart3DDiffuseMaterialArchive,
    TSCH3DArchives.Chart3DModulateMaterialArchive,
    TSCH3DArchives.Chart3DSpecularMaterialArchive,
    TSCH3DArchives.Chart3DShininessMaterialArchive,
    TSCH3DArchives.Chart3DEnvironmentMaterialArchive,
    TSCH3DArchives.Chart3DFixedFunctionLightingModelArchive,
    TSCH3DArchives.Chart3DPhongLightingModelArchive,
    TSCH3DArchives.Chart3DPhongMaterialPackageArchive,
    TSCH3DArchives.Chart3DTSPImageDataTextureArchive,
    TSCH3DArchives.Chart3DBaseImageTextureTilingArchive,
    TSCH3DArchives.Chart3DImageTextureTilingArchive,
    TSCH3DArchives.Chart3DVectorArchive,
];
exports.TSCHArchives$Archives = [
    TSCHArchives.ChartDrawableArchive,
    TSCHArchives.ChartArchive,
    TSCHArchives.ChartMultiDataIndexUpgrade,
    TSCHArchives.ChartGarlicMinMaxUpgrade,
    TSCHArchives.ChartGarlicLabelFormatUpgrade,
    TSCHArchives.ChartPasteboardAdditionsArchive,
    TSCHArchives.ChartPreserveAppearanceForPresetArchive,
    TSCHArchives.ChartSupportsProportionalBendedCalloutLinesArchive,
    TSCHArchives.ChartSupportsRoundedCornersArchive,
    TSCHArchives.ChartSupportsSeriesPropertySpacingArchive,
    TSCHArchives.ChartGridArchive,
    TSCHArchives.ChartGridArchive_ChartGridRowColumnIdMap,
    TSCHArchives.ChartGridArchive_ChartGridRowColumnIdMap_Entry,
    TSCHArchives.ChartMediatorArchive,
    TSCHArchives.ChartFillSetArchive,
    TSCHArchives.ChartStylePreset,
    TSCHArchives.ChartPresetsArchive,
    TSCHArchives.PropertyValueStorageContainerArchive,
    TSCHArchives.StylePasteboardDataArchive,
    TSCHArchives.ChartSelectionPathTypeArchive,
    TSCHArchives.ChartAxisIDArchive,
    TSCHArchives.ChartSelectionPathArgumentArchive,
    TSCHArchives.ChartSelectionPathArchive,
    TSCHArchives.ChartSelectionArchive,
    TSCHArchives.ChartCDESelectionArchive,
    TSCHArchives.ChartUIState,
    TSCHArchives.ChartUIStateMultiDataIndexUpgrade,
    TSCHArchives.ChartFormatStructExtensions,
    TSCHArchives.ChartReferenceLineNonStyleItem,
    TSCHArchives.ChartAxisReferenceLineNonStylesArchive,
    TSCHArchives.ChartAxisReferenceLineStylesArchive,
    TSCHArchives.ChartReferenceLinesArchive,
    TSCHArchives.ChartPresetReferenceLineStylesArchive,
    TSCHArchives.ChartAxisReferenceLineSparseNonStylesArchive,
    TSCHArchives.PropertyValueStorageContainerReferenceLinesArchive,
    TSCHArchives.CollaboratorCDECursorSubselectionArchive,
    TSCHArchives.CollaboratorChartTitleCursorSubselectionArchive,
];
exports.TSCHArchives_Common$Archives = [
    TSCHArchives_Common.RectArchive,
    TSCHArchives_Common.ChartsNSNumberDoubleArchive,
    TSCHArchives_Common.ChartsNSArrayOfNSNumberDoubleArchive,
    TSCHArchives_Common.DEPRECATEDChart3DFillArchive,
    TSCHArchives_Common.ChartStyleArchive,
    TSCHArchives_Common.ChartNonStyleArchive,
    TSCHArchives_Common.LegendStyleArchive,
    TSCHArchives_Common.LegendNonStyleArchive,
    TSCHArchives_Common.ChartAxisStyleArchive,
    TSCHArchives_Common.ChartAxisNonStyleArchive,
    TSCHArchives_Common.ChartSeriesStyleArchive,
    TSCHArchives_Common.ChartSeriesNonStyleArchive,
    TSCHArchives_Common.GridValue,
    TSCHArchives_Common.GridRow,
    TSCHArchives_Common.ReferenceLineStyleArchive,
    TSCHArchives_Common.ReferenceLineNonStyleArchive,
];
exports.TSCHArchives_GEN$Archives = [
    TSCHArchives_GEN.ChartStyleArchive,
    TSCHArchives_GEN.ChartNonStyleArchive,
    TSCHArchives_GEN.LegendStyleArchive,
    TSCHArchives_GEN.LegendNonStyleArchive,
    TSCHArchives_GEN.ChartAxisStyleArchive,
    TSCHArchives_GEN.ChartAxisNonStyleArchive,
    TSCHArchives_GEN.ChartSeriesStyleArchive,
    TSCHArchives_GEN.ChartSeriesNonStyleArchive,
    TSCHArchives_GEN.ReferenceLineStyleArchive,
    TSCHArchives_GEN.ReferenceLineNonStyleArchive,
    TSCHArchives_GEN.ChartGenericPropertyMapArchive,
    TSCHArchives_GEN.LegendGenericPropertyMapArchive,
    TSCHArchives_GEN.ChartAxisGenericPropertyMapArchive,
    TSCHArchives_GEN.ChartSeriesGenericPropertyMapArchive,
    TSCHArchives_GEN.ReferenceLineGenericPropertyMapArchive,
];
exports.TSCHCommandArchives$Archives = [
    TSCHCommandArchives.StyleOwnerPath,
    TSCHCommandArchives.CommandSetChartTypeArchive,
    TSCHCommandArchives.StyleSwapUndoTupleArchive,
    TSCHCommandArchives.StyleSwapUndoTuplesArchive,
    TSCHCommandArchives.CommandStyleSwapArchive,
    TSCHCommandArchives.CommandMutatePropertiesArchive,
    TSCHCommandArchives.CommandMutatePropertiesArchive_GenericPropertyDeltaMap,
    TSCHCommandArchives.CommandMutatePropertiesArchive_GenericPropertyDeltaMap_GenericMapUnion,
    TSCHCommandArchives.CommandMutatePropertiesArchive_ParagraphStyleDeltaMap,
    TSCHCommandArchives.CommandMutatePropertiesArchive_ParagraphStyleDeltaMap_ParagraphStylePropertyMapArchive,
    TSCHCommandArchives.CommandScaleAllTextArchive,
    TSCHCommandArchives.CommandSetFontFamilyArchive,
    TSCHCommandArchives.CommandSetFontFamilyArchive_StyleSemanticTag,
    TSCHCommandArchives.CommandSetFontFamilyArchive_SemanticParagraphStyleMutationMap,
    TSCHCommandArchives.CommandSetFontFamilyArchive_SemanticParagraphStyleMutationMap_ParagraphStylePropertyMapArchive,
    TSCHCommandArchives.CommandApplyFillSetArchive,
    TSCHCommandArchives.CommandSetSeriesNameArchive,
    TSCHCommandArchives.CommandSetCategoryNameArchive,
    TSCHCommandArchives.CommandAddGridRowsArchive,
    TSCHCommandArchives.CommandAddGridColumnsArchive,
    TSCHCommandArchives.CommandMoveGridRowsArchive,
    TSCHCommandArchives.CommandMoveGridColumnsArchive,
    TSCHCommandArchives.CommandDeleteGridRowsArchive,
    TSCHCommandArchives.CommandDeleteGridColumnsArchive,
    TSCHCommandArchives.CommandSetGridValueArchive,
    TSCHCommandArchives.CommandSetGridDirectionArchive,
    TSCHCommandArchives.CommandSetSeriesTypeArchive,
    TSCHCommandArchives.CommandSetScatterFormatArchive,
    TSCHCommandArchives.CommandSetMultiDataSetIndexArchive,
    TSCHCommandArchives.CommandSetLegendFrameArchive,
    TSCHCommandArchives.CommandSetPieWedgeExplosion,
    TSCHCommandArchives.CommandChartApplyTheme,
    TSCHCommandArchives.CommandChartApplyPreset,
    TSCHCommandArchives.CommandCleanupGeometryArchive,
    TSCHCommandArchives.ChartCommandArchive,
    TSCHCommandArchives.CommandReplaceGridValuesArchive,
    TSCHCommandArchives.CommandReplaceThemePresetArchive,
    TSCHCommandArchives.CommandReplaceThemePresetArchive_StyleAndIdentifierPair,
    TSCHCommandArchives.CommandInvalidateWPCaches,
    TSCHCommandArchives.CommandReplaceCustomFormatArchive,
    TSCHCommandArchives.CommandReplaceCustomFormatArchive_FormatObjectMutation,
    TSCHCommandArchives.CommandAddReferenceLineArchive,
    TSCHCommandArchives.CommandDeleteReferenceLineArchive,
    TSCHCommandArchives.CommandPasteStyleArchive,
    TSCHCommandArchives.CommandReplaceImageDataArchive,
    TSCHCommandArchives.CommandInducedReplaceChartGrid,
    TSCHCommandArchives.CommandInduced3DChartGeometry,
    TSCHCommandArchives.ChartTextSelectionTransformerArchive,
    TSCHCommandArchives.ChartDrawableSelectionTransformerArchive,
    TSCHCommandArchives.ChartSubselectionTransformerHelperArchive,
    TSCHCommandArchives.ChartRefLineSubselectionTransformerHelperArchive,
    TSCHCommandArchives.ChartSubselectionIdentityTransformerHelperArchive,
    TSCHCommandArchives.ChartSubselectionTransformerArchive,
    TSCHCommandArchives.CDESelectionTransformerArchive,
];
exports.TSCHArchives_sos$Archives = [
    TSCHArchives_sos.SpecSetChart3dLightingPackageArchive,
    TSCHArchives_sos.SpecSetChart3dVectorArchive,
    TSCHArchives_sos.SpecSetChartErrorBarCustomDataArchive,
    TSCHArchives_sos.SpecSetChartsOptionalDoubleArchiveArchive,
    TSCHArchives_sos.SpecSetFormatObjectArchive,
    TSCHArchives_sos.SpecSetNumberFormatArchive,
    TSCHArchives_sos.SpecChart3dLightingPackageArchive,
    TSCHArchives_sos.SpecChart3dVectorArchive,
    TSCHArchives_sos.SpecChartErrorBarCustomDataArchive,
    TSCHArchives_sos.SpecChartsOptionalDoubleArchiveArchive,
    TSCHArchives_sos.SpecFormatObjectArchive,
    TSCHArchives_sos.SpecNumberFormatArchive,
    TSCHArchives_sos.ChartAxisNonStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartAxisStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartLegendNonStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartLegendStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartNonStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartReferenceLineNonStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartReferenceLineStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartSeriesNonStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartSeriesStylePropertyChangeSetArchive,
    TSCHArchives_sos.ChartStylePropertyChangeSetArchive,
];
exports.TSCHPreUFFArchives$Archives = [
    TSCHPreUFFArchives.ChartInfoArchive,
    TSCHPreUFFArchives.LegendModelArchive,
    TSCHPreUFFArchives.ChartModelArchive,
    TSCHPreUFFArchives.ChartGridArchive,
    TSCHPreUFFArchives.ChartGridArchive_ValueRow,
    TSCHPreUFFArchives.ChartsNumberFormatArchive,
    TSCHPreUFFArchives.ChartStyleArchive,
    TSCHPreUFFArchives.ChartNonStyleArchive,
    TSCHPreUFFArchives.LegendStyleArchive,
    TSCHPreUFFArchives.LegendNonStyleArchive,
    TSCHPreUFFArchives.ChartAxisStyleArchive,
    TSCHPreUFFArchives.ChartAxisNonStyleArchive,
    TSCHPreUFFArchives.ChartSeriesStyleArchive,
    TSCHPreUFFArchives.ChartSeriesNonStyleArchive,
];
exports.TSDArchives$Archives = [
    TSDArchives.EdgeInsetsArchive,
    TSDArchives.GeometryArchive,
    TSDArchives.PointPathSourceArchive,
    TSDArchives.ScalarPathSourceArchive,
    TSDArchives.BezierPathSourceArchive,
    TSDArchives.CalloutPathSourceArchive,
    TSDArchives.ConnectionLinePathSourceArchive,
    TSDArchives.EditableBezierPathSourceArchive,
    TSDArchives.EditableBezierPathSourceArchive_Node,
    TSDArchives.EditableBezierPathSourceArchive_Subpath,
    TSDArchives.PathSourceArchive,
    TSDArchives.AngleGradientArchive,
    TSDArchives.TransformGradientArchive,
    TSDArchives.GradientArchive,
    TSDArchives.GradientArchive_GradientStop,
    TSDArchives.ImageFillArchive,
    TSDArchives.FillArchive,
    TSDArchives.StrokePatternArchive,
    TSDArchives.StrokeArchive,
    TSDArchives.SmartStrokeArchive,
    TSDArchives.FrameArchive,
    TSDArchives.PatternedStrokeArchive,
    TSDArchives.LineEndArchive,
    TSDArchives.ShadowArchive,
    TSDArchives.DropShadowArchive,
    TSDArchives.ContactShadowArchive,
    TSDArchives.CurvedShadowArchive,
    TSDArchives.ReflectionArchive,
    TSDArchives.ImageAdjustmentsArchive,
    TSDArchives.ShapeStylePropertiesArchive,
    TSDArchives.ShapeStyleArchive,
    TSDArchives.MediaStylePropertiesArchive,
    TSDArchives.MediaStyleArchive,
    TSDArchives.ThemePresetsArchive,
    TSDArchives.ThemeReplaceFillPresetCommandArchive,
    TSDArchives.DrawableArchive,
    TSDArchives.ContainerArchive,
    TSDArchives.GroupArchive,
    TSDArchives.FreehandDrawingAnimationArchive,
    TSDArchives.FreehandDrawingArchive,
    TSDArchives.ShapeArchive,
    TSDArchives.ConnectionLineArchive,
    TSDArchives.ImageArchive,
    TSDArchives.MaskArchive,
    TSDArchives.ImageDataAttributes,
    TSDArchives.MovieArchive,
    TSDArchives.ExteriorTextWrapArchive,
    TSDArchives.DrawableContentDescription,
    TSDArchives.FreehandDrawingContentDescription,
    TSDArchives.FreehandDrawingToolkitUIState,
    TSDArchives.StandinCaptionArchive,
    TSDArchives.GuideArchive,
    TSDArchives.UserDefinedGuideArchive,
    TSDArchives.GuideStorageArchive,
    TSDArchives.CanvasSelectionArchive,
    TSDArchives.DrawableSelectionArchive,
    TSDArchives.GroupSelectionArchive,
    TSDArchives.PathSelectionArchive,
    TSDArchives.InfoHyperlinkSelectionArchive,
    TSDArchives.CommentStorageArchive,
    TSDArchives.ReplaceAnnotationAuthorCommandArchive,
    TSDArchives.PencilAnnotationArchive,
    TSDArchives.PencilAnnotationSelectionArchive,
    TSDArchives.PencilAnnotationStorageArchive,
    TSDArchives.SpecColorFillSetColorArchive,
    TSDArchives.SpecFrameSetAssetScaleArchive,
    TSDArchives.SpecGradientFillSetAngleArchive,
    TSDArchives.SpecImageFillSetTechniqueArchive,
    TSDArchives.SpecReflectionSetOpacityArchive,
    TSDArchives.SpecShadowSetAngleArchive,
    TSDArchives.SpecShadowSetColorArchive,
    TSDArchives.SpecShadowSetOffsetArchive,
    TSDArchives.SpecShadowSetOpacityArchive,
    TSDArchives.SpecShadowSetRadiusArchive,
    TSDArchives.SpecStrokeSetColorArchive,
    TSDArchives.SpecStrokeSetPatternArchive,
    TSDArchives.SpecStrokeSetWidthArchive,
    TSDArchives.Attribution,
    TSDArchives.MovieFingerprint,
    TSDArchives.MovieFingerprintTrack,
];
exports.TSDCommandArchives$Archives = [
    TSDCommandArchives.UndoObjectArchive,
    TSDCommandArchives.GroupDrawablesCommandArchive,
    TSDCommandArchives.UngroupGroupCommandArchive,
    TSDCommandArchives.ContainerRemoveChildrenCommandArchive,
    TSDCommandArchives.ContainerRemoveDrawablesCommandArchive,
    TSDCommandArchives.ContainerInsertChildrenCommandArchive,
    TSDCommandArchives.ContainerInsertDrawablesCommandArchive,
    TSDCommandArchives.ContainerReorderChildrenCommandArchive,
    TSDCommandArchives.GroupUngroupInformativeCommandArchive,
    TSDCommandArchives.InfoCommandArchive,
    TSDCommandArchives.ConnectionLineConnectCommandArchive,
    TSDCommandArchives.InfoGeometryCommandArchive,
    TSDCommandArchives.DrawablePathSourceCommandArchive,
    TSDCommandArchives.InstantAlphaCommandArchive,
    TSDCommandArchives.DrawableApplyThemeCommandArchive,
    TSDCommandArchives.AbstractStyleCommandArchive,
    TSDCommandArchives.StyledInfoSetStyleCommandArchive,
    TSDCommandArchives.BaseApplyPresetCommandArchive,
    TSDCommandArchives.ShapeApplyPresetCommandArchive,
    TSDCommandArchives.MediaApplyPresetCommandArchive,
    TSDCommandArchives.ShapeStyleSetValueCommandArchive,
    TSDCommandArchives.BaseStyleSetValueCommandArchive,
    TSDCommandArchives.MovieSetValueCommandArchive,
    TSDCommandArchives.MovieSetValueCommandArchive_PropertyValue,
    TSDCommandArchives.MediaStyleSetValueCommandArchive,
    TSDCommandArchives.ImageMediaCommandArchive,
    TSDCommandArchives.MediaOriginalSizeCommandArchive,
    TSDCommandArchives.MediaInfoGeometryCommandArchive,
    TSDCommandArchives.ImageNaturalSizeCommandArchive,
    TSDCommandArchives.ImageMaskCommandArchive,
    TSDCommandArchives.ImageAdjustmentsCommandArchive,
    TSDCommandArchives.MediaFlagsCommandArchive,
    TSDCommandArchives.DrawablesCommandGroupArchive,
    TSDCommandArchives.ExteriorTextWrapCommandArchive,
    TSDCommandArchives.DrawableHyperlinkCommandArchive,
    TSDCommandArchives.CommentInvalidatingCommandSelectionBehaviorArchive,
    TSDCommandArchives.ImageReplaceCommandArchive,
    TSDCommandArchives.DrawableLockCommandArchive,
    TSDCommandArchives.DrawableInfoCommentCommandArchive,
    TSDCommandArchives.DrawablePencilAnnotationCommandArchive,
    TSDCommandArchives.AbstractGuideCommandArchive,
    TSDCommandArchives.GuideCommandArchive,
    TSDCommandArchives.DrawableAspectRatioLockedCommandArchive,
    TSDCommandArchives.DrawableAccessibilityDescriptionCommandArchive,
    TSDCommandArchives.PasteStyleCommandArchive,
    TSDCommandArchives.ImageInfoAbstractGeometryCommandArchive,
    TSDCommandArchives.ImageInfoGeometryCommandArchive,
    TSDCommandArchives.ImageInfoMaskGeometryCommandArchive,
    TSDCommandArchives.InfoCollectionSelectionTransformerHelperArchive,
    TSDCommandArchives.DrawableSelectionTransformerArchive,
    TSDCommandArchives.InfoHyperlinkSelectionTransformerArchive,
    TSDCommandArchives.CanvasSelectionTransformerArchive,
    TSDCommandArchives.PathSelectionTransformerArchive,
    TSDCommandArchives.ShapeSelectionTransformerArchive,
    TSDCommandArchives.GroupSelectionTransformerArchive,
    TSDCommandArchives.PencilAnnotationSelectionTransformerArchive,
    TSDCommandArchives.FreehandDrawingOpacityCommandArchive,
    TSDCommandArchives.FreehandDrawingAnimationCommandArchive,
    TSDCommandArchives.InsertCaptionOrTitleCommandArchive,
    TSDCommandArchives.RemoveCaptionOrTitleCommandArchive,
    TSDCommandArchives.SetCaptionOrTitleVisibilityCommandArchive,
];
exports.TSDArchives_sos$Archives = [
    TSDArchives_sos.SpecSetFillArchive,
    TSDArchives_sos.SpecSetLineEndArchive,
    TSDArchives_sos.SpecSetReflectionArchive,
    TSDArchives_sos.SpecSetShadowArchive,
    TSDArchives_sos.SpecSetStrokeArchive,
    TSDArchives_sos.SpecFillArchive,
    TSDArchives_sos.SpecLineEndArchive,
    TSDArchives_sos.SpecReflectionArchive,
    TSDArchives_sos.SpecShadowArchive,
    TSDArchives_sos.SpecStrokeArchive,
    TSDArchives_sos.BaseShapeStylePropertyChangeSetArchive,
    TSDArchives_sos.MediaStylePropertyChangeSetArchive,
];
exports.TSKArchives$Archives = [
    TSKArchives.TreeNode,
    TSKArchives.LocalCommandHistoryItem,
    TSKArchives.LocalCommandHistoryArray,
    TSKArchives.LocalCommandHistoryArraySegment,
    TSKArchives.LocalCommandHistory,
    TSKArchives.CollaborationCommandHistoryArray,
    TSKArchives.CollaborationCommandHistoryArraySegment,
    TSKArchives.CollaborationCommandHistory,
    TSKArchives.CollaborationCommandHistory_ItemList,
    TSKArchives.CollaborationCommandHistoryItem,
    TSKArchives.CollaborationCommandHistoryCoalescingGroup,
    TSKArchives.CollaborationCommandHistoryCoalescingGroupNode,
    TSKArchives.CollaborationCommandHistoryOriginatingCommandAcknowledgementObserver,
    TSKArchives.DocumentArchive,
    TSKArchives.FormattingSymbolsArchive,
    TSKArchives.FormattingSymbolsArchive_CurrencySymbol,
    TSKArchives.DocumentSupportCollaborationState,
    TSKArchives.DocumentSupportArchive,
    TSKArchives.ViewStateArchive,
    TSKArchives.CommandArchive,
    TSKArchives.CommandGroupArchive,
    TSKArchives.InducedCommandCollectionArchive,
    TSKArchives.PropagatedCommandCollectionArchive,
    TSKArchives.FinalCommandPairArchive,
    TSKArchives.CommandContainerArchive,
    TSKArchives.ProgressiveCommandGroupArchive,
    TSKArchives.FormatStructArchive,
    TSKArchives.CustomFormatArchive,
    TSKArchives.CustomFormatArchive_Condition,
    TSKArchives.CustomFormatListArchive,
    TSKArchives.AnnotationAuthorArchive,
    TSKArchives.DeprecatedChangeAuthorArchive,
    TSKArchives.AnnotationAuthorStorageArchive,
    TSKArchives.SetAnnotationAuthorColorCommandArchive,
    TSKArchives.SetActivityAuthorShareParticipantIDCommandArchive,
    TSKArchives.CommandBehaviorSelectionPathStorageArchive,
    TSKArchives.CommandBehaviorArchive,
    TSKArchives.CommandSelectionBehaviorArchive,
    TSKArchives.SelectionPathTransformerArchive,
    TSKArchives.SelectionPathArchive,
    TSKArchives.DocumentSelectionArchive,
    TSKArchives.IdOperationArgs,
    TSKArchives.AddIdOperationArgs,
    TSKArchives.RemoveIdOperationArgs,
    TSKArchives.RearrangeIdOperationArgs,
    TSKArchives.IdPlacementOperationArgs,
    TSKArchives.NullCommandArchive,
    TSKArchives.GroupCommitCommandArchive,
    TSKArchives.UpgradeDocPostProcessingCommandArchive,
    TSKArchives.InducedCommandCollectionCommitCommandArchive,
    TSKArchives.ActivityCommitCommandArchive,
    TSKArchives.ExecuteTestBetweenRollbackAndReapplyCommandArchive,
    TSKArchives.ChangeDocumentPackageTypeCommandArchive,
    TSKArchives.CreateLocalStorageSnapshotCommandArchive,
    TSKArchives.BlockDiffsAtCurrentRevisionCommand,
    TSKArchives.RangeAddress,
    TSKArchives.Operation,
    TSKArchives.OperationTransformer,
    TSKArchives.TransformerEntry,
    TSKArchives.OutgoingCommandQueueItem,
    TSKArchives.OutgoingCommandQueueItemUUIDToDataMapEntry,
    TSKArchives.CollaborationAppliedCommandDocumentRevisionMapping,
    TSKArchives.CollaborationDocumentSessionState,
    TSKArchives.CollaborationDocumentSessionState_AcknowledgementObserverEntry,
    TSKArchives.NativeContentDescription,
    TSKArchives.StructuredTextImportSettings,
    TSKArchives.OperationStorageCommandOperationsEntry,
    TSKArchives.OperationStorageEntry,
    TSKArchives.OperationStorageEntryArray,
    TSKArchives.OperationStorageEntryArraySegment,
    TSKArchives.OperationStorage,
    TSKArchives.OutgoingCommandQueue,
    TSKArchives.OutgoingCommandQueueSegment,
    TSKArchives.DataReferenceRecord,
    TSKArchives.DataReferenceRecord_ContainerUUIDToReferencedDataPair,
    TSKArchives.CommandAssetChunkArchive,
    TSKArchives.AssetUploadStatusCommandArchive,
    TSKArchives.AssetUploadStatusCommandArchive_AssetUploadStatusInfo,
    TSKArchives.AssetUnmaterializedOnServerCommandArchive,
    TSKArchives.PencilAnnotationUIState,
    TSKArchives.CollaboratorCursorArchive,
    TSKArchives.ActivityStreamArchive,
    TSKArchives.ActivityStreamActivityArray,
    TSKArchives.ActivityStreamActivityArraySegment,
    TSKArchives.ActivityArchive,
    TSKArchives.ActivityAuthorArchive,
    TSKArchives.CommandActivityBehaviorArchive,
    TSKArchives.ActivityCursorCollectionArchive,
    TSKArchives.ActivityCursorCollectionPersistenceWrapperArchive,
    TSKArchives.ActivityNavigationInfoArchive,
    TSKArchives.CommentActivityNavigationInfoArchive,
    TSKArchives.ActivityAuthorCacheArchive,
    TSKArchives.ActivityAuthorCacheArchive_ShareParticipantIDCache,
    TSKArchives.ActivityAuthorCacheArchive_PublicIDCache,
    TSKArchives.ActivityAuthorCacheArchive_IndexCache,
    TSKArchives.ActivityAuthorCacheArchive_FirstJoinCache,
    TSKArchives.ActivityOnlyCommandArchive,
    TSKArchives.ActivityNotificationItemArchive,
    TSKArchives.ActivityNotificationParticipantCacheArchive,
    TSKArchives.ActivityNotificationParticipantCacheArchive_UniqueIdentifierAndAttempts,
    TSKArchives.ActivityNotificationQueueArchive,
    TSKArchives.ActivityStreamTransformationStateArchive,
    TSKArchives.ActivityStreamActivityCounterArchive,
    TSKArchives.ActivityStreamActivityCounterArchive_ActionTypeCounter,
    TSKArchives.ActivityStreamActivityCounterArchive_CursorTypeCounter,
    TSKArchives.ActivityStreamRemovedAuthorAuditorPendingStateArchive,
    TSKArchives.ActivityStreamRemovedAuthorAuditorPendingStateArchive_DateToAuditAndType,
];
exports.TSKArchives_sos$Archives = [
    TSKArchives_sos.FixCorruptedDataCommandArchive,
    TSKArchives_sos.RemoveAuthorIdentifiersCommandArchive,
    TSKArchives_sos.ResetActivityStreamCommandArchive,
];
exports.TSPArchiveMessages$Archives = [
    TSPArchiveMessages.ArchiveInfo,
    TSPArchiveMessages.MessageInfo,
    TSPArchiveMessages.FieldInfo,
    TSPArchiveMessages.FieldPath,
    TSPArchiveMessages.ComponentInfo,
    TSPArchiveMessages.ComponentExternalReference,
    TSPArchiveMessages.ComponentDataReference,
    TSPArchiveMessages.ComponentDataReference_ObjectReference,
    TSPArchiveMessages.ObjectUUIDMapEntry,
    TSPArchiveMessages.FeatureInfo,
    TSPArchiveMessages.PackageMetadata,
    TSPArchiveMessages.DocumentRevision,
    TSPArchiveMessages.PasteboardMetadata,
    TSPArchiveMessages.DataInfo,
    TSPArchiveMessages.DataMetadataMap,
    TSPArchiveMessages.DataMetadataMap_DataMetadataMapEntry,
    TSPArchiveMessages.DataMetadata,
    TSPArchiveMessages.EncryptionInfo,
    TSPArchiveMessages.EncryptionBlockInfo,
    TSPArchiveMessages.ViewStateMetadata,
    TSPArchiveMessages.ObjectSerializationMetadata,
    TSPArchiveMessages.ObjectSerializationDirectory,
    TSPArchiveMessages.ObjectSerializationDirectory_Entry,
    TSPArchiveMessages.DataPropertiesEntryV1,
    TSPArchiveMessages.DataPropertiesV1,
    TSPArchiveMessages.DocumentMetadata,
    TSPArchiveMessages.SupportMetadata,
    TSPArchiveMessages.SupportMetadata_DataCollaborationProperties,
];
exports.TSPDatabaseMessages$Archives = [
    TSPDatabaseMessages.DatabaseData,
    TSPDatabaseMessages.DatabaseDataArchive,
    TSPDatabaseMessages.DatabaseImageDataArchive,
];
exports.TSPMessages$Archives = [
    TSPMessages.Reference,
    TSPMessages.DataReference,
    TSPMessages.SparseReferenceArray,
    TSPMessages.SparseReferenceArray_Entry,
    TSPMessages.Point,
    TSPMessages.Size,
    TSPMessages.Range,
    TSPMessages.Date,
    TSPMessages.IndexSet,
    TSPMessages.Color,
    TSPMessages.Path,
    TSPMessages.Path_Element,
    TSPMessages.ReferenceDictionary,
    TSPMessages.ReferenceDictionary_Entry,
    TSPMessages.UUID,
    TSPMessages.CFUUIDArchive,
    TSPMessages.UUIDSetArchive,
    TSPMessages.UUIDMapArchive,
    TSPMessages.UUIDMultiMapArchive,
    TSPMessages.UUIDCoordArchive,
    TSPMessages.UUIDRectArchive,
    TSPMessages.SparseUUIDArray,
    TSPMessages.SparseUUIDArray_Entry,
    TSPMessages.UUIDPath,
    TSPMessages.SparseUUIDPathArray,
    TSPMessages.SparseUUIDPathArray_Entry,
    TSPMessages.PasteboardObject,
    TSPMessages.ObjectCollection,
    TSPMessages.ObjectContainer,
    TSPMessages.DataAttributes,
    TSPMessages.LargeArraySegment,
    TSPMessages.LargeNumberArraySegment,
    TSPMessages.LargeStringArraySegment,
    TSPMessages.LargeStringArraySegment_OptionalElement,
    TSPMessages.LargeUUIDArraySegment,
    TSPMessages.LargeLazyObjectArraySegment,
    TSPMessages.LargeObjectArraySegment,
    TSPMessages.LargeArray,
    TSPMessages.LargeNumberArray,
    TSPMessages.LargeStringArray,
    TSPMessages.LargeLazyObjectArray,
    TSPMessages.LargeObjectArray,
    TSPMessages.LargeUUIDArray,
];
exports.TSSArchives$Archives = [
    TSSArchives.StyleArchive,
    TSSArchives.StylesheetArchive,
    TSSArchives.StylesheetArchive_IdentifiedStyleEntry,
    TSSArchives.StylesheetArchive_StyleChildrenEntry,
    TSSArchives.StylesheetArchive_VersionedStyles,
    TSSArchives.ThemeArchive,
    TSSArchives.ApplyThemeCommandArchive,
    TSSArchives.ApplyThemeChildCommandArchive,
    TSSArchives.StyleUpdatePropertyMapCommandArchive,
    TSSArchives.ThemeReplacePresetCommandArchive,
    TSSArchives.ThemeReplaceColorPresetCommandArchive,
    TSSArchives.ThemeAddStylePresetCommandArchive,
    TSSArchives.ThemeRemoveStylePresetCommandArchive,
    TSSArchives.ThemeMovePresetCommandArchive,
    TSSArchives.ThemeReplaceStylePresetAndDisconnectStylesCommandArchive,
    TSSArchives.CommandPropertyEntryArchive,
    TSSArchives.CommandPropertyMapArchive,
];
exports.TSSArchives_sos$Archives = [
    TSSArchives_sos.SpecSetBoolArchive,
    TSSArchives_sos.SpecSetColorArchive,
    TSSArchives_sos.SpecSetDoubleArchive,
    TSSArchives_sos.SpecSetIntegerArchive,
    TSSArchives_sos.SpecSetStringArchive,
    TSSArchives_sos.SpecBoolArchive,
    TSSArchives_sos.SpecColorArchive,
    TSSArchives_sos.SpecDoubleArchive,
    TSSArchives_sos.SpecIntegerArchive,
    TSSArchives_sos.SpecStringArchive,
];
exports.TSTArchives$Archives = [
    TSTArchives.CellID,
    TSTArchives.CellUIDLookupListArchive,
    TSTArchives.CellUIDListArchive,
    TSTArchives.TableSize,
    TSTArchives.ExpandedTableSize,
    TSTArchives.CellRange,
    TSTArchives.ExpandedCellRange,
    TSTArchives.TableSelection,
    TSTArchives.TileRowInfo,
    TSTArchives.Tile,
    TSTArchives.TileStorage,
    TSTArchives.TileStorage_Tile,
    TSTArchives.PopUpMenuModel,
    TSTArchives.PopUpMenuModel_CellValue,
    TSTArchives.ImportWarningSetArchive,
    TSTArchives.ImportWarningSetArchive_FormulaImportWarning,
    TSTArchives.CellRefImportWarningSetPairArchive,
    TSTArchives.ImportWarningSetByCellRefArchive,
    TSTArchives.TableDataList,
    TSTArchives.TableDataList_ListEntry,
    TSTArchives.TableDataListSegment,
    TSTArchives.TableRBTree,
    TSTArchives.TableRBTree_Node,
    TSTArchives.HeaderStorageBucket,
    TSTArchives.HeaderStorageBucket_Header,
    TSTArchives.HeaderStorage,
    TSTArchives.DataStore,
    TSTArchives.TableInfoArchive,
    TSTArchives.CategoryInfoArchive,
    TSTArchives.WPTableInfoArchive,
    TSTArchives.TableStyleNetworkArchive,
    TSTArchives.TableGroupSortOrderUIDArchive,
    TSTArchives.TableGroupSortOrderUIDArchive_GroupSortRuleUIDArchive,
    TSTArchives.TableSortOrderArchive,
    TSTArchives.TableSortOrderArchive_SortRuleArchive,
    TSTArchives.TableSortOrderUIDArchive,
    TSTArchives.TableSortOrderUIDArchive_SortRuleArchive,
    TSTArchives.SortRuleReferenceTrackerArchive,
    TSTArchives.CellRegion,
    TSTArchives.CellUIDRegionArchive,
    TSTArchives.StructuredTextImportRecord,
    TSTArchives.TableModelArchive,
    TSTArchives.SummaryModelArchive,
    TSTArchives.SummaryModelGroupByChangeStateArchive,
    TSTArchives.ColumnRowUIDMapArchive,
    TSTArchives.StrokeLayerArchive,
    TSTArchives.StrokeLayerArchive_StrokeRunArchive,
    TSTArchives.StrokeSidecarArchive,
    TSTArchives.DurationWrapperArchive,
    TSTArchives.Cell,
    TSTArchives.MergeRegionMapArchive,
    TSTArchives.CellMapArchive,
    TSTArchives.CellListArchive,
    TSTArchives.CellListArchive_OptionalCell,
    TSTArchives.ConcurrentCellMapArchive,
    TSTArchives.ConcurrentCellListArchive,
    TSTArchives.ConcurrentCellListArchive_OptionalCell,
    TSTArchives.CellFormatAndValueArchive,
    TSTArchives.CellSpecArchive,
    TSTArchives.CommentStorageWrapperArchive,
    TSTArchives.CellDiffArchive,
    TSTArchives.CellDiffArray,
    TSTArchives.CellDiffArraySegment,
    TSTArchives.CellDiffMapArchive,
    TSTArchives.HierarchicalCellDiffMapArchive,
    TSTArchives.HierarchicalCellDiffMapArchive_BoxedRow,
    TSTArchives.DoubleStyleMapArchive,
    TSTArchives.DoubleStyleMapArchive_DoubleStyleMapEntryArchive,
    TSTArchives.StyleTableMapArchive,
    TSTArchives.StyleTableMapArchive_StyleTableMapEntryArchive,
    TSTArchives.SelectionArchive,
    TSTArchives.DeathhawkRdar39989167CellSelectionArchive,
    TSTArchives.StrokeSelectionArchive,
    TSTArchives.ControlCellSelectionArchive,
    TSTArchives.StockCellSelectionArchive,
    TSTArchives.TableNameSelectionArchive,
    TSTArchives.AutofillSelectionArchive,
    TSTArchives.FilterRulePrePivotArchive,
    TSTArchives.FilterRuleArchive,
    TSTArchives.TableStyleArchive,
    TSTArchives.CellStyleArchive,
    TSTArchives.FormulaPredArgDataArchive,
    TSTArchives.FormulaPredArgArchive,
    TSTArchives.FormulaPredicatePrePivotArchive,
    TSTArchives.FormulaPredicateArchive,
    TSTArchives.ConditionalStyleSetArchive,
    TSTArchives.ConditionalStyleSetArchive_ConditionalStyleRulePrePivot,
    TSTArchives.ConditionalStyleSetArchive_ConditionalStyleRule,
    TSTArchives.ConditionalStyleSetArchive_ConditionalStyleRules,
    TSTArchives.FilterSetArchive,
    TSTArchives.UniqueIndexArchive,
    TSTArchives.UniqueIndexArchive_UniqueIndexEntryArchive,
    TSTArchives.HiddenStateExtentArchive,
    TSTArchives.HiddenStateExtentArchive_RowOrColumnState,
    TSTArchives.HiddenStatesArchive,
    TSTArchives.HiddenStatesOwnerArchive,
    TSTArchives.ExpandCollapseStateArchive,
    TSTArchives.TokenAttachmentArchive,
    TSTArchives.FormulaArchive,
    TSTArchives.ExpressionNodeArchive,
    TSTArchives.BooleanNodeArchive,
    TSTArchives.NumberNodeArchive,
    TSTArchives.StringNodeArchive,
    TSTArchives.IdentifierNodeArchive,
    TSTArchives.ArrayNodeArchive,
    TSTArchives.ListNodeArchive,
    TSTArchives.OperatorNodeArchive,
    TSTArchives.PostfixOperatorNodeArchive,
    TSTArchives.PrefixOperatorNodeArchive,
    TSTArchives.FunctionNodeArchive,
    TSTArchives.FunctionEndNodeArchive,
    TSTArchives.DateNodeArchive,
    TSTArchives.ReferenceNodeArchive,
    TSTArchives.DurationNodeArchive,
    TSTArchives.ArgumentPlaceholderNodeArchive,
    TSTArchives.EmptyExpressionNodeArchive,
    TSTArchives.LetNodeArchive,
    TSTArchives.InNodeArchive,
    TSTArchives.VariableNodeArchive,
    TSTArchives.LayoutHintArchive,
    TSTArchives.CompletionTokenAttachmentArchive,
    TSTArchives.HiddenStateFormulaOwnerArchive,
    TSTArchives.FormulaStoreArchive,
    TSTArchives.FormulaStoreArchive_FormulaStorePair,
    TSTArchives.MergeOperationArchive,
    TSTArchives.MergeOwnerArchive,
    TSTArchives.PencilAnnotationArchive,
    TSTArchives.PencilAnnotationOwnerArchive,
    TSTArchives.AccumulatorArchive,
    TSTArchives.GroupColumnArchive,
    TSTArchives.GroupColumnListArchive,
    TSTArchives.ColumnAggregateArchive,
    TSTArchives.ColumnAggregateListArchive,
    TSTArchives.GroupByArchive,
    TSTArchives.GroupByArchive_AggNodeArchive,
    TSTArchives.GroupByArchive_AggregatorArchive,
    TSTArchives.GroupByArchive_GroupNodeArchive,
    TSTArchives.GroupByArchive_GroupNodeArchive_FormatManagerArchive,
    TSTArchives.GroupByArchive_GroupNodeArchive_FormatManagerArchive_RowSetArchive,
    TSTArchives.CategoryOwnerArchive,
    TSTArchives.CategoryOwnerRefArchive,
    TSTArchives.PivotGroupingColumnOptionsMapArchive,
    TSTArchives.PivotOwnerArchive,
    TSTArchives.RichTextPayloadArchive,
    TSTArchives.FormulaEqualsTokenAttachmentArchive,
    TSTArchives.CellFillStandIn,
    TSTArchives.FormulaSelectionArchive,
    TSTArchives.CellBorderArchive,
    TSTArchives.DefaultCellStylesContainerArchive,
    TSTArchives.MultiTableRemapperArchive,
    TSTArchives.ChangePropagationMapWrapper,
    TSTArchives.SummaryCellVendorArchive,
    TSTArchives.SummaryCellVendorArchive_SummaryCellEntry,
    TSTArchives.CategoryOrderArchive,
    TSTArchives.PivotOrderArchive,
    TSTArchives.HeaderNameMgrTileArchive,
    TSTArchives.HeaderNameMgrTileArchive_NameFragmentArchive,
    TSTArchives.HeaderNameMgrArchive,
    TSTArchives.HeaderNameMgrArchive_PerTableArchive,
    TSTArchives.WidthHeightCacheFittingEntry,
    TSTArchives.WidthHeightCache,
    TSTArchives.LayoutEngineArchive,
    TSTArchives.CollaboratorTableCursorSubselectionArchive,
];
exports.TSTCommandArchives$Archives = [
    TSTCommandArchives.TableCommandArchive,
    TSTCommandArchives.CommandApplyCellContentsArchive,
    TSTCommandArchives.CommandApplyCellDiffMapArchive,
    TSTCommandArchives.CommandApplyCellMapArchive,
    TSTCommandArchives.CommandApplyConcurrentCellMapArchive,
    TSTCommandArchives.CommandChangeFreezeHeaderStateArchive,
    TSTCommandArchives.CommandDeleteCellsArchive,
    TSTCommandArchives.CommandDeleteCellContentsArchive,
    TSTCommandArchives.CommandMutateCellFormatArchive,
    TSTCommandArchives.CommandSetAutomaticDurationUnitsArchive,
    TSTCommandArchives.CommandSetDurationStyleArchive,
    TSTCommandArchives.CommandSetDurationUnitSmallestLargestArchive,
    TSTCommandArchives.CommandReplaceCustomFormatArchive,
    TSTCommandArchives.CommandMoveCellsArchive,
    TSTCommandArchives.ColumnRowMetadataArchive,
    TSTCommandArchives.CommandInsertColumnsOrRowsArchive,
    TSTCommandArchives.ColumnRowBundleArchive,
    TSTCommandArchives.CommandRemoveColumnsOrRowsArchive,
    TSTCommandArchives.CommandRowColumnRangeArchive,
    TSTCommandArchives.CommandResizeColumnOrRowArchive,
    TSTCommandArchives.CommandCategoryResizeColumnOrRowArchive,
    TSTCommandArchives.CommandHideShowArchive,
    TSTCommandArchives.CommandTextPreflightInsertCellArchive,
    TSTCommandArchives.CommandPostflightSetCellArchive,
    TSTCommandArchives.CommandChangeTableAreaForColumnOrRowArchive,
    TSTCommandArchives.CommandSetPencilAnnotationsArchive,
    TSTCommandArchives.CommandSetTableNameArchive,
    TSTCommandArchives.CommandSetTableNameEnabledArchive,
    TSTCommandArchives.CommandSetTableFontSizeArchive,
    TSTCommandArchives.CommandSetTableFontNameArchive,
    TSTCommandArchives.CommandSetTableNameHeightArchive,
    TSTCommandArchives.CommandSetNowArchive,
    TSTCommandArchives.CommandSetWasCutArchive,
    TSTCommandArchives.CommandStyleCellsArchive,
    TSTCommandArchives.CommandStyleTableArchive,
    TSTCommandArchives.CommandSetRepeatingHeaderEnabledArchive,
    TSTCommandArchives.CommandSetFiltersEnabledArchive,
    TSTCommandArchives.CommandAddTableStylePresetArchive,
    TSTCommandArchives.CommandRemoveTableStylePresetArchive,
    TSTCommandArchives.CommandReplaceTableStylePresetArchive,
    TSTCommandArchives.CommandApplyTableStylePresetArchive,
    TSTCommandArchives.CommandSetBaseArchive,
    TSTCommandArchives.CommandSetBasePlacesArchive,
    TSTCommandArchives.CommandSetBaseUseMinusSignArchive,
    TSTCommandArchives.CommandSetCurrencyCodeArchive,
    TSTCommandArchives.CommandSetFractionAccuracyArchive,
    TSTCommandArchives.CommandSetNegativeNumberStyleArchive,
    TSTCommandArchives.CommandSetNumberOfDecimalPlacesArchive,
    TSTCommandArchives.CommandSetDateTimeFormatArchive,
    TSTCommandArchives.CommandSetShowThousandsSeparatorArchive,
    TSTCommandArchives.CommandSetUseAccountingStyleArchive,
    TSTCommandArchives.CommandSetRangeControlMinMaxIncArchive,
    TSTCommandArchives.CommandMoveColumnsOrRowsArchive,
    TSTCommandArchives.CommandSortArchive,
    TSTCommandArchives.FormulaRewriteSpecArchive,
    TSTCommandArchives.CommandRewriteFormulasForTransposeArchive,
    TSTCommandArchives.CommandRewriteTableFormulasForRewriteSpecArchive,
    TSTCommandArchives.CommandRewriteMergeFormulasArchive,
    TSTCommandArchives.CommandRewriteCategoryFormulasArchive,
    TSTCommandArchives.CommandRewriteFilterFormulasForTableResizeArchive,
    TSTCommandArchives.CommandRewriteFilterFormulasForRewriteSpecArchive,
    TSTCommandArchives.CommandRewriteConditionalStylesForRewriteSpecArchive,
    TSTCommandArchives.CommandRewritePivotOwnerFormulasArchive,
    TSTCommandArchives.CommandMergeArchive,
    TSTCommandArchives.CommandInverseMergeArchive,
    TSTCommandArchives.CommandUnmergeArchive,
    TSTCommandArchives.CommandChooseTableIdRemapperArchive,
    TSTCommandArchives.CommandCategorySetGroupingColumnsArchive,
    TSTCommandArchives.FormulaEditingCommandGroupArchive,
    TSTCommandArchives.FormulaEditingCommandSelectionBehaviorArchive,
    TSTCommandArchives.TableCommandSelectionBehaviorArchive,
    TSTCommandArchives.CommandApplyCellCommentArchive,
    TSTCommandArchives.CommandSetFormulaTokenizationArchive,
    TSTCommandArchives.CommandSetFilterSetTypeArchive,
    TSTCommandArchives.CommandSetTextStyleArchive,
    TSTCommandArchives.CommandSetTextStylePropertiesArchive,
    TSTCommandArchives.CommandJustForNotifyingArchive,
    TSTCommandArchives.CommandSetStorageLanguageArchive,
    TSTCommandArchives.CommandSetSortOrderArchive,
    TSTCommandArchives.CommandSetGroupSortOrderArchive,
    TSTCommandArchives.CommandRewriteSortOrderForTableResizeArchive,
    TSTCommandArchives.CommandRewriteSortOrderForRewriteSpecArchive,
    TSTCommandArchives.CommandSetFilterSetArchive,
    TSTCommandArchives.ColumnRowRestoreDataArchive,
    TSTCommandArchives.CommandTransposeTableArchive,
    TSTCommandArchives.CommandSetStructuredTextImportRecordArchive,
    TSTCommandArchives.CommandCategoryCollapseExpandGroupArchive,
    TSTCommandArchives.CommandCategoryChangeSummaryAggregateType,
    TSTCommandArchives.CommandCategorySetLabelRowVisibility,
    TSTCommandArchives.CommandCategoryWillChangeGroupValue,
    TSTCommandArchives.IdempotentSelectionTransformerArchive,
    TSTCommandArchives.WPSelectionTransformerArchive,
    TSTCommandArchives.TableSubSelectionTransformerBaseArchive,
    TSTCommandArchives.TableNameSelectionTransformerArchive,
    TSTCommandArchives.ControlCellSelectionTransformerArchive,
    TSTCommandArchives.StockCellSelectionTransformerArchive,
    TSTCommandArchives.RegionSelectionTransformerArchive,
    TSTCommandArchives.StrokeSelectionTransformerArchive,
    TSTCommandArchives.RowColumnSelectionTransformerArchive,
    TSTCommandArchives.CommandCategoryMoveRowsArchive,
    TSTCommandArchives.CommandRewriteHiddenStatesForGroupByChangeArchive,
    TSTCommandArchives.CommandRewritePencilAnnotationFormulasArchive,
    TSTCommandArchives.CommandRewritePencilAnnotationFormulasArchive_AnnotationPair,
    TSTCommandArchives.CommandPivotSetPivotRulesArchive,
    TSTCommandArchives.CommandPivotSetGroupingColumnOptionsArchive,
    TSTCommandArchives.CommandPivotHideShowGrandTotalsArchive,
    TSTCommandArchives.CommandPivotSortArchive,
    TSTCommandArchives.CommandRewriteTrackedReferencesArchive,
    TSTCommandArchives.CommandExtendTableIDHistoryArchive,
];
exports.TSTStylePropertyArchiving$Archives = [
    TSTStylePropertyArchiving.Deprecated_TableStrokeArchive,
    TSTStylePropertyArchiving.CellStylePropertiesArchive,
    TSTStylePropertyArchiving.Deprecated_StrokePresetDataArchive,
    TSTStylePropertyArchiving.StrokePresetDataArchive,
    TSTStylePropertyArchiving.StrokePresetListArchive,
    TSTStylePropertyArchiving.TableStylePropertiesArchive,
    TSTStylePropertyArchiving.TableStylePresetArchive,
    TSTStylePropertyArchiving.TableStrokePresetArchive,
    TSTStylePropertyArchiving.ThemePresetsArchive,
];
exports.TSTArchives_sos$Archives = [
    TSTArchives_sos.SpecSetTableStrokePresetListArchive,
    TSTArchives_sos.SpecTableStrokePresetListArchive,
    TSTArchives_sos.TableCellStylePropertyChangeSetArchive,
    TSTArchives_sos.TableStylePropertyChangeSetArchive,
];
exports.TSWPArchives$Archives = [
    TSWPArchives.SelectionArchive,
    TSWPArchives.ObjectAttributeTable,
    TSWPArchives.ObjectAttributeTable_ObjectAttribute,
    TSWPArchives.StringAttributeTable,
    TSWPArchives.StringAttributeTable_StringAttribute,
    TSWPArchives.ParaDataAttributeTable,
    TSWPArchives.ParaDataAttributeTable_ParaDataAttribute,
    TSWPArchives.OverlappingFieldAttributeTable,
    TSWPArchives.OverlappingFieldAttributeTable_OverlappingFieldAttribute,
    TSWPArchives.StorageArchive,
    TSWPArchives.HighlightArchive,
    TSWPArchives.PencilAnnotationArchive,
    TSWPArchives.FontFeatureArchive,
    TSWPArchives.CharacterStylePropertiesArchive,
    TSWPArchives.CharacterStyleArchive,
    TSWPArchives.TabArchive,
    TSWPArchives.TabsArchive,
    TSWPArchives.LineSpacingArchive,
    TSWPArchives.ParagraphStylePropertiesArchive,
    TSWPArchives.ParagraphStyleArchive,
    TSWPArchives.ListStyleArchive,
    TSWPArchives.ListStyleArchive_LabelGeometry,
    TSWPArchives.ListStyleArchive_LabelImage,
    TSWPArchives.TextStylePresetArchive,
    TSWPArchives.ColumnsArchive,
    TSWPArchives.ColumnsArchive_EqualColumnsArchive,
    TSWPArchives.ColumnsArchive_NonEqualColumnsArchive,
    TSWPArchives.ColumnsArchive_NonEqualColumnsArchive_GapWidthArchive,
    TSWPArchives.PaddingArchive,
    TSWPArchives.ColumnStylePropertiesArchive,
    TSWPArchives.ColumnStyleArchive,
    TSWPArchives.ShapeStylePropertiesArchive,
    TSWPArchives.ShapeStyleArchive,
    TSWPArchives.ThemePresetsArchive,
    TSWPArchives.TextPresetDisplayItemArchive,
    TSWPArchives.TOCEntryStylePropertiesArchive,
    TSWPArchives.TOCEntryStyleArchive,
    TSWPArchives.TOCSettingsArchive,
    TSWPArchives.TOCSettingsArchive_TOCEntryData,
    TSWPArchives.TOCEntryInstanceArchive,
    TSWPArchives.UndoTransaction,
    TSWPArchives.UndoTransaction_GenericTransaction,
    TSWPArchives.UndoTransaction_TextTransaction,
    TSWPArchives.UndoTransaction_CharIndexTransaction,
    TSWPArchives.UndoTransaction_ReplaceCharIndexTransaction,
    TSWPArchives.UndoTransaction_AttributeIndexTransaction,
    TSWPArchives.UndoTransaction_InsertAttributeTransaction,
    TSWPArchives.UndoTransaction_InsertNilTransaction,
    TSWPArchives.UndoTransaction_CharDeltaTransaction,
    TSWPArchives.UndoTransaction_ParagraphDataTransaction,
    TSWPArchives.UndoTransaction_ObjectDOLCTransaction,
    TSWPArchives.UndoTransaction_CTDateTransaction,
    TSWPArchives.UndoTransaction_UnionTransaction,
    TSWPArchives.UndoTransaction_StorageAction,
    TSWPArchives.UndoTransaction_StorageActionGroup,
    TSWPArchives.UndoTransactionWrapperArchive,
    TSWPArchives.ShapeInfoArchive,
    TSWPArchives.CommentInfoArchive,
    TSWPArchives.TOCInfoArchive,
    TSWPArchives.TOCLayoutHintArchive,
    TSWPArchives.EquationInfoArchive,
    TSWPArchives.TextualAttachmentArchive,
    TSWPArchives.TSWPTOCPageNumberAttachmentArchive,
    TSWPArchives.UIGraphicalAttachment,
    TSWPArchives.DrawableAttachmentArchive,
    TSWPArchives.TOCAttachmentArchive,
    TSWPArchives.FootnoteReferenceAttachmentArchive,
    TSWPArchives.NumberAttachmentArchive,
    TSWPArchives.SmartFieldArchive,
    TSWPArchives.HyperlinkFieldArchive,
    TSWPArchives.PlaceholderSmartFieldArchive,
    TSWPArchives.UnsupportedHyperlinkFieldArchive,
    TSWPArchives.BibliographySmartFieldArchive,
    TSWPArchives.CitationRecordArchive,
    TSWPArchives.CitationSmartFieldArchive,
    TSWPArchives.DateTimeSmartFieldArchive,
    TSWPArchives.BookmarkFieldArchive,
    TSWPArchives.FilenameSmartFieldArchive,
    TSWPArchives.MergeFieldTypeArchive,
    TSWPArchives.MergeSmartFieldArchive,
    TSWPArchives.TOCSmartFieldArchive,
    TSWPArchives.TOCSmartFieldArchive_TOCEntry,
    TSWPArchives.RubyFieldArchive,
    TSWPArchives.TateChuYokoFieldArchive,
    TSWPArchives.ChangeArchive,
    TSWPArchives.ChangeSessionArchive,
    TSWPArchives.SectionPlaceholderArchive,
    TSWPArchives.HyperlinkSelectionArchive,
    TSWPArchives.FlowInfoArchive,
    TSWPArchives.FlowInfoContainerArchive,
    TSWPArchives.DropCapArchive,
    TSWPArchives.DropCapStylePropertiesArchive,
    TSWPArchives.DropCapStyleArchive,
    TSWPArchives.CollaboratorTextCursorSubselectionArchive,
];
exports.TSWPCommandArchives$Archives = [
    TSWPCommandArchives.DummyCommandArchive,
    TSWPCommandArchives.TextCommandArchive,
    TSWPCommandArchives.StorageActionCommandArchive,
    TSWPCommandArchives.SetColumnStyleCommandArchive,
    TSWPCommandArchives.ReplaceAllUsesOfStyleCommandArchive,
    TSWPCommandArchives.InsertColumnsCommandArchive,
    TSWPCommandArchives.InsertRowsCommandArchive,
    TSWPCommandArchives.RemoveColumnsCommandArchive,
    TSWPCommandArchives.RemoveRowsCommandArchive,
    TSWPCommandArchives.MergeCellsCommandArchive,
    TSWPCommandArchives.ApplyPlaceholderTextCommandArchive,
    TSWPCommandArchives.UpdateDateTimeFieldCommandArchive,
    TSWPCommandArchives.ApplyRubyTextCommandArchive,
    TSWPCommandArchives.ModifyRubyTextCommandArchive,
    TSWPCommandArchives.ModifyTOCSettingsBaseCommandArchive,
    TSWPCommandArchives.ModifyTOCSettingsForTOCInfoCommandArchive,
    TSWPCommandArchives.ModifyTOCSettingsPresetForThemeCommandArchive,
    TSWPCommandArchives.AnchorAttachmentCommandArchive,
    TSWPCommandArchives.TextApplyThemeCommandArchive,
    TSWPCommandArchives.MoveColumnsCommandArchive,
    TSWPCommandArchives.MoveRowsCommandArchive,
    TSWPCommandArchives.ShapeApplyPresetCommandArchive,
    TSWPCommandArchives.StyleBaseCommandArchive,
    TSWPCommandArchives.StyleCreateCommandArchive,
    TSWPCommandArchives.StyleRenameCommandArchive,
    TSWPCommandArchives.StyleUpdateCommandArchive,
    TSWPCommandArchives.StyleDeleteCommandArchive,
    TSWPCommandArchives.StyleReorderCommandArchive,
    TSWPCommandArchives.StyleUpdatePropertyMapCommandArchive,
    TSWPCommandArchives.ShapeStyleSetValueCommandArchive,
    TSWPCommandArchives.SelectionTransformerArchive,
    TSWPCommandArchives.PencilAnnotationSelectionTransformerArchive,
    TSWPCommandArchives.ShapeSelectionTransformerArchive,
    TSWPCommandArchives.ShapeContentDescription,
    TSWPCommandArchives.ObjectPropertyArchive,
    TSWPCommandArchives.SetObjectPropertiesCommandArchive,
    TSWPCommandArchives.UpdateFlowInfoCommandArchive,
    TSWPCommandArchives.AddFlowInfoCommandArchive,
    TSWPCommandArchives.RemoveFlowInfoCommandArchive,
    TSWPCommandArchives.TextCommentReplyCommandArchive,
    TSWPCommandArchives.ContainedObjectsCommandArchive,
    TSWPCommandArchives.ContainedObjectsCommandArchive_AddBehaviorArgs,
    TSWPCommandArchives.ContainedObjectsCommandArchive_RemoveBehaviorArgs,
    TSWPCommandArchives.ContainedObjectsCommandArchive_RearrangeBehaviorArgs,
    TSWPCommandArchives.EquationInfoGeometryCommandArchive,
    TSWPCommandArchives.CharacterStyleChangePropertyCommand_GArchive,
    TSWPCommandArchives.ParagraphStyleChangePropertyCommand_GArchive,
];
exports.TSWPArchives_sos$Archives = [
    TSWPArchives_sos.SpecSetCharacterStyleArchive,
    TSWPArchives_sos.SpecSetColumnsArchive,
    TSWPArchives_sos.SpecSetDropCapArchive,
    TSWPArchives_sos.SpecSetFontFeaturesArchive,
    TSWPArchives_sos.SpecSetLineSpacingArchive,
    TSWPArchives_sos.SpecSetListStyleArchive,
    TSWPArchives_sos.SpecSetListStyleBoolArrayArchive,
    TSWPArchives_sos.SpecSetListStyleFloatArrayArchive,
    TSWPArchives_sos.SpecSetListStyleImageDataArrayArchive,
    TSWPArchives_sos.SpecSetListStyleListLabelGeometryArrayArchive,
    TSWPArchives_sos.SpecSetListStyleListLabelTypeArrayArchive,
    TSWPArchives_sos.SpecSetListStyleListNumberTypeArrayArchive,
    TSWPArchives_sos.SpecSetListStyleStringArrayArchive,
    TSWPArchives_sos.SpecSetPaddingArchive,
    TSWPArchives_sos.SpecSetParagraphStyleArchive,
    TSWPArchives_sos.SpecSetRuleOffsetArchive,
    TSWPArchives_sos.SpecSetTabsArchive,
    TSWPArchives_sos.SpecCharacterStyleArchive,
    TSWPArchives_sos.SpecColumnsArchive,
    TSWPArchives_sos.SpecDropCapArchive,
    TSWPArchives_sos.SpecFontFeaturesArchive,
    TSWPArchives_sos.SpecLineSpacingArchive,
    TSWPArchives_sos.SpecListStyleArchive,
    TSWPArchives_sos.SpecListStyleBoolArrayArchive,
    TSWPArchives_sos.SpecListStyleFloatArrayArchive,
    TSWPArchives_sos.SpecListStyleImageDataArrayArchive,
    TSWPArchives_sos.SpecListStyleListLabelGeometryArrayArchive,
    TSWPArchives_sos.SpecListStyleListLabelTypeArrayArchive,
    TSWPArchives_sos.SpecListStyleListNumberTypeArrayArchive,
    TSWPArchives_sos.SpecListStyleStringArrayArchive,
    TSWPArchives_sos.SpecPaddingArchive,
    TSWPArchives_sos.SpecParagraphStyleArchive,
    TSWPArchives_sos.SpecRuleOffsetArchive,
    TSWPArchives_sos.SpecTabsArchive,
    TSWPArchives_sos.CharacterStylePropertyChangeSetArchive,
    TSWPArchives_sos.ColumnStylePropertyChangeSetArchive,
    TSWPArchives_sos.DropCapSpecificStylePropertyChangeSetArchive,
    TSWPArchives_sos.DropCapStylePropertyChangeSetArchive,
    TSWPArchives_sos.ListStylePropertyChangeSetArchive,
    TSWPArchives_sos.ParagraphSpecificStylePropertyChangeSetArchive,
    TSWPArchives_sos.ParagraphStylePropertyChangeSetArchive,
    TSWPArchives_sos.ShapeStylePropertyChangeSetArchive,
    TSWPArchives_sos.TOCEntryStylePropertyChangeSetArchive,
    TSWPArchives_sos.StyleDiffArchive,
];
exports.AllArchives = [
    ...exports.KNArchives$Archives,
    ...exports.KNCommandArchives$Archives,
    ...exports.KNArchives_sos$Archives,
    ...exports.KNCommandArchives_sos$Archives,
    ...exports.TSAArchives$Archives,
    ...exports.TSAArchives_sos$Archives,
    ...exports.TSACommandArchives_sos$Archives,
    ...exports.TSCEArchives$Archives,
    ...exports.TSCH3DArchives$Archives,
    ...exports.TSCHArchives$Archives,
    ...exports.TSCHArchives_Common$Archives,
    ...exports.TSCHArchives_GEN$Archives,
    ...exports.TSCHCommandArchives$Archives,
    ...exports.TSCHArchives_sos$Archives,
    ...exports.TSCHPreUFFArchives$Archives,
    ...exports.TSDArchives$Archives,
    ...exports.TSDCommandArchives$Archives,
    ...exports.TSDArchives_sos$Archives,
    ...exports.TSKArchives$Archives,
    ...exports.TSKArchives_sos$Archives,
    ...exports.TSPArchiveMessages$Archives,
    ...exports.TSPDatabaseMessages$Archives,
    ...exports.TSPMessages$Archives,
    ...exports.TSSArchives$Archives,
    ...exports.TSSArchives_sos$Archives,
    ...exports.TSTArchives$Archives,
    ...exports.TSTCommandArchives$Archives,
    ...exports.TSTStylePropertyArchiving$Archives,
    ...exports.TSTArchives_sos$Archives,
    ...exports.TSWPArchives$Archives,
    ...exports.TSWPCommandArchives$Archives,
    ...exports.TSWPArchives_sos$Archives,
];
