// keynote-archives/lib/generated/KNArchives_sos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SlideStylePropertyChangeSetArchive = exports.MotionBackgroundStylePropertyChangeSetArchive = exports.SpecTransitionAttributesArchive = exports.SpecSetTransitionAttributesArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSSArchives_sos_1 = require("./TSSArchives_sos");
const TSSArchives_sos_2 = require("./TSSArchives_sos");
const TSSArchives_sos_3 = require("./TSSArchives_sos");
const TSSArchives_sos_4 = require("./TSSArchives_sos");
const TSDArchives_sos_1 = require("./TSDArchives_sos");
const TSSArchives_sos_5 = require("./TSSArchives_sos");
const KNArchives_1 = require("./KNArchives");
// @generated message type with reflection information, may provide speed optimized methods
class SpecSetTransitionAttributesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("KNSOS.SpecSetTransitionAttributesArchive", [
            { no: 1, name: "transition_attributes", kind: "message", T: () => KNArchives_1.TransitionAttributesArchive },
            { no: 2, name: "unset", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { unset: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional KN.TransitionAttributesArchive transition_attributes */ 1:
                    message.transitionAttributes = KNArchives_1.TransitionAttributesArchive.internalBinaryRead(reader, reader.uint32(), options, message.transitionAttributes);
                    break;
                case /* bool unset */ 2:
                    message.unset = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional KN.TransitionAttributesArchive transition_attributes = 1; */
        if (message.transitionAttributes)
            KNArchives_1.TransitionAttributesArchive.internalBinaryWrite(message.transitionAttributes, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool unset = 2; */
        if (message.unset !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.unset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message KNSOS.SpecSetTransitionAttributesArchive
 */
exports.SpecSetTransitionAttributesArchive = new SpecSetTransitionAttributesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecTransitionAttributesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("KNSOS.SpecTransitionAttributesArchive", [
            { no: 1, name: "spec_set_transition_attributes", kind: "message", T: () => exports.SpecSetTransitionAttributesArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional KNSOS.SpecSetTransitionAttributesArchive spec_set_transition_attributes */ 1:
                    message.specSetTransitionAttributes = exports.SpecSetTransitionAttributesArchive.internalBinaryRead(reader, reader.uint32(), options, message.specSetTransitionAttributes);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional KNSOS.SpecSetTransitionAttributesArchive spec_set_transition_attributes = 1; */
        if (message.specSetTransitionAttributes)
            exports.SpecSetTransitionAttributesArchive.internalBinaryWrite(message.specSetTransitionAttributes, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message KNSOS.SpecTransitionAttributesArchive
 */
exports.SpecTransitionAttributesArchive = new SpecTransitionAttributesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MotionBackgroundStylePropertyChangeSetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("KNSOS.MotionBackgroundStylePropertyChangeSetArchive", [
            { no: 1, name: "name", kind: "message", T: () => TSSArchives_sos_5.SpecStringArchive },
            { no: 2, name: "name_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "variant", kind: "message", T: () => TSSArchives_sos_5.SpecStringArchive },
            { no: 4, name: "variant_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "color_gradient", kind: "message", T: () => TSDArchives_sos_1.SpecFillArchive },
            { no: 6, name: "color_gradient_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "random_seed", kind: "message", T: () => TSSArchives_sos_4.SpecIntegerArchive },
            { no: 8, name: "random_seed_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 9, name: "color_blur_sigma", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 10, name: "color_blur_sigma_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 11, name: "color_contrast", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 12, name: "color_contrast_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 13, name: "color_gamma", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 14, name: "color_gamma_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 15, name: "color_saturation", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 16, name: "color_saturation_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 17, name: "color_brightness", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 18, name: "color_brightness_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 19, name: "color_hue", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 20, name: "color_hue_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 21, name: "color_vibrance", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 22, name: "color_vibrance_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 23, name: "noise_amplitude", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 24, name: "noise_amplitude_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 25, name: "noise_movement", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 26, name: "noise_movement_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 27, name: "noise_octaves", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 28, name: "noise_octaves_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 29, name: "noise_scale", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 30, name: "noise_scale_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 31, name: "noise_scale_horizontal", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 32, name: "noise_scale_horizontal_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 33, name: "noise_scale_vertical", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 34, name: "noise_scale_vertical_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 35, name: "noise_gain", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 36, name: "noise_gain_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 37, name: "noise_lacunarity", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 38, name: "noise_lacunarity_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 39, name: "noise_speed", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 40, name: "noise_speed_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 41, name: "color_palette_offset", kind: "message", T: () => TSSArchives_sos_4.SpecIntegerArchive },
            { no: 42, name: "color_palette_offset_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 43, name: "zoom", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 44, name: "zoom_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 45, name: "invert", kind: "message", T: () => TSSArchives_sos_2.SpecBoolArchive },
            { no: 46, name: "invert_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 47, name: "blend_mode", kind: "message", T: () => TSSArchives_sos_5.SpecStringArchive },
            { no: 48, name: "blend_mode_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 49, name: "source_blend_factor", kind: "message", T: () => TSSArchives_sos_5.SpecStringArchive },
            { no: 50, name: "source_blend_factor_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 51, name: "dest_blend_factor", kind: "message", T: () => TSSArchives_sos_5.SpecStringArchive },
            { no: 52, name: "dest_blend_factor_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 53, name: "motion", kind: "message", T: () => TSSArchives_sos_5.SpecStringArchive },
            { no: 54, name: "motion_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 55, name: "background_color", kind: "message", T: () => TSSArchives_sos_1.SpecColorArchive },
            { no: 56, name: "background_color_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 57, name: "foreground_color", kind: "message", T: () => TSSArchives_sos_1.SpecColorArchive },
            { no: 58, name: "foreground_color_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 59, name: "line_color", kind: "message", T: () => TSSArchives_sos_1.SpecColorArchive },
            { no: 60, name: "line_color_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 61, name: "point_color", kind: "message", T: () => TSSArchives_sos_1.SpecColorArchive },
            { no: 62, name: "point_color_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 63, name: "alpha", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 64, name: "alpha_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 65, name: "border", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 66, name: "border_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 67, name: "blend", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 68, name: "blend_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 69, name: "brightness", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 70, name: "brightness_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 71, name: "border_lightness", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 72, name: "border_lightness_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 73, name: "contrast", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 74, name: "contrast_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 75, name: "damping", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 76, name: "damping_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 77, name: "density", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 78, name: "density_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 79, name: "diffuse_power", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 80, name: "diffuse_power_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 81, name: "distortion", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 82, name: "distortion_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 83, name: "entropy", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 84, name: "entropy_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 85, name: "exposure", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 86, name: "exposure_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 87, name: "filter", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 88, name: "filter_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 89, name: "fog_distance", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 90, name: "fog_distance_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 91, name: "fog_power", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 92, name: "fog_power_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 93, name: "gamma", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 94, name: "gamma_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 95, name: "grain_amount", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 96, name: "grain_amount_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 97, name: "grain_intensity", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 98, name: "grain_intensity_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 99, name: "hue", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 100, name: "hue_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 101, name: "lightness", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 102, name: "lightness_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 103, name: "line_alpha", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 104, name: "line_alpha_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 105, name: "line_center", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 106, name: "line_center_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 107, name: "line_fade", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 108, name: "line_fade_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 109, name: "line_fade_power", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 110, name: "line_fade_power_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 111, name: "line_length", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 112, name: "line_length_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 113, name: "line_width", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 114, name: "line_width_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 115, name: "noise_size_scale", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 116, name: "noise_size_scale_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 117, name: "noise_size_weight", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 118, name: "noise_size_weight_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 119, name: "noise_variance", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 120, name: "noise_variance_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 121, name: "offset", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 122, name: "offset_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 123, name: "motion_background_opacity", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 124, name: "motion_background_opacity_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 125, name: "point_depth_alpha", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 126, name: "point_depth_alpha_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 127, name: "point_size", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 128, name: "point_size_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 129, name: "power", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 130, name: "power_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 131, name: "radius", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 132, name: "radius_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 133, name: "refraction", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 134, name: "refraction_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 135, name: "resolution", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 136, name: "resolution_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 137, name: "saturation", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 138, name: "saturation_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 139, name: "scale", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 140, name: "scale_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 141, name: "shadow_alpha", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 142, name: "shadow_alpha_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 143, name: "shadow_angle", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 144, name: "shadow_angle_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 145, name: "shadow_offset", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 146, name: "shadow_offset_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 147, name: "shadow_scale", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 148, name: "shadow_scale_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 149, name: "shadow_value", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 150, name: "shadow_value_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 151, name: "speed", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 152, name: "speed_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 153, name: "texture_power", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 154, name: "texture_power_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 155, name: "time", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 156, name: "time_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 157, name: "draw_points", kind: "message", T: () => TSSArchives_sos_2.SpecBoolArchive },
            { no: 158, name: "draw_points_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 159, name: "draw_lines", kind: "message", T: () => TSSArchives_sos_2.SpecBoolArchive },
            { no: 160, name: "draw_lines_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 161, name: "draw_shadows", kind: "message", T: () => TSSArchives_sos_2.SpecBoolArchive },
            { no: 162, name: "draw_shadows_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 163, name: "pixels_per_division", kind: "message", T: () => TSSArchives_sos_4.SpecIntegerArchive },
            { no: 164, name: "pixels_per_division_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 165, name: "symmetry", kind: "message", T: () => TSSArchives_sos_2.SpecBoolArchive },
            { no: 166, name: "symmetry_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 167, name: "lock_focus", kind: "message", T: () => TSSArchives_sos_2.SpecBoolArchive },
            { no: 168, name: "lock_focus_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 169, name: "amplitude", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 170, name: "amplitude_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 171, name: "envelope", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 172, name: "envelope_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 173, name: "y_offset", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 174, name: "y_offset_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 175, name: "focus", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 176, name: "focus_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 177, name: "blur", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 178, name: "blur_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 179, name: "grain_speed", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 180, name: "grain_speed_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 181, name: "mesh_color", kind: "message", T: () => TSSArchives_sos_1.SpecColorArchive },
            { no: 182, name: "mesh_color_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 183, name: "angle", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 184, name: "angle_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 185, name: "mesh_line_width", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 186, name: "mesh_line_width_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 187, name: "delta_time", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 188, name: "delta_time_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 189, name: "threshold", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 190, name: "threshold_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 191, name: "time_start", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 192, name: "time_start_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 193, name: "instance_count", kind: "message", T: () => TSSArchives_sos_4.SpecIntegerArchive },
            { no: 194, name: "instance_count_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 195, name: "extrusion", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 196, name: "extrusion_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 197, name: "noise_offset", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 198, name: "noise_offset_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 199, name: "color_hue_offset", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 200, name: "color_hue_offset_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 201, name: "shadow_hue", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 202, name: "shadow_hue_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 203, name: "shadow_hue_offset", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 204, name: "shadow_hue_offset_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 205, name: "shadow_saturation", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 206, name: "shadow_saturation_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 207, name: "shadow_brightness", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 208, name: "shadow_brightness_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 209, name: "shadow_intensity", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 210, name: "shadow_intensity_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 211, name: "shadow_power", kind: "message", T: () => TSSArchives_sos_3.SpecDoubleArchive },
            { no: 212, name: "shadow_power_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 213, name: "noise_octaves_iterations", kind: "message", T: () => TSSArchives_sos_4.SpecIntegerArchive },
            { no: 214, name: "noise_octaves_iterations_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecStringArchive name */ 1:
                    message.name = TSSArchives_sos_5.SpecStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.name);
                    break;
                case /* optional bool name_undefined */ 2:
                    message.nameUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecStringArchive variant */ 3:
                    message.variant = TSSArchives_sos_5.SpecStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.variant);
                    break;
                case /* optional bool variant_undefined */ 4:
                    message.variantUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecFillArchive color_gradient */ 5:
                    message.colorGradient = TSDArchives_sos_1.SpecFillArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorGradient);
                    break;
                case /* optional bool color_gradient_undefined */ 6:
                    message.colorGradientUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive random_seed */ 7:
                    message.randomSeed = TSSArchives_sos_4.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.randomSeed);
                    break;
                case /* optional bool random_seed_undefined */ 8:
                    message.randomSeedUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_blur_sigma */ 9:
                    message.colorBlurSigma = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorBlurSigma);
                    break;
                case /* optional bool color_blur_sigma_undefined */ 10:
                    message.colorBlurSigmaUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_contrast */ 11:
                    message.colorContrast = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorContrast);
                    break;
                case /* optional bool color_contrast_undefined */ 12:
                    message.colorContrastUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_gamma */ 13:
                    message.colorGamma = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorGamma);
                    break;
                case /* optional bool color_gamma_undefined */ 14:
                    message.colorGammaUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_saturation */ 15:
                    message.colorSaturation = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorSaturation);
                    break;
                case /* optional bool color_saturation_undefined */ 16:
                    message.colorSaturationUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_brightness */ 17:
                    message.colorBrightness = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorBrightness);
                    break;
                case /* optional bool color_brightness_undefined */ 18:
                    message.colorBrightnessUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_hue */ 19:
                    message.colorHue = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorHue);
                    break;
                case /* optional bool color_hue_undefined */ 20:
                    message.colorHueUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_vibrance */ 21:
                    message.colorVibrance = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorVibrance);
                    break;
                case /* optional bool color_vibrance_undefined */ 22:
                    message.colorVibranceUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_amplitude */ 23:
                    message.noiseAmplitude = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseAmplitude);
                    break;
                case /* optional bool noise_amplitude_undefined */ 24:
                    message.noiseAmplitudeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_movement */ 25:
                    message.noiseMovement = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseMovement);
                    break;
                case /* optional bool noise_movement_undefined */ 26:
                    message.noiseMovementUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_octaves */ 27:
                    message.noiseOctaves = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseOctaves);
                    break;
                case /* optional bool noise_octaves_undefined */ 28:
                    message.noiseOctavesUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_scale */ 29:
                    message.noiseScale = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseScale);
                    break;
                case /* optional bool noise_scale_undefined */ 30:
                    message.noiseScaleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_scale_horizontal */ 31:
                    message.noiseScaleHorizontal = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseScaleHorizontal);
                    break;
                case /* optional bool noise_scale_horizontal_undefined */ 32:
                    message.noiseScaleHorizontalUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_scale_vertical */ 33:
                    message.noiseScaleVertical = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseScaleVertical);
                    break;
                case /* optional bool noise_scale_vertical_undefined */ 34:
                    message.noiseScaleVerticalUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_gain */ 35:
                    message.noiseGain = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseGain);
                    break;
                case /* optional bool noise_gain_undefined */ 36:
                    message.noiseGainUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_lacunarity */ 37:
                    message.noiseLacunarity = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseLacunarity);
                    break;
                case /* optional bool noise_lacunarity_undefined */ 38:
                    message.noiseLacunarityUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_speed */ 39:
                    message.noiseSpeed = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseSpeed);
                    break;
                case /* optional bool noise_speed_undefined */ 40:
                    message.noiseSpeedUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive color_palette_offset */ 41:
                    message.colorPaletteOffset = TSSArchives_sos_4.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorPaletteOffset);
                    break;
                case /* optional bool color_palette_offset_undefined */ 42:
                    message.colorPaletteOffsetUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive zoom */ 43:
                    message.zoom = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.zoom);
                    break;
                case /* optional bool zoom_undefined */ 44:
                    message.zoomUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive invert */ 45:
                    message.invert = TSSArchives_sos_2.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.invert);
                    break;
                case /* optional bool invert_undefined */ 46:
                    message.invertUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecStringArchive blend_mode */ 47:
                    message.blendMode = TSSArchives_sos_5.SpecStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.blendMode);
                    break;
                case /* optional bool blend_mode_undefined */ 48:
                    message.blendModeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecStringArchive source_blend_factor */ 49:
                    message.sourceBlendFactor = TSSArchives_sos_5.SpecStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.sourceBlendFactor);
                    break;
                case /* optional bool source_blend_factor_undefined */ 50:
                    message.sourceBlendFactorUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecStringArchive dest_blend_factor */ 51:
                    message.destBlendFactor = TSSArchives_sos_5.SpecStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.destBlendFactor);
                    break;
                case /* optional bool dest_blend_factor_undefined */ 52:
                    message.destBlendFactorUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecStringArchive motion */ 53:
                    message.motion = TSSArchives_sos_5.SpecStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.motion);
                    break;
                case /* optional bool motion_undefined */ 54:
                    message.motionUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecColorArchive background_color */ 55:
                    message.backgroundColor = TSSArchives_sos_1.SpecColorArchive.internalBinaryRead(reader, reader.uint32(), options, message.backgroundColor);
                    break;
                case /* optional bool background_color_undefined */ 56:
                    message.backgroundColorUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecColorArchive foreground_color */ 57:
                    message.foregroundColor = TSSArchives_sos_1.SpecColorArchive.internalBinaryRead(reader, reader.uint32(), options, message.foregroundColor);
                    break;
                case /* optional bool foreground_color_undefined */ 58:
                    message.foregroundColorUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecColorArchive line_color */ 59:
                    message.lineColor = TSSArchives_sos_1.SpecColorArchive.internalBinaryRead(reader, reader.uint32(), options, message.lineColor);
                    break;
                case /* optional bool line_color_undefined */ 60:
                    message.lineColorUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecColorArchive point_color */ 61:
                    message.pointColor = TSSArchives_sos_1.SpecColorArchive.internalBinaryRead(reader, reader.uint32(), options, message.pointColor);
                    break;
                case /* optional bool point_color_undefined */ 62:
                    message.pointColorUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive alpha */ 63:
                    message.alpha = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.alpha);
                    break;
                case /* optional bool alpha_undefined */ 64:
                    message.alphaUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive border */ 65:
                    message.border = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.border);
                    break;
                case /* optional bool border_undefined */ 66:
                    message.borderUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive blend */ 67:
                    message.blend = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.blend);
                    break;
                case /* optional bool blend_undefined */ 68:
                    message.blendUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive brightness */ 69:
                    message.brightness = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.brightness);
                    break;
                case /* optional bool brightness_undefined */ 70:
                    message.brightnessUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive border_lightness */ 71:
                    message.borderLightness = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.borderLightness);
                    break;
                case /* optional bool border_lightness_undefined */ 72:
                    message.borderLightnessUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive contrast */ 73:
                    message.contrast = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.contrast);
                    break;
                case /* optional bool contrast_undefined */ 74:
                    message.contrastUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive damping */ 75:
                    message.damping = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.damping);
                    break;
                case /* optional bool damping_undefined */ 76:
                    message.dampingUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive density */ 77:
                    message.density = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.density);
                    break;
                case /* optional bool density_undefined */ 78:
                    message.densityUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive diffuse_power */ 79:
                    message.diffusePower = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.diffusePower);
                    break;
                case /* optional bool diffuse_power_undefined */ 80:
                    message.diffusePowerUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive distortion */ 81:
                    message.distortion = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.distortion);
                    break;
                case /* optional bool distortion_undefined */ 82:
                    message.distortionUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive entropy */ 83:
                    message.entropy = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.entropy);
                    break;
                case /* optional bool entropy_undefined */ 84:
                    message.entropyUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive exposure */ 85:
                    message.exposure = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.exposure);
                    break;
                case /* optional bool exposure_undefined */ 86:
                    message.exposureUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive filter */ 87:
                    message.filter = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.filter);
                    break;
                case /* optional bool filter_undefined */ 88:
                    message.filterUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive fog_distance */ 89:
                    message.fogDistance = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.fogDistance);
                    break;
                case /* optional bool fog_distance_undefined */ 90:
                    message.fogDistanceUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive fog_power */ 91:
                    message.fogPower = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.fogPower);
                    break;
                case /* optional bool fog_power_undefined */ 92:
                    message.fogPowerUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive gamma */ 93:
                    message.gamma = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.gamma);
                    break;
                case /* optional bool gamma_undefined */ 94:
                    message.gammaUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive grain_amount */ 95:
                    message.grainAmount = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.grainAmount);
                    break;
                case /* optional bool grain_amount_undefined */ 96:
                    message.grainAmountUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive grain_intensity */ 97:
                    message.grainIntensity = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.grainIntensity);
                    break;
                case /* optional bool grain_intensity_undefined */ 98:
                    message.grainIntensityUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive hue */ 99:
                    message.hue = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.hue);
                    break;
                case /* optional bool hue_undefined */ 100:
                    message.hueUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive lightness */ 101:
                    message.lightness = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.lightness);
                    break;
                case /* optional bool lightness_undefined */ 102:
                    message.lightnessUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive line_alpha */ 103:
                    message.lineAlpha = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.lineAlpha);
                    break;
                case /* optional bool line_alpha_undefined */ 104:
                    message.lineAlphaUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive line_center */ 105:
                    message.lineCenter = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.lineCenter);
                    break;
                case /* optional bool line_center_undefined */ 106:
                    message.lineCenterUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive line_fade */ 107:
                    message.lineFade = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.lineFade);
                    break;
                case /* optional bool line_fade_undefined */ 108:
                    message.lineFadeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive line_fade_power */ 109:
                    message.lineFadePower = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.lineFadePower);
                    break;
                case /* optional bool line_fade_power_undefined */ 110:
                    message.lineFadePowerUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive line_length */ 111:
                    message.lineLength = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.lineLength);
                    break;
                case /* optional bool line_length_undefined */ 112:
                    message.lineLengthUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive line_width */ 113:
                    message.lineWidth = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.lineWidth);
                    break;
                case /* optional bool line_width_undefined */ 114:
                    message.lineWidthUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_size_scale */ 115:
                    message.noiseSizeScale = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseSizeScale);
                    break;
                case /* optional bool noise_size_scale_undefined */ 116:
                    message.noiseSizeScaleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_size_weight */ 117:
                    message.noiseSizeWeight = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseSizeWeight);
                    break;
                case /* optional bool noise_size_weight_undefined */ 118:
                    message.noiseSizeWeightUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_variance */ 119:
                    message.noiseVariance = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseVariance);
                    break;
                case /* optional bool noise_variance_undefined */ 120:
                    message.noiseVarianceUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive offset */ 121:
                    message.offset = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.offset);
                    break;
                case /* optional bool offset_undefined */ 122:
                    message.offsetUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive motion_background_opacity */ 123:
                    message.motionBackgroundOpacity = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.motionBackgroundOpacity);
                    break;
                case /* optional bool motion_background_opacity_undefined */ 124:
                    message.motionBackgroundOpacityUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive point_depth_alpha */ 125:
                    message.pointDepthAlpha = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.pointDepthAlpha);
                    break;
                case /* optional bool point_depth_alpha_undefined */ 126:
                    message.pointDepthAlphaUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive point_size */ 127:
                    message.pointSize = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.pointSize);
                    break;
                case /* optional bool point_size_undefined */ 128:
                    message.pointSizeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive power */ 129:
                    message.power = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.power);
                    break;
                case /* optional bool power_undefined */ 130:
                    message.powerUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive radius */ 131:
                    message.radius = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.radius);
                    break;
                case /* optional bool radius_undefined */ 132:
                    message.radiusUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive refraction */ 133:
                    message.refraction = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.refraction);
                    break;
                case /* optional bool refraction_undefined */ 134:
                    message.refractionUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive resolution */ 135:
                    message.resolution = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.resolution);
                    break;
                case /* optional bool resolution_undefined */ 136:
                    message.resolutionUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive saturation */ 137:
                    message.saturation = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.saturation);
                    break;
                case /* optional bool saturation_undefined */ 138:
                    message.saturationUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive scale */ 139:
                    message.scale = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.scale);
                    break;
                case /* optional bool scale_undefined */ 140:
                    message.scaleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_alpha */ 141:
                    message.shadowAlpha = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowAlpha);
                    break;
                case /* optional bool shadow_alpha_undefined */ 142:
                    message.shadowAlphaUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_angle */ 143:
                    message.shadowAngle = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowAngle);
                    break;
                case /* optional bool shadow_angle_undefined */ 144:
                    message.shadowAngleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_offset */ 145:
                    message.shadowOffset = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowOffset);
                    break;
                case /* optional bool shadow_offset_undefined */ 146:
                    message.shadowOffsetUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_scale */ 147:
                    message.shadowScale = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowScale);
                    break;
                case /* optional bool shadow_scale_undefined */ 148:
                    message.shadowScaleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_value */ 149:
                    message.shadowValue = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowValue);
                    break;
                case /* optional bool shadow_value_undefined */ 150:
                    message.shadowValueUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive speed */ 151:
                    message.speed = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.speed);
                    break;
                case /* optional bool speed_undefined */ 152:
                    message.speedUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive texture_power */ 153:
                    message.texturePower = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.texturePower);
                    break;
                case /* optional bool texture_power_undefined */ 154:
                    message.texturePowerUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive time */ 155:
                    message.time = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.time);
                    break;
                case /* optional bool time_undefined */ 156:
                    message.timeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive draw_points */ 157:
                    message.drawPoints = TSSArchives_sos_2.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.drawPoints);
                    break;
                case /* optional bool draw_points_undefined */ 158:
                    message.drawPointsUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive draw_lines */ 159:
                    message.drawLines = TSSArchives_sos_2.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.drawLines);
                    break;
                case /* optional bool draw_lines_undefined */ 160:
                    message.drawLinesUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive draw_shadows */ 161:
                    message.drawShadows = TSSArchives_sos_2.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.drawShadows);
                    break;
                case /* optional bool draw_shadows_undefined */ 162:
                    message.drawShadowsUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive pixels_per_division */ 163:
                    message.pixelsPerDivision = TSSArchives_sos_4.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.pixelsPerDivision);
                    break;
                case /* optional bool pixels_per_division_undefined */ 164:
                    message.pixelsPerDivisionUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive symmetry */ 165:
                    message.symmetry = TSSArchives_sos_2.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.symmetry);
                    break;
                case /* optional bool symmetry_undefined */ 166:
                    message.symmetryUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive lock_focus */ 167:
                    message.lockFocus = TSSArchives_sos_2.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.lockFocus);
                    break;
                case /* optional bool lock_focus_undefined */ 168:
                    message.lockFocusUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive amplitude */ 169:
                    message.amplitude = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.amplitude);
                    break;
                case /* optional bool amplitude_undefined */ 170:
                    message.amplitudeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive envelope */ 171:
                    message.envelope = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.envelope);
                    break;
                case /* optional bool envelope_undefined */ 172:
                    message.envelopeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive y_offset */ 173:
                    message.yOffset = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.yOffset);
                    break;
                case /* optional bool y_offset_undefined */ 174:
                    message.yOffsetUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive focus */ 175:
                    message.focus = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.focus);
                    break;
                case /* optional bool focus_undefined */ 176:
                    message.focusUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive blur */ 177:
                    message.blur = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.blur);
                    break;
                case /* optional bool blur_undefined */ 178:
                    message.blurUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive grain_speed */ 179:
                    message.grainSpeed = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.grainSpeed);
                    break;
                case /* optional bool grain_speed_undefined */ 180:
                    message.grainSpeedUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecColorArchive mesh_color */ 181:
                    message.meshColor = TSSArchives_sos_1.SpecColorArchive.internalBinaryRead(reader, reader.uint32(), options, message.meshColor);
                    break;
                case /* optional bool mesh_color_undefined */ 182:
                    message.meshColorUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive angle */ 183:
                    message.angle = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.angle);
                    break;
                case /* optional bool angle_undefined */ 184:
                    message.angleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive mesh_line_width */ 185:
                    message.meshLineWidth = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.meshLineWidth);
                    break;
                case /* optional bool mesh_line_width_undefined */ 186:
                    message.meshLineWidthUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive delta_time */ 187:
                    message.deltaTime = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.deltaTime);
                    break;
                case /* optional bool delta_time_undefined */ 188:
                    message.deltaTimeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive threshold */ 189:
                    message.threshold = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.threshold);
                    break;
                case /* optional bool threshold_undefined */ 190:
                    message.thresholdUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive time_start */ 191:
                    message.timeStart = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.timeStart);
                    break;
                case /* optional bool time_start_undefined */ 192:
                    message.timeStartUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive instance_count */ 193:
                    message.instanceCount = TSSArchives_sos_4.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.instanceCount);
                    break;
                case /* optional bool instance_count_undefined */ 194:
                    message.instanceCountUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive extrusion */ 195:
                    message.extrusion = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.extrusion);
                    break;
                case /* optional bool extrusion_undefined */ 196:
                    message.extrusionUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive noise_offset */ 197:
                    message.noiseOffset = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseOffset);
                    break;
                case /* optional bool noise_offset_undefined */ 198:
                    message.noiseOffsetUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive color_hue_offset */ 199:
                    message.colorHueOffset = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.colorHueOffset);
                    break;
                case /* optional bool color_hue_offset_undefined */ 200:
                    message.colorHueOffsetUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_hue */ 201:
                    message.shadowHue = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowHue);
                    break;
                case /* optional bool shadow_hue_undefined */ 202:
                    message.shadowHueUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_hue_offset */ 203:
                    message.shadowHueOffset = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowHueOffset);
                    break;
                case /* optional bool shadow_hue_offset_undefined */ 204:
                    message.shadowHueOffsetUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_saturation */ 205:
                    message.shadowSaturation = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowSaturation);
                    break;
                case /* optional bool shadow_saturation_undefined */ 206:
                    message.shadowSaturationUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_brightness */ 207:
                    message.shadowBrightness = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowBrightness);
                    break;
                case /* optional bool shadow_brightness_undefined */ 208:
                    message.shadowBrightnessUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_intensity */ 209:
                    message.shadowIntensity = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowIntensity);
                    break;
                case /* optional bool shadow_intensity_undefined */ 210:
                    message.shadowIntensityUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecDoubleArchive shadow_power */ 211:
                    message.shadowPower = TSSArchives_sos_3.SpecDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.shadowPower);
                    break;
                case /* optional bool shadow_power_undefined */ 212:
                    message.shadowPowerUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive noise_octaves_iterations */ 213:
                    message.noiseOctavesIterations = TSSArchives_sos_4.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.noiseOctavesIterations);
                    break;
                case /* optional bool noise_octaves_iterations_undefined */ 214:
                    message.noiseOctavesIterationsUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecStringArchive name = 1; */
        if (message.name)
            TSSArchives_sos_5.SpecStringArchive.internalBinaryWrite(message.name, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool name_undefined = 2; */
        if (message.nameUndefined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.nameUndefined);
        /* optional TSSSOS.SpecStringArchive variant = 3; */
        if (message.variant)
            TSSArchives_sos_5.SpecStringArchive.internalBinaryWrite(message.variant, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool variant_undefined = 4; */
        if (message.variantUndefined !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.variantUndefined);
        /* optional TSDSOS.SpecFillArchive color_gradient = 5; */
        if (message.colorGradient)
            TSDArchives_sos_1.SpecFillArchive.internalBinaryWrite(message.colorGradient, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_gradient_undefined = 6; */
        if (message.colorGradientUndefined !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.colorGradientUndefined);
        /* optional TSSSOS.SpecIntegerArchive random_seed = 7; */
        if (message.randomSeed)
            TSSArchives_sos_4.SpecIntegerArchive.internalBinaryWrite(message.randomSeed, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool random_seed_undefined = 8; */
        if (message.randomSeedUndefined !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).bool(message.randomSeedUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_blur_sigma = 9; */
        if (message.colorBlurSigma)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorBlurSigma, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_blur_sigma_undefined = 10; */
        if (message.colorBlurSigmaUndefined !== undefined)
            writer.tag(10, runtime_1.WireType.Varint).bool(message.colorBlurSigmaUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_contrast = 11; */
        if (message.colorContrast)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorContrast, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_contrast_undefined = 12; */
        if (message.colorContrastUndefined !== undefined)
            writer.tag(12, runtime_1.WireType.Varint).bool(message.colorContrastUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_gamma = 13; */
        if (message.colorGamma)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorGamma, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_gamma_undefined = 14; */
        if (message.colorGammaUndefined !== undefined)
            writer.tag(14, runtime_1.WireType.Varint).bool(message.colorGammaUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_saturation = 15; */
        if (message.colorSaturation)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorSaturation, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_saturation_undefined = 16; */
        if (message.colorSaturationUndefined !== undefined)
            writer.tag(16, runtime_1.WireType.Varint).bool(message.colorSaturationUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_brightness = 17; */
        if (message.colorBrightness)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorBrightness, writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_brightness_undefined = 18; */
        if (message.colorBrightnessUndefined !== undefined)
            writer.tag(18, runtime_1.WireType.Varint).bool(message.colorBrightnessUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_hue = 19; */
        if (message.colorHue)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorHue, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_hue_undefined = 20; */
        if (message.colorHueUndefined !== undefined)
            writer.tag(20, runtime_1.WireType.Varint).bool(message.colorHueUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_vibrance = 21; */
        if (message.colorVibrance)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorVibrance, writer.tag(21, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_vibrance_undefined = 22; */
        if (message.colorVibranceUndefined !== undefined)
            writer.tag(22, runtime_1.WireType.Varint).bool(message.colorVibranceUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_amplitude = 23; */
        if (message.noiseAmplitude)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseAmplitude, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_amplitude_undefined = 24; */
        if (message.noiseAmplitudeUndefined !== undefined)
            writer.tag(24, runtime_1.WireType.Varint).bool(message.noiseAmplitudeUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_movement = 25; */
        if (message.noiseMovement)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseMovement, writer.tag(25, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_movement_undefined = 26; */
        if (message.noiseMovementUndefined !== undefined)
            writer.tag(26, runtime_1.WireType.Varint).bool(message.noiseMovementUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_octaves = 27; */
        if (message.noiseOctaves)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseOctaves, writer.tag(27, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_octaves_undefined = 28; */
        if (message.noiseOctavesUndefined !== undefined)
            writer.tag(28, runtime_1.WireType.Varint).bool(message.noiseOctavesUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_scale = 29; */
        if (message.noiseScale)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseScale, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_scale_undefined = 30; */
        if (message.noiseScaleUndefined !== undefined)
            writer.tag(30, runtime_1.WireType.Varint).bool(message.noiseScaleUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_scale_horizontal = 31; */
        if (message.noiseScaleHorizontal)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseScaleHorizontal, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_scale_horizontal_undefined = 32; */
        if (message.noiseScaleHorizontalUndefined !== undefined)
            writer.tag(32, runtime_1.WireType.Varint).bool(message.noiseScaleHorizontalUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_scale_vertical = 33; */
        if (message.noiseScaleVertical)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseScaleVertical, writer.tag(33, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_scale_vertical_undefined = 34; */
        if (message.noiseScaleVerticalUndefined !== undefined)
            writer.tag(34, runtime_1.WireType.Varint).bool(message.noiseScaleVerticalUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_gain = 35; */
        if (message.noiseGain)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseGain, writer.tag(35, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_gain_undefined = 36; */
        if (message.noiseGainUndefined !== undefined)
            writer.tag(36, runtime_1.WireType.Varint).bool(message.noiseGainUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_lacunarity = 37; */
        if (message.noiseLacunarity)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseLacunarity, writer.tag(37, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_lacunarity_undefined = 38; */
        if (message.noiseLacunarityUndefined !== undefined)
            writer.tag(38, runtime_1.WireType.Varint).bool(message.noiseLacunarityUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_speed = 39; */
        if (message.noiseSpeed)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseSpeed, writer.tag(39, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_speed_undefined = 40; */
        if (message.noiseSpeedUndefined !== undefined)
            writer.tag(40, runtime_1.WireType.Varint).bool(message.noiseSpeedUndefined);
        /* optional TSSSOS.SpecIntegerArchive color_palette_offset = 41; */
        if (message.colorPaletteOffset)
            TSSArchives_sos_4.SpecIntegerArchive.internalBinaryWrite(message.colorPaletteOffset, writer.tag(41, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_palette_offset_undefined = 42; */
        if (message.colorPaletteOffsetUndefined !== undefined)
            writer.tag(42, runtime_1.WireType.Varint).bool(message.colorPaletteOffsetUndefined);
        /* optional TSSSOS.SpecDoubleArchive zoom = 43; */
        if (message.zoom)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.zoom, writer.tag(43, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool zoom_undefined = 44; */
        if (message.zoomUndefined !== undefined)
            writer.tag(44, runtime_1.WireType.Varint).bool(message.zoomUndefined);
        /* optional TSSSOS.SpecBoolArchive invert = 45; */
        if (message.invert)
            TSSArchives_sos_2.SpecBoolArchive.internalBinaryWrite(message.invert, writer.tag(45, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool invert_undefined = 46; */
        if (message.invertUndefined !== undefined)
            writer.tag(46, runtime_1.WireType.Varint).bool(message.invertUndefined);
        /* optional TSSSOS.SpecStringArchive blend_mode = 47; */
        if (message.blendMode)
            TSSArchives_sos_5.SpecStringArchive.internalBinaryWrite(message.blendMode, writer.tag(47, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool blend_mode_undefined = 48; */
        if (message.blendModeUndefined !== undefined)
            writer.tag(48, runtime_1.WireType.Varint).bool(message.blendModeUndefined);
        /* optional TSSSOS.SpecStringArchive source_blend_factor = 49; */
        if (message.sourceBlendFactor)
            TSSArchives_sos_5.SpecStringArchive.internalBinaryWrite(message.sourceBlendFactor, writer.tag(49, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool source_blend_factor_undefined = 50; */
        if (message.sourceBlendFactorUndefined !== undefined)
            writer.tag(50, runtime_1.WireType.Varint).bool(message.sourceBlendFactorUndefined);
        /* optional TSSSOS.SpecStringArchive dest_blend_factor = 51; */
        if (message.destBlendFactor)
            TSSArchives_sos_5.SpecStringArchive.internalBinaryWrite(message.destBlendFactor, writer.tag(51, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool dest_blend_factor_undefined = 52; */
        if (message.destBlendFactorUndefined !== undefined)
            writer.tag(52, runtime_1.WireType.Varint).bool(message.destBlendFactorUndefined);
        /* optional TSSSOS.SpecStringArchive motion = 53; */
        if (message.motion)
            TSSArchives_sos_5.SpecStringArchive.internalBinaryWrite(message.motion, writer.tag(53, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool motion_undefined = 54; */
        if (message.motionUndefined !== undefined)
            writer.tag(54, runtime_1.WireType.Varint).bool(message.motionUndefined);
        /* optional TSSSOS.SpecColorArchive background_color = 55; */
        if (message.backgroundColor)
            TSSArchives_sos_1.SpecColorArchive.internalBinaryWrite(message.backgroundColor, writer.tag(55, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool background_color_undefined = 56; */
        if (message.backgroundColorUndefined !== undefined)
            writer.tag(56, runtime_1.WireType.Varint).bool(message.backgroundColorUndefined);
        /* optional TSSSOS.SpecColorArchive foreground_color = 57; */
        if (message.foregroundColor)
            TSSArchives_sos_1.SpecColorArchive.internalBinaryWrite(message.foregroundColor, writer.tag(57, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool foreground_color_undefined = 58; */
        if (message.foregroundColorUndefined !== undefined)
            writer.tag(58, runtime_1.WireType.Varint).bool(message.foregroundColorUndefined);
        /* optional TSSSOS.SpecColorArchive line_color = 59; */
        if (message.lineColor)
            TSSArchives_sos_1.SpecColorArchive.internalBinaryWrite(message.lineColor, writer.tag(59, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool line_color_undefined = 60; */
        if (message.lineColorUndefined !== undefined)
            writer.tag(60, runtime_1.WireType.Varint).bool(message.lineColorUndefined);
        /* optional TSSSOS.SpecColorArchive point_color = 61; */
        if (message.pointColor)
            TSSArchives_sos_1.SpecColorArchive.internalBinaryWrite(message.pointColor, writer.tag(61, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool point_color_undefined = 62; */
        if (message.pointColorUndefined !== undefined)
            writer.tag(62, runtime_1.WireType.Varint).bool(message.pointColorUndefined);
        /* optional TSSSOS.SpecDoubleArchive alpha = 63; */
        if (message.alpha)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.alpha, writer.tag(63, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool alpha_undefined = 64; */
        if (message.alphaUndefined !== undefined)
            writer.tag(64, runtime_1.WireType.Varint).bool(message.alphaUndefined);
        /* optional TSSSOS.SpecDoubleArchive border = 65; */
        if (message.border)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.border, writer.tag(65, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool border_undefined = 66; */
        if (message.borderUndefined !== undefined)
            writer.tag(66, runtime_1.WireType.Varint).bool(message.borderUndefined);
        /* optional TSSSOS.SpecDoubleArchive blend = 67; */
        if (message.blend)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.blend, writer.tag(67, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool blend_undefined = 68; */
        if (message.blendUndefined !== undefined)
            writer.tag(68, runtime_1.WireType.Varint).bool(message.blendUndefined);
        /* optional TSSSOS.SpecDoubleArchive brightness = 69; */
        if (message.brightness)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.brightness, writer.tag(69, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool brightness_undefined = 70; */
        if (message.brightnessUndefined !== undefined)
            writer.tag(70, runtime_1.WireType.Varint).bool(message.brightnessUndefined);
        /* optional TSSSOS.SpecDoubleArchive border_lightness = 71; */
        if (message.borderLightness)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.borderLightness, writer.tag(71, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool border_lightness_undefined = 72; */
        if (message.borderLightnessUndefined !== undefined)
            writer.tag(72, runtime_1.WireType.Varint).bool(message.borderLightnessUndefined);
        /* optional TSSSOS.SpecDoubleArchive contrast = 73; */
        if (message.contrast)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.contrast, writer.tag(73, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool contrast_undefined = 74; */
        if (message.contrastUndefined !== undefined)
            writer.tag(74, runtime_1.WireType.Varint).bool(message.contrastUndefined);
        /* optional TSSSOS.SpecDoubleArchive damping = 75; */
        if (message.damping)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.damping, writer.tag(75, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool damping_undefined = 76; */
        if (message.dampingUndefined !== undefined)
            writer.tag(76, runtime_1.WireType.Varint).bool(message.dampingUndefined);
        /* optional TSSSOS.SpecDoubleArchive density = 77; */
        if (message.density)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.density, writer.tag(77, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool density_undefined = 78; */
        if (message.densityUndefined !== undefined)
            writer.tag(78, runtime_1.WireType.Varint).bool(message.densityUndefined);
        /* optional TSSSOS.SpecDoubleArchive diffuse_power = 79; */
        if (message.diffusePower)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.diffusePower, writer.tag(79, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool diffuse_power_undefined = 80; */
        if (message.diffusePowerUndefined !== undefined)
            writer.tag(80, runtime_1.WireType.Varint).bool(message.diffusePowerUndefined);
        /* optional TSSSOS.SpecDoubleArchive distortion = 81; */
        if (message.distortion)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.distortion, writer.tag(81, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool distortion_undefined = 82; */
        if (message.distortionUndefined !== undefined)
            writer.tag(82, runtime_1.WireType.Varint).bool(message.distortionUndefined);
        /* optional TSSSOS.SpecDoubleArchive entropy = 83; */
        if (message.entropy)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.entropy, writer.tag(83, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool entropy_undefined = 84; */
        if (message.entropyUndefined !== undefined)
            writer.tag(84, runtime_1.WireType.Varint).bool(message.entropyUndefined);
        /* optional TSSSOS.SpecDoubleArchive exposure = 85; */
        if (message.exposure)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.exposure, writer.tag(85, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool exposure_undefined = 86; */
        if (message.exposureUndefined !== undefined)
            writer.tag(86, runtime_1.WireType.Varint).bool(message.exposureUndefined);
        /* optional TSSSOS.SpecDoubleArchive filter = 87; */
        if (message.filter)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.filter, writer.tag(87, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool filter_undefined = 88; */
        if (message.filterUndefined !== undefined)
            writer.tag(88, runtime_1.WireType.Varint).bool(message.filterUndefined);
        /* optional TSSSOS.SpecDoubleArchive fog_distance = 89; */
        if (message.fogDistance)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.fogDistance, writer.tag(89, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool fog_distance_undefined = 90; */
        if (message.fogDistanceUndefined !== undefined)
            writer.tag(90, runtime_1.WireType.Varint).bool(message.fogDistanceUndefined);
        /* optional TSSSOS.SpecDoubleArchive fog_power = 91; */
        if (message.fogPower)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.fogPower, writer.tag(91, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool fog_power_undefined = 92; */
        if (message.fogPowerUndefined !== undefined)
            writer.tag(92, runtime_1.WireType.Varint).bool(message.fogPowerUndefined);
        /* optional TSSSOS.SpecDoubleArchive gamma = 93; */
        if (message.gamma)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.gamma, writer.tag(93, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool gamma_undefined = 94; */
        if (message.gammaUndefined !== undefined)
            writer.tag(94, runtime_1.WireType.Varint).bool(message.gammaUndefined);
        /* optional TSSSOS.SpecDoubleArchive grain_amount = 95; */
        if (message.grainAmount)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.grainAmount, writer.tag(95, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool grain_amount_undefined = 96; */
        if (message.grainAmountUndefined !== undefined)
            writer.tag(96, runtime_1.WireType.Varint).bool(message.grainAmountUndefined);
        /* optional TSSSOS.SpecDoubleArchive grain_intensity = 97; */
        if (message.grainIntensity)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.grainIntensity, writer.tag(97, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool grain_intensity_undefined = 98; */
        if (message.grainIntensityUndefined !== undefined)
            writer.tag(98, runtime_1.WireType.Varint).bool(message.grainIntensityUndefined);
        /* optional TSSSOS.SpecDoubleArchive hue = 99; */
        if (message.hue)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.hue, writer.tag(99, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool hue_undefined = 100; */
        if (message.hueUndefined !== undefined)
            writer.tag(100, runtime_1.WireType.Varint).bool(message.hueUndefined);
        /* optional TSSSOS.SpecDoubleArchive lightness = 101; */
        if (message.lightness)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.lightness, writer.tag(101, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool lightness_undefined = 102; */
        if (message.lightnessUndefined !== undefined)
            writer.tag(102, runtime_1.WireType.Varint).bool(message.lightnessUndefined);
        /* optional TSSSOS.SpecDoubleArchive line_alpha = 103; */
        if (message.lineAlpha)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.lineAlpha, writer.tag(103, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool line_alpha_undefined = 104; */
        if (message.lineAlphaUndefined !== undefined)
            writer.tag(104, runtime_1.WireType.Varint).bool(message.lineAlphaUndefined);
        /* optional TSSSOS.SpecDoubleArchive line_center = 105; */
        if (message.lineCenter)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.lineCenter, writer.tag(105, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool line_center_undefined = 106; */
        if (message.lineCenterUndefined !== undefined)
            writer.tag(106, runtime_1.WireType.Varint).bool(message.lineCenterUndefined);
        /* optional TSSSOS.SpecDoubleArchive line_fade = 107; */
        if (message.lineFade)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.lineFade, writer.tag(107, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool line_fade_undefined = 108; */
        if (message.lineFadeUndefined !== undefined)
            writer.tag(108, runtime_1.WireType.Varint).bool(message.lineFadeUndefined);
        /* optional TSSSOS.SpecDoubleArchive line_fade_power = 109; */
        if (message.lineFadePower)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.lineFadePower, writer.tag(109, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool line_fade_power_undefined = 110; */
        if (message.lineFadePowerUndefined !== undefined)
            writer.tag(110, runtime_1.WireType.Varint).bool(message.lineFadePowerUndefined);
        /* optional TSSSOS.SpecDoubleArchive line_length = 111; */
        if (message.lineLength)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.lineLength, writer.tag(111, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool line_length_undefined = 112; */
        if (message.lineLengthUndefined !== undefined)
            writer.tag(112, runtime_1.WireType.Varint).bool(message.lineLengthUndefined);
        /* optional TSSSOS.SpecDoubleArchive line_width = 113; */
        if (message.lineWidth)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.lineWidth, writer.tag(113, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool line_width_undefined = 114; */
        if (message.lineWidthUndefined !== undefined)
            writer.tag(114, runtime_1.WireType.Varint).bool(message.lineWidthUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_size_scale = 115; */
        if (message.noiseSizeScale)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseSizeScale, writer.tag(115, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_size_scale_undefined = 116; */
        if (message.noiseSizeScaleUndefined !== undefined)
            writer.tag(116, runtime_1.WireType.Varint).bool(message.noiseSizeScaleUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_size_weight = 117; */
        if (message.noiseSizeWeight)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseSizeWeight, writer.tag(117, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_size_weight_undefined = 118; */
        if (message.noiseSizeWeightUndefined !== undefined)
            writer.tag(118, runtime_1.WireType.Varint).bool(message.noiseSizeWeightUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_variance = 119; */
        if (message.noiseVariance)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseVariance, writer.tag(119, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_variance_undefined = 120; */
        if (message.noiseVarianceUndefined !== undefined)
            writer.tag(120, runtime_1.WireType.Varint).bool(message.noiseVarianceUndefined);
        /* optional TSSSOS.SpecDoubleArchive offset = 121; */
        if (message.offset)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.offset, writer.tag(121, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool offset_undefined = 122; */
        if (message.offsetUndefined !== undefined)
            writer.tag(122, runtime_1.WireType.Varint).bool(message.offsetUndefined);
        /* optional TSSSOS.SpecDoubleArchive motion_background_opacity = 123; */
        if (message.motionBackgroundOpacity)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.motionBackgroundOpacity, writer.tag(123, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool motion_background_opacity_undefined = 124; */
        if (message.motionBackgroundOpacityUndefined !== undefined)
            writer.tag(124, runtime_1.WireType.Varint).bool(message.motionBackgroundOpacityUndefined);
        /* optional TSSSOS.SpecDoubleArchive point_depth_alpha = 125; */
        if (message.pointDepthAlpha)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.pointDepthAlpha, writer.tag(125, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool point_depth_alpha_undefined = 126; */
        if (message.pointDepthAlphaUndefined !== undefined)
            writer.tag(126, runtime_1.WireType.Varint).bool(message.pointDepthAlphaUndefined);
        /* optional TSSSOS.SpecDoubleArchive point_size = 127; */
        if (message.pointSize)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.pointSize, writer.tag(127, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool point_size_undefined = 128; */
        if (message.pointSizeUndefined !== undefined)
            writer.tag(128, runtime_1.WireType.Varint).bool(message.pointSizeUndefined);
        /* optional TSSSOS.SpecDoubleArchive power = 129; */
        if (message.power)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.power, writer.tag(129, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool power_undefined = 130; */
        if (message.powerUndefined !== undefined)
            writer.tag(130, runtime_1.WireType.Varint).bool(message.powerUndefined);
        /* optional TSSSOS.SpecDoubleArchive radius = 131; */
        if (message.radius)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.radius, writer.tag(131, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool radius_undefined = 132; */
        if (message.radiusUndefined !== undefined)
            writer.tag(132, runtime_1.WireType.Varint).bool(message.radiusUndefined);
        /* optional TSSSOS.SpecDoubleArchive refraction = 133; */
        if (message.refraction)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.refraction, writer.tag(133, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool refraction_undefined = 134; */
        if (message.refractionUndefined !== undefined)
            writer.tag(134, runtime_1.WireType.Varint).bool(message.refractionUndefined);
        /* optional TSSSOS.SpecDoubleArchive resolution = 135; */
        if (message.resolution)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.resolution, writer.tag(135, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool resolution_undefined = 136; */
        if (message.resolutionUndefined !== undefined)
            writer.tag(136, runtime_1.WireType.Varint).bool(message.resolutionUndefined);
        /* optional TSSSOS.SpecDoubleArchive saturation = 137; */
        if (message.saturation)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.saturation, writer.tag(137, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool saturation_undefined = 138; */
        if (message.saturationUndefined !== undefined)
            writer.tag(138, runtime_1.WireType.Varint).bool(message.saturationUndefined);
        /* optional TSSSOS.SpecDoubleArchive scale = 139; */
        if (message.scale)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.scale, writer.tag(139, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool scale_undefined = 140; */
        if (message.scaleUndefined !== undefined)
            writer.tag(140, runtime_1.WireType.Varint).bool(message.scaleUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_alpha = 141; */
        if (message.shadowAlpha)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowAlpha, writer.tag(141, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_alpha_undefined = 142; */
        if (message.shadowAlphaUndefined !== undefined)
            writer.tag(142, runtime_1.WireType.Varint).bool(message.shadowAlphaUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_angle = 143; */
        if (message.shadowAngle)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowAngle, writer.tag(143, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_angle_undefined = 144; */
        if (message.shadowAngleUndefined !== undefined)
            writer.tag(144, runtime_1.WireType.Varint).bool(message.shadowAngleUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_offset = 145; */
        if (message.shadowOffset)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowOffset, writer.tag(145, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_offset_undefined = 146; */
        if (message.shadowOffsetUndefined !== undefined)
            writer.tag(146, runtime_1.WireType.Varint).bool(message.shadowOffsetUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_scale = 147; */
        if (message.shadowScale)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowScale, writer.tag(147, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_scale_undefined = 148; */
        if (message.shadowScaleUndefined !== undefined)
            writer.tag(148, runtime_1.WireType.Varint).bool(message.shadowScaleUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_value = 149; */
        if (message.shadowValue)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowValue, writer.tag(149, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_value_undefined = 150; */
        if (message.shadowValueUndefined !== undefined)
            writer.tag(150, runtime_1.WireType.Varint).bool(message.shadowValueUndefined);
        /* optional TSSSOS.SpecDoubleArchive speed = 151; */
        if (message.speed)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.speed, writer.tag(151, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool speed_undefined = 152; */
        if (message.speedUndefined !== undefined)
            writer.tag(152, runtime_1.WireType.Varint).bool(message.speedUndefined);
        /* optional TSSSOS.SpecDoubleArchive texture_power = 153; */
        if (message.texturePower)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.texturePower, writer.tag(153, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool texture_power_undefined = 154; */
        if (message.texturePowerUndefined !== undefined)
            writer.tag(154, runtime_1.WireType.Varint).bool(message.texturePowerUndefined);
        /* optional TSSSOS.SpecDoubleArchive time = 155; */
        if (message.time)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.time, writer.tag(155, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool time_undefined = 156; */
        if (message.timeUndefined !== undefined)
            writer.tag(156, runtime_1.WireType.Varint).bool(message.timeUndefined);
        /* optional TSSSOS.SpecBoolArchive draw_points = 157; */
        if (message.drawPoints)
            TSSArchives_sos_2.SpecBoolArchive.internalBinaryWrite(message.drawPoints, writer.tag(157, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool draw_points_undefined = 158; */
        if (message.drawPointsUndefined !== undefined)
            writer.tag(158, runtime_1.WireType.Varint).bool(message.drawPointsUndefined);
        /* optional TSSSOS.SpecBoolArchive draw_lines = 159; */
        if (message.drawLines)
            TSSArchives_sos_2.SpecBoolArchive.internalBinaryWrite(message.drawLines, writer.tag(159, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool draw_lines_undefined = 160; */
        if (message.drawLinesUndefined !== undefined)
            writer.tag(160, runtime_1.WireType.Varint).bool(message.drawLinesUndefined);
        /* optional TSSSOS.SpecBoolArchive draw_shadows = 161; */
        if (message.drawShadows)
            TSSArchives_sos_2.SpecBoolArchive.internalBinaryWrite(message.drawShadows, writer.tag(161, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool draw_shadows_undefined = 162; */
        if (message.drawShadowsUndefined !== undefined)
            writer.tag(162, runtime_1.WireType.Varint).bool(message.drawShadowsUndefined);
        /* optional TSSSOS.SpecIntegerArchive pixels_per_division = 163; */
        if (message.pixelsPerDivision)
            TSSArchives_sos_4.SpecIntegerArchive.internalBinaryWrite(message.pixelsPerDivision, writer.tag(163, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool pixels_per_division_undefined = 164; */
        if (message.pixelsPerDivisionUndefined !== undefined)
            writer.tag(164, runtime_1.WireType.Varint).bool(message.pixelsPerDivisionUndefined);
        /* optional TSSSOS.SpecBoolArchive symmetry = 165; */
        if (message.symmetry)
            TSSArchives_sos_2.SpecBoolArchive.internalBinaryWrite(message.symmetry, writer.tag(165, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool symmetry_undefined = 166; */
        if (message.symmetryUndefined !== undefined)
            writer.tag(166, runtime_1.WireType.Varint).bool(message.symmetryUndefined);
        /* optional TSSSOS.SpecBoolArchive lock_focus = 167; */
        if (message.lockFocus)
            TSSArchives_sos_2.SpecBoolArchive.internalBinaryWrite(message.lockFocus, writer.tag(167, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool lock_focus_undefined = 168; */
        if (message.lockFocusUndefined !== undefined)
            writer.tag(168, runtime_1.WireType.Varint).bool(message.lockFocusUndefined);
        /* optional TSSSOS.SpecDoubleArchive amplitude = 169; */
        if (message.amplitude)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.amplitude, writer.tag(169, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool amplitude_undefined = 170; */
        if (message.amplitudeUndefined !== undefined)
            writer.tag(170, runtime_1.WireType.Varint).bool(message.amplitudeUndefined);
        /* optional TSSSOS.SpecDoubleArchive envelope = 171; */
        if (message.envelope)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.envelope, writer.tag(171, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool envelope_undefined = 172; */
        if (message.envelopeUndefined !== undefined)
            writer.tag(172, runtime_1.WireType.Varint).bool(message.envelopeUndefined);
        /* optional TSSSOS.SpecDoubleArchive y_offset = 173; */
        if (message.yOffset)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.yOffset, writer.tag(173, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool y_offset_undefined = 174; */
        if (message.yOffsetUndefined !== undefined)
            writer.tag(174, runtime_1.WireType.Varint).bool(message.yOffsetUndefined);
        /* optional TSSSOS.SpecDoubleArchive focus = 175; */
        if (message.focus)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.focus, writer.tag(175, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool focus_undefined = 176; */
        if (message.focusUndefined !== undefined)
            writer.tag(176, runtime_1.WireType.Varint).bool(message.focusUndefined);
        /* optional TSSSOS.SpecDoubleArchive blur = 177; */
        if (message.blur)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.blur, writer.tag(177, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool blur_undefined = 178; */
        if (message.blurUndefined !== undefined)
            writer.tag(178, runtime_1.WireType.Varint).bool(message.blurUndefined);
        /* optional TSSSOS.SpecDoubleArchive grain_speed = 179; */
        if (message.grainSpeed)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.grainSpeed, writer.tag(179, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool grain_speed_undefined = 180; */
        if (message.grainSpeedUndefined !== undefined)
            writer.tag(180, runtime_1.WireType.Varint).bool(message.grainSpeedUndefined);
        /* optional TSSSOS.SpecColorArchive mesh_color = 181; */
        if (message.meshColor)
            TSSArchives_sos_1.SpecColorArchive.internalBinaryWrite(message.meshColor, writer.tag(181, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool mesh_color_undefined = 182; */
        if (message.meshColorUndefined !== undefined)
            writer.tag(182, runtime_1.WireType.Varint).bool(message.meshColorUndefined);
        /* optional TSSSOS.SpecDoubleArchive angle = 183; */
        if (message.angle)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.angle, writer.tag(183, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool angle_undefined = 184; */
        if (message.angleUndefined !== undefined)
            writer.tag(184, runtime_1.WireType.Varint).bool(message.angleUndefined);
        /* optional TSSSOS.SpecDoubleArchive mesh_line_width = 185; */
        if (message.meshLineWidth)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.meshLineWidth, writer.tag(185, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool mesh_line_width_undefined = 186; */
        if (message.meshLineWidthUndefined !== undefined)
            writer.tag(186, runtime_1.WireType.Varint).bool(message.meshLineWidthUndefined);
        /* optional TSSSOS.SpecDoubleArchive delta_time = 187; */
        if (message.deltaTime)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.deltaTime, writer.tag(187, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool delta_time_undefined = 188; */
        if (message.deltaTimeUndefined !== undefined)
            writer.tag(188, runtime_1.WireType.Varint).bool(message.deltaTimeUndefined);
        /* optional TSSSOS.SpecDoubleArchive threshold = 189; */
        if (message.threshold)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.threshold, writer.tag(189, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool threshold_undefined = 190; */
        if (message.thresholdUndefined !== undefined)
            writer.tag(190, runtime_1.WireType.Varint).bool(message.thresholdUndefined);
        /* optional TSSSOS.SpecDoubleArchive time_start = 191; */
        if (message.timeStart)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.timeStart, writer.tag(191, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool time_start_undefined = 192; */
        if (message.timeStartUndefined !== undefined)
            writer.tag(192, runtime_1.WireType.Varint).bool(message.timeStartUndefined);
        /* optional TSSSOS.SpecIntegerArchive instance_count = 193; */
        if (message.instanceCount)
            TSSArchives_sos_4.SpecIntegerArchive.internalBinaryWrite(message.instanceCount, writer.tag(193, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool instance_count_undefined = 194; */
        if (message.instanceCountUndefined !== undefined)
            writer.tag(194, runtime_1.WireType.Varint).bool(message.instanceCountUndefined);
        /* optional TSSSOS.SpecDoubleArchive extrusion = 195; */
        if (message.extrusion)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.extrusion, writer.tag(195, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool extrusion_undefined = 196; */
        if (message.extrusionUndefined !== undefined)
            writer.tag(196, runtime_1.WireType.Varint).bool(message.extrusionUndefined);
        /* optional TSSSOS.SpecDoubleArchive noise_offset = 197; */
        if (message.noiseOffset)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.noiseOffset, writer.tag(197, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_offset_undefined = 198; */
        if (message.noiseOffsetUndefined !== undefined)
            writer.tag(198, runtime_1.WireType.Varint).bool(message.noiseOffsetUndefined);
        /* optional TSSSOS.SpecDoubleArchive color_hue_offset = 199; */
        if (message.colorHueOffset)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.colorHueOffset, writer.tag(199, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool color_hue_offset_undefined = 200; */
        if (message.colorHueOffsetUndefined !== undefined)
            writer.tag(200, runtime_1.WireType.Varint).bool(message.colorHueOffsetUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_hue = 201; */
        if (message.shadowHue)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowHue, writer.tag(201, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_hue_undefined = 202; */
        if (message.shadowHueUndefined !== undefined)
            writer.tag(202, runtime_1.WireType.Varint).bool(message.shadowHueUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_hue_offset = 203; */
        if (message.shadowHueOffset)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowHueOffset, writer.tag(203, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_hue_offset_undefined = 204; */
        if (message.shadowHueOffsetUndefined !== undefined)
            writer.tag(204, runtime_1.WireType.Varint).bool(message.shadowHueOffsetUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_saturation = 205; */
        if (message.shadowSaturation)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowSaturation, writer.tag(205, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_saturation_undefined = 206; */
        if (message.shadowSaturationUndefined !== undefined)
            writer.tag(206, runtime_1.WireType.Varint).bool(message.shadowSaturationUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_brightness = 207; */
        if (message.shadowBrightness)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowBrightness, writer.tag(207, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_brightness_undefined = 208; */
        if (message.shadowBrightnessUndefined !== undefined)
            writer.tag(208, runtime_1.WireType.Varint).bool(message.shadowBrightnessUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_intensity = 209; */
        if (message.shadowIntensity)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowIntensity, writer.tag(209, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_intensity_undefined = 210; */
        if (message.shadowIntensityUndefined !== undefined)
            writer.tag(210, runtime_1.WireType.Varint).bool(message.shadowIntensityUndefined);
        /* optional TSSSOS.SpecDoubleArchive shadow_power = 211; */
        if (message.shadowPower)
            TSSArchives_sos_3.SpecDoubleArchive.internalBinaryWrite(message.shadowPower, writer.tag(211, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool shadow_power_undefined = 212; */
        if (message.shadowPowerUndefined !== undefined)
            writer.tag(212, runtime_1.WireType.Varint).bool(message.shadowPowerUndefined);
        /* optional TSSSOS.SpecIntegerArchive noise_octaves_iterations = 213; */
        if (message.noiseOctavesIterations)
            TSSArchives_sos_4.SpecIntegerArchive.internalBinaryWrite(message.noiseOctavesIterations, writer.tag(213, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool noise_octaves_iterations_undefined = 214; */
        if (message.noiseOctavesIterationsUndefined !== undefined)
            writer.tag(214, runtime_1.WireType.Varint).bool(message.noiseOctavesIterationsUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message KNSOS.MotionBackgroundStylePropertyChangeSetArchive
 */
exports.MotionBackgroundStylePropertyChangeSetArchive = new MotionBackgroundStylePropertyChangeSetArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SlideStylePropertyChangeSetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("KNSOS.SlideStylePropertyChangeSetArchive", [
            { no: 1, name: "fill", kind: "message", T: () => TSDArchives_sos_1.SpecFillArchive },
            { no: 2, name: "fill_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "slide_transition", kind: "message", T: () => exports.SpecTransitionAttributesArchive },
            { no: 4, name: "slide_transition_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSDSOS.SpecFillArchive fill */ 1:
                    message.fill = TSDArchives_sos_1.SpecFillArchive.internalBinaryRead(reader, reader.uint32(), options, message.fill);
                    break;
                case /* optional bool fill_undefined */ 2:
                    message.fillUndefined = reader.bool();
                    break;
                case /* optional KNSOS.SpecTransitionAttributesArchive slide_transition */ 3:
                    message.slideTransition = exports.SpecTransitionAttributesArchive.internalBinaryRead(reader, reader.uint32(), options, message.slideTransition);
                    break;
                case /* optional bool slide_transition_undefined */ 4:
                    message.slideTransitionUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSDSOS.SpecFillArchive fill = 1; */
        if (message.fill)
            TSDArchives_sos_1.SpecFillArchive.internalBinaryWrite(message.fill, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool fill_undefined = 2; */
        if (message.fillUndefined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.fillUndefined);
        /* optional KNSOS.SpecTransitionAttributesArchive slide_transition = 3; */
        if (message.slideTransition)
            exports.SpecTransitionAttributesArchive.internalBinaryWrite(message.slideTransition, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool slide_transition_undefined = 4; */
        if (message.slideTransitionUndefined !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.slideTransitionUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message KNSOS.SlideStylePropertyChangeSetArchive
 */
exports.SlideStylePropertyChangeSetArchive = new SlideStylePropertyChangeSetArchive$Type();
