// keynote-archives/lib/generated/KNCommandArchives_sos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommandSlideReapplyTemplateSlideArchive = exports.InducedVerifyDrawableZOrdersWithServerCommandArchive = exports.InducedVerifyDocumentWithServerCommandArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSACommandArchives_sos_1 = require("./TSACommandArchives_sos");
const TSACommandArchives_sos_2 = require("./TSACommandArchives_sos");
const TSPMessages_1 = require("./TSPMessages");
const TSKArchives_1 = require("./TSKArchives");
// @generated message type with reflection information, may provide speed optimized methods
class InducedVerifyDocumentWithServerCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("KNSOS.InducedVerifyDocumentWithServerCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "slide_node_id_list", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.UUID },
            { no: 3, name: "slide_node_id_list_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "template_slide_node_id_list", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.UUID },
            { no: 5, name: "template_slide_node_id_list_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { slideNodeIdList: [], templateSlideNodeIdList: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSP.UUID slide_node_id_list */ 2:
                    message.slideNodeIdList.push(TSPMessages_1.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool slide_node_id_list_undefined */ 3:
                    message.slideNodeIdListUndefined = reader.bool();
                    break;
                case /* repeated TSP.UUID template_slide_node_id_list */ 4:
                    message.templateSlideNodeIdList.push(TSPMessages_1.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool template_slide_node_id_list_undefined */ 5:
                    message.templateSlideNodeIdListUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUID slide_node_id_list = 2; */
        for (let i = 0; i < message.slideNodeIdList.length; i++)
            TSPMessages_1.UUID.internalBinaryWrite(message.slideNodeIdList[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool slide_node_id_list_undefined = 3; */
        if (message.slideNodeIdListUndefined !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.slideNodeIdListUndefined);
        /* repeated TSP.UUID template_slide_node_id_list = 4; */
        for (let i = 0; i < message.templateSlideNodeIdList.length; i++)
            TSPMessages_1.UUID.internalBinaryWrite(message.templateSlideNodeIdList[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool template_slide_node_id_list_undefined = 5; */
        if (message.templateSlideNodeIdListUndefined !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).bool(message.templateSlideNodeIdListUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message KNSOS.InducedVerifyDocumentWithServerCommandArchive
 */
exports.InducedVerifyDocumentWithServerCommandArchive = new InducedVerifyDocumentWithServerCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class InducedVerifyDrawableZOrdersWithServerCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("KNSOS.InducedVerifyDrawableZOrdersWithServerCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSACommandArchives_sos_2.InducedVerifyDrawableZOrdersWithServerCommandArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSASOS.InducedVerifyDrawableZOrdersWithServerCommandArchive super */ 1:
                    message.super = TSACommandArchives_sos_2.InducedVerifyDrawableZOrdersWithServerCommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSASOS.InducedVerifyDrawableZOrdersWithServerCommandArchive super = 1; */
        if (message.super)
            TSACommandArchives_sos_2.InducedVerifyDrawableZOrdersWithServerCommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message KNSOS.InducedVerifyDrawableZOrdersWithServerCommandArchive
 */
exports.InducedVerifyDrawableZOrdersWithServerCommandArchive = new InducedVerifyDrawableZOrdersWithServerCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CommandSlideReapplyTemplateSlideArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("KNSOS.CommandSlideReapplyTemplateSlideArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSACommandArchives_sos_1.CommandReapplyMasterArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSASOS.CommandReapplyMasterArchive super */ 1:
                    message.super = TSACommandArchives_sos_1.CommandReapplyMasterArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSASOS.CommandReapplyMasterArchive super = 1; */
        if (message.super)
            TSACommandArchives_sos_1.CommandReapplyMasterArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message KNSOS.CommandSlideReapplyTemplateSlideArchive
 */
exports.CommandSlideReapplyTemplateSlideArchive = new CommandSlideReapplyTemplateSlideArchive$Type();
