// keynote-archives/lib/generated/TSAArchives.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TitlePlacementCommandArchive = exports.CaptionInfoArchive = exports.CaptionPlacementArchive = exports.WebVideoInfo = exports.CollaboratorGalleryItemCursor = exports.GalleryItemSetValueCommand_PropertyValue = exports.GalleryItemSetValueCommand = exports.GalleryItemSelectionTransformerHelper = exports.GalleryItemSelectionTransformer = exports.GalleryItemSelection = exports.GallerySelectionTransformer = exports.GalleryInfo = exports.GalleryItem = exports.GalleryItemSetGeometryCommand = exports.GalleryInfoRemoveItemsCommandArchive = exports.GalleryInfoInsertItemsCommandArchive = exports.GalleryInfoSetValueCommandArchive_PropertyValue = exports.GalleryInfoSetValueCommandArchive = exports.RemoteDataChangeCommandArchive = exports.StyleUpdatePropertyMapCommandArchive = exports.InducedVerifyTransformHistoryWithServerCommandArchive = exports.InducedVerifyObjectsWithServerCommandArchive = exports.ChangeDocumentLocaleCommandArchive = exports.NeedsMediaCompatibilityUpgradeCommandArchive = exports.ReplaceCustomFormatCommandArchive = exports.UpdateCustomFormatCommandArchive = exports.AddCustomFormatCommandArchive = exports.ShortcutCommandArchive = exports.PropagatePresetCommandArchive = exports.ShortcutControllerArchive_ShortcutMapEntry = exports.ShortcutControllerArchive = exports.ThemePresetsArchive = exports.FunctionBrowserStateArchive = exports.DocumentArchive = exports.GalleryItemProperty = exports.GalleryCaptionMode = exports.GalleryInfoProperty = exports.GalleryInfoCaptionMode = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSDCommandArchives_1 = require("./TSDCommandArchives");
const TSDCommandArchives_2 = require("./TSDCommandArchives");
const TSWPArchives_1 = require("./TSWPArchives");
const TSDArchives_1 = require("./TSDArchives");
const TSDCommandArchives_3 = require("./TSDCommandArchives");
const TSDArchives_2 = require("./TSDArchives");
const TSPMessages_1 = require("./TSPMessages");
const TSPMessages_2 = require("./TSPMessages");
const TSDCommandArchives_4 = require("./TSDCommandArchives");
const TSDCommandArchives_5 = require("./TSDCommandArchives");
const TSPMessages_3 = require("./TSPMessages");
const TSCEArchives_1 = require("./TSCEArchives");
const TSCEArchives_2 = require("./TSCEArchives");
const TSWPCommandArchives_1 = require("./TSWPCommandArchives");
const TSKArchives_1 = require("./TSKArchives");
const TSKArchives_2 = require("./TSKArchives");
const TSKArchives_3 = require("./TSKArchives");
const TSPMessages_4 = require("./TSPMessages");
const TSKArchives_4 = require("./TSKArchives");
const TSKArchives_5 = require("./TSKArchives");
const TSPMessages_5 = require("./TSPMessages");
const TSWPArchives_2 = require("./TSWPArchives");
const TSKArchives_6 = require("./TSKArchives");
/**
 * @generated from protobuf enum TSA.GalleryInfoCaptionMode
 */
var GalleryInfoCaptionMode;
(function (GalleryInfoCaptionMode) {
    /**
     * @generated from protobuf enum value: GalleryInfoCaptionModeNone = 0;
     */
    GalleryInfoCaptionMode[GalleryInfoCaptionMode["GalleryInfoCaptionModeNone"] = 0] = "GalleryInfoCaptionModeNone";
    /**
     * @generated from protobuf enum value: GalleryInfoCaptionModePerImage = 1;
     */
    GalleryInfoCaptionMode[GalleryInfoCaptionMode["GalleryInfoCaptionModePerImage"] = 1] = "GalleryInfoCaptionModePerImage";
    /**
     * @generated from protobuf enum value: GalleryInfoCaptionModeAllImages = 2;
     */
    GalleryInfoCaptionMode[GalleryInfoCaptionMode["GalleryInfoCaptionModeAllImages"] = 2] = "GalleryInfoCaptionModeAllImages";
})(GalleryInfoCaptionMode = exports.GalleryInfoCaptionMode || (exports.GalleryInfoCaptionMode = {}));
/**
 * @generated from protobuf enum TSA.GalleryInfoProperty
 */
var GalleryInfoProperty;
(function (GalleryInfoProperty) {
    /**
     * @generated from protobuf enum value: GalleryInfoCaptionProperty = 0;
     */
    GalleryInfoProperty[GalleryInfoProperty["GalleryInfoCaptionProperty"] = 0] = "GalleryInfoCaptionProperty";
})(GalleryInfoProperty = exports.GalleryInfoProperty || (exports.GalleryInfoProperty = {}));
/**
 * @generated from protobuf enum TSA.GalleryCaptionMode
 */
var GalleryCaptionMode;
(function (GalleryCaptionMode) {
    /**
     * @generated from protobuf enum value: GalleryCaptionModeNone = 0;
     */
    GalleryCaptionMode[GalleryCaptionMode["GalleryCaptionModeNone"] = 0] = "GalleryCaptionModeNone";
    /**
     * @generated from protobuf enum value: GalleryCaptionModePerImage = 1;
     */
    GalleryCaptionMode[GalleryCaptionMode["GalleryCaptionModePerImage"] = 1] = "GalleryCaptionModePerImage";
    /**
     * @generated from protobuf enum value: GalleryCaptionModeAllImages = 2;
     */
    GalleryCaptionMode[GalleryCaptionMode["GalleryCaptionModeAllImages"] = 2] = "GalleryCaptionModeAllImages";
})(GalleryCaptionMode = exports.GalleryCaptionMode || (exports.GalleryCaptionMode = {}));
/**
 * @generated from protobuf enum TSA.GalleryItemProperty
 */
var GalleryItemProperty;
(function (GalleryItemProperty) {
    /**
     * @generated from protobuf enum value: GalleryItemPropertyImageData = 0;
     */
    GalleryItemProperty[GalleryItemProperty["GalleryItemPropertyImageData"] = 0] = "GalleryItemPropertyImageData";
    /**
     * @generated from protobuf enum value: GalleryItemPropertyThumbnailImageData = 1;
     */
    GalleryItemProperty[GalleryItemProperty["GalleryItemPropertyThumbnailImageData"] = 1] = "GalleryItemPropertyThumbnailImageData";
    /**
     * @generated from protobuf enum value: GalleryItemPropertyAccessibilityDescription = 2;
     */
    GalleryItemProperty[GalleryItemProperty["GalleryItemPropertyAccessibilityDescription"] = 2] = "GalleryItemPropertyAccessibilityDescription";
})(GalleryItemProperty = exports.GalleryItemProperty || (exports.GalleryItemProperty = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DocumentArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.DocumentArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_6.DocumentArchive },
            { no: 2, name: "text_preset_display_items", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSWPArchives_2.TextPresetDisplayItemArchive },
            { no: 3, name: "document_language", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "calculation_engine", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 5, name: "view_state", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 6, name: "function_browser_state", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 7, name: "tables_custom_format_list", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 8, name: "needs_media_compatibility_upgrade", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 9, name: "template_identifier", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 10, name: "shortcut_controller", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 11, name: "annotation_cache_deprecated", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 12, name: "custom_format_list", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 13, name: "annotation_cache_deprecated_2", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 14, name: "collaborative_media_compatibility_upgrade_did_fail", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 15, name: "can_use_hevc", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 16, name: "is_content_source", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { textPresetDisplayItems: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.DocumentArchive super */ 1:
                    message.super = TSKArchives_6.DocumentArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSWP.TextPresetDisplayItemArchive text_preset_display_items */ 2:
                    message.textPresetDisplayItems.push(TSWPArchives_2.TextPresetDisplayItemArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional string document_language */ 3:
                    message.documentLanguage = reader.string();
                    break;
                case /* optional TSP.Reference calculation_engine */ 4:
                    message.calculationEngine = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.calculationEngine);
                    break;
                case /* optional TSP.Reference view_state */ 5:
                    message.viewState = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.viewState);
                    break;
                case /* optional TSP.Reference function_browser_state */ 6:
                    message.functionBrowserState = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.functionBrowserState);
                    break;
                case /* optional TSP.Reference tables_custom_format_list */ 7:
                    message.tablesCustomFormatList = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.tablesCustomFormatList);
                    break;
                case /* optional bool needs_media_compatibility_upgrade */ 8:
                    message.needsMediaCompatibilityUpgrade = reader.bool();
                    break;
                case /* optional string template_identifier */ 9:
                    message.templateIdentifier = reader.string();
                    break;
                case /* optional TSP.Reference shortcut_controller */ 10:
                    message.shortcutController = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.shortcutController);
                    break;
                case /* optional TSP.Reference annotation_cache_deprecated */ 11:
                    message.annotationCacheDeprecated = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.annotationCacheDeprecated);
                    break;
                case /* optional TSP.Reference custom_format_list */ 12:
                    message.customFormatList = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.customFormatList);
                    break;
                case /* optional TSP.Reference annotation_cache_deprecated_2 */ 13:
                    message.annotationCacheDeprecated2 = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.annotationCacheDeprecated2);
                    break;
                case /* optional bool collaborative_media_compatibility_upgrade_did_fail */ 14:
                    message.collaborativeMediaCompatibilityUpgradeDidFail = reader.bool();
                    break;
                case /* optional bool can_use_hevc */ 15:
                    message.canUseHevc = reader.bool();
                    break;
                case /* optional bool is_content_source */ 16:
                    message.isContentSource = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.DocumentArchive super = 1; */
        if (message.super)
            TSKArchives_6.DocumentArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSWP.TextPresetDisplayItemArchive text_preset_display_items = 2; */
        for (let i = 0; i < message.textPresetDisplayItems.length; i++)
            TSWPArchives_2.TextPresetDisplayItemArchive.internalBinaryWrite(message.textPresetDisplayItems[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string document_language = 3; */
        if (message.documentLanguage !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.documentLanguage);
        /* optional TSP.Reference calculation_engine = 4; */
        if (message.calculationEngine)
            TSPMessages_5.Reference.internalBinaryWrite(message.calculationEngine, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference view_state = 5; */
        if (message.viewState)
            TSPMessages_5.Reference.internalBinaryWrite(message.viewState, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference function_browser_state = 6; */
        if (message.functionBrowserState)
            TSPMessages_5.Reference.internalBinaryWrite(message.functionBrowserState, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference tables_custom_format_list = 7; */
        if (message.tablesCustomFormatList)
            TSPMessages_5.Reference.internalBinaryWrite(message.tablesCustomFormatList, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool needs_media_compatibility_upgrade = 8; */
        if (message.needsMediaCompatibilityUpgrade !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).bool(message.needsMediaCompatibilityUpgrade);
        /* optional string template_identifier = 9; */
        if (message.templateIdentifier !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.templateIdentifier);
        /* optional TSP.Reference shortcut_controller = 10; */
        if (message.shortcutController)
            TSPMessages_5.Reference.internalBinaryWrite(message.shortcutController, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference annotation_cache_deprecated = 11; */
        if (message.annotationCacheDeprecated)
            TSPMessages_5.Reference.internalBinaryWrite(message.annotationCacheDeprecated, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference custom_format_list = 12; */
        if (message.customFormatList)
            TSPMessages_5.Reference.internalBinaryWrite(message.customFormatList, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference annotation_cache_deprecated_2 = 13; */
        if (message.annotationCacheDeprecated2)
            TSPMessages_5.Reference.internalBinaryWrite(message.annotationCacheDeprecated2, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool collaborative_media_compatibility_upgrade_did_fail = 14; */
        if (message.collaborativeMediaCompatibilityUpgradeDidFail !== undefined)
            writer.tag(14, runtime_1.WireType.Varint).bool(message.collaborativeMediaCompatibilityUpgradeDidFail);
        /* optional bool can_use_hevc = 15; */
        if (message.canUseHevc !== undefined)
            writer.tag(15, runtime_1.WireType.Varint).bool(message.canUseHevc);
        /* optional bool is_content_source = 16; */
        if (message.isContentSource !== undefined)
            writer.tag(16, runtime_1.WireType.Varint).bool(message.isContentSource);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.DocumentArchive
 */
exports.DocumentArchive = new DocumentArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class FunctionBrowserStateArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.FunctionBrowserStateArchive", [
            { no: 1, name: "recent_functions", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 13 /*ScalarType.UINT32*/ },
            { no: 2, name: "back_functions", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 13 /*ScalarType.UINT32*/ },
            { no: 3, name: "forward_functions", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 13 /*ScalarType.UINT32*/ },
            { no: 4, name: "current_function", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = { recentFunctions: [], backFunctions: [], forwardFunctions: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated uint32 recent_functions */ 1:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.recentFunctions.push(reader.uint32());
                    else
                        message.recentFunctions.push(reader.uint32());
                    break;
                case /* repeated uint32 back_functions */ 2:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.backFunctions.push(reader.uint32());
                    else
                        message.backFunctions.push(reader.uint32());
                    break;
                case /* repeated uint32 forward_functions */ 3:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.forwardFunctions.push(reader.uint32());
                    else
                        message.forwardFunctions.push(reader.uint32());
                    break;
                case /* optional uint32 current_function */ 4:
                    message.currentFunction = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated uint32 recent_functions = 1; */
        for (let i = 0; i < message.recentFunctions.length; i++)
            writer.tag(1, runtime_1.WireType.Varint).uint32(message.recentFunctions[i]);
        /* repeated uint32 back_functions = 2; */
        for (let i = 0; i < message.backFunctions.length; i++)
            writer.tag(2, runtime_1.WireType.Varint).uint32(message.backFunctions[i]);
        /* repeated uint32 forward_functions = 3; */
        for (let i = 0; i < message.forwardFunctions.length; i++)
            writer.tag(3, runtime_1.WireType.Varint).uint32(message.forwardFunctions[i]);
        /* optional uint32 current_function = 4; */
        if (message.currentFunction !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).uint32(message.currentFunction);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.FunctionBrowserStateArchive
 */
exports.FunctionBrowserStateArchive = new FunctionBrowserStateArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemePresetsArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.ThemePresetsArchive", [
            { no: 1, name: "caption_style_presets", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_5.Reference }
        ]);
    }
    create(value) {
        const message = { captionStylePresets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSP.Reference caption_style_presets */ 1:
                    message.captionStylePresets.push(TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSP.Reference caption_style_presets = 1; */
        for (let i = 0; i < message.captionStylePresets.length; i++)
            TSPMessages_5.Reference.internalBinaryWrite(message.captionStylePresets[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.ThemePresetsArchive
 */
exports.ThemePresetsArchive = new ThemePresetsArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ShortcutControllerArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.ShortcutControllerArchive", [
            { no: 1, name: "entries", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ShortcutControllerArchive_ShortcutMapEntry }
        ]);
    }
    create(value) {
        const message = { entries: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSA.ShortcutControllerArchive.ShortcutMapEntry entries */ 1:
                    message.entries.push(exports.ShortcutControllerArchive_ShortcutMapEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSA.ShortcutControllerArchive.ShortcutMapEntry entries = 1; */
        for (let i = 0; i < message.entries.length; i++)
            exports.ShortcutControllerArchive_ShortcutMapEntry.internalBinaryWrite(message.entries[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.ShortcutControllerArchive
 */
exports.ShortcutControllerArchive = new ShortcutControllerArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ShortcutControllerArchive_ShortcutMapEntry$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.ShortcutControllerArchive.ShortcutMapEntry", [
            { no: 1, name: "shortcut", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "style", kind: "message", T: () => TSPMessages_5.Reference }
        ]);
    }
    create(value) {
        const message = { shortcut: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string shortcut */ 1:
                    message.shortcut = reader.string();
                    break;
                case /* TSP.Reference style */ 2:
                    message.style = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.style);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string shortcut = 1; */
        if (message.shortcut !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.shortcut);
        /* TSP.Reference style = 2; */
        if (message.style)
            TSPMessages_5.Reference.internalBinaryWrite(message.style, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.ShortcutControllerArchive.ShortcutMapEntry
 */
exports.ShortcutControllerArchive_ShortcutMapEntry = new ShortcutControllerArchive_ShortcutMapEntry$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PropagatePresetCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.PropagatePresetCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "preset_change_command", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 3, name: "always_preserve_appearance", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { alwaysPreserveAppearance: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference preset_change_command */ 2:
                    message.presetChangeCommand = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.presetChangeCommand);
                    break;
                case /* bool always_preserve_appearance */ 3:
                    message.alwaysPreserveAppearance = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference preset_change_command = 2; */
        if (message.presetChangeCommand)
            TSPMessages_5.Reference.internalBinaryWrite(message.presetChangeCommand, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool always_preserve_appearance = 3; */
        if (message.alwaysPreserveAppearance !== false)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.alwaysPreserveAppearance);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.PropagatePresetCommandArchive
 */
exports.PropagatePresetCommandArchive = new PropagatePresetCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ShortcutCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.ShortcutCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "old_style", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 3, name: "new_style", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 4, name: "old_shortcut", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "new_shortcut", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.Reference old_style */ 2:
                    message.oldStyle = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.oldStyle);
                    break;
                case /* optional TSP.Reference new_style */ 3:
                    message.newStyle = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.newStyle);
                    break;
                case /* optional string old_shortcut */ 4:
                    message.oldShortcut = reader.string();
                    break;
                case /* optional string new_shortcut */ 5:
                    message.newShortcut = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference old_style = 2; */
        if (message.oldStyle)
            TSPMessages_5.Reference.internalBinaryWrite(message.oldStyle, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference new_style = 3; */
        if (message.newStyle)
            TSPMessages_5.Reference.internalBinaryWrite(message.newStyle, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string old_shortcut = 4; */
        if (message.oldShortcut !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.oldShortcut);
        /* optional string new_shortcut = 5; */
        if (message.newShortcut !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.newShortcut);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.ShortcutCommandArchive
 */
exports.ShortcutCommandArchive = new ShortcutCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class AddCustomFormatCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.AddCustomFormatCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "custom_format", kind: "message", T: () => TSKArchives_4.CustomFormatArchive },
            { no: 3, name: "custom_format_key", kind: "message", T: () => TSPMessages_4.UUID }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSK.CustomFormatArchive custom_format */ 2:
                    message.customFormat = TSKArchives_4.CustomFormatArchive.internalBinaryRead(reader, reader.uint32(), options, message.customFormat);
                    break;
                case /* optional TSP.UUID custom_format_key */ 3:
                    message.customFormatKey = TSPMessages_4.UUID.internalBinaryRead(reader, reader.uint32(), options, message.customFormatKey);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSK.CustomFormatArchive custom_format = 2; */
        if (message.customFormat)
            TSKArchives_4.CustomFormatArchive.internalBinaryWrite(message.customFormat, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUID custom_format_key = 3; */
        if (message.customFormatKey)
            TSPMessages_4.UUID.internalBinaryWrite(message.customFormatKey, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.AddCustomFormatCommandArchive
 */
exports.AddCustomFormatCommandArchive = new AddCustomFormatCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class UpdateCustomFormatCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.UpdateCustomFormatCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "custom_format", kind: "message", T: () => TSKArchives_4.CustomFormatArchive },
            { no: 3, name: "old_custom_format", kind: "message", T: () => TSKArchives_4.CustomFormatArchive },
            { no: 4, name: "custom_format_key", kind: "message", T: () => TSPMessages_4.UUID }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSK.CustomFormatArchive custom_format */ 2:
                    message.customFormat = TSKArchives_4.CustomFormatArchive.internalBinaryRead(reader, reader.uint32(), options, message.customFormat);
                    break;
                case /* optional TSK.CustomFormatArchive old_custom_format */ 3:
                    message.oldCustomFormat = TSKArchives_4.CustomFormatArchive.internalBinaryRead(reader, reader.uint32(), options, message.oldCustomFormat);
                    break;
                case /* optional TSP.UUID custom_format_key */ 4:
                    message.customFormatKey = TSPMessages_4.UUID.internalBinaryRead(reader, reader.uint32(), options, message.customFormatKey);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSK.CustomFormatArchive custom_format = 2; */
        if (message.customFormat)
            TSKArchives_4.CustomFormatArchive.internalBinaryWrite(message.customFormat, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSK.CustomFormatArchive old_custom_format = 3; */
        if (message.oldCustomFormat)
            TSKArchives_4.CustomFormatArchive.internalBinaryWrite(message.oldCustomFormat, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUID custom_format_key = 4; */
        if (message.customFormatKey)
            TSPMessages_4.UUID.internalBinaryWrite(message.customFormatKey, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.UpdateCustomFormatCommandArchive
 */
exports.UpdateCustomFormatCommandArchive = new UpdateCustomFormatCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ReplaceCustomFormatCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.ReplaceCustomFormatCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "custom_format", kind: "message", T: () => TSKArchives_4.CustomFormatArchive },
            { no: 3, name: "custom_format_key", kind: "message", T: () => TSPMessages_4.UUID },
            { no: 4, name: "replacement_format", kind: "message", T: () => TSKArchives_3.FormatStructArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSK.CustomFormatArchive custom_format */ 2:
                    message.customFormat = TSKArchives_4.CustomFormatArchive.internalBinaryRead(reader, reader.uint32(), options, message.customFormat);
                    break;
                case /* optional TSP.UUID custom_format_key */ 3:
                    message.customFormatKey = TSPMessages_4.UUID.internalBinaryRead(reader, reader.uint32(), options, message.customFormatKey);
                    break;
                case /* optional TSK.FormatStructArchive replacement_format */ 4:
                    message.replacementFormat = TSKArchives_3.FormatStructArchive.internalBinaryRead(reader, reader.uint32(), options, message.replacementFormat);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSK.CustomFormatArchive custom_format = 2; */
        if (message.customFormat)
            TSKArchives_4.CustomFormatArchive.internalBinaryWrite(message.customFormat, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUID custom_format_key = 3; */
        if (message.customFormatKey)
            TSPMessages_4.UUID.internalBinaryWrite(message.customFormatKey, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSK.FormatStructArchive replacement_format = 4; */
        if (message.replacementFormat)
            TSKArchives_3.FormatStructArchive.internalBinaryWrite(message.replacementFormat, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.ReplaceCustomFormatCommandArchive
 */
exports.ReplaceCustomFormatCommandArchive = new ReplaceCustomFormatCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class NeedsMediaCompatibilityUpgradeCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.NeedsMediaCompatibilityUpgradeCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "needs_media_compatibility_upgrade", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "old_needs_media_compatibility_upgrade", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "collaborative_media_compatibility_upgrade_did_fail", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "old_collaborative_media_compatibility_upgrade_did_fail", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 6, name: "can_use_hevc", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "old_can_use_hevc", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional bool needs_media_compatibility_upgrade */ 2:
                    message.needsMediaCompatibilityUpgrade = reader.bool();
                    break;
                case /* optional bool old_needs_media_compatibility_upgrade */ 3:
                    message.oldNeedsMediaCompatibilityUpgrade = reader.bool();
                    break;
                case /* optional bool collaborative_media_compatibility_upgrade_did_fail */ 4:
                    message.collaborativeMediaCompatibilityUpgradeDidFail = reader.bool();
                    break;
                case /* optional bool old_collaborative_media_compatibility_upgrade_did_fail */ 5:
                    message.oldCollaborativeMediaCompatibilityUpgradeDidFail = reader.bool();
                    break;
                case /* optional bool can_use_hevc */ 6:
                    message.canUseHevc = reader.bool();
                    break;
                case /* optional bool old_can_use_hevc */ 7:
                    message.oldCanUseHevc = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool needs_media_compatibility_upgrade = 2; */
        if (message.needsMediaCompatibilityUpgrade !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.needsMediaCompatibilityUpgrade);
        /* optional bool old_needs_media_compatibility_upgrade = 3; */
        if (message.oldNeedsMediaCompatibilityUpgrade !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.oldNeedsMediaCompatibilityUpgrade);
        /* optional bool collaborative_media_compatibility_upgrade_did_fail = 4; */
        if (message.collaborativeMediaCompatibilityUpgradeDidFail !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.collaborativeMediaCompatibilityUpgradeDidFail);
        /* optional bool old_collaborative_media_compatibility_upgrade_did_fail = 5; */
        if (message.oldCollaborativeMediaCompatibilityUpgradeDidFail !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).bool(message.oldCollaborativeMediaCompatibilityUpgradeDidFail);
        /* optional bool can_use_hevc = 6; */
        if (message.canUseHevc !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.canUseHevc);
        /* optional bool old_can_use_hevc = 7; */
        if (message.oldCanUseHevc !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).bool(message.oldCanUseHevc);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.NeedsMediaCompatibilityUpgradeCommandArchive
 */
exports.NeedsMediaCompatibilityUpgradeCommandArchive = new NeedsMediaCompatibilityUpgradeCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChangeDocumentLocaleCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.ChangeDocumentLocaleCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "old_locale_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "new_locale_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "old_formatting_symbols", kind: "message", T: () => TSKArchives_2.FormattingSymbolsArchive },
            { no: 5, name: "new_formatting_symbols", kind: "message", T: () => TSKArchives_2.FormattingSymbolsArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional string old_locale_id */ 2:
                    message.oldLocaleId = reader.string();
                    break;
                case /* optional string new_locale_id */ 3:
                    message.newLocaleId = reader.string();
                    break;
                case /* optional TSK.FormattingSymbolsArchive old_formatting_symbols */ 4:
                    message.oldFormattingSymbols = TSKArchives_2.FormattingSymbolsArchive.internalBinaryRead(reader, reader.uint32(), options, message.oldFormattingSymbols);
                    break;
                case /* optional TSK.FormattingSymbolsArchive new_formatting_symbols */ 5:
                    message.newFormattingSymbols = TSKArchives_2.FormattingSymbolsArchive.internalBinaryRead(reader, reader.uint32(), options, message.newFormattingSymbols);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string old_locale_id = 2; */
        if (message.oldLocaleId !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.oldLocaleId);
        /* optional string new_locale_id = 3; */
        if (message.newLocaleId !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.newLocaleId);
        /* optional TSK.FormattingSymbolsArchive old_formatting_symbols = 4; */
        if (message.oldFormattingSymbols)
            TSKArchives_2.FormattingSymbolsArchive.internalBinaryWrite(message.oldFormattingSymbols, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSK.FormattingSymbolsArchive new_formatting_symbols = 5; */
        if (message.newFormattingSymbols)
            TSKArchives_2.FormattingSymbolsArchive.internalBinaryWrite(message.newFormattingSymbols, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.ChangeDocumentLocaleCommandArchive
 */
exports.ChangeDocumentLocaleCommandArchive = new ChangeDocumentLocaleCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class InducedVerifyObjectsWithServerCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.InducedVerifyObjectsWithServerCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "object_id_list", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_4.UUID },
            { no: 3, name: "object_id_list_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "server_object_s_o_s_string_list", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "server_object_s_o_s_string_list_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 6, name: "pending_recalc", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "remote_data_sync_key", kind: "scalar", T: 1 /*ScalarType.DOUBLE*/ }
        ]);
    }
    create(value) {
        const message = { objectIdList: [], serverObjectSOSStringList: [], pendingRecalc: false, remoteDataSyncKey: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSP.UUID object_id_list */ 2:
                    message.objectIdList.push(TSPMessages_4.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool object_id_list_undefined */ 3:
                    message.objectIdListUndefined = reader.bool();
                    break;
                case /* repeated string server_object_s_o_s_string_list */ 4:
                    message.serverObjectSOSStringList.push(reader.string());
                    break;
                case /* optional bool server_object_s_o_s_string_list_undefined */ 5:
                    message.serverObjectSOSStringListUndefined = reader.bool();
                    break;
                case /* bool pending_recalc */ 6:
                    message.pendingRecalc = reader.bool();
                    break;
                case /* double remote_data_sync_key */ 7:
                    message.remoteDataSyncKey = reader.double();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUID object_id_list = 2; */
        for (let i = 0; i < message.objectIdList.length; i++)
            TSPMessages_4.UUID.internalBinaryWrite(message.objectIdList[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool object_id_list_undefined = 3; */
        if (message.objectIdListUndefined !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.objectIdListUndefined);
        /* repeated string server_object_s_o_s_string_list = 4; */
        for (let i = 0; i < message.serverObjectSOSStringList.length; i++)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.serverObjectSOSStringList[i]);
        /* optional bool server_object_s_o_s_string_list_undefined = 5; */
        if (message.serverObjectSOSStringListUndefined !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).bool(message.serverObjectSOSStringListUndefined);
        /* bool pending_recalc = 6; */
        if (message.pendingRecalc !== false)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.pendingRecalc);
        /* double remote_data_sync_key = 7; */
        if (message.remoteDataSyncKey !== 0)
            writer.tag(7, runtime_1.WireType.Bit64).double(message.remoteDataSyncKey);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.InducedVerifyObjectsWithServerCommandArchive
 */
exports.InducedVerifyObjectsWithServerCommandArchive = new InducedVerifyObjectsWithServerCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class InducedVerifyTransformHistoryWithServerCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.InducedVerifyTransformHistoryWithServerCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "server_operation_storage_entries", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSKArchives_1.OperationStorageEntry },
            { no: 3, name: "server_operation_storage_entries_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { serverOperationStorageEntries: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSK.OperationStorageEntry server_operation_storage_entries */ 2:
                    message.serverOperationStorageEntries.push(TSKArchives_1.OperationStorageEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool server_operation_storage_entries_undefined */ 3:
                    message.serverOperationStorageEntriesUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSK.OperationStorageEntry server_operation_storage_entries = 2; */
        for (let i = 0; i < message.serverOperationStorageEntries.length; i++)
            TSKArchives_1.OperationStorageEntry.internalBinaryWrite(message.serverOperationStorageEntries[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool server_operation_storage_entries_undefined = 3; */
        if (message.serverOperationStorageEntriesUndefined !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.serverOperationStorageEntriesUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.InducedVerifyTransformHistoryWithServerCommandArchive
 */
exports.InducedVerifyTransformHistoryWithServerCommandArchive = new InducedVerifyTransformHistoryWithServerCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StyleUpdatePropertyMapCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.StyleUpdatePropertyMapCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSWPCommandArchives_1.StyleUpdatePropertyMapCommandArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSWP.StyleUpdatePropertyMapCommandArchive super */ 1:
                    message.super = TSWPCommandArchives_1.StyleUpdatePropertyMapCommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSWP.StyleUpdatePropertyMapCommandArchive super = 1; */
        if (message.super)
            TSWPCommandArchives_1.StyleUpdatePropertyMapCommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.StyleUpdatePropertyMapCommandArchive
 */
exports.StyleUpdatePropertyMapCommandArchive = new StyleUpdatePropertyMapCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class RemoteDataChangeCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.RemoteDataChangeCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "value_map", kind: "message", T: () => TSCEArchives_2.RemoteDataValueMapArchive },
            { no: 3, name: "quotes", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSCEArchives_1.StockArchive },
            { no: 4, name: "remote_data_sync_key", kind: "scalar", T: 1 /*ScalarType.DOUBLE*/ }
        ]);
    }
    create(value) {
        const message = { quotes: [], remoteDataSyncKey: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSCE.RemoteDataValueMapArchive value_map */ 2:
                    message.valueMap = TSCEArchives_2.RemoteDataValueMapArchive.internalBinaryRead(reader, reader.uint32(), options, message.valueMap);
                    break;
                case /* repeated TSCE.StockArchive quotes */ 3:
                    message.quotes.push(TSCEArchives_1.StockArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* double remote_data_sync_key */ 4:
                    message.remoteDataSyncKey = reader.double();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSCE.RemoteDataValueMapArchive value_map = 2; */
        if (message.valueMap)
            TSCEArchives_2.RemoteDataValueMapArchive.internalBinaryWrite(message.valueMap, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCE.StockArchive quotes = 3; */
        for (let i = 0; i < message.quotes.length; i++)
            TSCEArchives_1.StockArchive.internalBinaryWrite(message.quotes[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* double remote_data_sync_key = 4; */
        if (message.remoteDataSyncKey !== 0)
            writer.tag(4, runtime_1.WireType.Bit64).double(message.remoteDataSyncKey);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.RemoteDataChangeCommandArchive
 */
exports.RemoteDataChangeCommandArchive = new RemoteDataChangeCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryInfoSetValueCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryInfoSetValueCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "info_id_path", kind: "message", T: () => TSPMessages_3.UUIDPath },
            { no: 3, name: "property", kind: "enum", opt: true, T: () => ["TSA.GalleryInfoProperty", GalleryInfoProperty] },
            { no: 4, name: "value", kind: "message", T: () => exports.GalleryInfoSetValueCommandArchive_PropertyValue },
            { no: 5, name: "old_value", kind: "message", T: () => exports.GalleryInfoSetValueCommandArchive_PropertyValue }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.UUIDPath info_id_path */ 2:
                    message.infoIdPath = TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options, message.infoIdPath);
                    break;
                case /* optional TSA.GalleryInfoProperty property */ 3:
                    message.property = reader.int32();
                    break;
                case /* optional TSA.GalleryInfoSetValueCommandArchive.PropertyValue value */ 4:
                    message.value = exports.GalleryInfoSetValueCommandArchive_PropertyValue.internalBinaryRead(reader, reader.uint32(), options, message.value);
                    break;
                case /* optional TSA.GalleryInfoSetValueCommandArchive.PropertyValue old_value */ 5:
                    message.oldValue = exports.GalleryInfoSetValueCommandArchive_PropertyValue.internalBinaryRead(reader, reader.uint32(), options, message.oldValue);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUIDPath info_id_path = 2; */
        if (message.infoIdPath)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.infoIdPath, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSA.GalleryInfoProperty property = 3; */
        if (message.property !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.property);
        /* optional TSA.GalleryInfoSetValueCommandArchive.PropertyValue value = 4; */
        if (message.value)
            exports.GalleryInfoSetValueCommandArchive_PropertyValue.internalBinaryWrite(message.value, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSA.GalleryInfoSetValueCommandArchive.PropertyValue old_value = 5; */
        if (message.oldValue)
            exports.GalleryInfoSetValueCommandArchive_PropertyValue.internalBinaryWrite(message.oldValue, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryInfoSetValueCommandArchive
 */
exports.GalleryInfoSetValueCommandArchive = new GalleryInfoSetValueCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryInfoSetValueCommandArchive_PropertyValue$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryInfoSetValueCommandArchive.PropertyValue", [
            { no: 1, name: "caption_mode", kind: "enum", opt: true, T: () => ["TSA.GalleryInfoCaptionMode", GalleryInfoCaptionMode] }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSA.GalleryInfoCaptionMode caption_mode */ 1:
                    message.captionMode = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSA.GalleryInfoCaptionMode caption_mode = 1; */
        if (message.captionMode !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.captionMode);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryInfoSetValueCommandArchive.PropertyValue
 */
exports.GalleryInfoSetValueCommandArchive_PropertyValue = new GalleryInfoSetValueCommandArchive_PropertyValue$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryInfoInsertItemsCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryInfoInsertItemsCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSDCommandArchives_5.ContainerInsertChildrenCommandArchive },
            { no: 2, name: "items", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_5.Reference }
        ]);
    }
    create(value) {
        const message = { items: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSD.ContainerInsertChildrenCommandArchive super */ 1:
                    message.super = TSDCommandArchives_5.ContainerInsertChildrenCommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSP.Reference items */ 2:
                    message.items.push(TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSD.ContainerInsertChildrenCommandArchive super = 1; */
        if (message.super)
            TSDCommandArchives_5.ContainerInsertChildrenCommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference items = 2; */
        for (let i = 0; i < message.items.length; i++)
            TSPMessages_5.Reference.internalBinaryWrite(message.items[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryInfoInsertItemsCommandArchive
 */
exports.GalleryInfoInsertItemsCommandArchive = new GalleryInfoInsertItemsCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryInfoRemoveItemsCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryInfoRemoveItemsCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSDCommandArchives_4.ContainerRemoveChildrenCommandArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSD.ContainerRemoveChildrenCommandArchive super */ 1:
                    message.super = TSDCommandArchives_4.ContainerRemoveChildrenCommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSD.ContainerRemoveChildrenCommandArchive super = 1; */
        if (message.super)
            TSDCommandArchives_4.ContainerRemoveChildrenCommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryInfoRemoveItemsCommandArchive
 */
exports.GalleryInfoRemoveItemsCommandArchive = new GalleryInfoRemoveItemsCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryItemSetGeometryCommand$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryItemSetGeometryCommand", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "item_id_path", kind: "message", T: () => TSPMessages_3.UUIDPath },
            { no: 3, name: "offset", kind: "message", T: () => TSPMessages_2.Point },
            { no: 4, name: "scale", kind: "scalar", opt: true, T: 2 /*ScalarType.FLOAT*/ },
            { no: 5, name: "old_offset", kind: "message", T: () => TSPMessages_2.Point },
            { no: 6, name: "old_scale", kind: "scalar", opt: true, T: 2 /*ScalarType.FLOAT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.UUIDPath item_id_path */ 2:
                    message.itemIdPath = TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options, message.itemIdPath);
                    break;
                case /* optional TSP.Point offset */ 3:
                    message.offset = TSPMessages_2.Point.internalBinaryRead(reader, reader.uint32(), options, message.offset);
                    break;
                case /* optional float scale */ 4:
                    message.scale = reader.float();
                    break;
                case /* optional TSP.Point old_offset */ 5:
                    message.oldOffset = TSPMessages_2.Point.internalBinaryRead(reader, reader.uint32(), options, message.oldOffset);
                    break;
                case /* optional float old_scale */ 6:
                    message.oldScale = reader.float();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUIDPath item_id_path = 2; */
        if (message.itemIdPath)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.itemIdPath, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Point offset = 3; */
        if (message.offset)
            TSPMessages_2.Point.internalBinaryWrite(message.offset, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional float scale = 4; */
        if (message.scale !== undefined)
            writer.tag(4, runtime_1.WireType.Bit32).float(message.scale);
        /* optional TSP.Point old_offset = 5; */
        if (message.oldOffset)
            TSPMessages_2.Point.internalBinaryWrite(message.oldOffset, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional float old_scale = 6; */
        if (message.oldScale !== undefined)
            writer.tag(6, runtime_1.WireType.Bit32).float(message.oldScale);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryItemSetGeometryCommand
 */
exports.GalleryItemSetGeometryCommand = new GalleryItemSetGeometryCommand$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryItem$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryItem", [
            { no: 1, name: "offset", kind: "message", T: () => TSPMessages_2.Point },
            { no: 2, name: "scale", kind: "scalar", opt: true, T: 2 /*ScalarType.FLOAT*/ },
            { no: 3, name: "image_data", kind: "message", T: () => TSPMessages_1.DataReference },
            { no: 4, name: "thumbnail_image_data", kind: "message", T: () => TSPMessages_1.DataReference },
            { no: 5, name: "image_adjustments", kind: "message", T: () => TSDArchives_2.ImageAdjustmentsArchive },
            { no: 6, name: "adjusted_image_data", kind: "message", T: () => TSPMessages_1.DataReference },
            { no: 7, name: "thumbnail_adjusted_image_data", kind: "message", T: () => TSPMessages_1.DataReference },
            { no: 8, name: "caption_storage", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 9, name: "accessibility_description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Point offset */ 1:
                    message.offset = TSPMessages_2.Point.internalBinaryRead(reader, reader.uint32(), options, message.offset);
                    break;
                case /* optional float scale */ 2:
                    message.scale = reader.float();
                    break;
                case /* optional TSP.DataReference image_data */ 3:
                    message.imageData = TSPMessages_1.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.imageData);
                    break;
                case /* optional TSP.DataReference thumbnail_image_data */ 4:
                    message.thumbnailImageData = TSPMessages_1.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.thumbnailImageData);
                    break;
                case /* optional TSD.ImageAdjustmentsArchive image_adjustments */ 5:
                    message.imageAdjustments = TSDArchives_2.ImageAdjustmentsArchive.internalBinaryRead(reader, reader.uint32(), options, message.imageAdjustments);
                    break;
                case /* optional TSP.DataReference adjusted_image_data */ 6:
                    message.adjustedImageData = TSPMessages_1.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.adjustedImageData);
                    break;
                case /* optional TSP.DataReference thumbnail_adjusted_image_data */ 7:
                    message.thumbnailAdjustedImageData = TSPMessages_1.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.thumbnailAdjustedImageData);
                    break;
                case /* optional TSP.Reference caption_storage */ 8:
                    message.captionStorage = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.captionStorage);
                    break;
                case /* optional string accessibility_description */ 9:
                    message.accessibilityDescription = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Point offset = 1; */
        if (message.offset)
            TSPMessages_2.Point.internalBinaryWrite(message.offset, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional float scale = 2; */
        if (message.scale !== undefined)
            writer.tag(2, runtime_1.WireType.Bit32).float(message.scale);
        /* optional TSP.DataReference image_data = 3; */
        if (message.imageData)
            TSPMessages_1.DataReference.internalBinaryWrite(message.imageData, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.DataReference thumbnail_image_data = 4; */
        if (message.thumbnailImageData)
            TSPMessages_1.DataReference.internalBinaryWrite(message.thumbnailImageData, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.ImageAdjustmentsArchive image_adjustments = 5; */
        if (message.imageAdjustments)
            TSDArchives_2.ImageAdjustmentsArchive.internalBinaryWrite(message.imageAdjustments, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.DataReference adjusted_image_data = 6; */
        if (message.adjustedImageData)
            TSPMessages_1.DataReference.internalBinaryWrite(message.adjustedImageData, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.DataReference thumbnail_adjusted_image_data = 7; */
        if (message.thumbnailAdjustedImageData)
            TSPMessages_1.DataReference.internalBinaryWrite(message.thumbnailAdjustedImageData, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference caption_storage = 8; */
        if (message.captionStorage)
            TSPMessages_5.Reference.internalBinaryWrite(message.captionStorage, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string accessibility_description = 9; */
        if (message.accessibilityDescription !== undefined)
            writer.tag(9, runtime_1.WireType.LengthDelimited).string(message.accessibilityDescription);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryItem
 */
exports.GalleryItem = new GalleryItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryInfo", [
            { no: 1, name: "items", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_5.Reference },
            { no: 2, name: "caption_mode", kind: "enum", opt: true, T: () => ["TSA.GalleryCaptionMode", GalleryCaptionMode] },
            { no: 3, name: "caption_storage", kind: "message", T: () => TSPMessages_5.Reference }
        ]);
    }
    create(value) {
        const message = { items: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSP.Reference items */ 1:
                    message.items.push(TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional TSA.GalleryCaptionMode caption_mode */ 2:
                    message.captionMode = reader.int32();
                    break;
                case /* optional TSP.Reference caption_storage */ 3:
                    message.captionStorage = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.captionStorage);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSP.Reference items = 1; */
        for (let i = 0; i < message.items.length; i++)
            TSPMessages_5.Reference.internalBinaryWrite(message.items[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSA.GalleryCaptionMode caption_mode = 2; */
        if (message.captionMode !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.captionMode);
        /* optional TSP.Reference caption_storage = 3; */
        if (message.captionStorage)
            TSPMessages_5.Reference.internalBinaryWrite(message.captionStorage, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryInfo
 */
exports.GalleryInfo = new GalleryInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GallerySelectionTransformer$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GallerySelectionTransformer", [
            { no: 1, name: "super", kind: "message", T: () => TSDCommandArchives_3.DrawableSelectionTransformerArchive },
            { no: 2, name: "gallery_uuid_path", kind: "message", T: () => TSPMessages_3.UUIDPath },
            { no: 3, name: "caption_storage_uuid_path", kind: "message", T: () => TSPMessages_3.UUIDPath }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSD.DrawableSelectionTransformerArchive super */ 1:
                    message.super = TSDCommandArchives_3.DrawableSelectionTransformerArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.UUIDPath gallery_uuid_path */ 2:
                    message.galleryUuidPath = TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options, message.galleryUuidPath);
                    break;
                case /* optional TSP.UUIDPath caption_storage_uuid_path */ 3:
                    message.captionStorageUuidPath = TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options, message.captionStorageUuidPath);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSD.DrawableSelectionTransformerArchive super = 1; */
        if (message.super)
            TSDCommandArchives_3.DrawableSelectionTransformerArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUIDPath gallery_uuid_path = 2; */
        if (message.galleryUuidPath)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.galleryUuidPath, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUIDPath caption_storage_uuid_path = 3; */
        if (message.captionStorageUuidPath)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.captionStorageUuidPath, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GallerySelectionTransformer
 */
exports.GallerySelectionTransformer = new GallerySelectionTransformer$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryItemSelection$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryItemSelection", [
            { no: 1, name: "displayed_item", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 2, name: "items", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_5.Reference }
        ]);
    }
    create(value) {
        const message = { items: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference displayed_item */ 1:
                    message.displayedItem = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.displayedItem);
                    break;
                case /* repeated TSP.Reference items */ 2:
                    message.items.push(TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference displayed_item = 1; */
        if (message.displayedItem)
            TSPMessages_5.Reference.internalBinaryWrite(message.displayedItem, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference items = 2; */
        for (let i = 0; i < message.items.length; i++)
            TSPMessages_5.Reference.internalBinaryWrite(message.items[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryItemSelection
 */
exports.GalleryItemSelection = new GalleryItemSelection$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryItemSelectionTransformer$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryItemSelectionTransformer", [
            { no: 1, name: "transformer_helper", kind: "message", T: () => exports.GalleryItemSelectionTransformerHelper },
            { no: 2, name: "caption_storage_uuid_path", kind: "message", T: () => TSPMessages_3.UUIDPath }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSA.GalleryItemSelectionTransformerHelper transformer_helper */ 1:
                    message.transformerHelper = exports.GalleryItemSelectionTransformerHelper.internalBinaryRead(reader, reader.uint32(), options, message.transformerHelper);
                    break;
                case /* optional TSP.UUIDPath caption_storage_uuid_path */ 2:
                    message.captionStorageUuidPath = TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options, message.captionStorageUuidPath);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSA.GalleryItemSelectionTransformerHelper transformer_helper = 1; */
        if (message.transformerHelper)
            exports.GalleryItemSelectionTransformerHelper.internalBinaryWrite(message.transformerHelper, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUIDPath caption_storage_uuid_path = 2; */
        if (message.captionStorageUuidPath)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.captionStorageUuidPath, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryItemSelectionTransformer
 */
exports.GalleryItemSelectionTransformer = new GalleryItemSelectionTransformer$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryItemSelectionTransformerHelper$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryItemSelectionTransformerHelper", [
            { no: 1, name: "displayed_item_uuid_path", kind: "message", T: () => TSPMessages_3.UUIDPath },
            { no: 2, name: "item_uuid_paths", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.UUIDPath }
        ]);
    }
    create(value) {
        const message = { itemUuidPaths: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.UUIDPath displayed_item_uuid_path */ 1:
                    message.displayedItemUuidPath = TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options, message.displayedItemUuidPath);
                    break;
                case /* repeated TSP.UUIDPath item_uuid_paths */ 2:
                    message.itemUuidPaths.push(TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.UUIDPath displayed_item_uuid_path = 1; */
        if (message.displayedItemUuidPath)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.displayedItemUuidPath, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUIDPath item_uuid_paths = 2; */
        for (let i = 0; i < message.itemUuidPaths.length; i++)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.itemUuidPaths[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryItemSelectionTransformerHelper
 */
exports.GalleryItemSelectionTransformerHelper = new GalleryItemSelectionTransformerHelper$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryItemSetValueCommand$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryItemSetValueCommand", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_5.CommandArchive },
            { no: 2, name: "item_id_path", kind: "message", T: () => TSPMessages_3.UUIDPath },
            { no: 3, name: "property", kind: "enum", opt: true, T: () => ["TSA.GalleryItemProperty", GalleryItemProperty] },
            { no: 4, name: "value", kind: "message", T: () => exports.GalleryItemSetValueCommand_PropertyValue },
            { no: 5, name: "old_value", kind: "message", T: () => exports.GalleryItemSetValueCommand_PropertyValue }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_5.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.UUIDPath item_id_path */ 2:
                    message.itemIdPath = TSPMessages_3.UUIDPath.internalBinaryRead(reader, reader.uint32(), options, message.itemIdPath);
                    break;
                case /* optional TSA.GalleryItemProperty property */ 3:
                    message.property = reader.int32();
                    break;
                case /* optional TSA.GalleryItemSetValueCommand.PropertyValue value */ 4:
                    message.value = exports.GalleryItemSetValueCommand_PropertyValue.internalBinaryRead(reader, reader.uint32(), options, message.value);
                    break;
                case /* optional TSA.GalleryItemSetValueCommand.PropertyValue old_value */ 5:
                    message.oldValue = exports.GalleryItemSetValueCommand_PropertyValue.internalBinaryRead(reader, reader.uint32(), options, message.oldValue);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_5.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.UUIDPath item_id_path = 2; */
        if (message.itemIdPath)
            TSPMessages_3.UUIDPath.internalBinaryWrite(message.itemIdPath, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSA.GalleryItemProperty property = 3; */
        if (message.property !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.property);
        /* optional TSA.GalleryItemSetValueCommand.PropertyValue value = 4; */
        if (message.value)
            exports.GalleryItemSetValueCommand_PropertyValue.internalBinaryWrite(message.value, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSA.GalleryItemSetValueCommand.PropertyValue old_value = 5; */
        if (message.oldValue)
            exports.GalleryItemSetValueCommand_PropertyValue.internalBinaryWrite(message.oldValue, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryItemSetValueCommand
 */
exports.GalleryItemSetValueCommand = new GalleryItemSetValueCommand$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GalleryItemSetValueCommand_PropertyValue$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.GalleryItemSetValueCommand.PropertyValue", [
            { no: 1, name: "image_data", kind: "message", T: () => TSPMessages_1.DataReference },
            { no: 2, name: "accessibility_description", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.DataReference image_data */ 1:
                    message.imageData = TSPMessages_1.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.imageData);
                    break;
                case /* optional string accessibility_description */ 2:
                    message.accessibilityDescription = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.DataReference image_data = 1; */
        if (message.imageData)
            TSPMessages_1.DataReference.internalBinaryWrite(message.imageData, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string accessibility_description = 2; */
        if (message.accessibilityDescription !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.accessibilityDescription);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.GalleryItemSetValueCommand.PropertyValue
 */
exports.GalleryItemSetValueCommand_PropertyValue = new GalleryItemSetValueCommand_PropertyValue$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CollaboratorGalleryItemCursor$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.CollaboratorGalleryItemCursor", [
            { no: 1, name: "displayed_item_id", kind: "message", T: () => TSPMessages_4.UUID },
            { no: 2, name: "item_ids", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_4.UUID }
        ]);
    }
    create(value) {
        const message = { itemIds: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.UUID displayed_item_id */ 1:
                    message.displayedItemId = TSPMessages_4.UUID.internalBinaryRead(reader, reader.uint32(), options, message.displayedItemId);
                    break;
                case /* repeated TSP.UUID item_ids */ 2:
                    message.itemIds.push(TSPMessages_4.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.UUID displayed_item_id = 1; */
        if (message.displayedItemId)
            TSPMessages_4.UUID.internalBinaryWrite(message.displayedItemId, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUID item_ids = 2; */
        for (let i = 0; i < message.itemIds.length; i++)
            TSPMessages_4.UUID.internalBinaryWrite(message.itemIds[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.CollaboratorGalleryItemCursor
 */
exports.CollaboratorGalleryItemCursor = new CollaboratorGalleryItemCursor$Type();
// @generated message type with reflection information, may provide speed optimized methods
class WebVideoInfo$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.WebVideoInfo", [
            { no: 1, name: "url", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "poster_image_data", kind: "message", T: () => TSPMessages_1.DataReference },
            { no: 3, name: "attribution", kind: "message", T: () => TSDArchives_1.Attribution }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string url */ 1:
                    message.url = reader.string();
                    break;
                case /* optional TSP.DataReference poster_image_data */ 2:
                    message.posterImageData = TSPMessages_1.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.posterImageData);
                    break;
                case /* optional TSD.Attribution attribution */ 3:
                    message.attribution = TSDArchives_1.Attribution.internalBinaryRead(reader, reader.uint32(), options, message.attribution);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string url = 1; */
        if (message.url !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.url);
        /* optional TSP.DataReference poster_image_data = 2; */
        if (message.posterImageData)
            TSPMessages_1.DataReference.internalBinaryWrite(message.posterImageData, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.Attribution attribution = 3; */
        if (message.attribution)
            TSDArchives_1.Attribution.internalBinaryWrite(message.attribution, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.WebVideoInfo
 */
exports.WebVideoInfo = new WebVideoInfo$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CaptionPlacementArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.CaptionPlacementArchive", [
            { no: 1, name: "caption_anchor_location", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 2, name: "drawable_anchor_location", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int32 caption_anchor_location */ 1:
                    message.captionAnchorLocation = reader.int32();
                    break;
                case /* optional int32 drawable_anchor_location */ 2:
                    message.drawableAnchorLocation = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int32 caption_anchor_location = 1; */
        if (message.captionAnchorLocation !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.captionAnchorLocation);
        /* optional int32 drawable_anchor_location = 2; */
        if (message.drawableAnchorLocation !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.drawableAnchorLocation);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.CaptionPlacementArchive
 */
exports.CaptionPlacementArchive = new CaptionPlacementArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CaptionInfoArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.CaptionInfoArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSWPArchives_1.ShapeInfoArchive },
            { no: 2, name: "placement", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 3, name: "childInfoKind", kind: "enum", opt: true, T: () => ["TSD.CaptionOrTitleKind", TSDCommandArchives_2.CaptionOrTitleKind] }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSWP.ShapeInfoArchive super */ 1:
                    message.super = TSWPArchives_1.ShapeInfoArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.Reference placement */ 2:
                    message.placement = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.placement);
                    break;
                case /* optional TSD.CaptionOrTitleKind childInfoKind */ 3:
                    message.childInfoKind = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSWP.ShapeInfoArchive super = 1; */
        if (message.super)
            TSWPArchives_1.ShapeInfoArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference placement = 2; */
        if (message.placement)
            TSPMessages_5.Reference.internalBinaryWrite(message.placement, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.CaptionOrTitleKind childInfoKind = 3; */
        if (message.childInfoKind !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.childInfoKind);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.CaptionInfoArchive
 */
exports.CaptionInfoArchive = new CaptionInfoArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TitlePlacementCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSA.TitlePlacementCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSDCommandArchives_1.InfoCommandArchive },
            { no: 2, name: "placing_child_info_kind", kind: "enum", opt: true, T: () => ["TSD.CaptionOrTitleKind", TSDCommandArchives_2.CaptionOrTitleKind] },
            { no: 3, name: "placement", kind: "message", T: () => TSPMessages_5.Reference },
            { no: 4, name: "old_placement", kind: "message", T: () => TSPMessages_5.Reference }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSD.InfoCommandArchive super */ 1:
                    message.super = TSDCommandArchives_1.InfoCommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSD.CaptionOrTitleKind placing_child_info_kind */ 2:
                    message.placingChildInfoKind = reader.int32();
                    break;
                case /* optional TSP.Reference placement */ 3:
                    message.placement = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.placement);
                    break;
                case /* optional TSP.Reference old_placement */ 4:
                    message.oldPlacement = TSPMessages_5.Reference.internalBinaryRead(reader, reader.uint32(), options, message.oldPlacement);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSD.InfoCommandArchive super = 1; */
        if (message.super)
            TSDCommandArchives_1.InfoCommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.CaptionOrTitleKind placing_child_info_kind = 2; */
        if (message.placingChildInfoKind !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.placingChildInfoKind);
        /* optional TSP.Reference placement = 3; */
        if (message.placement)
            TSPMessages_5.Reference.internalBinaryWrite(message.placement, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference old_placement = 4; */
        if (message.oldPlacement)
            TSPMessages_5.Reference.internalBinaryWrite(message.oldPlacement, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSA.TitlePlacementCommandArchive
 */
exports.TitlePlacementCommandArchive = new TitlePlacementCommandArchive$Type();
