// keynote-archives/lib/generated/TSAArchives_sos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrawableZOrderListArchive = exports.InducedVerifyActivityStreamWithServerCommandArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSPMessages_1 = require("./TSPMessages");
const TSPMessages_2 = require("./TSPMessages");
const TSKArchives_1 = require("./TSKArchives");
// @generated message type with reflection information, may provide speed optimized methods
class InducedVerifyActivityStreamWithServerCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSASOS.InducedVerifyActivityStreamWithServerCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "server_activities", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_2.Reference },
            { no: 3, name: "server_activities_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "timestamp_of_last_activity_when_last_activity_coalescing", kind: "scalar", T: 1 /*ScalarType.DOUBLE*/ },
            { no: 5, name: "next_activity_to_transform_index", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 6, name: "author_identifiers_to_verify", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.UUID },
            { no: 7, name: "author_identifiers_to_verify_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { serverActivities: [], timestampOfLastActivityWhenLastActivityCoalescing: 0, nextActivityToTransformIndex: 0, authorIdentifiersToVerify: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSP.Reference server_activities */ 2:
                    message.serverActivities.push(TSPMessages_2.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool server_activities_undefined */ 3:
                    message.serverActivitiesUndefined = reader.bool();
                    break;
                case /* double timestamp_of_last_activity_when_last_activity_coalescing */ 4:
                    message.timestampOfLastActivityWhenLastActivityCoalescing = reader.double();
                    break;
                case /* int32 next_activity_to_transform_index */ 5:
                    message.nextActivityToTransformIndex = reader.int32();
                    break;
                case /* repeated TSP.UUID author_identifiers_to_verify */ 6:
                    message.authorIdentifiersToVerify.push(TSPMessages_1.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool author_identifiers_to_verify_undefined */ 7:
                    message.authorIdentifiersToVerifyUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference server_activities = 2; */
        for (let i = 0; i < message.serverActivities.length; i++)
            TSPMessages_2.Reference.internalBinaryWrite(message.serverActivities[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool server_activities_undefined = 3; */
        if (message.serverActivitiesUndefined !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.serverActivitiesUndefined);
        /* double timestamp_of_last_activity_when_last_activity_coalescing = 4; */
        if (message.timestampOfLastActivityWhenLastActivityCoalescing !== 0)
            writer.tag(4, runtime_1.WireType.Bit64).double(message.timestampOfLastActivityWhenLastActivityCoalescing);
        /* int32 next_activity_to_transform_index = 5; */
        if (message.nextActivityToTransformIndex !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.nextActivityToTransformIndex);
        /* repeated TSP.UUID author_identifiers_to_verify = 6; */
        for (let i = 0; i < message.authorIdentifiersToVerify.length; i++)
            TSPMessages_1.UUID.internalBinaryWrite(message.authorIdentifiersToVerify[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool author_identifiers_to_verify_undefined = 7; */
        if (message.authorIdentifiersToVerifyUndefined !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).bool(message.authorIdentifiersToVerifyUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSASOS.InducedVerifyActivityStreamWithServerCommandArchive
 */
exports.InducedVerifyActivityStreamWithServerCommandArchive = new InducedVerifyActivityStreamWithServerCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DrawableZOrderListArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSASOS.DrawableZOrderListArchive", [
            { no: 1, name: "container_id", kind: "message", T: () => TSPMessages_1.UUID },
            { no: 2, name: "drawable_id_list", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.UUID },
            { no: 3, name: "drawable_id_list_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { drawableIdList: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.UUID container_id */ 1:
                    message.containerId = TSPMessages_1.UUID.internalBinaryRead(reader, reader.uint32(), options, message.containerId);
                    break;
                case /* repeated TSP.UUID drawable_id_list */ 2:
                    message.drawableIdList.push(TSPMessages_1.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool drawable_id_list_undefined */ 3:
                    message.drawableIdListUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.UUID container_id = 1; */
        if (message.containerId)
            TSPMessages_1.UUID.internalBinaryWrite(message.containerId, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUID drawable_id_list = 2; */
        for (let i = 0; i < message.drawableIdList.length; i++)
            TSPMessages_1.UUID.internalBinaryWrite(message.drawableIdList[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool drawable_id_list_undefined = 3; */
        if (message.drawableIdListUndefined !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.drawableIdListUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSASOS.DrawableZOrderListArchive
 */
exports.DrawableZOrderListArchive = new DrawableZOrderListArchive$Type();
