// keynote-archives/lib/generated/TSCHArchives.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CollaboratorChartTitleCursorSubselectionArchive = exports.CollaboratorCDECursorSubselectionArchive = exports.PropertyValueStorageContainerReferenceLinesArchive = exports.ChartAxisReferenceLineSparseNonStylesArchive = exports.ChartPresetReferenceLineStylesArchive = exports.ChartReferenceLinesArchive = exports.ChartAxisReferenceLineStylesArchive = exports.ChartAxisReferenceLineNonStylesArchive = exports.ChartReferenceLineNonStyleItem = exports.ChartFormatStructExtensions = exports.ChartUIStateMultiDataIndexUpgrade = exports.ChartUIState = exports.ChartCDESelectionArchive = exports.ChartSelectionArchive = exports.ChartSelectionPathArchive = exports.ChartSelectionPathArgumentArchive = exports.ChartAxisIDArchive = exports.ChartSelectionPathTypeArchive = exports.StylePasteboardDataArchive = exports.PropertyValueStorageContainerArchive = exports.ChartPresetsArchive = exports.ChartStylePreset = exports.ChartFillSetArchive = exports.ChartMediatorArchive = exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry = exports.ChartGridArchive_ChartGridRowColumnIdMap = exports.ChartGridArchive = exports.ChartSupportsSeriesPropertySpacingArchive = exports.ChartSupportsRoundedCornersArchive = exports.ChartSupportsProportionalBendedCalloutLinesArchive = exports.ChartPreserveAppearanceForPresetArchive = exports.ChartPasteboardAdditionsArchive = exports.ChartGarlicLabelFormatUpgrade = exports.ChartGarlicMinMaxUpgrade = exports.ChartMultiDataIndexUpgrade = exports.ChartArchive = exports.ChartDrawableArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSPMessages_1 = require("./TSPMessages");
const TSDArchives_1 = require("./TSDArchives");
const TSCHArchives_Common_1 = require("./TSCHArchives_Common");
const TSSArchives_1 = require("./TSSArchives");
const TSCHArchives_Common_2 = require("./TSCHArchives_Common");
const TSPMessages_2 = require("./TSPMessages");
const TSCHArchives_Common_3 = require("./TSCHArchives_Common");
const TSPMessages_3 = require("./TSPMessages");
const TSCHArchives_Common_4 = require("./TSCHArchives_Common");
const TSCHArchives_Common_5 = require("./TSCHArchives_Common");
const TSCHArchives_Common_6 = require("./TSCHArchives_Common");
const TSDArchives_2 = require("./TSDArchives");
// @generated message type with reflection information, may provide speed optimized methods
class ChartDrawableArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartDrawableArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSDArchives_2.DrawableArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSD.DrawableArchive super */ 1:
                    message.super = TSDArchives_2.DrawableArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSD.DrawableArchive super = 1; */
        if (message.super)
            TSDArchives_2.DrawableArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartDrawableArchive
 */
exports.ChartDrawableArchive = new ChartDrawableArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartArchive", [
            { no: 1, name: "chart_type", kind: "enum", opt: true, T: () => ["TSCH.ChartType", TSCHArchives_Common_6.ChartType] },
            { no: 2, name: "scatter_format", kind: "enum", opt: true, T: () => ["TSCH.ScatterFormat", TSCHArchives_Common_5.ScatterFormat] },
            { no: 3, name: "legend_frame", kind: "message", T: () => TSCHArchives_Common_4.RectArchive },
            { no: 4, name: "preset", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 5, name: "series_direction", kind: "enum", opt: true, T: () => ["TSCH.SeriesDirection", TSCHArchives_Common_3.SeriesDirection] },
            { no: 6, name: "contains_default_data", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "grid", kind: "message", T: () => exports.ChartGridArchive },
            { no: 8, name: "mediator", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 9, name: "chart_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 10, name: "chart_non_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 11, name: "legend_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 12, name: "legend_non_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 13, name: "value_axis_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 14, name: "value_axis_nonstyles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 15, name: "category_axis_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 16, name: "category_axis_nonstyles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 17, name: "series_theme_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 18, name: "series_private_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 19, name: "series_non_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 20, name: "paragraph_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 21, name: "multidataset_index", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ },
            { no: 22, name: "needs_calc_engine_deferred_import_action", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 23, name: "owned_preset", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 24, name: "is_dirty", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { valueAxisStyles: [], valueAxisNonstyles: [], categoryAxisStyles: [], categoryAxisNonstyles: [], seriesThemeStyles: [], paragraphStyles: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSCH.ChartType chart_type */ 1:
                    message.chartType = reader.int32();
                    break;
                case /* optional TSCH.ScatterFormat scatter_format */ 2:
                    message.scatterFormat = reader.int32();
                    break;
                case /* optional TSCH.RectArchive legend_frame */ 3:
                    message.legendFrame = TSCHArchives_Common_4.RectArchive.internalBinaryRead(reader, reader.uint32(), options, message.legendFrame);
                    break;
                case /* optional TSP.Reference preset */ 4:
                    message.preset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.preset);
                    break;
                case /* optional TSCH.SeriesDirection series_direction */ 5:
                    message.seriesDirection = reader.int32();
                    break;
                case /* optional bool contains_default_data */ 6:
                    message.containsDefaultData = reader.bool();
                    break;
                case /* optional TSCH.ChartGridArchive grid */ 7:
                    message.grid = exports.ChartGridArchive.internalBinaryRead(reader, reader.uint32(), options, message.grid);
                    break;
                case /* optional TSP.Reference mediator */ 8:
                    message.mediator = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.mediator);
                    break;
                case /* optional TSP.Reference chart_style */ 9:
                    message.chartStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chartStyle);
                    break;
                case /* optional TSP.Reference chart_non_style */ 10:
                    message.chartNonStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chartNonStyle);
                    break;
                case /* optional TSP.Reference legend_style */ 11:
                    message.legendStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.legendStyle);
                    break;
                case /* optional TSP.Reference legend_non_style */ 12:
                    message.legendNonStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.legendNonStyle);
                    break;
                case /* repeated TSP.Reference value_axis_styles */ 13:
                    message.valueAxisStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference value_axis_nonstyles */ 14:
                    message.valueAxisNonstyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference category_axis_styles */ 15:
                    message.categoryAxisStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference category_axis_nonstyles */ 16:
                    message.categoryAxisNonstyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference series_theme_styles */ 17:
                    message.seriesThemeStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional TSP.SparseReferenceArray series_private_styles */ 18:
                    message.seriesPrivateStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.seriesPrivateStyles);
                    break;
                case /* optional TSP.SparseReferenceArray series_non_styles */ 19:
                    message.seriesNonStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.seriesNonStyles);
                    break;
                case /* repeated TSP.Reference paragraph_styles */ 20:
                    message.paragraphStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional uint32 multidataset_index */ 21:
                    message.multidatasetIndex = reader.uint32();
                    break;
                case /* optional bool needs_calc_engine_deferred_import_action */ 22:
                    message.needsCalcEngineDeferredImportAction = reader.bool();
                    break;
                case /* optional TSP.Reference owned_preset */ 23:
                    message.ownedPreset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.ownedPreset);
                    break;
                case /* optional bool is_dirty */ 24:
                    message.isDirty = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSCH.ChartType chart_type = 1; */
        if (message.chartType !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.chartType);
        /* optional TSCH.ScatterFormat scatter_format = 2; */
        if (message.scatterFormat !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.scatterFormat);
        /* optional TSCH.RectArchive legend_frame = 3; */
        if (message.legendFrame)
            TSCHArchives_Common_4.RectArchive.internalBinaryWrite(message.legendFrame, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference preset = 4; */
        if (message.preset)
            TSPMessages_3.Reference.internalBinaryWrite(message.preset, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.SeriesDirection series_direction = 5; */
        if (message.seriesDirection !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.seriesDirection);
        /* optional bool contains_default_data = 6; */
        if (message.containsDefaultData !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.containsDefaultData);
        /* optional TSCH.ChartGridArchive grid = 7; */
        if (message.grid)
            exports.ChartGridArchive.internalBinaryWrite(message.grid, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference mediator = 8; */
        if (message.mediator)
            TSPMessages_3.Reference.internalBinaryWrite(message.mediator, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference chart_style = 9; */
        if (message.chartStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.chartStyle, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference chart_non_style = 10; */
        if (message.chartNonStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.chartNonStyle, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference legend_style = 11; */
        if (message.legendStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.legendStyle, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference legend_non_style = 12; */
        if (message.legendNonStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.legendNonStyle, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference value_axis_styles = 13; */
        for (let i = 0; i < message.valueAxisStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.valueAxisStyles[i], writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference value_axis_nonstyles = 14; */
        for (let i = 0; i < message.valueAxisNonstyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.valueAxisNonstyles[i], writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference category_axis_styles = 15; */
        for (let i = 0; i < message.categoryAxisStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.categoryAxisStyles[i], writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference category_axis_nonstyles = 16; */
        for (let i = 0; i < message.categoryAxisNonstyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.categoryAxisNonstyles[i], writer.tag(16, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference series_theme_styles = 17; */
        for (let i = 0; i < message.seriesThemeStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.seriesThemeStyles[i], writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray series_private_styles = 18; */
        if (message.seriesPrivateStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.seriesPrivateStyles, writer.tag(18, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray series_non_styles = 19; */
        if (message.seriesNonStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.seriesNonStyles, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference paragraph_styles = 20; */
        for (let i = 0; i < message.paragraphStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.paragraphStyles[i], writer.tag(20, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional uint32 multidataset_index = 21; */
        if (message.multidatasetIndex !== undefined)
            writer.tag(21, runtime_1.WireType.Varint).uint32(message.multidatasetIndex);
        /* optional bool needs_calc_engine_deferred_import_action = 22; */
        if (message.needsCalcEngineDeferredImportAction !== undefined)
            writer.tag(22, runtime_1.WireType.Varint).bool(message.needsCalcEngineDeferredImportAction);
        /* optional TSP.Reference owned_preset = 23; */
        if (message.ownedPreset)
            TSPMessages_3.Reference.internalBinaryWrite(message.ownedPreset, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool is_dirty = 24; */
        if (message.isDirty !== undefined)
            writer.tag(24, runtime_1.WireType.Varint).bool(message.isDirty);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartArchive
 */
exports.ChartArchive = new ChartArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartMultiDataIndexUpgrade$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartMultiDataIndexUpgrade", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartMultiDataIndexUpgrade
 */
exports.ChartMultiDataIndexUpgrade = new ChartMultiDataIndexUpgrade$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartGarlicMinMaxUpgrade$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartGarlicMinMaxUpgrade", [
            { no: 1, name: "horizontal_min", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 2, name: "horizontal_max", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 3, name: "value_min", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 4, name: "value_max", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional double horizontal_min */ 1:
                    message.horizontalMin = reader.double();
                    break;
                case /* optional double horizontal_max */ 2:
                    message.horizontalMax = reader.double();
                    break;
                case /* optional double value_min */ 3:
                    message.valueMin = reader.double();
                    break;
                case /* optional double value_max */ 4:
                    message.valueMax = reader.double();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional double horizontal_min = 1; */
        if (message.horizontalMin !== undefined)
            writer.tag(1, runtime_1.WireType.Bit64).double(message.horizontalMin);
        /* optional double horizontal_max = 2; */
        if (message.horizontalMax !== undefined)
            writer.tag(2, runtime_1.WireType.Bit64).double(message.horizontalMax);
        /* optional double value_min = 3; */
        if (message.valueMin !== undefined)
            writer.tag(3, runtime_1.WireType.Bit64).double(message.valueMin);
        /* optional double value_max = 4; */
        if (message.valueMax !== undefined)
            writer.tag(4, runtime_1.WireType.Bit64).double(message.valueMax);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartGarlicMinMaxUpgrade
 */
exports.ChartGarlicMinMaxUpgrade = new ChartGarlicMinMaxUpgrade$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartGarlicLabelFormatUpgrade$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartGarlicLabelFormatUpgrade", [
            { no: 1, name: "value_prefix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "value_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "value_use_separator", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "horizontal_prefix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "horizontal_suffix", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "horizontal_use_separator", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string value_prefix */ 1:
                    message.valuePrefix = reader.string();
                    break;
                case /* optional string value_suffix */ 2:
                    message.valueSuffix = reader.string();
                    break;
                case /* optional bool value_use_separator */ 3:
                    message.valueUseSeparator = reader.bool();
                    break;
                case /* optional string horizontal_prefix */ 4:
                    message.horizontalPrefix = reader.string();
                    break;
                case /* optional string horizontal_suffix */ 5:
                    message.horizontalSuffix = reader.string();
                    break;
                case /* optional bool horizontal_use_separator */ 6:
                    message.horizontalUseSeparator = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string value_prefix = 1; */
        if (message.valuePrefix !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.valuePrefix);
        /* optional string value_suffix = 2; */
        if (message.valueSuffix !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.valueSuffix);
        /* optional bool value_use_separator = 3; */
        if (message.valueUseSeparator !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.valueUseSeparator);
        /* optional string horizontal_prefix = 4; */
        if (message.horizontalPrefix !== undefined)
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.horizontalPrefix);
        /* optional string horizontal_suffix = 5; */
        if (message.horizontalSuffix !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.horizontalSuffix);
        /* optional bool horizontal_use_separator = 6; */
        if (message.horizontalUseSeparator !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.horizontalUseSeparator);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartGarlicLabelFormatUpgrade
 */
exports.ChartGarlicLabelFormatUpgrade = new ChartGarlicLabelFormatUpgrade$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartPasteboardAdditionsArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartPasteboardAdditionsArchive", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartPasteboardAdditionsArchive
 */
exports.ChartPasteboardAdditionsArchive = new ChartPasteboardAdditionsArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartPreserveAppearanceForPresetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartPreserveAppearanceForPresetArchive", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartPreserveAppearanceForPresetArchive
 */
exports.ChartPreserveAppearanceForPresetArchive = new ChartPreserveAppearanceForPresetArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartSupportsProportionalBendedCalloutLinesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartSupportsProportionalBendedCalloutLinesArchive", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartSupportsProportionalBendedCalloutLinesArchive
 */
exports.ChartSupportsProportionalBendedCalloutLinesArchive = new ChartSupportsProportionalBendedCalloutLinesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartSupportsRoundedCornersArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartSupportsRoundedCornersArchive", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartSupportsRoundedCornersArchive
 */
exports.ChartSupportsRoundedCornersArchive = new ChartSupportsRoundedCornersArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartSupportsSeriesPropertySpacingArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartSupportsSeriesPropertySpacingArchive", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartSupportsSeriesPropertySpacingArchive
 */
exports.ChartSupportsSeriesPropertySpacingArchive = new ChartSupportsSeriesPropertySpacingArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartGridArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartGridArchive", [
            { no: 1, name: "row_name", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "column_name", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "grid_row", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSCHArchives_Common_2.GridRow },
            { no: 4, name: "idMap", kind: "message", T: () => exports.ChartGridArchive_ChartGridRowColumnIdMap }
        ]);
    }
    create(value) {
        const message = { rowName: [], columnName: [], gridRow: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated string row_name */ 1:
                    message.rowName.push(reader.string());
                    break;
                case /* repeated string column_name */ 2:
                    message.columnName.push(reader.string());
                    break;
                case /* repeated TSCH.GridRow grid_row */ 3:
                    message.gridRow.push(TSCHArchives_Common_2.GridRow.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional TSCH.ChartGridArchive.ChartGridRowColumnIdMap idMap */ 4:
                    message.idMap = exports.ChartGridArchive_ChartGridRowColumnIdMap.internalBinaryRead(reader, reader.uint32(), options, message.idMap);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated string row_name = 1; */
        for (let i = 0; i < message.rowName.length; i++)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.rowName[i]);
        /* repeated string column_name = 2; */
        for (let i = 0; i < message.columnName.length; i++)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.columnName[i]);
        /* repeated TSCH.GridRow grid_row = 3; */
        for (let i = 0; i < message.gridRow.length; i++)
            TSCHArchives_Common_2.GridRow.internalBinaryWrite(message.gridRow[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.ChartGridArchive.ChartGridRowColumnIdMap idMap = 4; */
        if (message.idMap)
            exports.ChartGridArchive_ChartGridRowColumnIdMap.internalBinaryWrite(message.idMap, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartGridArchive
 */
exports.ChartGridArchive = new ChartGridArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartGridArchive_ChartGridRowColumnIdMap$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartGridArchive.ChartGridRowColumnIdMap", [
            { no: 1, name: "row_id_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry },
            { no: 2, name: "column_id_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry }
        ]);
    }
    create(value) {
        const message = { rowIdMap: [], columnIdMap: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSCH.ChartGridArchive.ChartGridRowColumnIdMap.Entry row_id_map */ 1:
                    message.rowIdMap.push(exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSCH.ChartGridArchive.ChartGridRowColumnIdMap.Entry column_id_map */ 2:
                    message.columnIdMap.push(exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSCH.ChartGridArchive.ChartGridRowColumnIdMap.Entry row_id_map = 1; */
        for (let i = 0; i < message.rowIdMap.length; i++)
            exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry.internalBinaryWrite(message.rowIdMap[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.ChartGridArchive.ChartGridRowColumnIdMap.Entry column_id_map = 2; */
        for (let i = 0; i < message.columnIdMap.length; i++)
            exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry.internalBinaryWrite(message.columnIdMap[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartGridArchive.ChartGridRowColumnIdMap
 */
exports.ChartGridArchive_ChartGridRowColumnIdMap = new ChartGridArchive_ChartGridRowColumnIdMap$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartGridArchive_ChartGridRowColumnIdMap_Entry$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartGridArchive.ChartGridRowColumnIdMap.Entry", [
            { no: 1, name: "uniqueId", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "index", kind: "scalar", T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = { uniqueId: "", index: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string uniqueId */ 1:
                    message.uniqueId = reader.string();
                    break;
                case /* uint32 index */ 2:
                    message.index = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string uniqueId = 1; */
        if (message.uniqueId !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.uniqueId);
        /* uint32 index = 2; */
        if (message.index !== 0)
            writer.tag(2, runtime_1.WireType.Varint).uint32(message.index);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartGridArchive.ChartGridRowColumnIdMap.Entry
 */
exports.ChartGridArchive_ChartGridRowColumnIdMap_Entry = new ChartGridArchive_ChartGridRowColumnIdMap_Entry$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartMediatorArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartMediatorArchive", [
            { no: 1, name: "info", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "local_series_indexes", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 13 /*ScalarType.UINT32*/ },
            { no: 3, name: "remote_series_indexes", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = { localSeriesIndexes: [], remoteSeriesIndexes: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference info */ 1:
                    message.info = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.info);
                    break;
                case /* repeated uint32 local_series_indexes */ 2:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.localSeriesIndexes.push(reader.uint32());
                    else
                        message.localSeriesIndexes.push(reader.uint32());
                    break;
                case /* repeated uint32 remote_series_indexes */ 3:
                    if (wireType === runtime_1.WireType.LengthDelimited)
                        for (let e = reader.int32() + reader.pos; reader.pos < e;)
                            message.remoteSeriesIndexes.push(reader.uint32());
                    else
                        message.remoteSeriesIndexes.push(reader.uint32());
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference info = 1; */
        if (message.info)
            TSPMessages_3.Reference.internalBinaryWrite(message.info, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated uint32 local_series_indexes = 2; */
        for (let i = 0; i < message.localSeriesIndexes.length; i++)
            writer.tag(2, runtime_1.WireType.Varint).uint32(message.localSeriesIndexes[i]);
        /* repeated uint32 remote_series_indexes = 3; */
        for (let i = 0; i < message.remoteSeriesIndexes.length; i++)
            writer.tag(3, runtime_1.WireType.Varint).uint32(message.remoteSeriesIndexes[i]);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartMediatorArchive
 */
exports.ChartMediatorArchive = new ChartMediatorArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartFillSetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartFillSetArchive", [
            { no: 1, name: "identifier", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "lookup_string", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "number_of_theme_series_styles", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ },
            { no: 4, name: "series_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { identifier: "", lookupString: "", seriesStyles: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string identifier */ 1:
                    message.identifier = reader.string();
                    break;
                case /* string lookup_string */ 2:
                    message.lookupString = reader.string();
                    break;
                case /* optional uint32 number_of_theme_series_styles */ 3:
                    message.numberOfThemeSeriesStyles = reader.uint32();
                    break;
                case /* repeated TSP.Reference series_styles */ 4:
                    message.seriesStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string identifier = 1; */
        if (message.identifier !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.identifier);
        /* string lookup_string = 2; */
        if (message.lookupString !== "")
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.lookupString);
        /* optional uint32 number_of_theme_series_styles = 3; */
        if (message.numberOfThemeSeriesStyles !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).uint32(message.numberOfThemeSeriesStyles);
        /* repeated TSP.Reference series_styles = 4; */
        for (let i = 0; i < message.seriesStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.seriesStyles[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartFillSetArchive
 */
exports.ChartFillSetArchive = new ChartFillSetArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartStylePreset$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartStylePreset", [
            { no: 1, name: "chart_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "legend_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "value_axis_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 4, name: "category_axis_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 5, name: "series_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 6, name: "paragraph_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 7, name: "uuid", kind: "scalar", opt: true, T: 12 /*ScalarType.BYTES*/ }
        ]);
    }
    create(value) {
        const message = { valueAxisStyles: [], categoryAxisStyles: [], seriesStyles: [], paragraphStyles: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference chart_style */ 1:
                    message.chartStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chartStyle);
                    break;
                case /* optional TSP.Reference legend_style */ 2:
                    message.legendStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.legendStyle);
                    break;
                case /* repeated TSP.Reference value_axis_styles */ 3:
                    message.valueAxisStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference category_axis_styles */ 4:
                    message.categoryAxisStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference series_styles */ 5:
                    message.seriesStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference paragraph_styles */ 6:
                    message.paragraphStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bytes uuid */ 7:
                    message.uuid = reader.bytes();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference chart_style = 1; */
        if (message.chartStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.chartStyle, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference legend_style = 2; */
        if (message.legendStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.legendStyle, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference value_axis_styles = 3; */
        for (let i = 0; i < message.valueAxisStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.valueAxisStyles[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference category_axis_styles = 4; */
        for (let i = 0; i < message.categoryAxisStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.categoryAxisStyles[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference series_styles = 5; */
        for (let i = 0; i < message.seriesStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.seriesStyles[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference paragraph_styles = 6; */
        for (let i = 0; i < message.paragraphStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.paragraphStyles[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bytes uuid = 7; */
        if (message.uuid !== undefined)
            writer.tag(7, runtime_1.WireType.LengthDelimited).bytes(message.uuid);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartStylePreset
 */
exports.ChartStylePreset = new ChartStylePreset$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartPresetsArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartPresetsArchive", [
            { no: 1, name: "chart_presets", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { chartPresets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSP.Reference chart_presets */ 1:
                    message.chartPresets.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSP.Reference chart_presets = 1; */
        for (let i = 0; i < message.chartPresets.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.chartPresets[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartPresetsArchive
 */
exports.ChartPresetsArchive = new ChartPresetsArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PropertyValueStorageContainerArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.PropertyValueStorageContainerArchive", [
            { no: 1, name: "chart_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "chart_nonstyle", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "legend_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "legend_nonstyle", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 5, name: "value_axis_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 6, name: "value_axis_nonstyles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 7, name: "category_axis_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 8, name: "category_axis_nonstyles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 9, name: "series_theme_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 10, name: "series_private_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 11, name: "series_nonstyles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray },
            { no: 12, name: "paragraph_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference chart_style */ 1:
                    message.chartStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chartStyle);
                    break;
                case /* optional TSP.Reference chart_nonstyle */ 2:
                    message.chartNonstyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chartNonstyle);
                    break;
                case /* optional TSP.Reference legend_style */ 3:
                    message.legendStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.legendStyle);
                    break;
                case /* optional TSP.Reference legend_nonstyle */ 4:
                    message.legendNonstyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.legendNonstyle);
                    break;
                case /* optional TSP.SparseReferenceArray value_axis_styles */ 5:
                    message.valueAxisStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.valueAxisStyles);
                    break;
                case /* optional TSP.SparseReferenceArray value_axis_nonstyles */ 6:
                    message.valueAxisNonstyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.valueAxisNonstyles);
                    break;
                case /* optional TSP.SparseReferenceArray category_axis_styles */ 7:
                    message.categoryAxisStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.categoryAxisStyles);
                    break;
                case /* optional TSP.SparseReferenceArray category_axis_nonstyles */ 8:
                    message.categoryAxisNonstyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.categoryAxisNonstyles);
                    break;
                case /* optional TSP.SparseReferenceArray series_theme_styles */ 9:
                    message.seriesThemeStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.seriesThemeStyles);
                    break;
                case /* optional TSP.SparseReferenceArray series_private_styles */ 10:
                    message.seriesPrivateStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.seriesPrivateStyles);
                    break;
                case /* optional TSP.SparseReferenceArray series_nonstyles */ 11:
                    message.seriesNonstyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.seriesNonstyles);
                    break;
                case /* optional TSP.SparseReferenceArray paragraph_styles */ 12:
                    message.paragraphStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.paragraphStyles);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference chart_style = 1; */
        if (message.chartStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.chartStyle, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference chart_nonstyle = 2; */
        if (message.chartNonstyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.chartNonstyle, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference legend_style = 3; */
        if (message.legendStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.legendStyle, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference legend_nonstyle = 4; */
        if (message.legendNonstyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.legendNonstyle, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray value_axis_styles = 5; */
        if (message.valueAxisStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.valueAxisStyles, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray value_axis_nonstyles = 6; */
        if (message.valueAxisNonstyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.valueAxisNonstyles, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray category_axis_styles = 7; */
        if (message.categoryAxisStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.categoryAxisStyles, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray category_axis_nonstyles = 8; */
        if (message.categoryAxisNonstyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.categoryAxisNonstyles, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray series_theme_styles = 9; */
        if (message.seriesThemeStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.seriesThemeStyles, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray series_private_styles = 10; */
        if (message.seriesPrivateStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.seriesPrivateStyles, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray series_nonstyles = 11; */
        if (message.seriesNonstyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.seriesNonstyles, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray paragraph_styles = 12; */
        if (message.paragraphStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.paragraphStyles, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.PropertyValueStorageContainerArchive
 */
exports.PropertyValueStorageContainerArchive = new PropertyValueStorageContainerArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StylePasteboardDataArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.StylePasteboardDataArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSSArchives_1.StyleArchive },
            { no: 2, name: "style_network", kind: "message", T: () => exports.PropertyValueStorageContainerArchive },
            { no: 3, name: "copied_from_entire_chart", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSS.StyleArchive super */ 1:
                    message.super = TSSArchives_1.StyleArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSCH.PropertyValueStorageContainerArchive style_network */ 2:
                    message.styleNetwork = exports.PropertyValueStorageContainerArchive.internalBinaryRead(reader, reader.uint32(), options, message.styleNetwork);
                    break;
                case /* optional bool copied_from_entire_chart */ 3:
                    message.copiedFromEntireChart = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSS.StyleArchive super = 1; */
        if (message.super)
            TSSArchives_1.StyleArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.PropertyValueStorageContainerArchive style_network = 2; */
        if (message.styleNetwork)
            exports.PropertyValueStorageContainerArchive.internalBinaryWrite(message.styleNetwork, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool copied_from_entire_chart = 3; */
        if (message.copiedFromEntireChart !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.copiedFromEntireChart);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.StylePasteboardDataArchive
 */
exports.StylePasteboardDataArchive = new StylePasteboardDataArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartSelectionPathTypeArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartSelectionPathTypeArchive", [
            { no: 1, name: "path_type", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "path_name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string path_type */ 1:
                    message.pathType = reader.string();
                    break;
                case /* optional string path_name */ 2:
                    message.pathName = reader.string();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string path_type = 1; */
        if (message.pathType !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.pathType);
        /* optional string path_name = 2; */
        if (message.pathName !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.pathName);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartSelectionPathTypeArchive
 */
exports.ChartSelectionPathTypeArchive = new ChartSelectionPathTypeArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartAxisIDArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartAxisIDArchive", [
            { no: 1, name: "axis_type", kind: "enum", opt: true, T: () => ["TSCH.AxisType", TSCHArchives_Common_1.AxisType] },
            { no: 2, name: "ordinal", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSCH.AxisType axis_type */ 1:
                    message.axisType = reader.int32();
                    break;
                case /* optional uint32 ordinal */ 2:
                    message.ordinal = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSCH.AxisType axis_type = 1; */
        if (message.axisType !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.axisType);
        /* optional uint32 ordinal = 2; */
        if (message.ordinal !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).uint32(message.ordinal);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartAxisIDArchive
 */
exports.ChartAxisIDArchive = new ChartAxisIDArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartSelectionPathArgumentArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartSelectionPathArgumentArchive", [
            { no: 1, name: "number", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ },
            { no: 2, name: "axis_id", kind: "message", T: () => exports.ChartAxisIDArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional uint32 number */ 1:
                    message.number = reader.uint32();
                    break;
                case /* optional TSCH.ChartAxisIDArchive axis_id */ 2:
                    message.axisId = exports.ChartAxisIDArchive.internalBinaryRead(reader, reader.uint32(), options, message.axisId);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional uint32 number = 1; */
        if (message.number !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).uint32(message.number);
        /* optional TSCH.ChartAxisIDArchive axis_id = 2; */
        if (message.axisId)
            exports.ChartAxisIDArchive.internalBinaryWrite(message.axisId, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartSelectionPathArgumentArchive
 */
exports.ChartSelectionPathArgumentArchive = new ChartSelectionPathArgumentArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartSelectionPathArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartSelectionPathArchive", [
            { no: 1, name: "path_type", kind: "message", T: () => exports.ChartSelectionPathTypeArchive },
            { no: 2, name: "sub_selection", kind: "message", T: () => exports.ChartSelectionPathArchive },
            { no: 3, name: "arguments", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartSelectionPathArgumentArchive }
        ]);
    }
    create(value) {
        const message = { arguments: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSCH.ChartSelectionPathTypeArchive path_type */ 1:
                    message.pathType = exports.ChartSelectionPathTypeArchive.internalBinaryRead(reader, reader.uint32(), options, message.pathType);
                    break;
                case /* optional TSCH.ChartSelectionPathArchive sub_selection */ 2:
                    message.subSelection = exports.ChartSelectionPathArchive.internalBinaryRead(reader, reader.uint32(), options, message.subSelection);
                    break;
                case /* repeated TSCH.ChartSelectionPathArgumentArchive arguments */ 3:
                    message.arguments.push(exports.ChartSelectionPathArgumentArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSCH.ChartSelectionPathTypeArchive path_type = 1; */
        if (message.pathType)
            exports.ChartSelectionPathTypeArchive.internalBinaryWrite(message.pathType, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.ChartSelectionPathArchive sub_selection = 2; */
        if (message.subSelection)
            exports.ChartSelectionPathArchive.internalBinaryWrite(message.subSelection, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.ChartSelectionPathArgumentArchive arguments = 3; */
        for (let i = 0; i < message.arguments.length; i++)
            exports.ChartSelectionPathArgumentArchive.internalBinaryWrite(message.arguments[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartSelectionPathArchive
 */
exports.ChartSelectionPathArchive = new ChartSelectionPathArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartSelectionArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartSelectionArchive", [
            { no: 3, name: "super", kind: "message", T: () => TSDArchives_1.DrawableSelectionArchive },
            { no: 1, name: "chart", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "paths", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartSelectionPathArchive }
        ]);
    }
    create(value) {
        const message = { paths: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSD.DrawableSelectionArchive super */ 3:
                    message.super = TSDArchives_1.DrawableSelectionArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.Reference chart */ 1:
                    message.chart = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chart);
                    break;
                case /* repeated TSCH.ChartSelectionPathArchive paths */ 2:
                    message.paths.push(exports.ChartSelectionPathArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSD.DrawableSelectionArchive super = 3; */
        if (message.super)
            TSDArchives_1.DrawableSelectionArchive.internalBinaryWrite(message.super, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference chart = 1; */
        if (message.chart)
            TSPMessages_3.Reference.internalBinaryWrite(message.chart, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.ChartSelectionPathArchive paths = 2; */
        for (let i = 0; i < message.paths.length; i++)
            exports.ChartSelectionPathArchive.internalBinaryWrite(message.paths[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartSelectionArchive
 */
exports.ChartSelectionArchive = new ChartSelectionArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartCDESelectionArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartCDESelectionArchive", [
            { no: 1, name: "chart", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "selection_type", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "row_location", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ },
            { no: 4, name: "row_length", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ },
            { no: 5, name: "column_location", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ },
            { no: 6, name: "column_length", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference chart */ 1:
                    message.chart = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chart);
                    break;
                case /* optional int32 selection_type */ 2:
                    message.selectionType = reader.int32();
                    break;
                case /* optional uint32 row_location */ 3:
                    message.rowLocation = reader.uint32();
                    break;
                case /* optional uint32 row_length */ 4:
                    message.rowLength = reader.uint32();
                    break;
                case /* optional uint32 column_location */ 5:
                    message.columnLocation = reader.uint32();
                    break;
                case /* optional uint32 column_length */ 6:
                    message.columnLength = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference chart = 1; */
        if (message.chart)
            TSPMessages_3.Reference.internalBinaryWrite(message.chart, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int32 selection_type = 2; */
        if (message.selectionType !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.selectionType);
        /* optional uint32 row_location = 3; */
        if (message.rowLocation !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).uint32(message.rowLocation);
        /* optional uint32 row_length = 4; */
        if (message.rowLength !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).uint32(message.rowLength);
        /* optional uint32 column_location = 5; */
        if (message.columnLocation !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).uint32(message.columnLocation);
        /* optional uint32 column_length = 6; */
        if (message.columnLength !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).uint32(message.columnLength);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartCDESelectionArchive
 */
exports.ChartCDESelectionArchive = new ChartCDESelectionArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartUIState$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartUIState", [
            { no: 1, name: "chart", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "cde_last_row_selected", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "cde_last_col_selected", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 4, name: "cde_last_row_count", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 5, name: "cde_last_col_count", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 6, name: "cde_use_full_keyboard", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "multi_data_set_index", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 8, name: "is_valid_multi_data_set_index", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference chart */ 1:
                    message.chart = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.chart);
                    break;
                case /* optional int32 cde_last_row_selected */ 2:
                    message.cdeLastRowSelected = reader.int32();
                    break;
                case /* optional int32 cde_last_col_selected */ 3:
                    message.cdeLastColSelected = reader.int32();
                    break;
                case /* optional int32 cde_last_row_count */ 4:
                    message.cdeLastRowCount = reader.int32();
                    break;
                case /* optional int32 cde_last_col_count */ 5:
                    message.cdeLastColCount = reader.int32();
                    break;
                case /* optional bool cde_use_full_keyboard */ 6:
                    message.cdeUseFullKeyboard = reader.bool();
                    break;
                case /* optional int32 multi_data_set_index */ 7:
                    message.multiDataSetIndex = reader.int32();
                    break;
                case /* optional bool is_valid_multi_data_set_index */ 8:
                    message.isValidMultiDataSetIndex = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference chart = 1; */
        if (message.chart)
            TSPMessages_3.Reference.internalBinaryWrite(message.chart, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int32 cde_last_row_selected = 2; */
        if (message.cdeLastRowSelected !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.cdeLastRowSelected);
        /* optional int32 cde_last_col_selected = 3; */
        if (message.cdeLastColSelected !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.cdeLastColSelected);
        /* optional int32 cde_last_row_count = 4; */
        if (message.cdeLastRowCount !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.cdeLastRowCount);
        /* optional int32 cde_last_col_count = 5; */
        if (message.cdeLastColCount !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.cdeLastColCount);
        /* optional bool cde_use_full_keyboard = 6; */
        if (message.cdeUseFullKeyboard !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.cdeUseFullKeyboard);
        /* optional int32 multi_data_set_index = 7; */
        if (message.multiDataSetIndex !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).int32(message.multiDataSetIndex);
        /* optional bool is_valid_multi_data_set_index = 8; */
        if (message.isValidMultiDataSetIndex !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).bool(message.isValidMultiDataSetIndex);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartUIState
 */
exports.ChartUIState = new ChartUIState$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartUIStateMultiDataIndexUpgrade$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartUIStateMultiDataIndexUpgrade", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartUIStateMultiDataIndexUpgrade
 */
exports.ChartUIStateMultiDataIndexUpgrade = new ChartUIStateMultiDataIndexUpgrade$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartFormatStructExtensions$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartFormatStructExtensions", []);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        return target ?? this.create();
    }
    internalBinaryWrite(message, writer, options) {
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartFormatStructExtensions
 */
exports.ChartFormatStructExtensions = new ChartFormatStructExtensions$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartReferenceLineNonStyleItem$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartReferenceLineNonStyleItem", [
            { no: 1, name: "non_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "uuid", kind: "message", T: () => TSPMessages_1.UUID }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSP.Reference non_style */ 1:
                    message.nonStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.nonStyle);
                    break;
                case /* TSP.UUID uuid */ 2:
                    message.uuid = TSPMessages_1.UUID.internalBinaryRead(reader, reader.uint32(), options, message.uuid);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSP.Reference non_style = 1; */
        if (message.nonStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.nonStyle, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.UUID uuid = 2; */
        if (message.uuid)
            TSPMessages_1.UUID.internalBinaryWrite(message.uuid, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartReferenceLineNonStyleItem
 */
exports.ChartReferenceLineNonStyleItem = new ChartReferenceLineNonStyleItem$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartAxisReferenceLineNonStylesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartAxisReferenceLineNonStylesArchive", [
            { no: 1, name: "axis_id", kind: "message", T: () => exports.ChartAxisIDArchive },
            { no: 2, name: "reference_line_non_style_items", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartReferenceLineNonStyleItem }
        ]);
    }
    create(value) {
        const message = { referenceLineNonStyleItems: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.ChartAxisIDArchive axis_id */ 1:
                    message.axisId = exports.ChartAxisIDArchive.internalBinaryRead(reader, reader.uint32(), options, message.axisId);
                    break;
                case /* repeated TSCH.ChartReferenceLineNonStyleItem reference_line_non_style_items */ 2:
                    message.referenceLineNonStyleItems.push(exports.ChartReferenceLineNonStyleItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.ChartAxisIDArchive axis_id = 1; */
        if (message.axisId)
            exports.ChartAxisIDArchive.internalBinaryWrite(message.axisId, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.ChartReferenceLineNonStyleItem reference_line_non_style_items = 2; */
        for (let i = 0; i < message.referenceLineNonStyleItems.length; i++)
            exports.ChartReferenceLineNonStyleItem.internalBinaryWrite(message.referenceLineNonStyleItems[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartAxisReferenceLineNonStylesArchive
 */
exports.ChartAxisReferenceLineNonStylesArchive = new ChartAxisReferenceLineNonStylesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartAxisReferenceLineStylesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartAxisReferenceLineStylesArchive", [
            { no: 1, name: "axis_id", kind: "message", T: () => exports.ChartAxisIDArchive },
            { no: 2, name: "reference_line_styles", kind: "message", T: () => TSPMessages_2.SparseReferenceArray }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.ChartAxisIDArchive axis_id */ 1:
                    message.axisId = exports.ChartAxisIDArchive.internalBinaryRead(reader, reader.uint32(), options, message.axisId);
                    break;
                case /* optional TSP.SparseReferenceArray reference_line_styles */ 2:
                    message.referenceLineStyles = TSPMessages_2.SparseReferenceArray.internalBinaryRead(reader, reader.uint32(), options, message.referenceLineStyles);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.ChartAxisIDArchive axis_id = 1; */
        if (message.axisId)
            exports.ChartAxisIDArchive.internalBinaryWrite(message.axisId, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.SparseReferenceArray reference_line_styles = 2; */
        if (message.referenceLineStyles)
            TSPMessages_2.SparseReferenceArray.internalBinaryWrite(message.referenceLineStyles, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartAxisReferenceLineStylesArchive
 */
exports.ChartAxisReferenceLineStylesArchive = new ChartAxisReferenceLineStylesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartReferenceLinesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartReferenceLinesArchive", [
            { no: 1, name: "reference_line_non_styles_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartAxisReferenceLineNonStylesArchive },
            { no: 2, name: "reference_line_styles_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartAxisReferenceLineStylesArchive },
            { no: 3, name: "theme_preset_reference_line_style", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { referenceLineNonStylesMap: [], referenceLineStylesMap: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSCH.ChartAxisReferenceLineNonStylesArchive reference_line_non_styles_map */ 1:
                    message.referenceLineNonStylesMap.push(exports.ChartAxisReferenceLineNonStylesArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSCH.ChartAxisReferenceLineStylesArchive reference_line_styles_map */ 2:
                    message.referenceLineStylesMap.push(exports.ChartAxisReferenceLineStylesArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional TSP.Reference theme_preset_reference_line_style */ 3:
                    message.themePresetReferenceLineStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.themePresetReferenceLineStyle);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSCH.ChartAxisReferenceLineNonStylesArchive reference_line_non_styles_map = 1; */
        for (let i = 0; i < message.referenceLineNonStylesMap.length; i++)
            exports.ChartAxisReferenceLineNonStylesArchive.internalBinaryWrite(message.referenceLineNonStylesMap[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.ChartAxisReferenceLineStylesArchive reference_line_styles_map = 2; */
        for (let i = 0; i < message.referenceLineStylesMap.length; i++)
            exports.ChartAxisReferenceLineStylesArchive.internalBinaryWrite(message.referenceLineStylesMap[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference theme_preset_reference_line_style = 3; */
        if (message.themePresetReferenceLineStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.themePresetReferenceLineStyle, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartReferenceLinesArchive
 */
exports.ChartReferenceLinesArchive = new ChartReferenceLinesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartPresetReferenceLineStylesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartPresetReferenceLineStylesArchive", [
            { no: 1, name: "reference_line_styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { referenceLineStyles: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSP.Reference reference_line_styles */ 1:
                    message.referenceLineStyles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSP.Reference reference_line_styles = 1; */
        for (let i = 0; i < message.referenceLineStyles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.referenceLineStyles[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartPresetReferenceLineStylesArchive
 */
exports.ChartPresetReferenceLineStylesArchive = new ChartPresetReferenceLineStylesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ChartAxisReferenceLineSparseNonStylesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.ChartAxisReferenceLineSparseNonStylesArchive", [
            { no: 1, name: "axis_id", kind: "message", T: () => exports.ChartAxisIDArchive },
            { no: 2, name: "reference_line_non_style_items", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartReferenceLineNonStyleItem }
        ]);
    }
    create(value) {
        const message = { referenceLineNonStyleItems: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.ChartAxisIDArchive axis_id */ 1:
                    message.axisId = exports.ChartAxisIDArchive.internalBinaryRead(reader, reader.uint32(), options, message.axisId);
                    break;
                case /* repeated TSCH.ChartReferenceLineNonStyleItem reference_line_non_style_items */ 2:
                    message.referenceLineNonStyleItems.push(exports.ChartReferenceLineNonStyleItem.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.ChartAxisIDArchive axis_id = 1; */
        if (message.axisId)
            exports.ChartAxisIDArchive.internalBinaryWrite(message.axisId, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.ChartReferenceLineNonStyleItem reference_line_non_style_items = 2; */
        for (let i = 0; i < message.referenceLineNonStyleItems.length; i++)
            exports.ChartReferenceLineNonStyleItem.internalBinaryWrite(message.referenceLineNonStyleItems[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.ChartAxisReferenceLineSparseNonStylesArchive
 */
exports.ChartAxisReferenceLineSparseNonStylesArchive = new ChartAxisReferenceLineSparseNonStylesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PropertyValueStorageContainerReferenceLinesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.PropertyValueStorageContainerReferenceLinesArchive", [
            { no: 1, name: "reference_line_non_styles_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartAxisReferenceLineSparseNonStylesArchive },
            { no: 2, name: "reference_line_styles_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.ChartAxisReferenceLineStylesArchive },
            { no: 3, name: "theme_preset_reference_line_style", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { referenceLineNonStylesMap: [], referenceLineStylesMap: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSCH.ChartAxisReferenceLineSparseNonStylesArchive reference_line_non_styles_map */ 1:
                    message.referenceLineNonStylesMap.push(exports.ChartAxisReferenceLineSparseNonStylesArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSCH.ChartAxisReferenceLineStylesArchive reference_line_styles_map */ 2:
                    message.referenceLineStylesMap.push(exports.ChartAxisReferenceLineStylesArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional TSP.Reference theme_preset_reference_line_style */ 3:
                    message.themePresetReferenceLineStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.themePresetReferenceLineStyle);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSCH.ChartAxisReferenceLineSparseNonStylesArchive reference_line_non_styles_map = 1; */
        for (let i = 0; i < message.referenceLineNonStylesMap.length; i++)
            exports.ChartAxisReferenceLineSparseNonStylesArchive.internalBinaryWrite(message.referenceLineNonStylesMap[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.ChartAxisReferenceLineStylesArchive reference_line_styles_map = 2; */
        for (let i = 0; i < message.referenceLineStylesMap.length; i++)
            exports.ChartAxisReferenceLineStylesArchive.internalBinaryWrite(message.referenceLineStylesMap[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference theme_preset_reference_line_style = 3; */
        if (message.themePresetReferenceLineStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.themePresetReferenceLineStyle, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.PropertyValueStorageContainerReferenceLinesArchive
 */
exports.PropertyValueStorageContainerReferenceLinesArchive = new PropertyValueStorageContainerReferenceLinesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CollaboratorCDECursorSubselectionArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.CollaboratorCDECursorSubselectionArchive", [
            { no: 1, name: "row_location", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 2, name: "row_length", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "column_location", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 4, name: "column_length", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int32 row_location */ 1:
                    message.rowLocation = reader.int32();
                    break;
                case /* optional int32 row_length */ 2:
                    message.rowLength = reader.int32();
                    break;
                case /* optional int32 column_location */ 3:
                    message.columnLocation = reader.int32();
                    break;
                case /* optional int32 column_length */ 4:
                    message.columnLength = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int32 row_location = 1; */
        if (message.rowLocation !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.rowLocation);
        /* optional int32 row_length = 2; */
        if (message.rowLength !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.rowLength);
        /* optional int32 column_location = 3; */
        if (message.columnLocation !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.columnLocation);
        /* optional int32 column_length = 4; */
        if (message.columnLength !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.columnLength);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.CollaboratorCDECursorSubselectionArchive
 */
exports.CollaboratorCDECursorSubselectionArchive = new CollaboratorCDECursorSubselectionArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CollaboratorChartTitleCursorSubselectionArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.CollaboratorChartTitleCursorSubselectionArchive", [
            { no: 1, name: "chart_title_selected", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional bool chart_title_selected */ 1:
                    message.chartTitleSelected = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional bool chart_title_selected = 1; */
        if (message.chartTitleSelected !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).bool(message.chartTitleSelected);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.CollaboratorChartTitleCursorSubselectionArchive
 */
exports.CollaboratorChartTitleCursorSubselectionArchive = new CollaboratorChartTitleCursorSubselectionArchive$Type();
