// keynote-archives/lib/generated/TSCH3DArchives.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chart3DVectorArchive = exports.Chart3DImageTextureTilingArchive = exports.Chart3DBaseImageTextureTilingArchive = exports.Chart3DTSPImageDataTextureArchive = exports.Chart3DPhongMaterialPackageArchive = exports.Chart3DPhongLightingModelArchive = exports.Chart3DFixedFunctionLightingModelArchive = exports.Chart3DEnvironmentMaterialArchive = exports.Chart3DShininessMaterialArchive = exports.Chart3DSpecularMaterialArchive = exports.Chart3DModulateMaterialArchive = exports.Chart3DDiffuseMaterialArchive = exports.Chart3DEmissiveMaterialArchive = exports.Chart3DTexturesMaterialArchive = exports.Chart3DLightingPackageArchive = exports.Chart3DLightingModelArchive = exports.Chart3DLightArchive = exports.Chart3DSpotLightArchive = exports.Chart3DDirectionalLightArchive = exports.Chart3DPointLightArchive = exports.Chart3DFillArchive = exports.Chart3DEnvironmentPackageArchive = exports.FillPropertyType = exports.TextureTilingContinuity = exports.TextureTilingYPosition = exports.TextureTilingXPosition = exports.TextureTilingWrap = exports.TextureTilingFace = exports.TextureTilingMode = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSPMessages_1 = require("./TSPMessages");
const TSPMessages_2 = require("./TSPMessages");
/**
 * @generated from protobuf enum TSCH.TextureTilingMode
 */
var TextureTilingMode;
(function (TextureTilingMode) {
    /**
     * @generated from protobuf enum value: textureTilingModeNone = 0;
     */
    TextureTilingMode[TextureTilingMode["textureTilingModeNone"] = 0] = "textureTilingModeNone";
    /**
     * @generated from protobuf enum value: textureTilingModeTallest = 1;
     */
    TextureTilingMode[TextureTilingMode["textureTilingModeTallest"] = 1] = "textureTilingModeTallest";
})(TextureTilingMode = exports.TextureTilingMode || (exports.TextureTilingMode = {}));
/**
 * @generated from protobuf enum TSCH.TextureTilingFace
 */
var TextureTilingFace;
(function (TextureTilingFace) {
    /**
     * @generated from protobuf enum value: textureTilingFaceAll = 0;
     */
    TextureTilingFace[TextureTilingFace["textureTilingFaceAll"] = 0] = "textureTilingFaceAll";
    /**
     * @generated from protobuf enum value: textureTilingFaceTopAndBottom = 1;
     */
    TextureTilingFace[TextureTilingFace["textureTilingFaceTopAndBottom"] = 1] = "textureTilingFaceTopAndBottom";
    /**
     * @generated from protobuf enum value: textureTilingFaceSide = 2;
     */
    TextureTilingFace[TextureTilingFace["textureTilingFaceSide"] = 2] = "textureTilingFaceSide";
})(TextureTilingFace = exports.TextureTilingFace || (exports.TextureTilingFace = {}));
/**
 * @generated from protobuf enum TSCH.TextureTilingWrap
 */
var TextureTilingWrap;
(function (TextureTilingWrap) {
    /**
     * @generated from protobuf enum value: textureTilingWrapProjected = 0;
     */
    TextureTilingWrap[TextureTilingWrap["textureTilingWrapProjected"] = 0] = "textureTilingWrapProjected";
    /**
     * @generated from protobuf enum value: textureTilingWrapFaceWrap = 1;
     */
    TextureTilingWrap[TextureTilingWrap["textureTilingWrapFaceWrap"] = 1] = "textureTilingWrapFaceWrap";
})(TextureTilingWrap = exports.TextureTilingWrap || (exports.TextureTilingWrap = {}));
/**
 * @generated from protobuf enum TSCH.TextureTilingXPosition
 */
var TextureTilingXPosition;
(function (TextureTilingXPosition) {
    /**
     * @generated from protobuf enum value: textureTilingXPositionLeft = 0;
     */
    TextureTilingXPosition[TextureTilingXPosition["textureTilingXPositionLeft"] = 0] = "textureTilingXPositionLeft";
    /**
     * @generated from protobuf enum value: textureTilingXPositionCenter = 1;
     */
    TextureTilingXPosition[TextureTilingXPosition["textureTilingXPositionCenter"] = 1] = "textureTilingXPositionCenter";
    /**
     * @generated from protobuf enum value: textureTilingXPositionRight = 2;
     */
    TextureTilingXPosition[TextureTilingXPosition["textureTilingXPositionRight"] = 2] = "textureTilingXPositionRight";
})(TextureTilingXPosition = exports.TextureTilingXPosition || (exports.TextureTilingXPosition = {}));
/**
 * @generated from protobuf enum TSCH.TextureTilingYPosition
 */
var TextureTilingYPosition;
(function (TextureTilingYPosition) {
    /**
     * @generated from protobuf enum value: textureTilingYPositionTop = 0;
     */
    TextureTilingYPosition[TextureTilingYPosition["textureTilingYPositionTop"] = 0] = "textureTilingYPositionTop";
    /**
     * @generated from protobuf enum value: textureTilingYPositionMiddle = 1;
     */
    TextureTilingYPosition[TextureTilingYPosition["textureTilingYPositionMiddle"] = 1] = "textureTilingYPositionMiddle";
    /**
     * @generated from protobuf enum value: textureTilingYPositionBottom = 2;
     */
    TextureTilingYPosition[TextureTilingYPosition["textureTilingYPositionBottom"] = 2] = "textureTilingYPositionBottom";
})(TextureTilingYPosition = exports.TextureTilingYPosition || (exports.TextureTilingYPosition = {}));
/**
 * @generated from protobuf enum TSCH.TextureTilingContinuity
 */
var TextureTilingContinuity;
(function (TextureTilingContinuity) {
    /**
     * @generated from protobuf enum value: textureTilingContinuityNone = 0;
     */
    TextureTilingContinuity[TextureTilingContinuity["textureTilingContinuityNone"] = 0] = "textureTilingContinuityNone";
    /**
     * @generated from protobuf enum value: textureTilingContinuityGlobal = 1;
     */
    TextureTilingContinuity[TextureTilingContinuity["textureTilingContinuityGlobal"] = 1] = "textureTilingContinuityGlobal";
    /**
     * @generated from protobuf enum value: textureTilingContinuitySeries = 2;
     */
    TextureTilingContinuity[TextureTilingContinuity["textureTilingContinuitySeries"] = 2] = "textureTilingContinuitySeries";
    /**
     * @generated from protobuf enum value: textureTilingContinuityJittered = 3;
     */
    TextureTilingContinuity[TextureTilingContinuity["textureTilingContinuityJittered"] = 3] = "textureTilingContinuityJittered";
})(TextureTilingContinuity = exports.TextureTilingContinuity || (exports.TextureTilingContinuity = {}));
/**
 * @generated from protobuf enum TSCH.FillPropertyType
 */
var FillPropertyType;
(function (FillPropertyType) {
    /**
     * @generated from protobuf enum value: fillPropertyTypeUndefined = 0;
     */
    FillPropertyType[FillPropertyType["fillPropertyTypeUndefined"] = 0] = "fillPropertyTypeUndefined";
    /**
     * @generated from protobuf enum value: fillPropertyTypeArea = 1;
     */
    FillPropertyType[FillPropertyType["fillPropertyTypeArea"] = 1] = "fillPropertyTypeArea";
    /**
     * @generated from protobuf enum value: fillPropertyTypeBar = 2;
     */
    FillPropertyType[FillPropertyType["fillPropertyTypeBar"] = 2] = "fillPropertyTypeBar";
    /**
     * @generated from protobuf enum value: fillPropertyTypeColumn = 3;
     */
    FillPropertyType[FillPropertyType["fillPropertyTypeColumn"] = 3] = "fillPropertyTypeColumn";
    /**
     * @generated from protobuf enum value: fillPropertyTypeLine = 4;
     */
    FillPropertyType[FillPropertyType["fillPropertyTypeLine"] = 4] = "fillPropertyTypeLine";
    /**
     * @generated from protobuf enum value: fillPropertyTypePie = 5;
     */
    FillPropertyType[FillPropertyType["fillPropertyTypePie"] = 5] = "fillPropertyTypePie";
})(FillPropertyType = exports.FillPropertyType || (exports.FillPropertyType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DEnvironmentPackageArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DEnvironmentPackageArchive", [
            { no: 1, name: "materials", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DEnvironmentMaterialArchive }
        ]);
    }
    create(value) {
        const message = { materials: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSCH.Chart3DEnvironmentMaterialArchive materials */ 1:
                    message.materials.push(exports.Chart3DEnvironmentMaterialArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSCH.Chart3DEnvironmentMaterialArchive materials = 1; */
        for (let i = 0; i < message.materials.length; i++)
            exports.Chart3DEnvironmentMaterialArchive.internalBinaryWrite(message.materials[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DEnvironmentPackageArchive
 */
exports.Chart3DEnvironmentPackageArchive = new Chart3DEnvironmentPackageArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DFillArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DFillArchive", [
            { no: 1, name: "lightingmodel", kind: "message", T: () => exports.Chart3DLightingModelArchive },
            { no: 2, name: "textureset_id", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "fill_type", kind: "enum", opt: true, T: () => ["TSCH.FillPropertyType", FillPropertyType] },
            { no: 4, name: "series_index", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSCH.Chart3DLightingModelArchive lightingmodel */ 1:
                    message.lightingmodel = exports.Chart3DLightingModelArchive.internalBinaryRead(reader, reader.uint32(), options, message.lightingmodel);
                    break;
                case /* optional string textureset_id */ 2:
                    message.texturesetId = reader.string();
                    break;
                case /* optional TSCH.FillPropertyType fill_type */ 3:
                    message.fillType = reader.int32();
                    break;
                case /* optional uint32 series_index */ 4:
                    message.seriesIndex = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSCH.Chart3DLightingModelArchive lightingmodel = 1; */
        if (message.lightingmodel)
            exports.Chart3DLightingModelArchive.internalBinaryWrite(message.lightingmodel, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string textureset_id = 2; */
        if (message.texturesetId !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.texturesetId);
        /* optional TSCH.FillPropertyType fill_type = 3; */
        if (message.fillType !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.fillType);
        /* optional uint32 series_index = 4; */
        if (message.seriesIndex !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).uint32(message.seriesIndex);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DFillArchive
 */
exports.Chart3DFillArchive = new Chart3DFillArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DPointLightArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DPointLightArchive", [
            { no: 1, name: "position", kind: "message", T: () => exports.Chart3DVectorArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DVectorArchive position */ 1:
                    message.position = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.position);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DVectorArchive position = 1; */
        if (message.position)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.position, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DPointLightArchive
 */
exports.Chart3DPointLightArchive = new Chart3DPointLightArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DDirectionalLightArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DDirectionalLightArchive", [
            { no: 1, name: "direction", kind: "message", T: () => exports.Chart3DVectorArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DVectorArchive direction */ 1:
                    message.direction = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.direction);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DVectorArchive direction = 1; */
        if (message.direction)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.direction, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DDirectionalLightArchive
 */
exports.Chart3DDirectionalLightArchive = new Chart3DDirectionalLightArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DSpotLightArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DSpotLightArchive", [
            { no: 1, name: "position", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 2, name: "direction", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 3, name: "cutoff", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ },
            { no: 4, name: "dropoff", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ }
        ]);
    }
    create(value) {
        const message = { cutoff: 0, dropoff: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DVectorArchive position */ 1:
                    message.position = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.position);
                    break;
                case /* TSCH.Chart3DVectorArchive direction */ 2:
                    message.direction = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.direction);
                    break;
                case /* float cutoff */ 3:
                    message.cutoff = reader.float();
                    break;
                case /* float dropoff */ 4:
                    message.dropoff = reader.float();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DVectorArchive position = 1; */
        if (message.position)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.position, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSCH.Chart3DVectorArchive direction = 2; */
        if (message.direction)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.direction, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* float cutoff = 3; */
        if (message.cutoff !== 0)
            writer.tag(3, runtime_1.WireType.Bit32).float(message.cutoff);
        /* float dropoff = 4; */
        if (message.dropoff !== 0)
            writer.tag(4, runtime_1.WireType.Bit32).float(message.dropoff);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DSpotLightArchive
 */
exports.Chart3DSpotLightArchive = new Chart3DSpotLightArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DLightArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DLightArchive", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "ambient_color", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 3, name: "diffuse_color", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 4, name: "specular_color", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 5, name: "intensity", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ },
            { no: 6, name: "attenuation", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 7, name: "coordinate_space", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 8, name: "enabled", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 9, name: "point_light", kind: "message", T: () => exports.Chart3DPointLightArchive },
            { no: 10, name: "directional_light", kind: "message", T: () => exports.Chart3DDirectionalLightArchive },
            { no: 11, name: "spot_light", kind: "message", T: () => exports.Chart3DSpotLightArchive }
        ]);
    }
    create(value) {
        const message = { name: "", intensity: 0, coordinateSpace: 0, enabled: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* TSCH.Chart3DVectorArchive ambient_color */ 2:
                    message.ambientColor = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.ambientColor);
                    break;
                case /* TSCH.Chart3DVectorArchive diffuse_color */ 3:
                    message.diffuseColor = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.diffuseColor);
                    break;
                case /* TSCH.Chart3DVectorArchive specular_color */ 4:
                    message.specularColor = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.specularColor);
                    break;
                case /* float intensity */ 5:
                    message.intensity = reader.float();
                    break;
                case /* TSCH.Chart3DVectorArchive attenuation */ 6:
                    message.attenuation = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.attenuation);
                    break;
                case /* uint32 coordinate_space */ 7:
                    message.coordinateSpace = reader.uint32();
                    break;
                case /* bool enabled */ 8:
                    message.enabled = reader.bool();
                    break;
                case /* optional TSCH.Chart3DPointLightArchive point_light */ 9:
                    message.pointLight = exports.Chart3DPointLightArchive.internalBinaryRead(reader, reader.uint32(), options, message.pointLight);
                    break;
                case /* optional TSCH.Chart3DDirectionalLightArchive directional_light */ 10:
                    message.directionalLight = exports.Chart3DDirectionalLightArchive.internalBinaryRead(reader, reader.uint32(), options, message.directionalLight);
                    break;
                case /* optional TSCH.Chart3DSpotLightArchive spot_light */ 11:
                    message.spotLight = exports.Chart3DSpotLightArchive.internalBinaryRead(reader, reader.uint32(), options, message.spotLight);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* TSCH.Chart3DVectorArchive ambient_color = 2; */
        if (message.ambientColor)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.ambientColor, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSCH.Chart3DVectorArchive diffuse_color = 3; */
        if (message.diffuseColor)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.diffuseColor, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSCH.Chart3DVectorArchive specular_color = 4; */
        if (message.specularColor)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.specularColor, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* float intensity = 5; */
        if (message.intensity !== 0)
            writer.tag(5, runtime_1.WireType.Bit32).float(message.intensity);
        /* TSCH.Chart3DVectorArchive attenuation = 6; */
        if (message.attenuation)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.attenuation, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* uint32 coordinate_space = 7; */
        if (message.coordinateSpace !== 0)
            writer.tag(7, runtime_1.WireType.Varint).uint32(message.coordinateSpace);
        /* bool enabled = 8; */
        if (message.enabled !== false)
            writer.tag(8, runtime_1.WireType.Varint).bool(message.enabled);
        /* optional TSCH.Chart3DPointLightArchive point_light = 9; */
        if (message.pointLight)
            exports.Chart3DPointLightArchive.internalBinaryWrite(message.pointLight, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DDirectionalLightArchive directional_light = 10; */
        if (message.directionalLight)
            exports.Chart3DDirectionalLightArchive.internalBinaryWrite(message.directionalLight, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DSpotLightArchive spot_light = 11; */
        if (message.spotLight)
            exports.Chart3DSpotLightArchive.internalBinaryWrite(message.spotLight, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DLightArchive
 */
exports.Chart3DLightArchive = new Chart3DLightArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DLightingModelArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DLightingModelArchive", [
            { no: 1, name: "phong", kind: "message", T: () => exports.Chart3DPhongLightingModelArchive },
            { no: 2, name: "fixed_function", kind: "message", T: () => exports.Chart3DFixedFunctionLightingModelArchive },
            { no: 3, name: "environment", kind: "message", T: () => exports.Chart3DEnvironmentPackageArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSCH.Chart3DPhongLightingModelArchive phong */ 1:
                    message.phong = exports.Chart3DPhongLightingModelArchive.internalBinaryRead(reader, reader.uint32(), options, message.phong);
                    break;
                case /* optional TSCH.Chart3DFixedFunctionLightingModelArchive fixed_function */ 2:
                    message.fixedFunction = exports.Chart3DFixedFunctionLightingModelArchive.internalBinaryRead(reader, reader.uint32(), options, message.fixedFunction);
                    break;
                case /* optional TSCH.Chart3DEnvironmentPackageArchive environment */ 3:
                    message.environment = exports.Chart3DEnvironmentPackageArchive.internalBinaryRead(reader, reader.uint32(), options, message.environment);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSCH.Chart3DPhongLightingModelArchive phong = 1; */
        if (message.phong)
            exports.Chart3DPhongLightingModelArchive.internalBinaryWrite(message.phong, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DFixedFunctionLightingModelArchive fixed_function = 2; */
        if (message.fixedFunction)
            exports.Chart3DFixedFunctionLightingModelArchive.internalBinaryWrite(message.fixedFunction, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DEnvironmentPackageArchive environment = 3; */
        if (message.environment)
            exports.Chart3DEnvironmentPackageArchive.internalBinaryWrite(message.environment, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DLightingModelArchive
 */
exports.Chart3DLightingModelArchive = new Chart3DLightingModelArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DLightingPackageArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DLightingPackageArchive", [
            { no: 1, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "lights", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DLightArchive }
        ]);
    }
    create(value) {
        const message = { name: "", lights: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string name */ 1:
                    message.name = reader.string();
                    break;
                case /* repeated TSCH.Chart3DLightArchive lights */ 2:
                    message.lights.push(exports.Chart3DLightArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string name = 1; */
        if (message.name !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* repeated TSCH.Chart3DLightArchive lights = 2; */
        for (let i = 0; i < message.lights.length; i++)
            exports.Chart3DLightArchive.internalBinaryWrite(message.lights[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DLightingPackageArchive
 */
exports.Chart3DLightingPackageArchive = new Chart3DLightingPackageArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DTexturesMaterialArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DTexturesMaterialArchive", [
            { no: 1, name: "color", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 2, name: "textures", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DTSPImageDataTextureArchive }
        ]);
    }
    create(value) {
        const message = { textures: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DVectorArchive color */ 1:
                    message.color = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.color);
                    break;
                case /* repeated TSCH.Chart3DTSPImageDataTextureArchive textures */ 2:
                    message.textures.push(exports.Chart3DTSPImageDataTextureArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DVectorArchive color = 1; */
        if (message.color)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.color, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.Chart3DTSPImageDataTextureArchive textures = 2; */
        for (let i = 0; i < message.textures.length; i++)
            exports.Chart3DTSPImageDataTextureArchive.internalBinaryWrite(message.textures[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DTexturesMaterialArchive
 */
exports.Chart3DTexturesMaterialArchive = new Chart3DTexturesMaterialArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DEmissiveMaterialArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DEmissiveMaterialArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.Chart3DTexturesMaterialArchive },
            { no: 2, name: "tilings", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DImageTextureTilingArchive }
        ]);
    }
    create(value) {
        const message = { tilings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DTexturesMaterialArchive super */ 1:
                    message.super = exports.Chart3DTexturesMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSCH.Chart3DImageTextureTilingArchive tilings */ 2:
                    message.tilings.push(exports.Chart3DImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DTexturesMaterialArchive super = 1; */
        if (message.super)
            exports.Chart3DTexturesMaterialArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.Chart3DImageTextureTilingArchive tilings = 2; */
        for (let i = 0; i < message.tilings.length; i++)
            exports.Chart3DImageTextureTilingArchive.internalBinaryWrite(message.tilings[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DEmissiveMaterialArchive
 */
exports.Chart3DEmissiveMaterialArchive = new Chart3DEmissiveMaterialArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DDiffuseMaterialArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DDiffuseMaterialArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.Chart3DTexturesMaterialArchive },
            { no: 2, name: "tilings", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DImageTextureTilingArchive }
        ]);
    }
    create(value) {
        const message = { tilings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DTexturesMaterialArchive super */ 1:
                    message.super = exports.Chart3DTexturesMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSCH.Chart3DImageTextureTilingArchive tilings */ 2:
                    message.tilings.push(exports.Chart3DImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DTexturesMaterialArchive super = 1; */
        if (message.super)
            exports.Chart3DTexturesMaterialArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.Chart3DImageTextureTilingArchive tilings = 2; */
        for (let i = 0; i < message.tilings.length; i++)
            exports.Chart3DImageTextureTilingArchive.internalBinaryWrite(message.tilings[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DDiffuseMaterialArchive
 */
exports.Chart3DDiffuseMaterialArchive = new Chart3DDiffuseMaterialArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DModulateMaterialArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DModulateMaterialArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.Chart3DTexturesMaterialArchive },
            { no: 2, name: "tilings", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DImageTextureTilingArchive }
        ]);
    }
    create(value) {
        const message = { tilings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DTexturesMaterialArchive super */ 1:
                    message.super = exports.Chart3DTexturesMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSCH.Chart3DImageTextureTilingArchive tilings */ 2:
                    message.tilings.push(exports.Chart3DImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DTexturesMaterialArchive super = 1; */
        if (message.super)
            exports.Chart3DTexturesMaterialArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.Chart3DImageTextureTilingArchive tilings = 2; */
        for (let i = 0; i < message.tilings.length; i++)
            exports.Chart3DImageTextureTilingArchive.internalBinaryWrite(message.tilings[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DModulateMaterialArchive
 */
exports.Chart3DModulateMaterialArchive = new Chart3DModulateMaterialArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DSpecularMaterialArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DSpecularMaterialArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.Chart3DTexturesMaterialArchive },
            { no: 2, name: "tilings", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DImageTextureTilingArchive }
        ]);
    }
    create(value) {
        const message = { tilings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DTexturesMaterialArchive super */ 1:
                    message.super = exports.Chart3DTexturesMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSCH.Chart3DImageTextureTilingArchive tilings */ 2:
                    message.tilings.push(exports.Chart3DImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DTexturesMaterialArchive super = 1; */
        if (message.super)
            exports.Chart3DTexturesMaterialArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.Chart3DImageTextureTilingArchive tilings = 2; */
        for (let i = 0; i < message.tilings.length; i++)
            exports.Chart3DImageTextureTilingArchive.internalBinaryWrite(message.tilings[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DSpecularMaterialArchive
 */
exports.Chart3DSpecularMaterialArchive = new Chart3DSpecularMaterialArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DShininessMaterialArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DShininessMaterialArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.Chart3DTexturesMaterialArchive },
            { no: 2, name: "tilings", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DImageTextureTilingArchive }
        ]);
    }
    create(value) {
        const message = { tilings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DTexturesMaterialArchive super */ 1:
                    message.super = exports.Chart3DTexturesMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSCH.Chart3DImageTextureTilingArchive tilings */ 2:
                    message.tilings.push(exports.Chart3DImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DTexturesMaterialArchive super = 1; */
        if (message.super)
            exports.Chart3DTexturesMaterialArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.Chart3DImageTextureTilingArchive tilings = 2; */
        for (let i = 0; i < message.tilings.length; i++)
            exports.Chart3DImageTextureTilingArchive.internalBinaryWrite(message.tilings[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DShininessMaterialArchive
 */
exports.Chart3DShininessMaterialArchive = new Chart3DShininessMaterialArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DEnvironmentMaterialArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DEnvironmentMaterialArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.Chart3DTexturesMaterialArchive },
            { no: 2, name: "OBSOLETE_tilings", kind: "message", jsonName: "OBSOLETETilings", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DBaseImageTextureTilingArchive },
            { no: 3, name: "decalMode", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "tilings", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Chart3DImageTextureTilingArchive }
        ]);
    }
    create(value) {
        const message = { oBSOLETETilings: [], tilings: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DTexturesMaterialArchive super */ 1:
                    message.super = exports.Chart3DTexturesMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSCH.Chart3DBaseImageTextureTilingArchive OBSOLETE_tilings = 2 [json_name = "OBSOLETETilings"];*/ 2:
                    message.oBSOLETETilings.push(exports.Chart3DBaseImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool decalMode */ 3:
                    message.decalMode = reader.bool();
                    break;
                case /* repeated TSCH.Chart3DImageTextureTilingArchive tilings */ 4:
                    message.tilings.push(exports.Chart3DImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DTexturesMaterialArchive super = 1; */
        if (message.super)
            exports.Chart3DTexturesMaterialArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSCH.Chart3DBaseImageTextureTilingArchive OBSOLETE_tilings = 2 [json_name = "OBSOLETETilings"]; */
        for (let i = 0; i < message.oBSOLETETilings.length; i++)
            exports.Chart3DBaseImageTextureTilingArchive.internalBinaryWrite(message.oBSOLETETilings[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool decalMode = 3; */
        if (message.decalMode !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.decalMode);
        /* repeated TSCH.Chart3DImageTextureTilingArchive tilings = 4; */
        for (let i = 0; i < message.tilings.length; i++)
            exports.Chart3DImageTextureTilingArchive.internalBinaryWrite(message.tilings[i], writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DEnvironmentMaterialArchive
 */
exports.Chart3DEnvironmentMaterialArchive = new Chart3DEnvironmentMaterialArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DFixedFunctionLightingModelArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DFixedFunctionLightingModelArchive", [
            { no: 1, name: "materials", kind: "message", T: () => exports.Chart3DPhongMaterialPackageArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DPhongMaterialPackageArchive materials */ 1:
                    message.materials = exports.Chart3DPhongMaterialPackageArchive.internalBinaryRead(reader, reader.uint32(), options, message.materials);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DPhongMaterialPackageArchive materials = 1; */
        if (message.materials)
            exports.Chart3DPhongMaterialPackageArchive.internalBinaryWrite(message.materials, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DFixedFunctionLightingModelArchive
 */
exports.Chart3DFixedFunctionLightingModelArchive = new Chart3DFixedFunctionLightingModelArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DPhongLightingModelArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DPhongLightingModelArchive", [
            { no: 1, name: "materials", kind: "message", T: () => exports.Chart3DPhongMaterialPackageArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DPhongMaterialPackageArchive materials */ 1:
                    message.materials = exports.Chart3DPhongMaterialPackageArchive.internalBinaryRead(reader, reader.uint32(), options, message.materials);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DPhongMaterialPackageArchive materials = 1; */
        if (message.materials)
            exports.Chart3DPhongMaterialPackageArchive.internalBinaryWrite(message.materials, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DPhongLightingModelArchive
 */
exports.Chart3DPhongLightingModelArchive = new Chart3DPhongLightingModelArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DPhongMaterialPackageArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DPhongMaterialPackageArchive", [
            { no: 1, name: "emissive", kind: "message", T: () => exports.Chart3DEmissiveMaterialArchive },
            { no: 2, name: "diffuse", kind: "message", T: () => exports.Chart3DDiffuseMaterialArchive },
            { no: 3, name: "modulate", kind: "message", T: () => exports.Chart3DModulateMaterialArchive },
            { no: 4, name: "specular", kind: "message", T: () => exports.Chart3DSpecularMaterialArchive },
            { no: 5, name: "shininess", kind: "message", T: () => exports.Chart3DShininessMaterialArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSCH.Chart3DEmissiveMaterialArchive emissive */ 1:
                    message.emissive = exports.Chart3DEmissiveMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.emissive);
                    break;
                case /* optional TSCH.Chart3DDiffuseMaterialArchive diffuse */ 2:
                    message.diffuse = exports.Chart3DDiffuseMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.diffuse);
                    break;
                case /* optional TSCH.Chart3DModulateMaterialArchive modulate */ 3:
                    message.modulate = exports.Chart3DModulateMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.modulate);
                    break;
                case /* optional TSCH.Chart3DSpecularMaterialArchive specular */ 4:
                    message.specular = exports.Chart3DSpecularMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.specular);
                    break;
                case /* optional TSCH.Chart3DShininessMaterialArchive shininess */ 5:
                    message.shininess = exports.Chart3DShininessMaterialArchive.internalBinaryRead(reader, reader.uint32(), options, message.shininess);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSCH.Chart3DEmissiveMaterialArchive emissive = 1; */
        if (message.emissive)
            exports.Chart3DEmissiveMaterialArchive.internalBinaryWrite(message.emissive, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DDiffuseMaterialArchive diffuse = 2; */
        if (message.diffuse)
            exports.Chart3DDiffuseMaterialArchive.internalBinaryWrite(message.diffuse, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DModulateMaterialArchive modulate = 3; */
        if (message.modulate)
            exports.Chart3DModulateMaterialArchive.internalBinaryWrite(message.modulate, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DSpecularMaterialArchive specular = 4; */
        if (message.specular)
            exports.Chart3DSpecularMaterialArchive.internalBinaryWrite(message.specular, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.Chart3DShininessMaterialArchive shininess = 5; */
        if (message.shininess)
            exports.Chart3DShininessMaterialArchive.internalBinaryWrite(message.shininess, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DPhongMaterialPackageArchive
 */
exports.Chart3DPhongMaterialPackageArchive = new Chart3DPhongMaterialPackageArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DTSPImageDataTextureArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DTSPImageDataTextureArchive", [
            { no: 3, name: "data", kind: "message", T: () => TSPMessages_2.DataReference },
            { no: 4, name: "mipmapdata", kind: "message", T: () => TSPMessages_2.DataReference },
            { no: 1, name: "database_data", kind: "message", T: () => TSPMessages_1.Reference },
            { no: 2, name: "database_mipmapdata", kind: "message", T: () => TSPMessages_1.Reference }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.DataReference data */ 3:
                    message.data = TSPMessages_2.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.data);
                    break;
                case /* optional TSP.DataReference mipmapdata */ 4:
                    message.mipmapdata = TSPMessages_2.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.mipmapdata);
                    break;
                case /* optional TSP.Reference database_data */ 1:
                    message.databaseData = TSPMessages_1.Reference.internalBinaryRead(reader, reader.uint32(), options, message.databaseData);
                    break;
                case /* optional TSP.Reference database_mipmapdata */ 2:
                    message.databaseMipmapdata = TSPMessages_1.Reference.internalBinaryRead(reader, reader.uint32(), options, message.databaseMipmapdata);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.DataReference data = 3; */
        if (message.data)
            TSPMessages_2.DataReference.internalBinaryWrite(message.data, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.DataReference mipmapdata = 4; */
        if (message.mipmapdata)
            TSPMessages_2.DataReference.internalBinaryWrite(message.mipmapdata, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference database_data = 1; */
        if (message.databaseData)
            TSPMessages_1.Reference.internalBinaryWrite(message.databaseData, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference database_mipmapdata = 2; */
        if (message.databaseMipmapdata)
            TSPMessages_1.Reference.internalBinaryWrite(message.databaseMipmapdata, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DTSPImageDataTextureArchive
 */
exports.Chart3DTSPImageDataTextureArchive = new Chart3DTSPImageDataTextureArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DBaseImageTextureTilingArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DBaseImageTextureTilingArchive", [
            { no: 1, name: "scale", kind: "message", T: () => exports.Chart3DVectorArchive },
            { no: 2, name: "rotation", kind: "scalar", opt: true, T: 2 /*ScalarType.FLOAT*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSCH.Chart3DVectorArchive scale */ 1:
                    message.scale = exports.Chart3DVectorArchive.internalBinaryRead(reader, reader.uint32(), options, message.scale);
                    break;
                case /* optional float rotation */ 2:
                    message.rotation = reader.float();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSCH.Chart3DVectorArchive scale = 1; */
        if (message.scale)
            exports.Chart3DVectorArchive.internalBinaryWrite(message.scale, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional float rotation = 2; */
        if (message.rotation !== undefined)
            writer.tag(2, runtime_1.WireType.Bit32).float(message.rotation);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DBaseImageTextureTilingArchive
 */
exports.Chart3DBaseImageTextureTilingArchive = new Chart3DBaseImageTextureTilingArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DImageTextureTilingArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DImageTextureTilingArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.Chart3DBaseImageTextureTilingArchive },
            { no: 2, name: "mode", kind: "enum", opt: true, T: () => ["TSCH.TextureTilingMode", TextureTilingMode] },
            { no: 3, name: "wrap", kind: "enum", opt: true, T: () => ["TSCH.TextureTilingWrap", TextureTilingWrap] },
            { no: 4, name: "face", kind: "enum", opt: true, T: () => ["TSCH.TextureTilingFace", TextureTilingFace] },
            { no: 5, name: "xposition", kind: "enum", opt: true, T: () => ["TSCH.TextureTilingXPosition", TextureTilingXPosition] },
            { no: 6, name: "yposition", kind: "enum", opt: true, T: () => ["TSCH.TextureTilingYPosition", TextureTilingYPosition] },
            { no: 7, name: "scontinuity", kind: "enum", opt: true, T: () => ["TSCH.TextureTilingContinuity", TextureTilingContinuity] },
            { no: 8, name: "tcontinuity", kind: "enum", opt: true, T: () => ["TSCH.TextureTilingContinuity", TextureTilingContinuity] },
            { no: 9, name: "reveal", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSCH.Chart3DBaseImageTextureTilingArchive super */ 1:
                    message.super = exports.Chart3DBaseImageTextureTilingArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSCH.TextureTilingMode mode */ 2:
                    message.mode = reader.int32();
                    break;
                case /* optional TSCH.TextureTilingWrap wrap */ 3:
                    message.wrap = reader.int32();
                    break;
                case /* optional TSCH.TextureTilingFace face */ 4:
                    message.face = reader.int32();
                    break;
                case /* optional TSCH.TextureTilingXPosition xposition */ 5:
                    message.xposition = reader.int32();
                    break;
                case /* optional TSCH.TextureTilingYPosition yposition */ 6:
                    message.yposition = reader.int32();
                    break;
                case /* optional TSCH.TextureTilingContinuity scontinuity */ 7:
                    message.scontinuity = reader.int32();
                    break;
                case /* optional TSCH.TextureTilingContinuity tcontinuity */ 8:
                    message.tcontinuity = reader.int32();
                    break;
                case /* optional bool reveal */ 9:
                    message.reveal = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSCH.Chart3DBaseImageTextureTilingArchive super = 1; */
        if (message.super)
            exports.Chart3DBaseImageTextureTilingArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSCH.TextureTilingMode mode = 2; */
        if (message.mode !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.mode);
        /* optional TSCH.TextureTilingWrap wrap = 3; */
        if (message.wrap !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.wrap);
        /* optional TSCH.TextureTilingFace face = 4; */
        if (message.face !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.face);
        /* optional TSCH.TextureTilingXPosition xposition = 5; */
        if (message.xposition !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.xposition);
        /* optional TSCH.TextureTilingYPosition yposition = 6; */
        if (message.yposition !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).int32(message.yposition);
        /* optional TSCH.TextureTilingContinuity scontinuity = 7; */
        if (message.scontinuity !== undefined)
            writer.tag(7, runtime_1.WireType.Varint).int32(message.scontinuity);
        /* optional TSCH.TextureTilingContinuity tcontinuity = 8; */
        if (message.tcontinuity !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).int32(message.tcontinuity);
        /* optional bool reveal = 9; */
        if (message.reveal !== undefined)
            writer.tag(9, runtime_1.WireType.Varint).bool(message.reveal);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DImageTextureTilingArchive
 */
exports.Chart3DImageTextureTilingArchive = new Chart3DImageTextureTilingArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Chart3DVectorArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSCH.Chart3DVectorArchive", [
            { no: 1, name: "x", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ },
            { no: 2, name: "y", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ },
            { no: 3, name: "z", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ },
            { no: 4, name: "w", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ }
        ]);
    }
    create(value) {
        const message = { x: 0, y: 0, z: 0, w: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* float x */ 1:
                    message.x = reader.float();
                    break;
                case /* float y */ 2:
                    message.y = reader.float();
                    break;
                case /* float z */ 3:
                    message.z = reader.float();
                    break;
                case /* float w */ 4:
                    message.w = reader.float();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* float x = 1; */
        if (message.x !== 0)
            writer.tag(1, runtime_1.WireType.Bit32).float(message.x);
        /* float y = 2; */
        if (message.y !== 0)
            writer.tag(2, runtime_1.WireType.Bit32).float(message.y);
        /* float z = 3; */
        if (message.z !== 0)
            writer.tag(3, runtime_1.WireType.Bit32).float(message.z);
        /* float w = 4; */
        if (message.w !== 0)
            writer.tag(4, runtime_1.WireType.Bit32).float(message.w);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSCH.Chart3DVectorArchive
 */
exports.Chart3DVectorArchive = new Chart3DVectorArchive$Type();
