// keynote-archives/lib/generated/TSKArchives_sos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResetActivityStreamCommandArchive = exports.RemoveAuthorIdentifiersCommandArchive = exports.FixCorruptedDataCommandArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSPMessages_1 = require("./TSPMessages");
const TSKArchives_1 = require("./TSKArchives");
// @generated message type with reflection information, may provide speed optimized methods
class FixCorruptedDataCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSKSOS.FixCorruptedDataCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "corrupted_digest_list", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "corrupted_digest_list_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { corruptedDigestList: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated string corrupted_digest_list */ 2:
                    message.corruptedDigestList.push(reader.string());
                    break;
                case /* optional bool corrupted_digest_list_undefined */ 3:
                    message.corruptedDigestListUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated string corrupted_digest_list = 2; */
        for (let i = 0; i < message.corruptedDigestList.length; i++)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.corruptedDigestList[i]);
        /* optional bool corrupted_digest_list_undefined = 3; */
        if (message.corruptedDigestListUndefined !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.corruptedDigestListUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSKSOS.FixCorruptedDataCommandArchive
 */
exports.FixCorruptedDataCommandArchive = new FixCorruptedDataCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class RemoveAuthorIdentifiersCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSKSOS.RemoveAuthorIdentifiersCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "author_identifiers", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.UUID },
            { no: 3, name: "author_identifiers_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { authorIdentifiers: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSP.UUID author_identifiers */ 2:
                    message.authorIdentifiers.push(TSPMessages_1.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool author_identifiers_undefined */ 3:
                    message.authorIdentifiersUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUID author_identifiers = 2; */
        for (let i = 0; i < message.authorIdentifiers.length; i++)
            TSPMessages_1.UUID.internalBinaryWrite(message.authorIdentifiers[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool author_identifiers_undefined = 3; */
        if (message.authorIdentifiersUndefined !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.authorIdentifiersUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSKSOS.RemoveAuthorIdentifiersCommandArchive
 */
exports.RemoveAuthorIdentifiersCommandArchive = new RemoveAuthorIdentifiersCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ResetActivityStreamCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSKSOS.ResetActivityStreamCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSKSOS.ResetActivityStreamCommandArchive
 */
exports.ResetActivityStreamCommandArchive = new ResetActivityStreamCommandArchive$Type();
