// keynote-archives/lib/generated/TSPDatabaseMessages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseImageDataArchive = exports.DatabaseDataArchive = exports.DatabaseData = exports.DatabaseImageDataArchive_ImageType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSPMessages_1 = require("./TSPMessages");
const TSPMessages_2 = require("./TSPMessages");
/**
 * @generated from protobuf enum TSP.DatabaseImageDataArchive.ImageType
 */
var DatabaseImageDataArchive_ImageType;
(function (DatabaseImageDataArchive_ImageType) {
    /**
     * @generated from protobuf enum value: unknown = 0;
     */
    DatabaseImageDataArchive_ImageType[DatabaseImageDataArchive_ImageType["unknown"] = 0] = "unknown";
    /**
     * @generated from protobuf enum value: bitmap = 1;
     */
    DatabaseImageDataArchive_ImageType[DatabaseImageDataArchive_ImageType["bitmap"] = 1] = "bitmap";
    /**
     * @generated from protobuf enum value: pdf = 2;
     */
    DatabaseImageDataArchive_ImageType[DatabaseImageDataArchive_ImageType["pdf"] = 2] = "pdf";
})(DatabaseImageDataArchive_ImageType = exports.DatabaseImageDataArchive_ImageType || (exports.DatabaseImageDataArchive_ImageType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class DatabaseData$Type extends runtime_5.MessageType {
    constructor() {
        super("TSP.DatabaseData", [
            { no: 1, name: "data", kind: "message", T: () => TSPMessages_2.DataReference }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSP.DataReference data */ 1:
                    message.data = TSPMessages_2.DataReference.internalBinaryRead(reader, reader.uint32(), options, message.data);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSP.DataReference data = 1; */
        if (message.data)
            TSPMessages_2.DataReference.internalBinaryWrite(message.data, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSP.DatabaseData
 */
exports.DatabaseData = new DatabaseData$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DatabaseDataArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSP.DatabaseDataArchive", [
            { no: 1, name: "data", kind: "message", T: () => TSPMessages_1.Reference },
            { no: 2, name: "app_relative_path", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "display_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "length", kind: "scalar", opt: true, T: 4 /*ScalarType.UINT64*/, L: 0 /*LongType.BIGINT*/ },
            { no: 5, name: "hash", kind: "scalar", opt: true, T: 13 /*ScalarType.UINT32*/ },
            { no: 6, name: "sharable", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { displayName: "", sharable: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference data */ 1:
                    message.data = TSPMessages_1.Reference.internalBinaryRead(reader, reader.uint32(), options, message.data);
                    break;
                case /* optional string app_relative_path */ 2:
                    message.appRelativePath = reader.string();
                    break;
                case /* string display_name */ 3:
                    message.displayName = reader.string();
                    break;
                case /* optional uint64 length */ 4:
                    message.length = reader.uint64().toBigInt();
                    break;
                case /* optional uint32 hash */ 5:
                    message.hash = reader.uint32();
                    break;
                case /* bool sharable */ 6:
                    message.sharable = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference data = 1; */
        if (message.data)
            TSPMessages_1.Reference.internalBinaryWrite(message.data, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string app_relative_path = 2; */
        if (message.appRelativePath !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.appRelativePath);
        /* string display_name = 3; */
        if (message.displayName !== "")
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.displayName);
        /* optional uint64 length = 4; */
        if (message.length !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).uint64(message.length);
        /* optional uint32 hash = 5; */
        if (message.hash !== undefined)
            writer.tag(5, runtime_1.WireType.Varint).uint32(message.hash);
        /* bool sharable = 6; */
        if (message.sharable !== false)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.sharable);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSP.DatabaseDataArchive
 */
exports.DatabaseDataArchive = new DatabaseDataArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DatabaseImageDataArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSP.DatabaseImageDataArchive", [
            { no: 1, name: "super", kind: "message", T: () => exports.DatabaseDataArchive },
            { no: 2, name: "type", kind: "enum", T: () => ["TSP.DatabaseImageDataArchive.ImageType", DatabaseImageDataArchive_ImageType] }
        ]);
    }
    create(value) {
        const message = { type: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSP.DatabaseDataArchive super */ 1:
                    message.super = exports.DatabaseDataArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.DatabaseImageDataArchive.ImageType type */ 2:
                    message.type = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSP.DatabaseDataArchive super = 1; */
        if (message.super)
            exports.DatabaseDataArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.DatabaseImageDataArchive.ImageType type = 2; */
        if (message.type !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.type);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSP.DatabaseImageDataArchive
 */
exports.DatabaseImageDataArchive = new DatabaseImageDataArchive$Type();
