// keynote-archives/lib/generated/TSSArchives.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommandPropertyMapArchive = exports.CommandPropertyEntryArchive = exports.ThemeReplaceStylePresetAndDisconnectStylesCommandArchive = exports.ThemeMovePresetCommandArchive = exports.ThemeRemoveStylePresetCommandArchive = exports.ThemeAddStylePresetCommandArchive = exports.ThemeReplaceColorPresetCommandArchive = exports.ThemeReplacePresetCommandArchive = exports.StyleUpdatePropertyMapCommandArchive = exports.ApplyThemeChildCommandArchive = exports.ApplyThemeCommandArchive = exports.ThemeArchive = exports.StylesheetArchive_VersionedStyles = exports.StylesheetArchive_StyleChildrenEntry = exports.StylesheetArchive_IdentifiedStyleEntry = exports.StylesheetArchive = exports.StyleArchive = exports.PropertyType = exports.ValueType = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSKArchives_1 = require("./TSKArchives");
const TSPMessages_1 = require("./TSPMessages");
const TSPMessages_2 = require("./TSPMessages");
const TSPMessages_3 = require("./TSPMessages");
/**
 * @generated from protobuf enum TSS.ValueType
 */
var ValueType;
(function (ValueType) {
    /**
     * @generated from protobuf enum value: ObjectType = 0;
     */
    ValueType[ValueType["ObjectType"] = 0] = "ObjectType";
    /**
     * @generated from protobuf enum value: IntType = 1;
     */
    ValueType[ValueType["IntType"] = 1] = "IntType";
    /**
     * @generated from protobuf enum value: FloatType = 2;
     */
    ValueType[ValueType["FloatType"] = 2] = "FloatType";
    /**
     * @generated from protobuf enum value: DoubleType = 3;
     */
    ValueType[ValueType["DoubleType"] = 3] = "DoubleType";
})(ValueType = exports.ValueType || (exports.ValueType = {}));
/**
 * @generated from protobuf enum TSS.PropertyType
 */
var PropertyType;
(function (PropertyType) {
    /**
     * @generated synthetic value - protobuf-ts requires all enums to have a 0 value
     */
    PropertyType[PropertyType["UNSPECIFIED$"] = 0] = "UNSPECIFIED$";
    /**
     * @generated from protobuf enum value: InvalidPropertyType = 1;
     */
    PropertyType[PropertyType["InvalidPropertyType"] = 1] = "InvalidPropertyType";
    /**
     * @generated from protobuf enum value: NullPropertyType = 2;
     */
    PropertyType[PropertyType["NullPropertyType"] = 2] = "NullPropertyType";
    /**
     * @generated from protobuf enum value: IntegerPropertyType = 3;
     */
    PropertyType[PropertyType["IntegerPropertyType"] = 3] = "IntegerPropertyType";
    /**
     * @generated from protobuf enum value: FloatPropertyType = 4;
     */
    PropertyType[PropertyType["FloatPropertyType"] = 4] = "FloatPropertyType";
    /**
     * @generated from protobuf enum value: DoublePropertyType = 5;
     */
    PropertyType[PropertyType["DoublePropertyType"] = 5] = "DoublePropertyType";
    /**
     * @generated from protobuf enum value: NSStringPropertyType = 6;
     */
    PropertyType[PropertyType["NSStringPropertyType"] = 6] = "NSStringPropertyType";
    /**
     * @generated from protobuf enum value: TSPObjectPropertyType = 7;
     */
    PropertyType[PropertyType["TSPObjectPropertyType"] = 7] = "TSPObjectPropertyType";
})(PropertyType = exports.PropertyType || (exports.PropertyType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class StyleArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.StyleArchive", [
            { no: 1, name: "name", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "style_identifier", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "parent", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "is_variation", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "stylesheet", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string name */ 1:
                    message.name = reader.string();
                    break;
                case /* optional string style_identifier */ 2:
                    message.styleIdentifier = reader.string();
                    break;
                case /* optional TSP.Reference parent */ 3:
                    message.parent = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.parent);
                    break;
                case /* optional bool is_variation */ 4:
                    message.isVariation = reader.bool();
                    break;
                case /* optional TSP.Reference stylesheet */ 5:
                    message.stylesheet = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.stylesheet);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string name = 1; */
        if (message.name !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.name);
        /* optional string style_identifier = 2; */
        if (message.styleIdentifier !== undefined)
            writer.tag(2, runtime_1.WireType.LengthDelimited).string(message.styleIdentifier);
        /* optional TSP.Reference parent = 3; */
        if (message.parent)
            TSPMessages_3.Reference.internalBinaryWrite(message.parent, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool is_variation = 4; */
        if (message.isVariation !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.isVariation);
        /* optional TSP.Reference stylesheet = 5; */
        if (message.stylesheet)
            TSPMessages_3.Reference.internalBinaryWrite(message.stylesheet, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.StyleArchive
 */
exports.StyleArchive = new StyleArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StylesheetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.StylesheetArchive", [
            { no: 1, name: "styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 2, name: "identifier_to_style_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.StylesheetArchive_IdentifiedStyleEntry },
            { no: 3, name: "parent", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "is_locked", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "parent_to_children_style_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.StylesheetArchive_StyleChildrenEntry },
            { no: 6, name: "can_cull_styles", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "styles_for_10_0", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 8, name: "styles_for_10_1", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 9, name: "styles_for_10_2", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 10, name: "styles_for_11_0", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 11, name: "styles_for_11_1", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 12, name: "styles_for_11_2", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 13, name: "styles_for_12_0", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 14, name: "styles_for_12_1", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles },
            { no: 15, name: "styles_for_12_2", kind: "message", T: () => exports.StylesheetArchive_VersionedStyles }
        ]);
    }
    create(value) {
        const message = { styles: [], identifierToStyleMap: [], parentToChildrenStyleMap: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSP.Reference styles */ 1:
                    message.styles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSS.StylesheetArchive.IdentifiedStyleEntry identifier_to_style_map */ 2:
                    message.identifierToStyleMap.push(exports.StylesheetArchive_IdentifiedStyleEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional TSP.Reference parent */ 3:
                    message.parent = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.parent);
                    break;
                case /* optional bool is_locked */ 4:
                    message.isLocked = reader.bool();
                    break;
                case /* repeated TSS.StylesheetArchive.StyleChildrenEntry parent_to_children_style_map */ 5:
                    message.parentToChildrenStyleMap.push(exports.StylesheetArchive_StyleChildrenEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional bool can_cull_styles */ 6:
                    message.canCullStyles = reader.bool();
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_10_0 */ 7:
                    message.stylesFor100 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor100);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_10_1 */ 8:
                    message.stylesFor101 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor101);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_10_2 */ 9:
                    message.stylesFor102 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor102);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_11_0 */ 10:
                    message.stylesFor110 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor110);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_11_1 */ 11:
                    message.stylesFor111 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor111);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_11_2 */ 12:
                    message.stylesFor112 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor112);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_12_0 */ 13:
                    message.stylesFor120 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor120);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_12_1 */ 14:
                    message.stylesFor121 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor121);
                    break;
                case /* optional TSS.StylesheetArchive.VersionedStyles styles_for_12_2 */ 15:
                    message.stylesFor122 = exports.StylesheetArchive_VersionedStyles.internalBinaryRead(reader, reader.uint32(), options, message.stylesFor122);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSP.Reference styles = 1; */
        for (let i = 0; i < message.styles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.styles[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSS.StylesheetArchive.IdentifiedStyleEntry identifier_to_style_map = 2; */
        for (let i = 0; i < message.identifierToStyleMap.length; i++)
            exports.StylesheetArchive_IdentifiedStyleEntry.internalBinaryWrite(message.identifierToStyleMap[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference parent = 3; */
        if (message.parent)
            TSPMessages_3.Reference.internalBinaryWrite(message.parent, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool is_locked = 4; */
        if (message.isLocked !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.isLocked);
        /* repeated TSS.StylesheetArchive.StyleChildrenEntry parent_to_children_style_map = 5; */
        for (let i = 0; i < message.parentToChildrenStyleMap.length; i++)
            exports.StylesheetArchive_StyleChildrenEntry.internalBinaryWrite(message.parentToChildrenStyleMap[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool can_cull_styles = 6; */
        if (message.canCullStyles !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.canCullStyles);
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_10_0 = 7; */
        if (message.stylesFor100)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor100, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_10_1 = 8; */
        if (message.stylesFor101)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor101, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_10_2 = 9; */
        if (message.stylesFor102)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor102, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_11_0 = 10; */
        if (message.stylesFor110)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor110, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_11_1 = 11; */
        if (message.stylesFor111)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor111, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_11_2 = 12; */
        if (message.stylesFor112)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor112, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_12_0 = 13; */
        if (message.stylesFor120)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor120, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_12_1 = 14; */
        if (message.stylesFor121)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor121, writer.tag(14, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSS.StylesheetArchive.VersionedStyles styles_for_12_2 = 15; */
        if (message.stylesFor122)
            exports.StylesheetArchive_VersionedStyles.internalBinaryWrite(message.stylesFor122, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.StylesheetArchive
 */
exports.StylesheetArchive = new StylesheetArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StylesheetArchive_IdentifiedStyleEntry$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.StylesheetArchive.IdentifiedStyleEntry", [
            { no: 1, name: "identifier", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "style", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { identifier: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* string identifier */ 1:
                    message.identifier = reader.string();
                    break;
                case /* TSP.Reference style */ 2:
                    message.style = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.style);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* string identifier = 1; */
        if (message.identifier !== "")
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.identifier);
        /* TSP.Reference style = 2; */
        if (message.style)
            TSPMessages_3.Reference.internalBinaryWrite(message.style, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.StylesheetArchive.IdentifiedStyleEntry
 */
exports.StylesheetArchive_IdentifiedStyleEntry = new StylesheetArchive_IdentifiedStyleEntry$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StylesheetArchive_StyleChildrenEntry$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.StylesheetArchive.StyleChildrenEntry", [
            { no: 1, name: "parent", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 2, name: "children", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { children: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSP.Reference parent */ 1:
                    message.parent = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.parent);
                    break;
                case /* repeated TSP.Reference children */ 2:
                    message.children.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSP.Reference parent = 1; */
        if (message.parent)
            TSPMessages_3.Reference.internalBinaryWrite(message.parent, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference children = 2; */
        for (let i = 0; i < message.children.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.children[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.StylesheetArchive.StyleChildrenEntry
 */
exports.StylesheetArchive_StyleChildrenEntry = new StylesheetArchive_StyleChildrenEntry$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StylesheetArchive_VersionedStyles$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.StylesheetArchive.VersionedStyles", [
            { no: 1, name: "styles", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 2, name: "identifier_to_style_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.StylesheetArchive_IdentifiedStyleEntry },
            { no: 3, name: "parent_to_children_style_map", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.StylesheetArchive_StyleChildrenEntry }
        ]);
    }
    create(value) {
        const message = { styles: [], identifierToStyleMap: [], parentToChildrenStyleMap: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSP.Reference styles */ 1:
                    message.styles.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSS.StylesheetArchive.IdentifiedStyleEntry identifier_to_style_map */ 2:
                    message.identifierToStyleMap.push(exports.StylesheetArchive_IdentifiedStyleEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSS.StylesheetArchive.StyleChildrenEntry parent_to_children_style_map */ 3:
                    message.parentToChildrenStyleMap.push(exports.StylesheetArchive_StyleChildrenEntry.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSP.Reference styles = 1; */
        for (let i = 0; i < message.styles.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.styles[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSS.StylesheetArchive.IdentifiedStyleEntry identifier_to_style_map = 2; */
        for (let i = 0; i < message.identifierToStyleMap.length; i++)
            exports.StylesheetArchive_IdentifiedStyleEntry.internalBinaryWrite(message.identifierToStyleMap[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSS.StylesheetArchive.StyleChildrenEntry parent_to_children_style_map = 3; */
        for (let i = 0; i < message.parentToChildrenStyleMap.length; i++)
            exports.StylesheetArchive_StyleChildrenEntry.internalBinaryWrite(message.parentToChildrenStyleMap[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.StylesheetArchive.VersionedStyles
 */
exports.StylesheetArchive_VersionedStyles = new StylesheetArchive_VersionedStyles$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemeArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ThemeArchive", [
            { no: 1, name: "legacy_stylesheet", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "theme_identifier", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "document_stylesheet", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 5, name: "old_uuids_for_preset_replacements", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_2.UUID },
            { no: 6, name: "new_uuids_for_preset_replacements", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_2.UUID },
            { no: 10, name: "color_presets", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.Color }
        ]);
    }
    create(value) {
        const message = { oldUuidsForPresetReplacements: [], newUuidsForPresetReplacements: [], colorPresets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Reference legacy_stylesheet */ 1:
                    message.legacyStylesheet = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.legacyStylesheet);
                    break;
                case /* optional string theme_identifier */ 3:
                    message.themeIdentifier = reader.string();
                    break;
                case /* optional TSP.Reference document_stylesheet */ 4:
                    message.documentStylesheet = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.documentStylesheet);
                    break;
                case /* repeated TSP.UUID old_uuids_for_preset_replacements */ 5:
                    message.oldUuidsForPresetReplacements.push(TSPMessages_2.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.UUID new_uuids_for_preset_replacements */ 6:
                    message.newUuidsForPresetReplacements.push(TSPMessages_2.UUID.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Color color_presets */ 10:
                    message.colorPresets.push(TSPMessages_1.Color.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Reference legacy_stylesheet = 1; */
        if (message.legacyStylesheet)
            TSPMessages_3.Reference.internalBinaryWrite(message.legacyStylesheet, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional string theme_identifier = 3; */
        if (message.themeIdentifier !== undefined)
            writer.tag(3, runtime_1.WireType.LengthDelimited).string(message.themeIdentifier);
        /* optional TSP.Reference document_stylesheet = 4; */
        if (message.documentStylesheet)
            TSPMessages_3.Reference.internalBinaryWrite(message.documentStylesheet, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUID old_uuids_for_preset_replacements = 5; */
        for (let i = 0; i < message.oldUuidsForPresetReplacements.length; i++)
            TSPMessages_2.UUID.internalBinaryWrite(message.oldUuidsForPresetReplacements[i], writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.UUID new_uuids_for_preset_replacements = 6; */
        for (let i = 0; i < message.newUuidsForPresetReplacements.length; i++)
            TSPMessages_2.UUID.internalBinaryWrite(message.newUuidsForPresetReplacements[i], writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Color color_presets = 10; */
        for (let i = 0; i < message.colorPresets.length; i++)
            TSPMessages_1.Color.internalBinaryWrite(message.colorPresets[i], writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ThemeArchive
 */
exports.ThemeArchive = new ThemeArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyThemeCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ApplyThemeCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "commands", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_3.Reference },
            { no: 3, name: "old_theme", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "new_theme", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { commands: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* repeated TSP.Reference commands */ 2:
                    message.commands.push(TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* optional TSP.Reference old_theme */ 3:
                    message.oldTheme = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.oldTheme);
                    break;
                case /* optional TSP.Reference new_theme */ 4:
                    message.newTheme = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.newTheme);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference commands = 2; */
        for (let i = 0; i < message.commands.length; i++)
            TSPMessages_3.Reference.internalBinaryWrite(message.commands[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference old_theme = 3; */
        if (message.oldTheme)
            TSPMessages_3.Reference.internalBinaryWrite(message.oldTheme, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference new_theme = 4; */
        if (message.newTheme)
            TSPMessages_3.Reference.internalBinaryWrite(message.newTheme, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ApplyThemeCommandArchive
 */
exports.ApplyThemeCommandArchive = new ApplyThemeCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ApplyThemeChildCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ApplyThemeChildCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "parent", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* optional TSP.Reference parent */ 2:
                    message.parent = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.parent);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference parent = 2; */
        if (message.parent)
            TSPMessages_3.Reference.internalBinaryWrite(message.parent, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ApplyThemeChildCommandArchive
 */
exports.ApplyThemeChildCommandArchive = new ApplyThemeChildCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StyleUpdatePropertyMapCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.StyleUpdatePropertyMapCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "current_style", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "style_with_old_property_map", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "style_with_new_property_map", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 7, name: "style_diff", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 6, name: "notify_for_style_clients", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference current_style */ 2:
                    message.currentStyle = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.currentStyle);
                    break;
                case /* optional TSP.Reference style_with_old_property_map */ 3:
                    message.styleWithOldPropertyMap = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.styleWithOldPropertyMap);
                    break;
                case /* optional TSP.Reference style_with_new_property_map */ 4:
                    message.styleWithNewPropertyMap = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.styleWithNewPropertyMap);
                    break;
                case /* optional TSP.Reference style_diff */ 7:
                    message.styleDiff = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.styleDiff);
                    break;
                case /* optional bool notify_for_style_clients */ 6:
                    message.notifyForStyleClients = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference current_style = 2; */
        if (message.currentStyle)
            TSPMessages_3.Reference.internalBinaryWrite(message.currentStyle, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference style_with_old_property_map = 3; */
        if (message.styleWithOldPropertyMap)
            TSPMessages_3.Reference.internalBinaryWrite(message.styleWithOldPropertyMap, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference style_with_new_property_map = 4; */
        if (message.styleWithNewPropertyMap)
            TSPMessages_3.Reference.internalBinaryWrite(message.styleWithNewPropertyMap, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference style_diff = 7; */
        if (message.styleDiff)
            TSPMessages_3.Reference.internalBinaryWrite(message.styleDiff, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool notify_for_style_clients = 6; */
        if (message.notifyForStyleClients !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.notifyForStyleClients);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.StyleUpdatePropertyMapCommandArchive
 */
exports.StyleUpdatePropertyMapCommandArchive = new StyleUpdatePropertyMapCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemeReplacePresetCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ThemeReplacePresetCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 3, name: "preset", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "oldPreset", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 5, name: "index", kind: "scalar", T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = { index: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference preset */ 3:
                    message.preset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.preset);
                    break;
                case /* optional TSP.Reference oldPreset */ 4:
                    message.oldPreset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.oldPreset);
                    break;
                case /* uint32 index */ 5:
                    message.index = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference preset = 3; */
        if (message.preset)
            TSPMessages_3.Reference.internalBinaryWrite(message.preset, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference oldPreset = 4; */
        if (message.oldPreset)
            TSPMessages_3.Reference.internalBinaryWrite(message.oldPreset, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* uint32 index = 5; */
        if (message.index !== 0)
            writer.tag(5, runtime_1.WireType.Varint).uint32(message.index);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ThemeReplacePresetCommandArchive
 */
exports.ThemeReplacePresetCommandArchive = new ThemeReplacePresetCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemeReplaceColorPresetCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ThemeReplaceColorPresetCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "theme", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "color", kind: "message", T: () => TSPMessages_1.Color },
            { no: 4, name: "old_color", kind: "message", T: () => TSPMessages_1.Color },
            { no: 5, name: "index", kind: "scalar", T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = { index: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference theme */ 2:
                    message.theme = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.theme);
                    break;
                case /* TSP.Color color */ 3:
                    message.color = TSPMessages_1.Color.internalBinaryRead(reader, reader.uint32(), options, message.color);
                    break;
                case /* TSP.Color old_color */ 4:
                    message.oldColor = TSPMessages_1.Color.internalBinaryRead(reader, reader.uint32(), options, message.oldColor);
                    break;
                case /* uint32 index */ 5:
                    message.index = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference theme = 2; */
        if (message.theme)
            TSPMessages_3.Reference.internalBinaryWrite(message.theme, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Color color = 3; */
        if (message.color)
            TSPMessages_1.Color.internalBinaryWrite(message.color, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Color old_color = 4; */
        if (message.oldColor)
            TSPMessages_1.Color.internalBinaryWrite(message.oldColor, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* uint32 index = 5; */
        if (message.index !== 0)
            writer.tag(5, runtime_1.WireType.Varint).uint32(message.index);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ThemeReplaceColorPresetCommandArchive
 */
exports.ThemeReplaceColorPresetCommandArchive = new ThemeReplaceColorPresetCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemeAddStylePresetCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ThemeAddStylePresetCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "theme", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "preset", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "preset_kind", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "identifier", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "add_preset_to_stylesheet", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { presetKind: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference theme */ 2:
                    message.theme = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.theme);
                    break;
                case /* TSP.Reference preset */ 3:
                    message.preset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.preset);
                    break;
                case /* string preset_kind */ 4:
                    message.presetKind = reader.string();
                    break;
                case /* optional string identifier */ 5:
                    message.identifier = reader.string();
                    break;
                case /* optional bool add_preset_to_stylesheet */ 6:
                    message.addPresetToStylesheet = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference theme = 2; */
        if (message.theme)
            TSPMessages_3.Reference.internalBinaryWrite(message.theme, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference preset = 3; */
        if (message.preset)
            TSPMessages_3.Reference.internalBinaryWrite(message.preset, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* string preset_kind = 4; */
        if (message.presetKind !== "")
            writer.tag(4, runtime_1.WireType.LengthDelimited).string(message.presetKind);
        /* optional string identifier = 5; */
        if (message.identifier !== undefined)
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.identifier);
        /* optional bool add_preset_to_stylesheet = 6; */
        if (message.addPresetToStylesheet !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.addPresetToStylesheet);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ThemeAddStylePresetCommandArchive
 */
exports.ThemeAddStylePresetCommandArchive = new ThemeAddStylePresetCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemeRemoveStylePresetCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ThemeRemoveStylePresetCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "theme", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "preset", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 4, name: "preset_index", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 5, name: "preset_kind", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 6, name: "identifier", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "replacement_preset", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { presetIndex: 0, presetKind: "" };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference theme */ 2:
                    message.theme = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.theme);
                    break;
                case /* TSP.Reference preset */ 3:
                    message.preset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.preset);
                    break;
                case /* uint32 preset_index */ 4:
                    message.presetIndex = reader.uint32();
                    break;
                case /* string preset_kind */ 5:
                    message.presetKind = reader.string();
                    break;
                case /* optional string identifier */ 6:
                    message.identifier = reader.string();
                    break;
                case /* optional TSP.Reference replacement_preset */ 7:
                    message.replacementPreset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.replacementPreset);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference theme = 2; */
        if (message.theme)
            TSPMessages_3.Reference.internalBinaryWrite(message.theme, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference preset = 3; */
        if (message.preset)
            TSPMessages_3.Reference.internalBinaryWrite(message.preset, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* uint32 preset_index = 4; */
        if (message.presetIndex !== 0)
            writer.tag(4, runtime_1.WireType.Varint).uint32(message.presetIndex);
        /* string preset_kind = 5; */
        if (message.presetKind !== "")
            writer.tag(5, runtime_1.WireType.LengthDelimited).string(message.presetKind);
        /* optional string identifier = 6; */
        if (message.identifier !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.identifier);
        /* optional TSP.Reference replacement_preset = 7; */
        if (message.replacementPreset)
            TSPMessages_3.Reference.internalBinaryWrite(message.replacementPreset, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ThemeRemoveStylePresetCommandArchive
 */
exports.ThemeRemoveStylePresetCommandArchive = new ThemeRemoveStylePresetCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemeMovePresetCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ThemeMovePresetCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "theme", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "preset_id", kind: "message", T: () => TSPMessages_2.UUID },
            { no: 4, name: "new_index", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 5, name: "old_index", kind: "scalar", T: 13 /*ScalarType.UINT32*/ }
        ]);
    }
    create(value) {
        const message = { newIndex: 0, oldIndex: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference theme */ 2:
                    message.theme = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.theme);
                    break;
                case /* TSP.UUID preset_id */ 3:
                    message.presetId = TSPMessages_2.UUID.internalBinaryRead(reader, reader.uint32(), options, message.presetId);
                    break;
                case /* uint32 new_index */ 4:
                    message.newIndex = reader.uint32();
                    break;
                case /* uint32 old_index */ 5:
                    message.oldIndex = reader.uint32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference theme = 2; */
        if (message.theme)
            TSPMessages_3.Reference.internalBinaryWrite(message.theme, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.UUID preset_id = 3; */
        if (message.presetId)
            TSPMessages_2.UUID.internalBinaryWrite(message.presetId, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* uint32 new_index = 4; */
        if (message.newIndex !== 0)
            writer.tag(4, runtime_1.WireType.Varint).uint32(message.newIndex);
        /* uint32 old_index = 5; */
        if (message.oldIndex !== 0)
            writer.tag(5, runtime_1.WireType.Varint).uint32(message.oldIndex);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ThemeMovePresetCommandArchive
 */
exports.ThemeMovePresetCommandArchive = new ThemeMovePresetCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemeReplaceStylePresetAndDisconnectStylesCommandArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.ThemeReplaceStylePresetAndDisconnectStylesCommandArchive", [
            { no: 1, name: "super", kind: "message", T: () => TSKArchives_1.CommandArchive },
            { no: 2, name: "preset", kind: "message", T: () => TSPMessages_3.Reference },
            { no: 3, name: "replacement_preset", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TSK.CommandArchive super */ 1:
                    message.super = TSKArchives_1.CommandArchive.internalBinaryRead(reader, reader.uint32(), options, message.super);
                    break;
                case /* TSP.Reference preset */ 2:
                    message.preset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.preset);
                    break;
                case /* TSP.Reference replacement_preset */ 3:
                    message.replacementPreset = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.replacementPreset);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TSK.CommandArchive super = 1; */
        if (message.super)
            TSKArchives_1.CommandArchive.internalBinaryWrite(message.super, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference preset = 2; */
        if (message.preset)
            TSPMessages_3.Reference.internalBinaryWrite(message.preset, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TSP.Reference replacement_preset = 3; */
        if (message.replacementPreset)
            TSPMessages_3.Reference.internalBinaryWrite(message.replacementPreset, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.ThemeReplaceStylePresetAndDisconnectStylesCommandArchive
 */
exports.ThemeReplaceStylePresetAndDisconnectStylesCommandArchive = new ThemeReplaceStylePresetAndDisconnectStylesCommandArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CommandPropertyEntryArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.CommandPropertyEntryArchive", [
            { no: 1, name: "property", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 2, name: "type", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 3, name: "integer_value", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 4, name: "float_value", kind: "scalar", opt: true, T: 2 /*ScalarType.FLOAT*/ },
            { no: 5, name: "double_value", kind: "scalar", opt: true, T: 1 /*ScalarType.DOUBLE*/ },
            { no: 6, name: "string_value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "tsp_reference", kind: "message", T: () => TSPMessages_3.Reference }
        ]);
    }
    create(value) {
        const message = { property: 0, type: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* uint32 property */ 1:
                    message.property = reader.uint32();
                    break;
                case /* int32 type */ 2:
                    message.type = reader.int32();
                    break;
                case /* optional int32 integer_value */ 3:
                    message.integerValue = reader.int32();
                    break;
                case /* optional float float_value */ 4:
                    message.floatValue = reader.float();
                    break;
                case /* optional double double_value */ 5:
                    message.doubleValue = reader.double();
                    break;
                case /* optional string string_value */ 6:
                    message.stringValue = reader.string();
                    break;
                case /* optional TSP.Reference tsp_reference */ 7:
                    message.tspReference = TSPMessages_3.Reference.internalBinaryRead(reader, reader.uint32(), options, message.tspReference);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* uint32 property = 1; */
        if (message.property !== 0)
            writer.tag(1, runtime_1.WireType.Varint).uint32(message.property);
        /* int32 type = 2; */
        if (message.type !== 0)
            writer.tag(2, runtime_1.WireType.Varint).int32(message.type);
        /* optional int32 integer_value = 3; */
        if (message.integerValue !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).int32(message.integerValue);
        /* optional float float_value = 4; */
        if (message.floatValue !== undefined)
            writer.tag(4, runtime_1.WireType.Bit32).float(message.floatValue);
        /* optional double double_value = 5; */
        if (message.doubleValue !== undefined)
            writer.tag(5, runtime_1.WireType.Bit64).double(message.doubleValue);
        /* optional string string_value = 6; */
        if (message.stringValue !== undefined)
            writer.tag(6, runtime_1.WireType.LengthDelimited).string(message.stringValue);
        /* optional TSP.Reference tsp_reference = 7; */
        if (message.tspReference)
            TSPMessages_3.Reference.internalBinaryWrite(message.tspReference, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.CommandPropertyEntryArchive
 */
exports.CommandPropertyEntryArchive = new CommandPropertyEntryArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CommandPropertyMapArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSS.CommandPropertyMapArchive", [
            { no: 1, name: "property_entries", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.CommandPropertyEntryArchive }
        ]);
    }
    create(value) {
        const message = { propertyEntries: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSS.CommandPropertyEntryArchive property_entries */ 1:
                    message.propertyEntries.push(exports.CommandPropertyEntryArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSS.CommandPropertyEntryArchive property_entries = 1; */
        for (let i = 0; i < message.propertyEntries.length; i++)
            exports.CommandPropertyEntryArchive.internalBinaryWrite(message.propertyEntries[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSS.CommandPropertyMapArchive
 */
exports.CommandPropertyMapArchive = new CommandPropertyMapArchive$Type();
