// keynote-archives/lib/generated/TSSArchives_sos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpecStringArchive = exports.SpecIntegerArchive = exports.SpecDoubleArchive = exports.SpecColorArchive = exports.SpecBoolArchive = exports.SpecSetStringArchive = exports.SpecSetIntegerArchive = exports.SpecSetDoubleArchive = exports.SpecSetColorArchive = exports.SpecSetBoolArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSPMessages_1 = require("./TSPMessages");
// @generated message type with reflection information, may provide speed optimized methods
class SpecSetBoolArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecSetBoolArchive", [
            { no: 1, name: "value", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 2, name: "unset", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { value: false, unset: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* bool value */ 1:
                    message.value = reader.bool();
                    break;
                case /* bool unset */ 2:
                    message.unset = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* bool value = 1; */
        if (message.value !== false)
            writer.tag(1, runtime_1.WireType.Varint).bool(message.value);
        /* bool unset = 2; */
        if (message.unset !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.unset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecSetBoolArchive
 */
exports.SpecSetBoolArchive = new SpecSetBoolArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecSetColorArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecSetColorArchive", [
            { no: 1, name: "color", kind: "message", T: () => TSPMessages_1.Color },
            { no: 2, name: "unset", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { unset: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSP.Color color */ 1:
                    message.color = TSPMessages_1.Color.internalBinaryRead(reader, reader.uint32(), options, message.color);
                    break;
                case /* bool unset */ 2:
                    message.unset = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSP.Color color = 1; */
        if (message.color)
            TSPMessages_1.Color.internalBinaryWrite(message.color, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool unset = 2; */
        if (message.unset !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.unset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecSetColorArchive
 */
exports.SpecSetColorArchive = new SpecSetColorArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecSetDoubleArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecSetDoubleArchive", [
            { no: 1, name: "value", kind: "scalar", T: 1 /*ScalarType.DOUBLE*/ },
            { no: 2, name: "unset", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { value: 0, unset: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* double value */ 1:
                    message.value = reader.double();
                    break;
                case /* bool unset */ 2:
                    message.unset = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* double value = 1; */
        if (message.value !== 0)
            writer.tag(1, runtime_1.WireType.Bit64).double(message.value);
        /* bool unset = 2; */
        if (message.unset !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.unset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecSetDoubleArchive
 */
exports.SpecSetDoubleArchive = new SpecSetDoubleArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecSetIntegerArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecSetIntegerArchive", [
            { no: 1, name: "value", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 2, name: "unset", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { value: 0, unset: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int32 value */ 1:
                    message.value = reader.int32();
                    break;
                case /* bool unset */ 2:
                    message.unset = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int32 value = 1; */
        if (message.value !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.value);
        /* bool unset = 2; */
        if (message.unset !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.unset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecSetIntegerArchive
 */
exports.SpecSetIntegerArchive = new SpecSetIntegerArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecSetStringArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecSetStringArchive", [
            { no: 1, name: "value", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "unset", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { unset: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional string value */ 1:
                    message.value = reader.string();
                    break;
                case /* bool unset */ 2:
                    message.unset = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional string value = 1; */
        if (message.value !== undefined)
            writer.tag(1, runtime_1.WireType.LengthDelimited).string(message.value);
        /* bool unset = 2; */
        if (message.unset !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.unset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecSetStringArchive
 */
exports.SpecSetStringArchive = new SpecSetStringArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecBoolArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecBoolArchive", [
            { no: 1, name: "spec_set_bool", kind: "message", T: () => exports.SpecSetBoolArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecSetBoolArchive spec_set_bool */ 1:
                    message.specSetBool = exports.SpecSetBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.specSetBool);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecSetBoolArchive spec_set_bool = 1; */
        if (message.specSetBool)
            exports.SpecSetBoolArchive.internalBinaryWrite(message.specSetBool, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecBoolArchive
 */
exports.SpecBoolArchive = new SpecBoolArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecColorArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecColorArchive", [
            { no: 1, name: "spec_set_color", kind: "message", T: () => exports.SpecSetColorArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecSetColorArchive spec_set_color */ 1:
                    message.specSetColor = exports.SpecSetColorArchive.internalBinaryRead(reader, reader.uint32(), options, message.specSetColor);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecSetColorArchive spec_set_color = 1; */
        if (message.specSetColor)
            exports.SpecSetColorArchive.internalBinaryWrite(message.specSetColor, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecColorArchive
 */
exports.SpecColorArchive = new SpecColorArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecDoubleArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecDoubleArchive", [
            { no: 1, name: "spec_set_double", kind: "message", T: () => exports.SpecSetDoubleArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecSetDoubleArchive spec_set_double */ 1:
                    message.specSetDouble = exports.SpecSetDoubleArchive.internalBinaryRead(reader, reader.uint32(), options, message.specSetDouble);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecSetDoubleArchive spec_set_double = 1; */
        if (message.specSetDouble)
            exports.SpecSetDoubleArchive.internalBinaryWrite(message.specSetDouble, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecDoubleArchive
 */
exports.SpecDoubleArchive = new SpecDoubleArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecIntegerArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecIntegerArchive", [
            { no: 1, name: "spec_set_integer", kind: "message", T: () => exports.SpecSetIntegerArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecSetIntegerArchive spec_set_integer */ 1:
                    message.specSetInteger = exports.SpecSetIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.specSetInteger);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecSetIntegerArchive spec_set_integer = 1; */
        if (message.specSetInteger)
            exports.SpecSetIntegerArchive.internalBinaryWrite(message.specSetInteger, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecIntegerArchive
 */
exports.SpecIntegerArchive = new SpecIntegerArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecStringArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSSSOS.SpecStringArchive", [
            { no: 1, name: "spec_set_string", kind: "message", T: () => exports.SpecSetStringArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecSetStringArchive spec_set_string */ 1:
                    message.specSetString = exports.SpecSetStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.specSetString);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecSetStringArchive spec_set_string = 1; */
        if (message.specSetString)
            exports.SpecSetStringArchive.internalBinaryWrite(message.specSetString, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSSSOS.SpecStringArchive
 */
exports.SpecStringArchive = new SpecStringArchive$Type();
