// keynote-archives/lib/generated/TSTArchives_sos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableStylePropertyChangeSetArchive = exports.TableCellStylePropertyChangeSetArchive = exports.SpecTableStrokePresetListArchive = exports.SpecSetTableStrokePresetListArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSSArchives_sos_1 = require("./TSSArchives_sos");
const TSDArchives_sos_1 = require("./TSDArchives_sos");
const TSWPArchives_sos_1 = require("./TSWPArchives_sos");
const TSDArchives_sos_2 = require("./TSDArchives_sos");
const TSSArchives_sos_2 = require("./TSSArchives_sos");
const TSSArchives_sos_3 = require("./TSSArchives_sos");
const TSTStylePropertyArchiving_1 = require("./TSTStylePropertyArchiving");
// @generated message type with reflection information, may provide speed optimized methods
class SpecSetTableStrokePresetListArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSTSOS.SpecSetTableStrokePresetListArchive", [
            { no: 1, name: "table_stroke_preset_list", kind: "message", T: () => TSTStylePropertyArchiving_1.StrokePresetListArchive },
            { no: 2, name: "unset", kind: "scalar", T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = { unset: false };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TST.StrokePresetListArchive table_stroke_preset_list */ 1:
                    message.tableStrokePresetList = TSTStylePropertyArchiving_1.StrokePresetListArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableStrokePresetList);
                    break;
                case /* bool unset */ 2:
                    message.unset = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TST.StrokePresetListArchive table_stroke_preset_list = 1; */
        if (message.tableStrokePresetList)
            TSTStylePropertyArchiving_1.StrokePresetListArchive.internalBinaryWrite(message.tableStrokePresetList, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* bool unset = 2; */
        if (message.unset !== false)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.unset);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSTSOS.SpecSetTableStrokePresetListArchive
 */
exports.SpecSetTableStrokePresetListArchive = new SpecSetTableStrokePresetListArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class SpecTableStrokePresetListArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSTSOS.SpecTableStrokePresetListArchive", [
            { no: 1, name: "spec_set_table_stroke_preset_list", kind: "message", T: () => exports.SpecSetTableStrokePresetListArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSTSOS.SpecSetTableStrokePresetListArchive spec_set_table_stroke_preset_list */ 1:
                    message.specSetTableStrokePresetList = exports.SpecSetTableStrokePresetListArchive.internalBinaryRead(reader, reader.uint32(), options, message.specSetTableStrokePresetList);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSTSOS.SpecSetTableStrokePresetListArchive spec_set_table_stroke_preset_list = 1; */
        if (message.specSetTableStrokePresetList)
            exports.SpecSetTableStrokePresetListArchive.internalBinaryWrite(message.specSetTableStrokePresetList, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSTSOS.SpecTableStrokePresetListArchive
 */
exports.SpecTableStrokePresetListArchive = new SpecTableStrokePresetListArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TableCellStylePropertyChangeSetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSTSOS.TableCellStylePropertyChangeSetArchive", [
            { no: 1, name: "text_wrap", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 2, name: "text_wrap_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "type", kind: "message", T: () => TSSArchives_sos_2.SpecIntegerArchive },
            { no: 4, name: "type_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "fill", kind: "message", T: () => TSDArchives_sos_2.SpecFillArchive },
            { no: 6, name: "fill_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "vertical_alignment", kind: "message", T: () => TSSArchives_sos_2.SpecIntegerArchive },
            { no: 8, name: "vertical_alignment_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 9, name: "padding", kind: "message", T: () => TSWPArchives_sos_1.SpecPaddingArchive },
            { no: 10, name: "padding_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecBoolArchive text_wrap */ 1:
                    message.textWrap = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.textWrap);
                    break;
                case /* optional bool text_wrap_undefined */ 2:
                    message.textWrapUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive type */ 3:
                    message.type = TSSArchives_sos_2.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.type);
                    break;
                case /* optional bool type_undefined */ 4:
                    message.typeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecFillArchive fill */ 5:
                    message.fill = TSDArchives_sos_2.SpecFillArchive.internalBinaryRead(reader, reader.uint32(), options, message.fill);
                    break;
                case /* optional bool fill_undefined */ 6:
                    message.fillUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive vertical_alignment */ 7:
                    message.verticalAlignment = TSSArchives_sos_2.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.verticalAlignment);
                    break;
                case /* optional bool vertical_alignment_undefined */ 8:
                    message.verticalAlignmentUndefined = reader.bool();
                    break;
                case /* optional TSWPSOS.SpecPaddingArchive padding */ 9:
                    message.padding = TSWPArchives_sos_1.SpecPaddingArchive.internalBinaryRead(reader, reader.uint32(), options, message.padding);
                    break;
                case /* optional bool padding_undefined */ 10:
                    message.paddingUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecBoolArchive text_wrap = 1; */
        if (message.textWrap)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.textWrap, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool text_wrap_undefined = 2; */
        if (message.textWrapUndefined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.textWrapUndefined);
        /* optional TSSSOS.SpecIntegerArchive type = 3; */
        if (message.type)
            TSSArchives_sos_2.SpecIntegerArchive.internalBinaryWrite(message.type, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool type_undefined = 4; */
        if (message.typeUndefined !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.typeUndefined);
        /* optional TSDSOS.SpecFillArchive fill = 5; */
        if (message.fill)
            TSDArchives_sos_2.SpecFillArchive.internalBinaryWrite(message.fill, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool fill_undefined = 6; */
        if (message.fillUndefined !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.fillUndefined);
        /* optional TSSSOS.SpecIntegerArchive vertical_alignment = 7; */
        if (message.verticalAlignment)
            TSSArchives_sos_2.SpecIntegerArchive.internalBinaryWrite(message.verticalAlignment, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool vertical_alignment_undefined = 8; */
        if (message.verticalAlignmentUndefined !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).bool(message.verticalAlignmentUndefined);
        /* optional TSWPSOS.SpecPaddingArchive padding = 9; */
        if (message.padding)
            TSWPArchives_sos_1.SpecPaddingArchive.internalBinaryWrite(message.padding, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool padding_undefined = 10; */
        if (message.paddingUndefined !== undefined)
            writer.tag(10, runtime_1.WireType.Varint).bool(message.paddingUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSTSOS.TableCellStylePropertyChangeSetArchive
 */
exports.TableCellStylePropertyChangeSetArchive = new TableCellStylePropertyChangeSetArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TableStylePropertyChangeSetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TSTSOS.TableStylePropertyChangeSetArchive", [
            { no: 1, name: "auto_resize", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 2, name: "auto_resize_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "banded_rows", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 4, name: "banded_rows_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 5, name: "banded_cell_fill", kind: "message", T: () => TSDArchives_sos_2.SpecFillArchive },
            { no: 6, name: "banded_cell_fill_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 7, name: "behaves_like_spreadsheet", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 8, name: "behaves_like_spreadsheet_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 9, name: "body_horizontal_border_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 10, name: "body_horizontal_border_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 11, name: "body_vertical_border_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 12, name: "body_vertical_border_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 13, name: "body_horizontal_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 14, name: "body_horizontal_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 15, name: "body_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 16, name: "body_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 17, name: "header_column_border_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 18, name: "header_column_border_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 19, name: "header_column_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 20, name: "header_column_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 21, name: "header_column_horizontal_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 22, name: "header_column_horizontal_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 23, name: "header_column_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 24, name: "header_column_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 25, name: "header_row_border_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 26, name: "header_row_border_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 27, name: "header_row_horizontal_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 28, name: "header_row_horizontal_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 29, name: "header_row_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 30, name: "header_row_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 31, name: "header_row_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 32, name: "header_row_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 33, name: "footer_row_border_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 34, name: "footer_row_border_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 35, name: "footer_row_horizontal_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 36, name: "footer_row_horizontal_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 37, name: "footer_row_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 38, name: "footer_row_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 39, name: "footer_row_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 40, name: "footer_row_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 41, name: "stroke_preset_list_object", kind: "message", T: () => exports.SpecTableStrokePresetListArchive },
            { no: 42, name: "stroke_preset_list_object_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 43, name: "vertical_strokes_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 44, name: "vertical_strokes_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 45, name: "horizontal_strokes_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 46, name: "horizontal_strokes_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 47, name: "header_row_sep_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 48, name: "header_row_sep_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 49, name: "header_row_verticals_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 50, name: "header_row_verticals_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 51, name: "header_col_sep_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 52, name: "header_col_sep_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 53, name: "header_col_horizontals_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 54, name: "header_col_horizontals_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 55, name: "footer_sep_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 56, name: "footer_sep_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 57, name: "footer_verticals_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 58, name: "footer_verticals_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 59, name: "border_strokes_visible", kind: "message", T: () => TSSArchives_sos_3.SpecBoolArchive },
            { no: 60, name: "border_strokes_visible_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 61, name: "master_font_family", kind: "message", T: () => TSSArchives_sos_1.SpecStringArchive },
            { no: 62, name: "master_font_family_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 63, name: "writing_direction", kind: "message", T: () => TSSArchives_sos_2.SpecIntegerArchive },
            { no: 64, name: "writing_direction_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 65, name: "category_level1_top_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 66, name: "category_level1_top_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 67, name: "category_level2_top_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 68, name: "category_level2_top_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 69, name: "category_level3_top_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 70, name: "category_level3_top_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 71, name: "category_level4_top_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 72, name: "category_level4_top_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 73, name: "category_level5_top_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 74, name: "category_level5_top_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 75, name: "category_level1_label_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 76, name: "category_level1_label_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 77, name: "category_level2_label_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 78, name: "category_level2_label_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 79, name: "category_level3_label_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 80, name: "category_level3_label_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 81, name: "category_level4_label_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 82, name: "category_level4_label_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 83, name: "category_level5_label_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 84, name: "category_level5_label_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 85, name: "category_level1_bottom_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 86, name: "category_level1_bottom_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 87, name: "category_level2_bottom_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 88, name: "category_level2_bottom_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 89, name: "category_level3_bottom_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 90, name: "category_level3_bottom_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 91, name: "category_level4_bottom_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 92, name: "category_level4_bottom_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 93, name: "category_level5_bottom_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 94, name: "category_level5_bottom_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 95, name: "category_level1_interior_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 96, name: "category_level1_interior_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 97, name: "category_level2_interior_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 98, name: "category_level2_interior_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 99, name: "category_level3_interior_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 100, name: "category_level3_interior_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 101, name: "category_level4_interior_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 102, name: "category_level4_interior_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 103, name: "category_level5_interior_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 104, name: "category_level5_interior_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 105, name: "body_pivot_group_horizontal_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 106, name: "body_pivot_group_horizontal_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 107, name: "body_pivot_group_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 108, name: "body_pivot_group_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 109, name: "body_pivot_de_emphasis_horizontal_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 110, name: "body_pivot_de_emphasis_horizontal_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 111, name: "body_pivot_emphasis_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 112, name: "body_pivot_emphasis_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 113, name: "header_column_pivot_group_horizontal_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 114, name: "header_column_pivot_group_horizontal_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 115, name: "header_column_pivot_group_de_emphasis_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 116, name: "header_column_pivot_group_de_emphasis_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 117, name: "header_column_pivot_separator_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 118, name: "header_column_pivot_separator_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 119, name: "header_row_pivot_group_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 120, name: "header_row_pivot_group_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 121, name: "header_row_pivot_group_de_emphasis_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 122, name: "header_row_pivot_group_de_emphasis_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 123, name: "header_row_pivot_title_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 124, name: "header_row_pivot_title_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 125, name: "footer_row_pivot_group_vertical_stroke", kind: "message", T: () => TSDArchives_sos_1.SpecStrokeArchive },
            { no: 126, name: "footer_row_pivot_group_vertical_stroke_undefined", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSSSOS.SpecBoolArchive auto_resize */ 1:
                    message.autoResize = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.autoResize);
                    break;
                case /* optional bool auto_resize_undefined */ 2:
                    message.autoResizeUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive banded_rows */ 3:
                    message.bandedRows = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.bandedRows);
                    break;
                case /* optional bool banded_rows_undefined */ 4:
                    message.bandedRowsUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecFillArchive banded_cell_fill */ 5:
                    message.bandedCellFill = TSDArchives_sos_2.SpecFillArchive.internalBinaryRead(reader, reader.uint32(), options, message.bandedCellFill);
                    break;
                case /* optional bool banded_cell_fill_undefined */ 6:
                    message.bandedCellFillUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive behaves_like_spreadsheet */ 7:
                    message.behavesLikeSpreadsheet = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.behavesLikeSpreadsheet);
                    break;
                case /* optional bool behaves_like_spreadsheet_undefined */ 8:
                    message.behavesLikeSpreadsheetUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_horizontal_border_stroke */ 9:
                    message.bodyHorizontalBorderStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyHorizontalBorderStroke);
                    break;
                case /* optional bool body_horizontal_border_stroke_undefined */ 10:
                    message.bodyHorizontalBorderStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_vertical_border_stroke */ 11:
                    message.bodyVerticalBorderStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyVerticalBorderStroke);
                    break;
                case /* optional bool body_vertical_border_stroke_undefined */ 12:
                    message.bodyVerticalBorderStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_horizontal_stroke */ 13:
                    message.bodyHorizontalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyHorizontalStroke);
                    break;
                case /* optional bool body_horizontal_stroke_undefined */ 14:
                    message.bodyHorizontalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_vertical_stroke */ 15:
                    message.bodyVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyVerticalStroke);
                    break;
                case /* optional bool body_vertical_stroke_undefined */ 16:
                    message.bodyVerticalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_column_border_stroke */ 17:
                    message.headerColumnBorderStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnBorderStroke);
                    break;
                case /* optional bool header_column_border_stroke_undefined */ 18:
                    message.headerColumnBorderStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_column_vertical_stroke */ 19:
                    message.headerColumnVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnVerticalStroke);
                    break;
                case /* optional bool header_column_vertical_stroke_undefined */ 20:
                    message.headerColumnVerticalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_column_horizontal_stroke */ 21:
                    message.headerColumnHorizontalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnHorizontalStroke);
                    break;
                case /* optional bool header_column_horizontal_stroke_undefined */ 22:
                    message.headerColumnHorizontalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_column_separator_stroke */ 23:
                    message.headerColumnSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnSeparatorStroke);
                    break;
                case /* optional bool header_column_separator_stroke_undefined */ 24:
                    message.headerColumnSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_row_border_stroke */ 25:
                    message.headerRowBorderStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowBorderStroke);
                    break;
                case /* optional bool header_row_border_stroke_undefined */ 26:
                    message.headerRowBorderStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_row_horizontal_stroke */ 27:
                    message.headerRowHorizontalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowHorizontalStroke);
                    break;
                case /* optional bool header_row_horizontal_stroke_undefined */ 28:
                    message.headerRowHorizontalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_row_vertical_stroke */ 29:
                    message.headerRowVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowVerticalStroke);
                    break;
                case /* optional bool header_row_vertical_stroke_undefined */ 30:
                    message.headerRowVerticalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_row_separator_stroke */ 31:
                    message.headerRowSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowSeparatorStroke);
                    break;
                case /* optional bool header_row_separator_stroke_undefined */ 32:
                    message.headerRowSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive footer_row_border_stroke */ 33:
                    message.footerRowBorderStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowBorderStroke);
                    break;
                case /* optional bool footer_row_border_stroke_undefined */ 34:
                    message.footerRowBorderStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive footer_row_horizontal_stroke */ 35:
                    message.footerRowHorizontalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowHorizontalStroke);
                    break;
                case /* optional bool footer_row_horizontal_stroke_undefined */ 36:
                    message.footerRowHorizontalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive footer_row_vertical_stroke */ 37:
                    message.footerRowVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowVerticalStroke);
                    break;
                case /* optional bool footer_row_vertical_stroke_undefined */ 38:
                    message.footerRowVerticalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive footer_row_separator_stroke */ 39:
                    message.footerRowSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowSeparatorStroke);
                    break;
                case /* optional bool footer_row_separator_stroke_undefined */ 40:
                    message.footerRowSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSTSOS.SpecTableStrokePresetListArchive stroke_preset_list_object */ 41:
                    message.strokePresetListObject = exports.SpecTableStrokePresetListArchive.internalBinaryRead(reader, reader.uint32(), options, message.strokePresetListObject);
                    break;
                case /* optional bool stroke_preset_list_object_undefined */ 42:
                    message.strokePresetListObjectUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive vertical_strokes_visible */ 43:
                    message.verticalStrokesVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.verticalStrokesVisible);
                    break;
                case /* optional bool vertical_strokes_visible_undefined */ 44:
                    message.verticalStrokesVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive horizontal_strokes_visible */ 45:
                    message.horizontalStrokesVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.horizontalStrokesVisible);
                    break;
                case /* optional bool horizontal_strokes_visible_undefined */ 46:
                    message.horizontalStrokesVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive header_row_sep_visible */ 47:
                    message.headerRowSepVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowSepVisible);
                    break;
                case /* optional bool header_row_sep_visible_undefined */ 48:
                    message.headerRowSepVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive header_row_verticals_visible */ 49:
                    message.headerRowVerticalsVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowVerticalsVisible);
                    break;
                case /* optional bool header_row_verticals_visible_undefined */ 50:
                    message.headerRowVerticalsVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive header_col_sep_visible */ 51:
                    message.headerColSepVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColSepVisible);
                    break;
                case /* optional bool header_col_sep_visible_undefined */ 52:
                    message.headerColSepVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive header_col_horizontals_visible */ 53:
                    message.headerColHorizontalsVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColHorizontalsVisible);
                    break;
                case /* optional bool header_col_horizontals_visible_undefined */ 54:
                    message.headerColHorizontalsVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive footer_sep_visible */ 55:
                    message.footerSepVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerSepVisible);
                    break;
                case /* optional bool footer_sep_visible_undefined */ 56:
                    message.footerSepVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive footer_verticals_visible */ 57:
                    message.footerVerticalsVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerVerticalsVisible);
                    break;
                case /* optional bool footer_verticals_visible_undefined */ 58:
                    message.footerVerticalsVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecBoolArchive border_strokes_visible */ 59:
                    message.borderStrokesVisible = TSSArchives_sos_3.SpecBoolArchive.internalBinaryRead(reader, reader.uint32(), options, message.borderStrokesVisible);
                    break;
                case /* optional bool border_strokes_visible_undefined */ 60:
                    message.borderStrokesVisibleUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecStringArchive master_font_family */ 61:
                    message.masterFontFamily = TSSArchives_sos_1.SpecStringArchive.internalBinaryRead(reader, reader.uint32(), options, message.masterFontFamily);
                    break;
                case /* optional bool master_font_family_undefined */ 62:
                    message.masterFontFamilyUndefined = reader.bool();
                    break;
                case /* optional TSSSOS.SpecIntegerArchive writing_direction */ 63:
                    message.writingDirection = TSSArchives_sos_2.SpecIntegerArchive.internalBinaryRead(reader, reader.uint32(), options, message.writingDirection);
                    break;
                case /* optional bool writing_direction_undefined */ 64:
                    message.writingDirectionUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level1_top_stroke */ 65:
                    message.categoryLevel1TopStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1TopStroke);
                    break;
                case /* optional bool category_level1_top_stroke_undefined */ 66:
                    message.categoryLevel1TopStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level2_top_stroke */ 67:
                    message.categoryLevel2TopStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2TopStroke);
                    break;
                case /* optional bool category_level2_top_stroke_undefined */ 68:
                    message.categoryLevel2TopStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level3_top_stroke */ 69:
                    message.categoryLevel3TopStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3TopStroke);
                    break;
                case /* optional bool category_level3_top_stroke_undefined */ 70:
                    message.categoryLevel3TopStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level4_top_stroke */ 71:
                    message.categoryLevel4TopStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4TopStroke);
                    break;
                case /* optional bool category_level4_top_stroke_undefined */ 72:
                    message.categoryLevel4TopStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level5_top_stroke */ 73:
                    message.categoryLevel5TopStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5TopStroke);
                    break;
                case /* optional bool category_level5_top_stroke_undefined */ 74:
                    message.categoryLevel5TopStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level1_label_separator_stroke */ 75:
                    message.categoryLevel1LabelSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1LabelSeparatorStroke);
                    break;
                case /* optional bool category_level1_label_separator_stroke_undefined */ 76:
                    message.categoryLevel1LabelSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level2_label_separator_stroke */ 77:
                    message.categoryLevel2LabelSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2LabelSeparatorStroke);
                    break;
                case /* optional bool category_level2_label_separator_stroke_undefined */ 78:
                    message.categoryLevel2LabelSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level3_label_separator_stroke */ 79:
                    message.categoryLevel3LabelSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3LabelSeparatorStroke);
                    break;
                case /* optional bool category_level3_label_separator_stroke_undefined */ 80:
                    message.categoryLevel3LabelSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level4_label_separator_stroke */ 81:
                    message.categoryLevel4LabelSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4LabelSeparatorStroke);
                    break;
                case /* optional bool category_level4_label_separator_stroke_undefined */ 82:
                    message.categoryLevel4LabelSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level5_label_separator_stroke */ 83:
                    message.categoryLevel5LabelSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5LabelSeparatorStroke);
                    break;
                case /* optional bool category_level5_label_separator_stroke_undefined */ 84:
                    message.categoryLevel5LabelSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level1_bottom_stroke */ 85:
                    message.categoryLevel1BottomStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1BottomStroke);
                    break;
                case /* optional bool category_level1_bottom_stroke_undefined */ 86:
                    message.categoryLevel1BottomStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level2_bottom_stroke */ 87:
                    message.categoryLevel2BottomStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2BottomStroke);
                    break;
                case /* optional bool category_level2_bottom_stroke_undefined */ 88:
                    message.categoryLevel2BottomStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level3_bottom_stroke */ 89:
                    message.categoryLevel3BottomStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3BottomStroke);
                    break;
                case /* optional bool category_level3_bottom_stroke_undefined */ 90:
                    message.categoryLevel3BottomStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level4_bottom_stroke */ 91:
                    message.categoryLevel4BottomStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4BottomStroke);
                    break;
                case /* optional bool category_level4_bottom_stroke_undefined */ 92:
                    message.categoryLevel4BottomStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level5_bottom_stroke */ 93:
                    message.categoryLevel5BottomStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5BottomStroke);
                    break;
                case /* optional bool category_level5_bottom_stroke_undefined */ 94:
                    message.categoryLevel5BottomStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level1_interior_stroke */ 95:
                    message.categoryLevel1InteriorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1InteriorStroke);
                    break;
                case /* optional bool category_level1_interior_stroke_undefined */ 96:
                    message.categoryLevel1InteriorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level2_interior_stroke */ 97:
                    message.categoryLevel2InteriorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2InteriorStroke);
                    break;
                case /* optional bool category_level2_interior_stroke_undefined */ 98:
                    message.categoryLevel2InteriorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level3_interior_stroke */ 99:
                    message.categoryLevel3InteriorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3InteriorStroke);
                    break;
                case /* optional bool category_level3_interior_stroke_undefined */ 100:
                    message.categoryLevel3InteriorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level4_interior_stroke */ 101:
                    message.categoryLevel4InteriorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4InteriorStroke);
                    break;
                case /* optional bool category_level4_interior_stroke_undefined */ 102:
                    message.categoryLevel4InteriorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive category_level5_interior_stroke */ 103:
                    message.categoryLevel5InteriorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5InteriorStroke);
                    break;
                case /* optional bool category_level5_interior_stroke_undefined */ 104:
                    message.categoryLevel5InteriorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_pivot_group_horizontal_stroke */ 105:
                    message.bodyPivotGroupHorizontalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyPivotGroupHorizontalStroke);
                    break;
                case /* optional bool body_pivot_group_horizontal_stroke_undefined */ 106:
                    message.bodyPivotGroupHorizontalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_pivot_group_vertical_stroke */ 107:
                    message.bodyPivotGroupVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyPivotGroupVerticalStroke);
                    break;
                case /* optional bool body_pivot_group_vertical_stroke_undefined */ 108:
                    message.bodyPivotGroupVerticalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_pivot_de_emphasis_horizontal_stroke */ 109:
                    message.bodyPivotDeEmphasisHorizontalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyPivotDeEmphasisHorizontalStroke);
                    break;
                case /* optional bool body_pivot_de_emphasis_horizontal_stroke_undefined */ 110:
                    message.bodyPivotDeEmphasisHorizontalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive body_pivot_emphasis_vertical_stroke */ 111:
                    message.bodyPivotEmphasisVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bodyPivotEmphasisVerticalStroke);
                    break;
                case /* optional bool body_pivot_emphasis_vertical_stroke_undefined */ 112:
                    message.bodyPivotEmphasisVerticalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_column_pivot_group_horizontal_stroke */ 113:
                    message.headerColumnPivotGroupHorizontalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnPivotGroupHorizontalStroke);
                    break;
                case /* optional bool header_column_pivot_group_horizontal_stroke_undefined */ 114:
                    message.headerColumnPivotGroupHorizontalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_column_pivot_group_de_emphasis_stroke */ 115:
                    message.headerColumnPivotGroupDeEmphasisStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnPivotGroupDeEmphasisStroke);
                    break;
                case /* optional bool header_column_pivot_group_de_emphasis_stroke_undefined */ 116:
                    message.headerColumnPivotGroupDeEmphasisStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_column_pivot_separator_stroke */ 117:
                    message.headerColumnPivotSeparatorStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnPivotSeparatorStroke);
                    break;
                case /* optional bool header_column_pivot_separator_stroke_undefined */ 118:
                    message.headerColumnPivotSeparatorStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_row_pivot_group_vertical_stroke */ 119:
                    message.headerRowPivotGroupVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowPivotGroupVerticalStroke);
                    break;
                case /* optional bool header_row_pivot_group_vertical_stroke_undefined */ 120:
                    message.headerRowPivotGroupVerticalStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_row_pivot_group_de_emphasis_stroke */ 121:
                    message.headerRowPivotGroupDeEmphasisStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowPivotGroupDeEmphasisStroke);
                    break;
                case /* optional bool header_row_pivot_group_de_emphasis_stroke_undefined */ 122:
                    message.headerRowPivotGroupDeEmphasisStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive header_row_pivot_title_stroke */ 123:
                    message.headerRowPivotTitleStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowPivotTitleStroke);
                    break;
                case /* optional bool header_row_pivot_title_stroke_undefined */ 124:
                    message.headerRowPivotTitleStrokeUndefined = reader.bool();
                    break;
                case /* optional TSDSOS.SpecStrokeArchive footer_row_pivot_group_vertical_stroke */ 125:
                    message.footerRowPivotGroupVerticalStroke = TSDArchives_sos_1.SpecStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowPivotGroupVerticalStroke);
                    break;
                case /* optional bool footer_row_pivot_group_vertical_stroke_undefined */ 126:
                    message.footerRowPivotGroupVerticalStrokeUndefined = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSSSOS.SpecBoolArchive auto_resize = 1; */
        if (message.autoResize)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.autoResize, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool auto_resize_undefined = 2; */
        if (message.autoResizeUndefined !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.autoResizeUndefined);
        /* optional TSSSOS.SpecBoolArchive banded_rows = 3; */
        if (message.bandedRows)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.bandedRows, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool banded_rows_undefined = 4; */
        if (message.bandedRowsUndefined !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.bandedRowsUndefined);
        /* optional TSDSOS.SpecFillArchive banded_cell_fill = 5; */
        if (message.bandedCellFill)
            TSDArchives_sos_2.SpecFillArchive.internalBinaryWrite(message.bandedCellFill, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool banded_cell_fill_undefined = 6; */
        if (message.bandedCellFillUndefined !== undefined)
            writer.tag(6, runtime_1.WireType.Varint).bool(message.bandedCellFillUndefined);
        /* optional TSSSOS.SpecBoolArchive behaves_like_spreadsheet = 7; */
        if (message.behavesLikeSpreadsheet)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.behavesLikeSpreadsheet, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool behaves_like_spreadsheet_undefined = 8; */
        if (message.behavesLikeSpreadsheetUndefined !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).bool(message.behavesLikeSpreadsheetUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_horizontal_border_stroke = 9; */
        if (message.bodyHorizontalBorderStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyHorizontalBorderStroke, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_horizontal_border_stroke_undefined = 10; */
        if (message.bodyHorizontalBorderStrokeUndefined !== undefined)
            writer.tag(10, runtime_1.WireType.Varint).bool(message.bodyHorizontalBorderStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_vertical_border_stroke = 11; */
        if (message.bodyVerticalBorderStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyVerticalBorderStroke, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_vertical_border_stroke_undefined = 12; */
        if (message.bodyVerticalBorderStrokeUndefined !== undefined)
            writer.tag(12, runtime_1.WireType.Varint).bool(message.bodyVerticalBorderStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_horizontal_stroke = 13; */
        if (message.bodyHorizontalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyHorizontalStroke, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_horizontal_stroke_undefined = 14; */
        if (message.bodyHorizontalStrokeUndefined !== undefined)
            writer.tag(14, runtime_1.WireType.Varint).bool(message.bodyHorizontalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_vertical_stroke = 15; */
        if (message.bodyVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyVerticalStroke, writer.tag(15, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_vertical_stroke_undefined = 16; */
        if (message.bodyVerticalStrokeUndefined !== undefined)
            writer.tag(16, runtime_1.WireType.Varint).bool(message.bodyVerticalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_column_border_stroke = 17; */
        if (message.headerColumnBorderStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerColumnBorderStroke, writer.tag(17, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_column_border_stroke_undefined = 18; */
        if (message.headerColumnBorderStrokeUndefined !== undefined)
            writer.tag(18, runtime_1.WireType.Varint).bool(message.headerColumnBorderStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_column_vertical_stroke = 19; */
        if (message.headerColumnVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerColumnVerticalStroke, writer.tag(19, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_column_vertical_stroke_undefined = 20; */
        if (message.headerColumnVerticalStrokeUndefined !== undefined)
            writer.tag(20, runtime_1.WireType.Varint).bool(message.headerColumnVerticalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_column_horizontal_stroke = 21; */
        if (message.headerColumnHorizontalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerColumnHorizontalStroke, writer.tag(21, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_column_horizontal_stroke_undefined = 22; */
        if (message.headerColumnHorizontalStrokeUndefined !== undefined)
            writer.tag(22, runtime_1.WireType.Varint).bool(message.headerColumnHorizontalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_column_separator_stroke = 23; */
        if (message.headerColumnSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerColumnSeparatorStroke, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_column_separator_stroke_undefined = 24; */
        if (message.headerColumnSeparatorStrokeUndefined !== undefined)
            writer.tag(24, runtime_1.WireType.Varint).bool(message.headerColumnSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_row_border_stroke = 25; */
        if (message.headerRowBorderStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerRowBorderStroke, writer.tag(25, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_border_stroke_undefined = 26; */
        if (message.headerRowBorderStrokeUndefined !== undefined)
            writer.tag(26, runtime_1.WireType.Varint).bool(message.headerRowBorderStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_row_horizontal_stroke = 27; */
        if (message.headerRowHorizontalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerRowHorizontalStroke, writer.tag(27, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_horizontal_stroke_undefined = 28; */
        if (message.headerRowHorizontalStrokeUndefined !== undefined)
            writer.tag(28, runtime_1.WireType.Varint).bool(message.headerRowHorizontalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_row_vertical_stroke = 29; */
        if (message.headerRowVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerRowVerticalStroke, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_vertical_stroke_undefined = 30; */
        if (message.headerRowVerticalStrokeUndefined !== undefined)
            writer.tag(30, runtime_1.WireType.Varint).bool(message.headerRowVerticalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_row_separator_stroke = 31; */
        if (message.headerRowSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerRowSeparatorStroke, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_separator_stroke_undefined = 32; */
        if (message.headerRowSeparatorStrokeUndefined !== undefined)
            writer.tag(32, runtime_1.WireType.Varint).bool(message.headerRowSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive footer_row_border_stroke = 33; */
        if (message.footerRowBorderStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.footerRowBorderStroke, writer.tag(33, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool footer_row_border_stroke_undefined = 34; */
        if (message.footerRowBorderStrokeUndefined !== undefined)
            writer.tag(34, runtime_1.WireType.Varint).bool(message.footerRowBorderStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive footer_row_horizontal_stroke = 35; */
        if (message.footerRowHorizontalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.footerRowHorizontalStroke, writer.tag(35, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool footer_row_horizontal_stroke_undefined = 36; */
        if (message.footerRowHorizontalStrokeUndefined !== undefined)
            writer.tag(36, runtime_1.WireType.Varint).bool(message.footerRowHorizontalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive footer_row_vertical_stroke = 37; */
        if (message.footerRowVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.footerRowVerticalStroke, writer.tag(37, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool footer_row_vertical_stroke_undefined = 38; */
        if (message.footerRowVerticalStrokeUndefined !== undefined)
            writer.tag(38, runtime_1.WireType.Varint).bool(message.footerRowVerticalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive footer_row_separator_stroke = 39; */
        if (message.footerRowSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.footerRowSeparatorStroke, writer.tag(39, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool footer_row_separator_stroke_undefined = 40; */
        if (message.footerRowSeparatorStrokeUndefined !== undefined)
            writer.tag(40, runtime_1.WireType.Varint).bool(message.footerRowSeparatorStrokeUndefined);
        /* optional TSTSOS.SpecTableStrokePresetListArchive stroke_preset_list_object = 41; */
        if (message.strokePresetListObject)
            exports.SpecTableStrokePresetListArchive.internalBinaryWrite(message.strokePresetListObject, writer.tag(41, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool stroke_preset_list_object_undefined = 42; */
        if (message.strokePresetListObjectUndefined !== undefined)
            writer.tag(42, runtime_1.WireType.Varint).bool(message.strokePresetListObjectUndefined);
        /* optional TSSSOS.SpecBoolArchive vertical_strokes_visible = 43; */
        if (message.verticalStrokesVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.verticalStrokesVisible, writer.tag(43, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool vertical_strokes_visible_undefined = 44; */
        if (message.verticalStrokesVisibleUndefined !== undefined)
            writer.tag(44, runtime_1.WireType.Varint).bool(message.verticalStrokesVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive horizontal_strokes_visible = 45; */
        if (message.horizontalStrokesVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.horizontalStrokesVisible, writer.tag(45, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool horizontal_strokes_visible_undefined = 46; */
        if (message.horizontalStrokesVisibleUndefined !== undefined)
            writer.tag(46, runtime_1.WireType.Varint).bool(message.horizontalStrokesVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive header_row_sep_visible = 47; */
        if (message.headerRowSepVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.headerRowSepVisible, writer.tag(47, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_sep_visible_undefined = 48; */
        if (message.headerRowSepVisibleUndefined !== undefined)
            writer.tag(48, runtime_1.WireType.Varint).bool(message.headerRowSepVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive header_row_verticals_visible = 49; */
        if (message.headerRowVerticalsVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.headerRowVerticalsVisible, writer.tag(49, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_verticals_visible_undefined = 50; */
        if (message.headerRowVerticalsVisibleUndefined !== undefined)
            writer.tag(50, runtime_1.WireType.Varint).bool(message.headerRowVerticalsVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive header_col_sep_visible = 51; */
        if (message.headerColSepVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.headerColSepVisible, writer.tag(51, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_col_sep_visible_undefined = 52; */
        if (message.headerColSepVisibleUndefined !== undefined)
            writer.tag(52, runtime_1.WireType.Varint).bool(message.headerColSepVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive header_col_horizontals_visible = 53; */
        if (message.headerColHorizontalsVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.headerColHorizontalsVisible, writer.tag(53, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_col_horizontals_visible_undefined = 54; */
        if (message.headerColHorizontalsVisibleUndefined !== undefined)
            writer.tag(54, runtime_1.WireType.Varint).bool(message.headerColHorizontalsVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive footer_sep_visible = 55; */
        if (message.footerSepVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.footerSepVisible, writer.tag(55, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool footer_sep_visible_undefined = 56; */
        if (message.footerSepVisibleUndefined !== undefined)
            writer.tag(56, runtime_1.WireType.Varint).bool(message.footerSepVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive footer_verticals_visible = 57; */
        if (message.footerVerticalsVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.footerVerticalsVisible, writer.tag(57, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool footer_verticals_visible_undefined = 58; */
        if (message.footerVerticalsVisibleUndefined !== undefined)
            writer.tag(58, runtime_1.WireType.Varint).bool(message.footerVerticalsVisibleUndefined);
        /* optional TSSSOS.SpecBoolArchive border_strokes_visible = 59; */
        if (message.borderStrokesVisible)
            TSSArchives_sos_3.SpecBoolArchive.internalBinaryWrite(message.borderStrokesVisible, writer.tag(59, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool border_strokes_visible_undefined = 60; */
        if (message.borderStrokesVisibleUndefined !== undefined)
            writer.tag(60, runtime_1.WireType.Varint).bool(message.borderStrokesVisibleUndefined);
        /* optional TSSSOS.SpecStringArchive master_font_family = 61; */
        if (message.masterFontFamily)
            TSSArchives_sos_1.SpecStringArchive.internalBinaryWrite(message.masterFontFamily, writer.tag(61, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool master_font_family_undefined = 62; */
        if (message.masterFontFamilyUndefined !== undefined)
            writer.tag(62, runtime_1.WireType.Varint).bool(message.masterFontFamilyUndefined);
        /* optional TSSSOS.SpecIntegerArchive writing_direction = 63; */
        if (message.writingDirection)
            TSSArchives_sos_2.SpecIntegerArchive.internalBinaryWrite(message.writingDirection, writer.tag(63, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool writing_direction_undefined = 64; */
        if (message.writingDirectionUndefined !== undefined)
            writer.tag(64, runtime_1.WireType.Varint).bool(message.writingDirectionUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level1_top_stroke = 65; */
        if (message.categoryLevel1TopStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel1TopStroke, writer.tag(65, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level1_top_stroke_undefined = 66; */
        if (message.categoryLevel1TopStrokeUndefined !== undefined)
            writer.tag(66, runtime_1.WireType.Varint).bool(message.categoryLevel1TopStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level2_top_stroke = 67; */
        if (message.categoryLevel2TopStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel2TopStroke, writer.tag(67, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level2_top_stroke_undefined = 68; */
        if (message.categoryLevel2TopStrokeUndefined !== undefined)
            writer.tag(68, runtime_1.WireType.Varint).bool(message.categoryLevel2TopStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level3_top_stroke = 69; */
        if (message.categoryLevel3TopStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel3TopStroke, writer.tag(69, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level3_top_stroke_undefined = 70; */
        if (message.categoryLevel3TopStrokeUndefined !== undefined)
            writer.tag(70, runtime_1.WireType.Varint).bool(message.categoryLevel3TopStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level4_top_stroke = 71; */
        if (message.categoryLevel4TopStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel4TopStroke, writer.tag(71, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level4_top_stroke_undefined = 72; */
        if (message.categoryLevel4TopStrokeUndefined !== undefined)
            writer.tag(72, runtime_1.WireType.Varint).bool(message.categoryLevel4TopStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level5_top_stroke = 73; */
        if (message.categoryLevel5TopStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel5TopStroke, writer.tag(73, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level5_top_stroke_undefined = 74; */
        if (message.categoryLevel5TopStrokeUndefined !== undefined)
            writer.tag(74, runtime_1.WireType.Varint).bool(message.categoryLevel5TopStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level1_label_separator_stroke = 75; */
        if (message.categoryLevel1LabelSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel1LabelSeparatorStroke, writer.tag(75, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level1_label_separator_stroke_undefined = 76; */
        if (message.categoryLevel1LabelSeparatorStrokeUndefined !== undefined)
            writer.tag(76, runtime_1.WireType.Varint).bool(message.categoryLevel1LabelSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level2_label_separator_stroke = 77; */
        if (message.categoryLevel2LabelSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel2LabelSeparatorStroke, writer.tag(77, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level2_label_separator_stroke_undefined = 78; */
        if (message.categoryLevel2LabelSeparatorStrokeUndefined !== undefined)
            writer.tag(78, runtime_1.WireType.Varint).bool(message.categoryLevel2LabelSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level3_label_separator_stroke = 79; */
        if (message.categoryLevel3LabelSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel3LabelSeparatorStroke, writer.tag(79, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level3_label_separator_stroke_undefined = 80; */
        if (message.categoryLevel3LabelSeparatorStrokeUndefined !== undefined)
            writer.tag(80, runtime_1.WireType.Varint).bool(message.categoryLevel3LabelSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level4_label_separator_stroke = 81; */
        if (message.categoryLevel4LabelSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel4LabelSeparatorStroke, writer.tag(81, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level4_label_separator_stroke_undefined = 82; */
        if (message.categoryLevel4LabelSeparatorStrokeUndefined !== undefined)
            writer.tag(82, runtime_1.WireType.Varint).bool(message.categoryLevel4LabelSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level5_label_separator_stroke = 83; */
        if (message.categoryLevel5LabelSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel5LabelSeparatorStroke, writer.tag(83, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level5_label_separator_stroke_undefined = 84; */
        if (message.categoryLevel5LabelSeparatorStrokeUndefined !== undefined)
            writer.tag(84, runtime_1.WireType.Varint).bool(message.categoryLevel5LabelSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level1_bottom_stroke = 85; */
        if (message.categoryLevel1BottomStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel1BottomStroke, writer.tag(85, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level1_bottom_stroke_undefined = 86; */
        if (message.categoryLevel1BottomStrokeUndefined !== undefined)
            writer.tag(86, runtime_1.WireType.Varint).bool(message.categoryLevel1BottomStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level2_bottom_stroke = 87; */
        if (message.categoryLevel2BottomStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel2BottomStroke, writer.tag(87, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level2_bottom_stroke_undefined = 88; */
        if (message.categoryLevel2BottomStrokeUndefined !== undefined)
            writer.tag(88, runtime_1.WireType.Varint).bool(message.categoryLevel2BottomStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level3_bottom_stroke = 89; */
        if (message.categoryLevel3BottomStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel3BottomStroke, writer.tag(89, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level3_bottom_stroke_undefined = 90; */
        if (message.categoryLevel3BottomStrokeUndefined !== undefined)
            writer.tag(90, runtime_1.WireType.Varint).bool(message.categoryLevel3BottomStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level4_bottom_stroke = 91; */
        if (message.categoryLevel4BottomStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel4BottomStroke, writer.tag(91, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level4_bottom_stroke_undefined = 92; */
        if (message.categoryLevel4BottomStrokeUndefined !== undefined)
            writer.tag(92, runtime_1.WireType.Varint).bool(message.categoryLevel4BottomStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level5_bottom_stroke = 93; */
        if (message.categoryLevel5BottomStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel5BottomStroke, writer.tag(93, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level5_bottom_stroke_undefined = 94; */
        if (message.categoryLevel5BottomStrokeUndefined !== undefined)
            writer.tag(94, runtime_1.WireType.Varint).bool(message.categoryLevel5BottomStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level1_interior_stroke = 95; */
        if (message.categoryLevel1InteriorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel1InteriorStroke, writer.tag(95, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level1_interior_stroke_undefined = 96; */
        if (message.categoryLevel1InteriorStrokeUndefined !== undefined)
            writer.tag(96, runtime_1.WireType.Varint).bool(message.categoryLevel1InteriorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level2_interior_stroke = 97; */
        if (message.categoryLevel2InteriorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel2InteriorStroke, writer.tag(97, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level2_interior_stroke_undefined = 98; */
        if (message.categoryLevel2InteriorStrokeUndefined !== undefined)
            writer.tag(98, runtime_1.WireType.Varint).bool(message.categoryLevel2InteriorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level3_interior_stroke = 99; */
        if (message.categoryLevel3InteriorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel3InteriorStroke, writer.tag(99, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level3_interior_stroke_undefined = 100; */
        if (message.categoryLevel3InteriorStrokeUndefined !== undefined)
            writer.tag(100, runtime_1.WireType.Varint).bool(message.categoryLevel3InteriorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level4_interior_stroke = 101; */
        if (message.categoryLevel4InteriorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel4InteriorStroke, writer.tag(101, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level4_interior_stroke_undefined = 102; */
        if (message.categoryLevel4InteriorStrokeUndefined !== undefined)
            writer.tag(102, runtime_1.WireType.Varint).bool(message.categoryLevel4InteriorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive category_level5_interior_stroke = 103; */
        if (message.categoryLevel5InteriorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.categoryLevel5InteriorStroke, writer.tag(103, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool category_level5_interior_stroke_undefined = 104; */
        if (message.categoryLevel5InteriorStrokeUndefined !== undefined)
            writer.tag(104, runtime_1.WireType.Varint).bool(message.categoryLevel5InteriorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_pivot_group_horizontal_stroke = 105; */
        if (message.bodyPivotGroupHorizontalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyPivotGroupHorizontalStroke, writer.tag(105, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_pivot_group_horizontal_stroke_undefined = 106; */
        if (message.bodyPivotGroupHorizontalStrokeUndefined !== undefined)
            writer.tag(106, runtime_1.WireType.Varint).bool(message.bodyPivotGroupHorizontalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_pivot_group_vertical_stroke = 107; */
        if (message.bodyPivotGroupVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyPivotGroupVerticalStroke, writer.tag(107, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_pivot_group_vertical_stroke_undefined = 108; */
        if (message.bodyPivotGroupVerticalStrokeUndefined !== undefined)
            writer.tag(108, runtime_1.WireType.Varint).bool(message.bodyPivotGroupVerticalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_pivot_de_emphasis_horizontal_stroke = 109; */
        if (message.bodyPivotDeEmphasisHorizontalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyPivotDeEmphasisHorizontalStroke, writer.tag(109, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_pivot_de_emphasis_horizontal_stroke_undefined = 110; */
        if (message.bodyPivotDeEmphasisHorizontalStrokeUndefined !== undefined)
            writer.tag(110, runtime_1.WireType.Varint).bool(message.bodyPivotDeEmphasisHorizontalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive body_pivot_emphasis_vertical_stroke = 111; */
        if (message.bodyPivotEmphasisVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.bodyPivotEmphasisVerticalStroke, writer.tag(111, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool body_pivot_emphasis_vertical_stroke_undefined = 112; */
        if (message.bodyPivotEmphasisVerticalStrokeUndefined !== undefined)
            writer.tag(112, runtime_1.WireType.Varint).bool(message.bodyPivotEmphasisVerticalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_column_pivot_group_horizontal_stroke = 113; */
        if (message.headerColumnPivotGroupHorizontalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerColumnPivotGroupHorizontalStroke, writer.tag(113, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_column_pivot_group_horizontal_stroke_undefined = 114; */
        if (message.headerColumnPivotGroupHorizontalStrokeUndefined !== undefined)
            writer.tag(114, runtime_1.WireType.Varint).bool(message.headerColumnPivotGroupHorizontalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_column_pivot_group_de_emphasis_stroke = 115; */
        if (message.headerColumnPivotGroupDeEmphasisStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerColumnPivotGroupDeEmphasisStroke, writer.tag(115, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_column_pivot_group_de_emphasis_stroke_undefined = 116; */
        if (message.headerColumnPivotGroupDeEmphasisStrokeUndefined !== undefined)
            writer.tag(116, runtime_1.WireType.Varint).bool(message.headerColumnPivotGroupDeEmphasisStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_column_pivot_separator_stroke = 117; */
        if (message.headerColumnPivotSeparatorStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerColumnPivotSeparatorStroke, writer.tag(117, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_column_pivot_separator_stroke_undefined = 118; */
        if (message.headerColumnPivotSeparatorStrokeUndefined !== undefined)
            writer.tag(118, runtime_1.WireType.Varint).bool(message.headerColumnPivotSeparatorStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_row_pivot_group_vertical_stroke = 119; */
        if (message.headerRowPivotGroupVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerRowPivotGroupVerticalStroke, writer.tag(119, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_pivot_group_vertical_stroke_undefined = 120; */
        if (message.headerRowPivotGroupVerticalStrokeUndefined !== undefined)
            writer.tag(120, runtime_1.WireType.Varint).bool(message.headerRowPivotGroupVerticalStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_row_pivot_group_de_emphasis_stroke = 121; */
        if (message.headerRowPivotGroupDeEmphasisStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerRowPivotGroupDeEmphasisStroke, writer.tag(121, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_pivot_group_de_emphasis_stroke_undefined = 122; */
        if (message.headerRowPivotGroupDeEmphasisStrokeUndefined !== undefined)
            writer.tag(122, runtime_1.WireType.Varint).bool(message.headerRowPivotGroupDeEmphasisStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive header_row_pivot_title_stroke = 123; */
        if (message.headerRowPivotTitleStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.headerRowPivotTitleStroke, writer.tag(123, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool header_row_pivot_title_stroke_undefined = 124; */
        if (message.headerRowPivotTitleStrokeUndefined !== undefined)
            writer.tag(124, runtime_1.WireType.Varint).bool(message.headerRowPivotTitleStrokeUndefined);
        /* optional TSDSOS.SpecStrokeArchive footer_row_pivot_group_vertical_stroke = 125; */
        if (message.footerRowPivotGroupVerticalStroke)
            TSDArchives_sos_1.SpecStrokeArchive.internalBinaryWrite(message.footerRowPivotGroupVerticalStroke, writer.tag(125, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool footer_row_pivot_group_vertical_stroke_undefined = 126; */
        if (message.footerRowPivotGroupVerticalStrokeUndefined !== undefined)
            writer.tag(126, runtime_1.WireType.Varint).bool(message.footerRowPivotGroupVerticalStrokeUndefined);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TSTSOS.TableStylePropertyChangeSetArchive
 */
exports.TableStylePropertyChangeSetArchive = new TableStylePropertyChangeSetArchive$Type();
