// keynote-archives/lib/generated/TSTStylePropertyArchiving.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThemePresetsArchive = exports.TableStrokePresetArchive = exports.TableStylePresetArchive = exports.TableStylePropertiesArchive = exports.StrokePresetListArchive = exports.StrokePresetDataArchive = exports.Deprecated_StrokePresetDataArchive = exports.CellStylePropertiesArchive = exports.Deprecated_TableStrokeArchive = void 0;
const runtime_1 = require("@protobuf-ts/runtime");
const runtime_2 = require("@protobuf-ts/runtime");
const runtime_3 = require("@protobuf-ts/runtime");
const runtime_4 = require("@protobuf-ts/runtime");
const runtime_5 = require("@protobuf-ts/runtime");
const TSPMessages_1 = require("./TSPMessages");
const TSWPArchives_1 = require("./TSWPArchives");
const TSWPArchives_2 = require("./TSWPArchives");
const TSDArchives_1 = require("./TSDArchives");
const TSDArchives_2 = require("./TSDArchives");
// @generated message type with reflection information, may provide speed optimized methods
class Deprecated_TableStrokeArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.Deprecated_TableStrokeArchive", [
            { no: 1, name: "stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 2, name: "background", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 3, name: "opacity", kind: "scalar", opt: true, T: 2 /*ScalarType.FLOAT*/ },
            { no: 4, name: "empty", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSD.StrokeArchive stroke */ 1:
                    message.stroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.stroke);
                    break;
                case /* optional bool background = 2 [deprecated = true];*/ 2:
                    message.background = reader.bool();
                    break;
                case /* optional float opacity = 3 [deprecated = true];*/ 3:
                    message.opacity = reader.float();
                    break;
                case /* optional bool empty = 4 [deprecated = true];*/ 4:
                    message.empty = reader.bool();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSD.StrokeArchive stroke = 1; */
        if (message.stroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.stroke, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool background = 2 [deprecated = true]; */
        if (message.background !== undefined)
            writer.tag(2, runtime_1.WireType.Varint).bool(message.background);
        /* optional float opacity = 3 [deprecated = true]; */
        if (message.opacity !== undefined)
            writer.tag(3, runtime_1.WireType.Bit32).float(message.opacity);
        /* optional bool empty = 4 [deprecated = true]; */
        if (message.empty !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).bool(message.empty);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.Deprecated_TableStrokeArchive
 */
exports.Deprecated_TableStrokeArchive = new Deprecated_TableStrokeArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CellStylePropertiesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.CellStylePropertiesArchive", [
            { no: 1, name: "cell_fill", kind: "message", T: () => TSDArchives_1.FillArchive },
            { no: 3, name: "text_wrap", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "deprecated_top_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 5, name: "deprecated_right_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 6, name: "deprecated_bottom_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 7, name: "deprecated_left_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 8, name: "vertical_alignment", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 9, name: "padding", kind: "message", T: () => TSWPArchives_2.PaddingArchive },
            { no: 10, name: "top_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 11, name: "right_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 12, name: "bottom_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 13, name: "left_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSD.FillArchive cell_fill */ 1:
                    message.cellFill = TSDArchives_1.FillArchive.internalBinaryRead(reader, reader.uint32(), options, message.cellFill);
                    break;
                case /* optional bool text_wrap */ 3:
                    message.textWrap = reader.bool();
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_top_stroke */ 4:
                    message.deprecatedTopStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedTopStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_right_stroke */ 5:
                    message.deprecatedRightStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedRightStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_bottom_stroke */ 6:
                    message.deprecatedBottomStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedBottomStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_left_stroke */ 7:
                    message.deprecatedLeftStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedLeftStroke);
                    break;
                case /* optional int32 vertical_alignment */ 8:
                    message.verticalAlignment = reader.int32();
                    break;
                case /* optional TSWP.PaddingArchive padding */ 9:
                    message.padding = TSWPArchives_2.PaddingArchive.internalBinaryRead(reader, reader.uint32(), options, message.padding);
                    break;
                case /* optional TSD.StrokeArchive top_stroke */ 10:
                    message.topStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.topStroke);
                    break;
                case /* optional TSD.StrokeArchive right_stroke */ 11:
                    message.rightStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.rightStroke);
                    break;
                case /* optional TSD.StrokeArchive bottom_stroke */ 12:
                    message.bottomStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.bottomStroke);
                    break;
                case /* optional TSD.StrokeArchive left_stroke */ 13:
                    message.leftStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.leftStroke);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSD.FillArchive cell_fill = 1; */
        if (message.cellFill)
            TSDArchives_1.FillArchive.internalBinaryWrite(message.cellFill, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool text_wrap = 3; */
        if (message.textWrap !== undefined)
            writer.tag(3, runtime_1.WireType.Varint).bool(message.textWrap);
        /* optional TST.Deprecated_TableStrokeArchive deprecated_top_stroke = 4; */
        if (message.deprecatedTopStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedTopStroke, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_right_stroke = 5; */
        if (message.deprecatedRightStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedRightStroke, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_bottom_stroke = 6; */
        if (message.deprecatedBottomStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedBottomStroke, writer.tag(6, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_left_stroke = 7; */
        if (message.deprecatedLeftStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedLeftStroke, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int32 vertical_alignment = 8; */
        if (message.verticalAlignment !== undefined)
            writer.tag(8, runtime_1.WireType.Varint).int32(message.verticalAlignment);
        /* optional TSWP.PaddingArchive padding = 9; */
        if (message.padding)
            TSWPArchives_2.PaddingArchive.internalBinaryWrite(message.padding, writer.tag(9, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive top_stroke = 10; */
        if (message.topStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.topStroke, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive right_stroke = 11; */
        if (message.rightStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.rightStroke, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive bottom_stroke = 12; */
        if (message.bottomStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.bottomStroke, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive left_stroke = 13; */
        if (message.leftStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.leftStroke, writer.tag(13, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.CellStylePropertiesArchive
 */
exports.CellStylePropertiesArchive = new CellStylePropertiesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Deprecated_StrokePresetDataArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.Deprecated_StrokePresetDataArchive", [
            { no: 2, name: "deprecated_horizontal_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 1, name: "deprecated_vertical_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 3, name: "deprecated_exterior_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 5, name: "deprecated_visible_mask", kind: "scalar", T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = { deprecatedVisibleMask: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* TST.Deprecated_TableStrokeArchive deprecated_horizontal_stroke */ 2:
                    message.deprecatedHorizontalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHorizontalStroke);
                    break;
                case /* TST.Deprecated_TableStrokeArchive deprecated_vertical_stroke */ 1:
                    message.deprecatedVerticalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedVerticalStroke);
                    break;
                case /* TST.Deprecated_TableStrokeArchive deprecated_exterior_stroke */ 3:
                    message.deprecatedExteriorStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedExteriorStroke);
                    break;
                case /* int32 deprecated_visible_mask */ 5:
                    message.deprecatedVisibleMask = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* TST.Deprecated_TableStrokeArchive deprecated_horizontal_stroke = 2; */
        if (message.deprecatedHorizontalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHorizontalStroke, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TST.Deprecated_TableStrokeArchive deprecated_vertical_stroke = 1; */
        if (message.deprecatedVerticalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedVerticalStroke, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* TST.Deprecated_TableStrokeArchive deprecated_exterior_stroke = 3; */
        if (message.deprecatedExteriorStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedExteriorStroke, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* int32 deprecated_visible_mask = 5; */
        if (message.deprecatedVisibleMask !== 0)
            writer.tag(5, runtime_1.WireType.Varint).int32(message.deprecatedVisibleMask);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.Deprecated_StrokePresetDataArchive
 */
exports.Deprecated_StrokePresetDataArchive = new Deprecated_StrokePresetDataArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StrokePresetDataArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.StrokePresetDataArchive", [
            { no: 1, name: "horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 2, name: "vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 3, name: "exterior_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 4, name: "visible_mask", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional TSD.StrokeArchive horizontal_stroke */ 1:
                    message.horizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.horizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive vertical_stroke */ 2:
                    message.verticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.verticalStroke);
                    break;
                case /* optional TSD.StrokeArchive exterior_stroke */ 3:
                    message.exteriorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.exteriorStroke);
                    break;
                case /* optional int32 visible_mask */ 4:
                    message.visibleMask = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional TSD.StrokeArchive horizontal_stroke = 1; */
        if (message.horizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.horizontalStroke, writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive vertical_stroke = 2; */
        if (message.verticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.verticalStroke, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive exterior_stroke = 3; */
        if (message.exteriorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.exteriorStroke, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional int32 visible_mask = 4; */
        if (message.visibleMask !== undefined)
            writer.tag(4, runtime_1.WireType.Varint).int32(message.visibleMask);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.StrokePresetDataArchive
 */
exports.StrokePresetDataArchive = new StrokePresetDataArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class StrokePresetListArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.StrokePresetListArchive", [
            { no: 1, name: "count", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 2, name: "deprecated_preset", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.Deprecated_StrokePresetDataArchive },
            { no: 3, name: "preset", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => exports.StrokePresetDataArchive }
        ]);
    }
    create(value) {
        const message = { count: 0, deprecatedPreset: [], preset: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int32 count */ 1:
                    message.count = reader.int32();
                    break;
                case /* repeated TST.Deprecated_StrokePresetDataArchive deprecated_preset */ 2:
                    message.deprecatedPreset.push(exports.Deprecated_StrokePresetDataArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TST.StrokePresetDataArchive preset */ 3:
                    message.preset.push(exports.StrokePresetDataArchive.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int32 count = 1; */
        if (message.count !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.count);
        /* repeated TST.Deprecated_StrokePresetDataArchive deprecated_preset = 2; */
        for (let i = 0; i < message.deprecatedPreset.length; i++)
            exports.Deprecated_StrokePresetDataArchive.internalBinaryWrite(message.deprecatedPreset[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TST.StrokePresetDataArchive preset = 3; */
        for (let i = 0; i < message.preset.length; i++)
            exports.StrokePresetDataArchive.internalBinaryWrite(message.preset[i], writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.StrokePresetListArchive
 */
exports.StrokePresetListArchive = new StrokePresetListArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TableStylePropertiesArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.TableStylePropertiesArchive", [
            { no: 1, name: "banded_rows", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 2, name: "banded_fill", kind: "message", T: () => TSDArchives_1.FillArchive },
            { no: 21, name: "behaves_like_spreadsheet", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 22, name: "auto_resize", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 4, name: "deprecated_header_row_separator_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 5, name: "deprecated_header_row_border_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 23, name: "deprecated_header_row_horizontal_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 24, name: "deprecated_header_row_vertical_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 7, name: "deprecated_header_column_border_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 8, name: "deprecated_header_column_separator_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 25, name: "deprecated_header_column_horizontal_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 26, name: "deprecated_header_column_vertical_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 10, name: "deprecated_footer_row_separator_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 11, name: "deprecated_footer_row_border_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 27, name: "deprecated_footer_row_horizontal_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 28, name: "deprecated_footer_row_vertical_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 12, name: "deprecated_table_body_horizontal_border_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 29, name: "deprecated_table_body_vertical_border_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 30, name: "deprecated_table_body_horizontal_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 31, name: "deprecated_table_body_vertical_stroke", kind: "message", T: () => exports.Deprecated_TableStrokeArchive },
            { no: 32, name: "stroke_preset_list", kind: "message", T: () => exports.StrokePresetListArchive },
            { no: 33, name: "v_strokes_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 34, name: "h_strokes_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 35, name: "hr_separator_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 36, name: "hc_separator_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 37, name: "footer_separator_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 38, name: "table_border_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 39, name: "table_header_border_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 42, name: "table_hc_divider_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 43, name: "table_hr_divider_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 44, name: "table_footer_divider_visible", kind: "scalar", opt: true, T: 8 /*ScalarType.BOOL*/ },
            { no: 40, name: "OBSOLETE_master_font_size", kind: "scalar", jsonName: "OBSOLETEMasterFontSize", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 41, name: "master_font_family", kind: "scalar", opt: true, T: 9 /*ScalarType.STRING*/ },
            { no: 45, name: "writing_direction", kind: "enum", opt: true, T: () => ["TSWP.WritingDirectionType", TSWPArchives_1.WritingDirectionType] },
            { no: 46, name: "header_row_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 47, name: "header_row_border_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 48, name: "header_row_horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 49, name: "header_row_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 50, name: "header_column_border_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 51, name: "header_column_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 52, name: "header_column_horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 53, name: "header_column_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 54, name: "footer_row_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 55, name: "footer_row_border_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 56, name: "footer_row_horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 57, name: "footer_row_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 58, name: "table_body_horizontal_border_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 59, name: "table_body_vertical_border_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 60, name: "table_body_horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 61, name: "table_body_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 62, name: "category_level1_top_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 63, name: "category_level2_top_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 64, name: "category_level3_top_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 65, name: "category_level4_top_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 66, name: "category_level5_top_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 67, name: "category_level1_bottom_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 68, name: "category_level2_bottom_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 69, name: "category_level3_bottom_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 70, name: "category_level4_bottom_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 71, name: "category_level5_bottom_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 72, name: "category_level1_interior_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 73, name: "category_level2_interior_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 74, name: "category_level3_interior_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 75, name: "category_level4_interior_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 76, name: "category_level5_interior_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 77, name: "category_level1_label_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 78, name: "category_level2_label_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 79, name: "category_level3_label_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 80, name: "category_level4_label_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 81, name: "category_level5_label_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 82, name: "table_body_pivot_group_horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 83, name: "table_body_pivot_group_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 84, name: "table_body_pivot_deemphasis_horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 85, name: "table_body_pivot_emphasis_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 86, name: "header_column_pivot_group_horizontal_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 87, name: "header_column_pivot_group_deemphasis_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 88, name: "header_column_pivot_separator_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 89, name: "header_row_pivot_group_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 90, name: "header_row_pivot_group_deemphasis_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 91, name: "header_row_pivot_title_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive },
            { no: 92, name: "footer_row_pivot_group_vertical_stroke", kind: "message", T: () => TSDArchives_2.StrokeArchive }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional bool banded_rows */ 1:
                    message.bandedRows = reader.bool();
                    break;
                case /* optional TSD.FillArchive banded_fill */ 2:
                    message.bandedFill = TSDArchives_1.FillArchive.internalBinaryRead(reader, reader.uint32(), options, message.bandedFill);
                    break;
                case /* optional bool behaves_like_spreadsheet */ 21:
                    message.behavesLikeSpreadsheet = reader.bool();
                    break;
                case /* optional bool auto_resize */ 22:
                    message.autoResize = reader.bool();
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_separator_stroke */ 4:
                    message.deprecatedHeaderRowSeparatorStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderRowSeparatorStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_border_stroke */ 5:
                    message.deprecatedHeaderRowBorderStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderRowBorderStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_horizontal_stroke */ 23:
                    message.deprecatedHeaderRowHorizontalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderRowHorizontalStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_vertical_stroke */ 24:
                    message.deprecatedHeaderRowVerticalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderRowVerticalStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_border_stroke */ 7:
                    message.deprecatedHeaderColumnBorderStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderColumnBorderStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_separator_stroke */ 8:
                    message.deprecatedHeaderColumnSeparatorStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderColumnSeparatorStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_horizontal_stroke */ 25:
                    message.deprecatedHeaderColumnHorizontalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderColumnHorizontalStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_vertical_stroke */ 26:
                    message.deprecatedHeaderColumnVerticalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedHeaderColumnVerticalStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_separator_stroke */ 10:
                    message.deprecatedFooterRowSeparatorStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedFooterRowSeparatorStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_border_stroke */ 11:
                    message.deprecatedFooterRowBorderStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedFooterRowBorderStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_horizontal_stroke */ 27:
                    message.deprecatedFooterRowHorizontalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedFooterRowHorizontalStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_vertical_stroke */ 28:
                    message.deprecatedFooterRowVerticalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedFooterRowVerticalStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_horizontal_border_stroke */ 12:
                    message.deprecatedTableBodyHorizontalBorderStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedTableBodyHorizontalBorderStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_vertical_border_stroke */ 29:
                    message.deprecatedTableBodyVerticalBorderStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedTableBodyVerticalBorderStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_horizontal_stroke */ 30:
                    message.deprecatedTableBodyHorizontalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedTableBodyHorizontalStroke);
                    break;
                case /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_vertical_stroke */ 31:
                    message.deprecatedTableBodyVerticalStroke = exports.Deprecated_TableStrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.deprecatedTableBodyVerticalStroke);
                    break;
                case /* optional TST.StrokePresetListArchive stroke_preset_list */ 32:
                    message.strokePresetList = exports.StrokePresetListArchive.internalBinaryRead(reader, reader.uint32(), options, message.strokePresetList);
                    break;
                case /* optional bool v_strokes_visible */ 33:
                    message.vStrokesVisible = reader.bool();
                    break;
                case /* optional bool h_strokes_visible */ 34:
                    message.hStrokesVisible = reader.bool();
                    break;
                case /* optional bool hr_separator_visible */ 35:
                    message.hrSeparatorVisible = reader.bool();
                    break;
                case /* optional bool hc_separator_visible */ 36:
                    message.hcSeparatorVisible = reader.bool();
                    break;
                case /* optional bool footer_separator_visible */ 37:
                    message.footerSeparatorVisible = reader.bool();
                    break;
                case /* optional bool table_border_visible */ 38:
                    message.tableBorderVisible = reader.bool();
                    break;
                case /* optional bool table_header_border_visible */ 39:
                    message.tableHeaderBorderVisible = reader.bool();
                    break;
                case /* optional bool table_hc_divider_visible */ 42:
                    message.tableHcDividerVisible = reader.bool();
                    break;
                case /* optional bool table_hr_divider_visible */ 43:
                    message.tableHrDividerVisible = reader.bool();
                    break;
                case /* optional bool table_footer_divider_visible */ 44:
                    message.tableFooterDividerVisible = reader.bool();
                    break;
                case /* optional int32 OBSOLETE_master_font_size = 40 [json_name = "OBSOLETEMasterFontSize"];*/ 40:
                    message.oBSOLETEMasterFontSize = reader.int32();
                    break;
                case /* optional string master_font_family */ 41:
                    message.masterFontFamily = reader.string();
                    break;
                case /* optional TSWP.WritingDirectionType writing_direction */ 45:
                    message.writingDirection = reader.int32();
                    break;
                case /* optional TSD.StrokeArchive header_row_separator_stroke */ 46:
                    message.headerRowSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive header_row_border_stroke */ 47:
                    message.headerRowBorderStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowBorderStroke);
                    break;
                case /* optional TSD.StrokeArchive header_row_horizontal_stroke */ 48:
                    message.headerRowHorizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowHorizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive header_row_vertical_stroke */ 49:
                    message.headerRowVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowVerticalStroke);
                    break;
                case /* optional TSD.StrokeArchive header_column_border_stroke */ 50:
                    message.headerColumnBorderStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnBorderStroke);
                    break;
                case /* optional TSD.StrokeArchive header_column_separator_stroke */ 51:
                    message.headerColumnSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive header_column_horizontal_stroke */ 52:
                    message.headerColumnHorizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnHorizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive header_column_vertical_stroke */ 53:
                    message.headerColumnVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnVerticalStroke);
                    break;
                case /* optional TSD.StrokeArchive footer_row_separator_stroke */ 54:
                    message.footerRowSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive footer_row_border_stroke */ 55:
                    message.footerRowBorderStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowBorderStroke);
                    break;
                case /* optional TSD.StrokeArchive footer_row_horizontal_stroke */ 56:
                    message.footerRowHorizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowHorizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive footer_row_vertical_stroke */ 57:
                    message.footerRowVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowVerticalStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_horizontal_border_stroke */ 58:
                    message.tableBodyHorizontalBorderStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyHorizontalBorderStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_vertical_border_stroke */ 59:
                    message.tableBodyVerticalBorderStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyVerticalBorderStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_horizontal_stroke */ 60:
                    message.tableBodyHorizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyHorizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_vertical_stroke */ 61:
                    message.tableBodyVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyVerticalStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level1_top_stroke */ 62:
                    message.categoryLevel1TopStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1TopStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level2_top_stroke */ 63:
                    message.categoryLevel2TopStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2TopStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level3_top_stroke */ 64:
                    message.categoryLevel3TopStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3TopStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level4_top_stroke */ 65:
                    message.categoryLevel4TopStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4TopStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level5_top_stroke */ 66:
                    message.categoryLevel5TopStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5TopStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level1_bottom_stroke */ 67:
                    message.categoryLevel1BottomStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1BottomStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level2_bottom_stroke */ 68:
                    message.categoryLevel2BottomStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2BottomStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level3_bottom_stroke */ 69:
                    message.categoryLevel3BottomStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3BottomStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level4_bottom_stroke */ 70:
                    message.categoryLevel4BottomStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4BottomStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level5_bottom_stroke */ 71:
                    message.categoryLevel5BottomStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5BottomStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level1_interior_stroke */ 72:
                    message.categoryLevel1InteriorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1InteriorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level2_interior_stroke */ 73:
                    message.categoryLevel2InteriorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2InteriorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level3_interior_stroke */ 74:
                    message.categoryLevel3InteriorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3InteriorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level4_interior_stroke */ 75:
                    message.categoryLevel4InteriorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4InteriorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level5_interior_stroke */ 76:
                    message.categoryLevel5InteriorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5InteriorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level1_label_separator_stroke */ 77:
                    message.categoryLevel1LabelSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel1LabelSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level2_label_separator_stroke */ 78:
                    message.categoryLevel2LabelSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel2LabelSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level3_label_separator_stroke */ 79:
                    message.categoryLevel3LabelSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel3LabelSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level4_label_separator_stroke */ 80:
                    message.categoryLevel4LabelSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel4LabelSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive category_level5_label_separator_stroke */ 81:
                    message.categoryLevel5LabelSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.categoryLevel5LabelSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_pivot_group_horizontal_stroke */ 82:
                    message.tableBodyPivotGroupHorizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyPivotGroupHorizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_pivot_group_vertical_stroke */ 83:
                    message.tableBodyPivotGroupVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyPivotGroupVerticalStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_pivot_deemphasis_horizontal_stroke */ 84:
                    message.tableBodyPivotDeemphasisHorizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyPivotDeemphasisHorizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive table_body_pivot_emphasis_vertical_stroke */ 85:
                    message.tableBodyPivotEmphasisVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.tableBodyPivotEmphasisVerticalStroke);
                    break;
                case /* optional TSD.StrokeArchive header_column_pivot_group_horizontal_stroke */ 86:
                    message.headerColumnPivotGroupHorizontalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnPivotGroupHorizontalStroke);
                    break;
                case /* optional TSD.StrokeArchive header_column_pivot_group_deemphasis_stroke */ 87:
                    message.headerColumnPivotGroupDeemphasisStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnPivotGroupDeemphasisStroke);
                    break;
                case /* optional TSD.StrokeArchive header_column_pivot_separator_stroke */ 88:
                    message.headerColumnPivotSeparatorStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerColumnPivotSeparatorStroke);
                    break;
                case /* optional TSD.StrokeArchive header_row_pivot_group_vertical_stroke */ 89:
                    message.headerRowPivotGroupVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowPivotGroupVerticalStroke);
                    break;
                case /* optional TSD.StrokeArchive header_row_pivot_group_deemphasis_stroke */ 90:
                    message.headerRowPivotGroupDeemphasisStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowPivotGroupDeemphasisStroke);
                    break;
                case /* optional TSD.StrokeArchive header_row_pivot_title_stroke */ 91:
                    message.headerRowPivotTitleStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.headerRowPivotTitleStroke);
                    break;
                case /* optional TSD.StrokeArchive footer_row_pivot_group_vertical_stroke */ 92:
                    message.footerRowPivotGroupVerticalStroke = TSDArchives_2.StrokeArchive.internalBinaryRead(reader, reader.uint32(), options, message.footerRowPivotGroupVerticalStroke);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional bool banded_rows = 1; */
        if (message.bandedRows !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).bool(message.bandedRows);
        /* optional TSD.FillArchive banded_fill = 2; */
        if (message.bandedFill)
            TSDArchives_1.FillArchive.internalBinaryWrite(message.bandedFill, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool behaves_like_spreadsheet = 21; */
        if (message.behavesLikeSpreadsheet !== undefined)
            writer.tag(21, runtime_1.WireType.Varint).bool(message.behavesLikeSpreadsheet);
        /* optional bool auto_resize = 22; */
        if (message.autoResize !== undefined)
            writer.tag(22, runtime_1.WireType.Varint).bool(message.autoResize);
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_separator_stroke = 4; */
        if (message.deprecatedHeaderRowSeparatorStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderRowSeparatorStroke, writer.tag(4, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_border_stroke = 5; */
        if (message.deprecatedHeaderRowBorderStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderRowBorderStroke, writer.tag(5, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_horizontal_stroke = 23; */
        if (message.deprecatedHeaderRowHorizontalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderRowHorizontalStroke, writer.tag(23, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_row_vertical_stroke = 24; */
        if (message.deprecatedHeaderRowVerticalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderRowVerticalStroke, writer.tag(24, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_border_stroke = 7; */
        if (message.deprecatedHeaderColumnBorderStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderColumnBorderStroke, writer.tag(7, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_separator_stroke = 8; */
        if (message.deprecatedHeaderColumnSeparatorStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderColumnSeparatorStroke, writer.tag(8, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_horizontal_stroke = 25; */
        if (message.deprecatedHeaderColumnHorizontalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderColumnHorizontalStroke, writer.tag(25, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_header_column_vertical_stroke = 26; */
        if (message.deprecatedHeaderColumnVerticalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedHeaderColumnVerticalStroke, writer.tag(26, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_separator_stroke = 10; */
        if (message.deprecatedFooterRowSeparatorStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedFooterRowSeparatorStroke, writer.tag(10, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_border_stroke = 11; */
        if (message.deprecatedFooterRowBorderStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedFooterRowBorderStroke, writer.tag(11, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_horizontal_stroke = 27; */
        if (message.deprecatedFooterRowHorizontalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedFooterRowHorizontalStroke, writer.tag(27, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_footer_row_vertical_stroke = 28; */
        if (message.deprecatedFooterRowVerticalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedFooterRowVerticalStroke, writer.tag(28, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_horizontal_border_stroke = 12; */
        if (message.deprecatedTableBodyHorizontalBorderStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedTableBodyHorizontalBorderStroke, writer.tag(12, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_vertical_border_stroke = 29; */
        if (message.deprecatedTableBodyVerticalBorderStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedTableBodyVerticalBorderStroke, writer.tag(29, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_horizontal_stroke = 30; */
        if (message.deprecatedTableBodyHorizontalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedTableBodyHorizontalStroke, writer.tag(30, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.Deprecated_TableStrokeArchive deprecated_table_body_vertical_stroke = 31; */
        if (message.deprecatedTableBodyVerticalStroke)
            exports.Deprecated_TableStrokeArchive.internalBinaryWrite(message.deprecatedTableBodyVerticalStroke, writer.tag(31, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TST.StrokePresetListArchive stroke_preset_list = 32; */
        if (message.strokePresetList)
            exports.StrokePresetListArchive.internalBinaryWrite(message.strokePresetList, writer.tag(32, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional bool v_strokes_visible = 33; */
        if (message.vStrokesVisible !== undefined)
            writer.tag(33, runtime_1.WireType.Varint).bool(message.vStrokesVisible);
        /* optional bool h_strokes_visible = 34; */
        if (message.hStrokesVisible !== undefined)
            writer.tag(34, runtime_1.WireType.Varint).bool(message.hStrokesVisible);
        /* optional bool hr_separator_visible = 35; */
        if (message.hrSeparatorVisible !== undefined)
            writer.tag(35, runtime_1.WireType.Varint).bool(message.hrSeparatorVisible);
        /* optional bool hc_separator_visible = 36; */
        if (message.hcSeparatorVisible !== undefined)
            writer.tag(36, runtime_1.WireType.Varint).bool(message.hcSeparatorVisible);
        /* optional bool footer_separator_visible = 37; */
        if (message.footerSeparatorVisible !== undefined)
            writer.tag(37, runtime_1.WireType.Varint).bool(message.footerSeparatorVisible);
        /* optional bool table_border_visible = 38; */
        if (message.tableBorderVisible !== undefined)
            writer.tag(38, runtime_1.WireType.Varint).bool(message.tableBorderVisible);
        /* optional bool table_header_border_visible = 39; */
        if (message.tableHeaderBorderVisible !== undefined)
            writer.tag(39, runtime_1.WireType.Varint).bool(message.tableHeaderBorderVisible);
        /* optional bool table_hc_divider_visible = 42; */
        if (message.tableHcDividerVisible !== undefined)
            writer.tag(42, runtime_1.WireType.Varint).bool(message.tableHcDividerVisible);
        /* optional bool table_hr_divider_visible = 43; */
        if (message.tableHrDividerVisible !== undefined)
            writer.tag(43, runtime_1.WireType.Varint).bool(message.tableHrDividerVisible);
        /* optional bool table_footer_divider_visible = 44; */
        if (message.tableFooterDividerVisible !== undefined)
            writer.tag(44, runtime_1.WireType.Varint).bool(message.tableFooterDividerVisible);
        /* optional int32 OBSOLETE_master_font_size = 40 [json_name = "OBSOLETEMasterFontSize"]; */
        if (message.oBSOLETEMasterFontSize !== undefined)
            writer.tag(40, runtime_1.WireType.Varint).int32(message.oBSOLETEMasterFontSize);
        /* optional string master_font_family = 41; */
        if (message.masterFontFamily !== undefined)
            writer.tag(41, runtime_1.WireType.LengthDelimited).string(message.masterFontFamily);
        /* optional TSWP.WritingDirectionType writing_direction = 45; */
        if (message.writingDirection !== undefined)
            writer.tag(45, runtime_1.WireType.Varint).int32(message.writingDirection);
        /* optional TSD.StrokeArchive header_row_separator_stroke = 46; */
        if (message.headerRowSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerRowSeparatorStroke, writer.tag(46, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_row_border_stroke = 47; */
        if (message.headerRowBorderStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerRowBorderStroke, writer.tag(47, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_row_horizontal_stroke = 48; */
        if (message.headerRowHorizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerRowHorizontalStroke, writer.tag(48, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_row_vertical_stroke = 49; */
        if (message.headerRowVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerRowVerticalStroke, writer.tag(49, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_column_border_stroke = 50; */
        if (message.headerColumnBorderStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerColumnBorderStroke, writer.tag(50, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_column_separator_stroke = 51; */
        if (message.headerColumnSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerColumnSeparatorStroke, writer.tag(51, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_column_horizontal_stroke = 52; */
        if (message.headerColumnHorizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerColumnHorizontalStroke, writer.tag(52, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_column_vertical_stroke = 53; */
        if (message.headerColumnVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerColumnVerticalStroke, writer.tag(53, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive footer_row_separator_stroke = 54; */
        if (message.footerRowSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.footerRowSeparatorStroke, writer.tag(54, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive footer_row_border_stroke = 55; */
        if (message.footerRowBorderStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.footerRowBorderStroke, writer.tag(55, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive footer_row_horizontal_stroke = 56; */
        if (message.footerRowHorizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.footerRowHorizontalStroke, writer.tag(56, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive footer_row_vertical_stroke = 57; */
        if (message.footerRowVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.footerRowVerticalStroke, writer.tag(57, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_horizontal_border_stroke = 58; */
        if (message.tableBodyHorizontalBorderStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyHorizontalBorderStroke, writer.tag(58, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_vertical_border_stroke = 59; */
        if (message.tableBodyVerticalBorderStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyVerticalBorderStroke, writer.tag(59, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_horizontal_stroke = 60; */
        if (message.tableBodyHorizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyHorizontalStroke, writer.tag(60, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_vertical_stroke = 61; */
        if (message.tableBodyVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyVerticalStroke, writer.tag(61, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level1_top_stroke = 62; */
        if (message.categoryLevel1TopStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel1TopStroke, writer.tag(62, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level2_top_stroke = 63; */
        if (message.categoryLevel2TopStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel2TopStroke, writer.tag(63, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level3_top_stroke = 64; */
        if (message.categoryLevel3TopStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel3TopStroke, writer.tag(64, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level4_top_stroke = 65; */
        if (message.categoryLevel4TopStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel4TopStroke, writer.tag(65, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level5_top_stroke = 66; */
        if (message.categoryLevel5TopStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel5TopStroke, writer.tag(66, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level1_bottom_stroke = 67; */
        if (message.categoryLevel1BottomStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel1BottomStroke, writer.tag(67, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level2_bottom_stroke = 68; */
        if (message.categoryLevel2BottomStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel2BottomStroke, writer.tag(68, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level3_bottom_stroke = 69; */
        if (message.categoryLevel3BottomStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel3BottomStroke, writer.tag(69, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level4_bottom_stroke = 70; */
        if (message.categoryLevel4BottomStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel4BottomStroke, writer.tag(70, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level5_bottom_stroke = 71; */
        if (message.categoryLevel5BottomStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel5BottomStroke, writer.tag(71, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level1_interior_stroke = 72; */
        if (message.categoryLevel1InteriorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel1InteriorStroke, writer.tag(72, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level2_interior_stroke = 73; */
        if (message.categoryLevel2InteriorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel2InteriorStroke, writer.tag(73, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level3_interior_stroke = 74; */
        if (message.categoryLevel3InteriorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel3InteriorStroke, writer.tag(74, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level4_interior_stroke = 75; */
        if (message.categoryLevel4InteriorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel4InteriorStroke, writer.tag(75, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level5_interior_stroke = 76; */
        if (message.categoryLevel5InteriorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel5InteriorStroke, writer.tag(76, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level1_label_separator_stroke = 77; */
        if (message.categoryLevel1LabelSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel1LabelSeparatorStroke, writer.tag(77, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level2_label_separator_stroke = 78; */
        if (message.categoryLevel2LabelSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel2LabelSeparatorStroke, writer.tag(78, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level3_label_separator_stroke = 79; */
        if (message.categoryLevel3LabelSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel3LabelSeparatorStroke, writer.tag(79, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level4_label_separator_stroke = 80; */
        if (message.categoryLevel4LabelSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel4LabelSeparatorStroke, writer.tag(80, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive category_level5_label_separator_stroke = 81; */
        if (message.categoryLevel5LabelSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.categoryLevel5LabelSeparatorStroke, writer.tag(81, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_pivot_group_horizontal_stroke = 82; */
        if (message.tableBodyPivotGroupHorizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyPivotGroupHorizontalStroke, writer.tag(82, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_pivot_group_vertical_stroke = 83; */
        if (message.tableBodyPivotGroupVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyPivotGroupVerticalStroke, writer.tag(83, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_pivot_deemphasis_horizontal_stroke = 84; */
        if (message.tableBodyPivotDeemphasisHorizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyPivotDeemphasisHorizontalStroke, writer.tag(84, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive table_body_pivot_emphasis_vertical_stroke = 85; */
        if (message.tableBodyPivotEmphasisVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.tableBodyPivotEmphasisVerticalStroke, writer.tag(85, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_column_pivot_group_horizontal_stroke = 86; */
        if (message.headerColumnPivotGroupHorizontalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerColumnPivotGroupHorizontalStroke, writer.tag(86, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_column_pivot_group_deemphasis_stroke = 87; */
        if (message.headerColumnPivotGroupDeemphasisStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerColumnPivotGroupDeemphasisStroke, writer.tag(87, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_column_pivot_separator_stroke = 88; */
        if (message.headerColumnPivotSeparatorStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerColumnPivotSeparatorStroke, writer.tag(88, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_row_pivot_group_vertical_stroke = 89; */
        if (message.headerRowPivotGroupVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerRowPivotGroupVerticalStroke, writer.tag(89, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_row_pivot_group_deemphasis_stroke = 90; */
        if (message.headerRowPivotGroupDeemphasisStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerRowPivotGroupDeemphasisStroke, writer.tag(90, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive header_row_pivot_title_stroke = 91; */
        if (message.headerRowPivotTitleStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.headerRowPivotTitleStroke, writer.tag(91, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSD.StrokeArchive footer_row_pivot_group_vertical_stroke = 92; */
        if (message.footerRowPivotGroupVerticalStroke)
            TSDArchives_2.StrokeArchive.internalBinaryWrite(message.footerRowPivotGroupVerticalStroke, writer.tag(92, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.TableStylePropertiesArchive
 */
exports.TableStylePropertiesArchive = new TableStylePropertiesArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TableStylePresetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.TableStylePresetArchive", [
            { no: 1, name: "index", kind: "scalar", opt: true, T: 5 /*ScalarType.INT32*/ },
            { no: 2, name: "image", kind: "message", T: () => TSPMessages_1.Reference },
            { no: 3, name: "style_network", kind: "message", T: () => TSPMessages_1.Reference }
        ]);
    }
    create(value) {
        const message = {};
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* optional int32 index */ 1:
                    message.index = reader.int32();
                    break;
                case /* optional TSP.Reference image */ 2:
                    message.image = TSPMessages_1.Reference.internalBinaryRead(reader, reader.uint32(), options, message.image);
                    break;
                case /* optional TSP.Reference style_network */ 3:
                    message.styleNetwork = TSPMessages_1.Reference.internalBinaryRead(reader, reader.uint32(), options, message.styleNetwork);
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* optional int32 index = 1; */
        if (message.index !== undefined)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.index);
        /* optional TSP.Reference image = 2; */
        if (message.image)
            TSPMessages_1.Reference.internalBinaryWrite(message.image, writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* optional TSP.Reference style_network = 3; */
        if (message.styleNetwork)
            TSPMessages_1.Reference.internalBinaryWrite(message.styleNetwork, writer.tag(3, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.TableStylePresetArchive
 */
exports.TableStylePresetArchive = new TableStylePresetArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TableStrokePresetArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.TableStrokePresetArchive", [
            { no: 1, name: "index", kind: "scalar", T: 5 /*ScalarType.INT32*/ }
        ]);
    }
    create(value) {
        const message = { index: 0 };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* int32 index */ 1:
                    message.index = reader.int32();
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* int32 index = 1; */
        if (message.index !== 0)
            writer.tag(1, runtime_1.WireType.Varint).int32(message.index);
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.TableStrokePresetArchive
 */
exports.TableStrokePresetArchive = new TableStrokePresetArchive$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ThemePresetsArchive$Type extends runtime_5.MessageType {
    constructor() {
        super("TST.ThemePresetsArchive", [
            { no: 1, name: "table_style_presets", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.Reference },
            { no: 2, name: "table_cell_stroke_presets", kind: "message", repeat: 2 /*RepeatType.UNPACKED*/, T: () => TSPMessages_1.Reference }
        ]);
    }
    create(value) {
        const message = { tableStylePresets: [], tableCellStrokePresets: [] };
        globalThis.Object.defineProperty(message, runtime_4.MESSAGE_TYPE, { enumerable: false, value: this });
        if (value !== undefined)
            (0, runtime_3.reflectionMergePartial)(this, message, value);
        return message;
    }
    internalBinaryRead(reader, length, options, target) {
        let message = target ?? this.create(), end = reader.pos + length;
        while (reader.pos < end) {
            let [fieldNo, wireType] = reader.tag();
            switch (fieldNo) {
                case /* repeated TSP.Reference table_style_presets */ 1:
                    message.tableStylePresets.push(TSPMessages_1.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                case /* repeated TSP.Reference table_cell_stroke_presets */ 2:
                    message.tableCellStrokePresets.push(TSPMessages_1.Reference.internalBinaryRead(reader, reader.uint32(), options));
                    break;
                default:
                    let u = options.readUnknownField;
                    if (u === "throw")
                        throw new globalThis.Error(`Unknown field ${fieldNo} (wire type ${wireType}) for ${this.typeName}`);
                    let d = reader.skip(wireType);
                    if (u !== false)
                        (u === true ? runtime_2.UnknownFieldHandler.onRead : u)(this.typeName, message, fieldNo, wireType, d);
            }
        }
        return message;
    }
    internalBinaryWrite(message, writer, options) {
        /* repeated TSP.Reference table_style_presets = 1; */
        for (let i = 0; i < message.tableStylePresets.length; i++)
            TSPMessages_1.Reference.internalBinaryWrite(message.tableStylePresets[i], writer.tag(1, runtime_1.WireType.LengthDelimited).fork(), options).join();
        /* repeated TSP.Reference table_cell_stroke_presets = 2; */
        for (let i = 0; i < message.tableCellStrokePresets.length; i++)
            TSPMessages_1.Reference.internalBinaryWrite(message.tableCellStrokePresets[i], writer.tag(2, runtime_1.WireType.LengthDelimited).fork(), options).join();
        let u = options.writeUnknownFields;
        if (u !== false)
            (u == true ? runtime_2.UnknownFieldHandler.onWrite : u)(this.typeName, message, writer);
        return writer;
    }
}
/**
 * @generated MessageType for protobuf message TST.ThemePresetsArchive
 */
exports.ThemePresetsArchive = new ThemePresetsArchive$Type();
